<html lang="en">
<head>
<title>R FAQ</title>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
<meta name="description" content="R FAQ">
<meta name="generator" content="makeinfo 4.8">
<link title="Top" rel="top" href="#Top">
<link href="http://www.gnu.org/software/texinfo/" rel="generator-home" title="Texinfo Homepage">
<meta http-equiv="Content-Style-Type" content="text/css">
<style type="text/css"><!--
  pre.display { font-family:inherit }
  pre.format  { font-family:inherit }
  pre.smalldisplay { font-family:inherit; font-size:smaller }
  pre.smallformat  { font-family:inherit; font-size:smaller }
  pre.smallexample { font-size:smaller }
  pre.smalllisp    { font-size:smaller }
  span.sc    { font-variant:small-caps }
  span.roman { font-family:serif; font-weight:normal; } 
  span.sansserif { font-family:sans-serif; font-weight:normal; } 
body {margin-left: 5%; margin-right: 5%;}

H1{             background: white;
                color: rgb(25%, 25%, 25%);
                font-family: monospace;
                font-size: xx-large;
                text-align: center }

H2{             background: white;
                color: rgb(40%, 40%, 40%);
                font-family: monospace;
                font-size: x-large;
                text-align: center }

H3{             background: white;
                color: rgb(40%, 40%, 40%);
                font-family: monospace;
                font-size: large }

H4{             background: white;
                color: rgb(40%, 40%, 40%);
                font-family: monospace }

span.samp{font-family: monospace}
span.command{font-family: monospace}
span.option{font-family: monospace}
span.file{font-family: monospace}
span.env{font-family: monospace}

ul.menu{list-style-type: decimal}
--></style>
</head>
<body>
<h1 class="settitle">R FAQ</h1>
<div class="node">
<p><hr>
<a name="Top"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Introduction">Introduction</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#dir">(dir)</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#dir">(dir)</a>

</div>

<h2 class="unnumbered">R FAQ</h2>

<p><h2>Frequently Asked Questions on R</h2>
<h2>Version 2.7.2008-08-22</h2>
<h2>ISBN 3-900051-08-9</h2>
<address>Kurt Hornik</address>
<p><p><hr><p>

<ul class="menu">
<li><a accesskey="1" href="#Introduction">Introduction</a>
<li><a accesskey="2" href="#R-Basics">R Basics</a>
<li><a accesskey="3" href="#R-and-S">R and S</a>
<li><a accesskey="4" href="#R-Web-Interfaces">R Web Interfaces</a>
<li><a accesskey="5" href="#R-Add_002dOn-Packages">R Add-On Packages</a>
<li><a accesskey="6" href="#R-and-Emacs">R and Emacs</a>
<li><a accesskey="7" href="#R-Miscellanea">R Miscellanea</a>
<li><a accesskey="8" href="#R-Programming">R Programming</a>
<li><a accesskey="9" href="#R-Bugs">R Bugs</a>
<li><a href="#Acknowledgments">Acknowledgments</a>
</ul>

<div class="node">
<p><hr>
<a name="Introduction"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#R-Basics">R Basics</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Top">Top</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Top">Top</a>

</div>

<h2 class="chapter">1 Introduction</h2>

<p>This document contains answers to some of the most frequently asked
questions about R.

<ul class="menu">
<li><a accesskey="1" href="#Legalese">Legalese</a>
<li><a accesskey="2" href="#Obtaining-this-document">Obtaining this document</a>
<li><a accesskey="3" href="#Citing-this-document">Citing this document</a>
<li><a accesskey="4" href="#Notation">Notation</a>
<li><a accesskey="5" href="#Feedback">Feedback</a>
</ul>

<div class="node">
<p><hr>
<a name="Legalese"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Obtaining-this-document">Obtaining this document</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Introduction">Introduction</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Introduction">Introduction</a>

</div>

<h3 class="section">1.1 Legalese</h3>

<p>This document is copyright &copy; 1998&ndash;2008 by Kurt
Hornik.

   <p>This document is free software; you can redistribute it and/or modify it
under the terms of the <acronym>GNU</acronym> General Public License as published
by the Free Software Foundation; either version 2, or (at your option)
any later version.

   <p>This document is distributed in the hope that it will be useful, but
WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
<acronym>GNU</acronym> General Public License for more details.

   <p>A copy of the <acronym>GNU</acronym> General Public License is available via WWW
at

<pre class="display">     <a href="http://www.gnu.org/copyleft/gpl.html">http://www.gnu.org/copyleft/gpl.html</a>.
</pre>
   <p class="noindent">You can also obtain it by writing to the Free Software Foundation, Inc.,
51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, U.S.A.

<div class="node">
<p><hr>
<a name="Obtaining-this-document"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Citing-this-document">Citing this document</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Legalese">Legalese</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Introduction">Introduction</a>

</div>

<h3 class="section">1.2 Obtaining this document</h3>

<p>The latest version of this document is always available from

<pre class="display">     <a href="http://CRAN.R-project.org/doc/FAQ/">http://CRAN.R-project.org/doc/FAQ/</a>
</pre>
   <p>From there, you can obtain versions converted to
<a href="http://CRAN.R-project.org/doc/FAQ/R-FAQ.txt">plain <acronym>ASCII</acronym> text</a>,
<a href="http://CRAN.R-project.org/doc/FAQ/R-FAQ.dvi.gz">DVI</a>,
<a href="http://CRAN.R-project.org/doc/FAQ/R-FAQ.info.gz"><acronym>GNU</acronym> info</a>, <a href="http://CRAN.R-project.org/doc/FAQ/R-FAQ.html"><acronym>HTML</acronym></a>,
<a href="http://CRAN.R-project.org/doc/FAQ/R-FAQ.pdf">PDF</a>,
<a href="http://CRAN.R-project.org/doc/FAQ/R-FAQ.ps.gz">PostScript</a> as
well as the <a href="http://CRAN.R-project.org/doc/FAQ/R-FAQ.texi">Texinfo source</a> used for creating all these formats using the
<a href="http://texinfo.org/"><acronym>GNU</acronym> Texinfo system</a>.

   <p>You can also obtain the R <acronym>FAQ</acronym> from the <samp><span class="file">doc/FAQ</span></samp>
subdirectory of a <acronym>CRAN</acronym> site (see <a href="#What-is-CRAN_003f">What is CRAN?</a>).

<div class="node">
<p><hr>
<a name="Citing-this-document"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Notation">Notation</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Obtaining-this-document">Obtaining this document</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Introduction">Introduction</a>

</div>

<h3 class="section">1.3 Citing this document</h3>

<p>In publications, please refer to this <acronym>FAQ</acronym> as Hornik
(2008), &ldquo;The R <acronym>FAQ</acronym>&rdquo;, and give the above,
<em>official</em> <acronym>URL</acronym> and the ISBN 3-900051-08-9:

<pre class="example">     @Misc{,
       author        = {Kurt Hornik},
       title         = {The {R} {FAQ}},
       year          = {2008},
       note          = {{ISBN} 3-900051-08-9},
       url           = {http://CRAN.R-project.org/doc/FAQ/R-FAQ.html}
     }
</pre>
   <div class="node">
<p><hr>
<a name="Notation"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Feedback">Feedback</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Citing-this-document">Citing this document</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Introduction">Introduction</a>

</div>

<h3 class="section">1.4 Notation</h3>

<p>Everything should be pretty standard.  `<samp><span class="samp">R&gt;</span></samp>' is used for the R
prompt, and a `<samp><span class="samp">$</span></samp>' for the shell prompt (where applicable).

<div class="node">
<p><hr>
<a name="Feedback"></a>
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Notation">Notation</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Introduction">Introduction</a>

</div>

<h3 class="section">1.5 Feedback</h3>

<p>Feedback via email to <a href="mailto:Kurt.Hornik@R-project.org">Kurt.Hornik@R-project.org</a> is of course
most welcome.

   <p>In particular, note that I do not have access to Windows or Macintosh
systems.  Features specific to the Windows and Mac OS X ports of R are
described in the
<a href="http://CRAN.R-project.org/bin/windows/base/rw-FAQ.html">&ldquo;R for Windows <acronym>FAQ</acronym>&rdquo;</a> and the
<a href="http://CRAN.R-project.org/bin/macosx/RMacOSX-FAQ.html">&ldquo;R for Mac OS X <acronym>FAQ</acronym></a>.  If you have information on Macintosh or Windows
systems that you think should be added to this document, please let me
know.

<!-- <FIXME> -->
<!-- Should we maybe have direct links inside the R tree to the various -->
<!-- rw-FAQ versions? -->
<!-- </FIXME> -->
<div class="node">
<p><hr>
<a name="R-Basics"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#R-and-S">R and S</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Introduction">Introduction</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Top">Top</a>

</div>

<h2 class="chapter">2 R Basics</h2>

<ul class="menu">
<li><a accesskey="1" href="#What-is-R_003f">What is R?</a>
<li><a accesskey="2" href="#What-machines-does-R-run-on_003f">What machines does R run on?</a>
<li><a accesskey="3" href="#What-is-the-current-version-of-R_003f">What is the current version of R?</a>
<li><a accesskey="4" href="#How-can-R-be-obtained_003f">How can R be obtained?</a>
<li><a accesskey="5" href="#How-can-R-be-installed_003f">How can R be installed?</a>
<li><a accesskey="6" href="#Are-there-Unix-binaries-for-R_003f">Are there Unix binaries for R?</a>
<li><a accesskey="7" href="#What-documentation-exists-for-R_003f">What documentation exists for R?</a>
<li><a accesskey="8" href="#Citing-R">Citing R</a>
<li><a accesskey="9" href="#What-mailing-lists-exist-for-R_003f">What mailing lists exist for R?</a>
<li><a href="#What-is-CRAN_003f">What is CRAN?</a>
<li><a href="#Can-I-use-R-for-commercial-purposes_003f">Can I use R for commercial purposes?</a>
<li><a href="#Why-is-R-named-R_003f">Why is R named R?</a>
<li><a href="#What-is-the-R-Foundation_003f">What is the R Foundation?</a>
</ul>

<div class="node">
<p><hr>
<a name="What-is-R%3f"></a>
<a name="What-is-R_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#What-machines-does-R-run-on_003f">What machines does R run on?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#R-Basics">R Basics</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Basics">R Basics</a>

</div>

<h3 class="section">2.1 What is R?</h3>

<p>R is a system for statistical computation and graphics.  It consists of
a language plus a run-time environment with graphics, a debugger, access
to certain system functions, and the ability to run programs stored in
script files.

   <p>The design of R has been heavily influenced by two existing languages:
Becker, Chambers &amp; Wilks' S (see <a href="#What-is-S_003f">What is S?</a>) and Sussman's
<a href="http://www.cs.indiana.edu/scheme-repository/home.html">Scheme</a>. 
Whereas the resulting language is very similar in appearance to S, the
underlying implementation and semantics are derived from Scheme. 
See <a href="#What-are-the-differences-between-R-and-S_003f">What are the differences between R and S?</a>, for further details.

   <p>The core of R is an interpreted computer language which allows branching
and looping as well as modular programming using functions.  Most of the
user-visible functions in R are written in R.  It is possible for the
user to interface to procedures written in the C, C++, or FORTRAN
languages for efficiency.  The R distribution contains functionality for
a large number of statistical procedures.  Among these are: linear and
generalized linear models, nonlinear regression models, time series
analysis, classical parametric and nonparametric tests, clustering and
smoothing.  There is also a large set of functions which provide a
flexible graphical environment for creating various kinds of data
presentations.  Additional modules (&ldquo;add-on packages&rdquo;) are available
for a variety of specific purposes (see <a href="#R-Add_002dOn-Packages">R Add-On Packages</a>).

   <p>R was initially written by <a href="mailto:Ross.Ihaka@R-project.org">Ross Ihaka</a>
and <a href="mailto:Robert.Gentleman@R-project.org">Robert Gentleman</a> at the
Department of Statistics of the University of Auckland in Auckland, New
Zealand.  In addition, a large group of individuals has contributed to R
by sending code and bug reports.

   <p>Since mid-1997 there has been a core group (the &ldquo;R Core Team&rdquo;) who can
modify the R source code archive.  The group currently consists of Doug
Bates, John Chambers, Peter Dalgaard, Robert Gentleman, Kurt Hornik,
Stefano Iacus, Ross Ihaka, Friedrich Leisch, Thomas Lumley, Martin
Maechler, Duncan Murdoch, Paul Murrell, Martyn Plummer, Brian Ripley,
Duncan Temple Lang, Luke Tierney, and Simon Urbanek.

   <p>R has a home page at <a href="http://www.R-project.org/">http://www.R-project.org/</a>.  It is
<a href="http://www.gnu.org/philosophy/free-sw.html">free software</a>
distributed under a <acronym>GNU</acronym>-style
<a href="http://www.gnu.org/copyleft/copyleft.html">copyleft</a>, and an
official part of the <a href="http://www.gnu.org/"><acronym>GNU</acronym></a> project
(&ldquo;<acronym>GNU</acronym> S&rdquo;).

<div class="node">
<p><hr>
<a name="What-machines-does-R-run-on%3f"></a>
<a name="What-machines-does-R-run-on_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#What-is-the-current-version-of-R_003f">What is the current version of R?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#What-is-R_003f">What is R?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Basics">R Basics</a>

</div>

<h3 class="section">2.2 What machines does R run on?</h3>

<p>R is being developed for the Unix, Windows and Mac families of operating
systems.  Support for Mac OS Classic ended with R 1.7.1.

   <p>The current version of R will configure and build under a number of
common Unix platforms including <var>cpu</var>-linux-gnu for the i386, alpha,
arm, hppa, ia64, m68k, mips/mipsel, powerpc, s390, sparc (e.g.,
<a href="http://buildd.debian.org/build.php?&amp;pkg=r-base">http://buildd.debian.org/build.php?&amp;pkg=r-base</a>), and x86_64 CPUs,
powerpc-apple-darwin, mips-sgi-irix,
<!-- <FIXME> -->
<!-- Not sure anymore ... -->
<!-- alpha-dec-osf4, -->
<!-- i386-freebsd, -->
<!-- i386-sun-solaris, -->
<!-- hppa-hp-hpux, -->
<!-- </FIXME> -->
rs6000-ibm-aix, and sparc-sun-solaris.

<!-- and according to @email{jlindsey@@luc.ac.be, Jim Lindsey} also on -->
<!-- Mac, Amiga and Atari under m68k-linux. -->
<p>If you know about other platforms, please drop us a note.

<div class="node">
<p><hr>
<a name="What-is-the-current-version-of-R%3f"></a>
<a name="What-is-the-current-version-of-R_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-R-be-obtained_003f">How can R be obtained?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#What-machines-does-R-run-on_003f">What machines does R run on?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Basics">R Basics</a>

</div>

<h3 class="section">2.3 What is the current version of R?</h3>

<p>The current released version is 2.7.2.  Based on this
`major.minor.patchlevel' numbering scheme, there are two development
versions of R, a patched version of the current release (`r-patched')
and one working towards the next minor or eventually major (`r-devel')
releases of R, respectively.  Version r-patched is for bug fixes mostly. 
New features are typically introduced in r-devel.

<div class="node">
<p><hr>
<a name="How-can-R-be-obtained%3f"></a>
<a name="How-can-R-be-obtained_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-R-be-installed_003f">How can R be installed?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#What-is-the-current-version-of-R_003f">What is the current version of R?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Basics">R Basics</a>

</div>

<h3 class="section">2.4 How can R be obtained?</h3>

<p>Sources, binaries and documentation for R can be obtained via <acronym>CRAN</acronym>,
the &ldquo;Comprehensive R Archive Network&rdquo; (see <a href="#What-is-CRAN_003f">What is CRAN?</a>).

   <p>Sources are also available via <a href="https://svn.R-project.org/R/">https://svn.R-project.org/R/</a>, the
R Subversion repository, but currently not via anonymous rsync (nor
CVS).

   <p>Tarballs with daily snapshots of the r-devel and r-patched development
versions of R can be found at
<a href="ftp://ftp.stat.math.ethz.ch/Software/R">ftp://ftp.stat.math.ethz.ch/Software/R</a>.

<!-- Sources are also available via anonymous rsync.  Use -->
<!-- @example -->
<!-- rsync -rptC -delete rsync.R-project.org::@var{module} R -->
<!-- @end example -->
<!-- @noindent -->
<!-- to create a copy of the source tree specified by @var{module} in the -->
<!-- subdirectory @file{R} of the current directory, where @var{module} -->
<!-- specifies one of the three existing flavors of the R sources, and can be -->
<!-- one of @samp{r-release} (current released version), @samp{r-patched} -->
<!-- (patched released version), and @samp{r-devel} (development version). -->
<!-- The rsync trees are created directly from the master CVS archive and are -->
<!-- updated hourly.  The @option{-C} and in the @command{rsync} command -->
<!-- is to cause it to skip the CVS directories.  Further information on -->
<!-- @command{rsync} is available at @url{http://rsync.samba.org/rsync/}. -->
<!-- @c <NOTE> -->
<!-- @c Keep in sync with R-admin. -->
<!-- Note that the sources available via rsync do not include the recommended -->
<!-- packages, whereas these are included in the tarballs of released -->
<!-- versions.  To install the appropriate sources for the recommended -->
<!-- packages, run @command{./tools/rsync-recommended} from the top-level of -->
<!-- the R sources that you pulled by rsync. -->
<!-- @c </NOTE> -->
<!-- The sources of the development version are also available via anonymous -->
<!-- CVS.  See @url{http://anoncvs.R-project.org} for more information. -->
<div class="node">
<p><hr>
<a name="How-can-R-be-installed%3f"></a>
<a name="How-can-R-be-installed_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Are-there-Unix-binaries-for-R_003f">Are there Unix binaries for R?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-R-be-obtained_003f">How can R be obtained?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Basics">R Basics</a>

</div>

<h3 class="section">2.5 How can R be installed?</h3>

<ul class="menu">
<li><a accesskey="1" href="#How-can-R-be-installed-_0028Unix_0029">How can R be installed (Unix)</a>
<li><a accesskey="2" href="#How-can-R-be-installed-_0028Windows_0029">How can R be installed (Windows)</a>
<li><a accesskey="3" href="#How-can-R-be-installed-_0028Macintosh_0029">How can R be installed (Macintosh)</a>
</ul>

<div class="node">
<p><hr>
<a name="How-can-R-be-installed-(Unix)"></a>
<a name="How-can-R-be-installed-_0028Unix_0029"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-R-be-installed-_0028Windows_0029">How can R be installed (Windows)</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-R-be-installed_003f">How can R be installed?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#How-can-R-be-installed_003f">How can R be installed?</a>

</div>

<h4 class="subsection">2.5.1 How can R be installed (Unix)</h4>

<p>If R is already installed, it can be started by typing <kbd>R</kbd> at the
shell prompt (of course, provided that the executable is in your path).

   <p>If binaries are available for your platform (see <a href="#Are-there-Unix-binaries-for-R_003f">Are there Unix binaries for R?</a>), you can use these, following the instructions that
come with them.

   <p>Otherwise, you can compile and install R yourself, which can be done
very easily under a number of common Unix platforms (see <a href="#What-machines-does-R-run-on_003f">What machines does R run on?</a>).  The file <samp><span class="file">INSTALL</span></samp> that comes with the
R distribution contains a brief introduction, and the &ldquo;R Installation
and Administration&rdquo; guide (see <a href="#What-documentation-exists-for-R_003f">What documentation exists for R?</a>)
has full details.

   <p>Note that you need a FORTRAN compiler or perhaps <samp><span class="command">f2c</span></samp> in
addition to a C compiler to build R.  Also, you need Perl version 5 to
build the R object documentations.  (If this is not available on your
system, you can obtain a PDF version of the object reference manual via
<acronym>CRAN</acronym>.)

   <p>In the simplest case, untar the R source code, change to the directory
thus created, and issue the following commands (at the shell prompt):

<pre class="example">     $ ./configure
     $ make
</pre>
   <p>If these commands execute successfully, the R binary and a shell script
front-end called <samp><span class="file">R</span></samp> are created and copied to the <samp><span class="file">bin</span></samp>
directory.  You can copy the script to a place where users can invoke
it, for example to <samp><span class="file">/usr/local/bin</span></samp>.  In addition, plain text help
pages as well as <acronym>HTML</acronym> and LaTeX versions of the documentation are
built.

   <p>Use <kbd>make dvi</kbd> to create DVI versions of the R manuals, such as
<samp><span class="file">refman.dvi</span></samp> (an R object reference index) and <samp><span class="file">R-exts.dvi</span></samp>,
the &ldquo;R Extension Writers Guide&rdquo;, in the <samp><span class="file">doc/manual</span></samp>
subdirectory.  These files can be previewed and printed using standard
programs such as <samp><span class="command">xdvi</span></samp> and <samp><span class="command">dvips</span></samp>.  You can also use
<kbd>make pdf</kbd> to build PDF (Portable Document Format) version of the
manuals, and view these using e.g. Acrobat.  Manuals written in the
<acronym>GNU</acronym> Texinfo system can also be converted to info files
suitable for reading online with Emacs or stand-alone <acronym>GNU</acronym>
Info; use <kbd>make info</kbd> to create these versions (note that this
requires Makeinfo version 4.5).

   <p>Finally, use <kbd>make check</kbd> to find out whether your R system works
correctly.

   <p>You can also perform a &ldquo;system-wide&rdquo; installation using <kbd>make
install</kbd>.  By default, this will install to the following directories:

     <dl>
<dt><samp><span class="file">${prefix}/bin</span></samp><dd>the front-end shell script
<br><dt><samp><span class="file">${prefix}/man/man1</span></samp><dd>the man page
<br><dt><samp><span class="file">${prefix}/lib/R</span></samp><dd>all the rest (libraries, on-line help system, <small class="dots">...</small>).  This is the &ldquo;R
Home Directory&rdquo; (<samp><span class="env">R_HOME</span></samp>) of the installed system. 
</dl>

<p class="noindent">In the above, <code>prefix</code> is determined during configuration
(typically <samp><span class="file">/usr/local</span></samp>) and can be set by running
<samp><span class="command">configure</span></samp> with the option

<pre class="example">     $ ./configure --prefix=/where/you/want/R/to/go
</pre>
   <p class="noindent">(E.g., the R executable will then be installed into
<samp><span class="file">/where/you/want/R/to/go/bin</span></samp>.)

   <p>To install DVI, info and PDF versions of the manuals, use <kbd>make
install-dvi</kbd>, <kbd>make install-info</kbd> and <kbd>make install-pdf</kbd>,
respectively.

<div class="node">
<p><hr>
<a name="How-can-R-be-installed-(Windows)"></a>
<a name="How-can-R-be-installed-_0028Windows_0029"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-R-be-installed-_0028Macintosh_0029">How can R be installed (Macintosh)</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-R-be-installed-_0028Unix_0029">How can R be installed (Unix)</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#How-can-R-be-installed_003f">How can R be installed?</a>

</div>

<h4 class="subsection">2.5.2 How can R be installed (Windows)</h4>

<p>The <samp><span class="file">bin/windows</span></samp> directory of a <acronym>CRAN</acronym> site contains binaries for
a base distribution and a large number of add-on packages from <acronym>CRAN</acronym>
to run on Windows 2000 and later (including 64-bit versions of Windows)
on ix86 and x86_64 chips. The Windows version of R was created by Robert
Gentleman and Guido Masarotto, and is now being developed and maintained
by <a href="mailto:murdoch@stats.uwo.ca">Duncan Murdoch</a> and
<a href="mailto:Brian.Ripley@R-project.org">Brian D. Ripley</a>.

<!-- Note that when uncompressing the zip files, the pkunzip program needs to -->
<!-- be invoked with the @samp{-D} flag to create subdirectories.  Also, be -->
<!-- aware that some decompression programs do not preserve long file names -->
<!-- properly. -->
<p>For most installations the Windows installer program will be the easiest
tool to use.

   <p>See the <a href="http://CRAN.R-project.org/bin/windows/base/rw-FAQ.html">&ldquo;R for Windows <acronym>FAQ</acronym>&rdquo;</a> for more details.

<div class="node">
<p><hr>
<a name="How-can-R-be-installed-(Macintosh)"></a>
<a name="How-can-R-be-installed-_0028Macintosh_0029"></a>
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-R-be-installed-_0028Windows_0029">How can R be installed (Windows)</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#How-can-R-be-installed_003f">How can R be installed?</a>

</div>

<h4 class="subsection">2.5.3 How can R be installed (Macintosh)</h4>

<p>The <samp><span class="file">bin/macosx</span></samp> directory of a <acronym>CRAN</acronym> site contains a standard
Apple installer package inside a disk image named <samp><span class="file">R.dmg</span></samp>.  Once
downloaded and executed, the installer will install the current
non-developer release of R.  RAqua is a native Mac OS X Darwin version
of R with a R.app Mac OS X <acronym>GUI</acronym>.  Inside
<samp><span class="file">bin/macosx/powerpc/contrib/</span><var>x</var><span class="file">.</span><var>y</var></samp> there are prebuilt
binary packages (for powerpc version of Mac OS X) to be used with RAqua
corresponding to the &ldquo;<var>x</var>.<var>y</var>&rdquo; release of R. The installation
of these packages is available through the &ldquo;Package&rdquo; menu of the R.app
<acronym>GUI</acronym>.  This port of R for Mac OS X is maintained by
<a href="mailto:Stefano.Iacus@R-project.org">Stefano Iacus</a>.  The
<a href="http://CRAN.R-project.org/bin/macosx/RMacOSX-FAQ.html">&ldquo;R for Mac OS X <acronym>FAQ</acronym></a> has more details.

   <p>The <samp><span class="file">bin/macos</span></samp> directory of a <acronym>CRAN</acronym> site contains bin-hexed
(<samp><span class="file">hqx</span></samp>) and stuffit (<samp><span class="file">sit</span></samp>) archives for a base distribution
and a large number of add-on packages of R 1.7.1 to run under Mac OS 8.6
to Mac OS 9.2.2.  This port of R for Macintosh is no longer supported.

<div class="node">
<p><hr>
<a name="Are-there-Unix-binaries-for-R%3f"></a>
<a name="Are-there-Unix-binaries-for-R_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#What-documentation-exists-for-R_003f">What documentation exists for R?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-R-be-installed_003f">How can R be installed?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Basics">R Basics</a>

</div>

<h3 class="section">2.6 Are there Unix binaries for R?</h3>

<!-- Linux binaries as of 2008-06-24: -->
<!-- debian -->
<!-- oldstable -->
<!-- i386   2.4.1  Johannes Ranke -->
<!-- amd64  2.4.1  Johannes Ranke -->
<!-- stable  i386   2.7.0  Johannes Ranke -->
<!-- amd64  2.7.0  Johannes Ranke -->
<!-- redhat -->
<!-- el4     i386   2.6.1  Bob Kinney <rhel_cran@hmdc.harvard.edu> -->
<!-- x86_64 2.6.1  Bob Kinney <rhel_cran@hmdc.harvard.edu> -->
<!-- el5     i386   2.6.1  Bob Kinney <rhel_cran@hmdc.harvard.edu> -->
<!-- x86_64 2.6.1  Bob Kinney <rhel_cran@hmdc.harvard.edu> -->
<!-- fedora7 i386   2.6.1  Martyn Plummer <Martyn.Plummer@R-project.org> -->
<!-- x86_64 2.6.1  Martyn Plummer <Martyn.Plummer@R-project.org> -->
<!-- fedora8 i386   2.6.1  Martyn Plummer <Martyn.Plummer@R-project.org> -->
<!-- x86_64 2.6.1  Martyn Plummer <Martyn.Plummer@R-project.org> -->
<!-- suse -->
<!-- 10.0    i586   2.7.0  Detlef Steuer <steuer@hsu-hh.de> -->
<!-- x86_64 2.7.0  Detlef Steuer <steuer@hsu-hh.de -->
<!-- 10.1    i586   2.7.0  Detlef Steuer <steuer@hsu-hh.de> -->
<!-- x86_64 2.7.0  Detlef Steuer <steuer@hsu-hh.de> -->
<!-- 10.2    i586   2.7.0  Detlef Steuer <steuer@hsu-hh.de> -->
<!-- x86_64 2.7.0  Detlef Steuer <steuer@hsu-hh.de> -->
<!-- 10.3    i586   2.7.0  Detlef Steuer <steuer@hsu-hh.de> -->
<!-- x86_64 2.7.0  Detlef Steuer <steuer@hsu-hh.de> -->
<!-- ubuntu -->
<!-- hardy   [Hardy Heron, 8.04] -->
<!-- i386   2.7.1  Vincent Goulet -->
<!-- amd64  2.7.1  Michael Rutter -->
<!-- gutsy   [Gutsy Gibbon, 7.10] -->
<!-- i386   2.7.1  Vincent Goulet -->
<!-- amd64  2.7.1  Michael Rutter -->
<!-- feisty  [Feisty Fawn,  7.04] -->
<!-- i386   2.7.1  Vincent Goulet -->
<!-- amd64  2.7.1  Michael Rutter -->
<!-- dapper  [Dapper Drake, 6.06] -->
<!-- i386   2.7.1  Vincent Goulet -->
<!-- amd64  2.7.1  Michael Rutter -->
<p>The <samp><span class="file">bin/linux</span></samp> directory of a <acronym>CRAN</acronym> site contains the following
packages.

   <blockquote>
        <p><table summary=""><tr align="left"><th valign="top"></th><th valign="top">CPU </th><th valign="top">Versions </th><th valign="top">Provider
<br></th></tr><tr align="left"><td valign="top">Debian   </td><td valign="top">i386/amd64 </td><td valign="top">stable/oldstable </td><td valign="top">Johannes Ranke
<!-- @item          @tab amd64 @tab stable/oldstable @tab Johannes Ranke -->
<br></td></tr><tr align="left"><td valign="top">Red Hat  </td><td valign="top">i386/x86_64 </td><td valign="top">fedora7/fedora8 </td><td valign="top">Martyn Plummer
<!-- @item          @tab x86_86 @tab fc5/fc6/fedora7/fedora8 @tab Martyn Plummer -->
<br></td></tr><tr align="left"><td valign="top"></td><td valign="top">i386/x86_64 </td><td valign="top">el4/el5 </td><td valign="top">Bob Kinney
<!-- @item          @tab x86_64 @tab Enterprise Linux @tab Matthew P. Cox -->
<br></td></tr><tr align="left"><td valign="top">SuSE     </td><td valign="top">i586/x86_64 </td><td valign="top">10.0/10.1/10.2/10.3 </td><td valign="top">Detlef Steuer
<!-- @item          @tab x86_64 @tab 10.0/10.1/10.2/10.3 @tab Detlef Steuer -->
<br></td></tr><tr align="left"><td valign="top">Ubuntu   </td><td valign="top">i386   </td><td valign="top">dapper/feisty/gutsy/hardy </td><td valign="top">Vincent Goulet
<br></td></tr><tr align="left"><td valign="top"></td><td valign="top">amd64  </td><td valign="top">dapper/feisty/gutsy/hardy </td><td valign="top">Michael Rutter
        <br></td></tr></table>
</blockquote>

   <p>Debian packages, maintained by Dirk Eddelbuettel and Doug Bates, have
long been part of the Debian distribution, and can be accessed through
APT, the Debian package maintenance tool.  Use e.g. <code>apt-get
install r-base r-recommended</code> to install the R environment and
recommended packages.  If you also want to build R packages from source,
also run <code>apt-get install r-base-dev</code> to obtain the additional
tools required for this.  So-called &ldquo;backports&rdquo; of the current R
packages for the <dfn>stable</dfn> distribution of Debian are provided by
Johannes Ranke, and available from CRAN.  Simply add the line

<pre class="example">     deb http://CRAN.R-project.org/bin/linux/debian stable/
</pre>
   <p class="noindent">(feel free to use a <acronym>CRAN</acronym> mirror instead of the master) to the file
<samp><span class="file">/etc/apt/sources.list</span></samp>, and install as usual.  More details on
installing and administering R on Debian Linux can be found at
<a href="http://CRAN.R-project.org/bin/linux/debian/README">http://CRAN.R-project.org/bin/linux/debian/README</a>.  These
backports should also be suitable for other Debian derivatives.  Native
backports for Ubuntu are provided by Vincent Goulet and Michael Rutter.

   <p>On SUSE, you can set up an installation source for R within Yast by
setting (e.g.)

<pre class="example">     Protocol: HTTP
     Server name: software.openSUSE.org
     Directory: /download/home:/dsteuer/openSUSE_10.2/
</pre>
   <p>With this setting, online updates will check for new versions of R.

<!-- Once you have added that line the programs @command{apt-get}, -->
<!-- @command{apt-cache}, and @command{dselect} (using the apt access -->
<!-- method) will automatically detect and install updates of the R -->
<!-- packages. -->
<!-- The @file{bin/osf} directory of a @CRAN{} site contains RPMs -->
<!-- by Albrecht Gebhardt for alpha systems running Alpha Unix -->
<!-- (OSF/Tru64). -->
<!-- There are also `tar' distributions for NEXTSTEP on the i386 and m68k -->
<!-- platforms in @file{bin/nextstep/i386} and @file{bin/nextstep/m68k}, -->
<!-- created by Stephen Shiboski <steve@biostat.ucsf.edu>. -->
<p>The <samp><span class="file">bin/solaris</span></samp> directory of a <acronym>CRAN</acronym> site contains binary
packages for Solaris on the SPARC and x64 platforms, provided by Mithun
Sridharan.

   <p>No other binary distributions are currently publically available via
<acronym>CRAN</acronym>.

   <p>A &ldquo;live&rdquo; Linux distribution with a particular focus on R is
<dfn>Quantian</dfn>, which provides a directly bootable and self-configuring
&ldquo;Live DVD&rdquo; containing numerous applications of interests to scientists
and researchers, including several hundred <acronym>CRAN</acronym> and Bioconductor
packages, the &ldquo;ESS&rdquo; extensions for Emacs, the &ldquo;JGR&rdquo; Java
<acronym>GUI</acronym> for R, the Ggobi visualization tool as well as several
other R interfaces. The <dfn>Quantian</dfn> website at
<a href="http://dirk.eddelbuettel.com/quantian/">http://dirk.eddelbuettel.com/quantian/</a> contains more details as
well download information.

<div class="node">
<p><hr>
<a name="What-documentation-exists-for-R%3f"></a>
<a name="What-documentation-exists-for-R_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Citing-R">Citing R</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Are-there-Unix-binaries-for-R_003f">Are there Unix binaries for R?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Basics">R Basics</a>

</div>

<h3 class="section">2.7 What documentation exists for R?</h3>

<p>Online documentation for most of the functions and variables in R
exists, and can be printed on-screen by typing <kbd>help(</kbd><var>name</var><kbd>)</kbd>
(or <kbd>?</kbd><var>name</var>) at the R prompt, where <var>name</var> is the name of
the topic help is sought for.  (In the case of unary and binary
operators and control-flow special forms, the name may need to be be
quoted.)

   <p>This documentation can also be made available as one reference manual
for on-line reading in <acronym>HTML</acronym> and PDF formats, and as hardcopy via
LaTeX, see <a href="#How-can-R-be-installed_003f">How can R be installed?</a>.  An up-to-date <acronym>HTML</acronym>
version is always available for web browsing at
<a href="http://stat.ethz.ch/R-manual/">http://stat.ethz.ch/R-manual/</a>.

   <p>Printed copies of the R reference manual for some version(s) are
available from Network Theory Ltd, at
<a href="http://www.network-theory.co.uk/R/base/">http://www.network-theory.co.uk/R/base/</a>.  For each set of manuals
sold, the publisher donates USD 10 to the R Foundation (see <a href="#What-is-the-R-Foundation_003f">What is the R Foundation?</a>).

   <p>The R distribution also comes with the following manuals.

     <ul>
<li>&ldquo;An Introduction to R&rdquo; (<samp><span class="file">R-intro</span></samp>)
includes information on data types, programming elements, statistical
modeling and graphics.  This document is based on the &ldquo;Notes on
<span class="sc">S-Plus</span>&rdquo; by Bill Venables and David Smith. 
<li>&ldquo;Writing R Extensions&rdquo; (<samp><span class="file">R-exts</span></samp>)
currently describes the process of creating R add-on packages, writing R
documentation, R's system and foreign language interfaces, and the R
<acronym>API</acronym>. 
<li>&ldquo;R Data Import/Export&rdquo; (<samp><span class="file">R-data</span></samp>)
is a guide to importing and exporting data to and from R. 
<li>&ldquo;The R Language Definition&rdquo; (<samp><span class="file">R-lang</span></samp>),
a first version of the &ldquo;Kernighan &amp; Ritchie of R&rdquo;, explains
evaluation, parsing, object oriented programming, computing on the
language, and so forth. 
<li>&ldquo;R Installation and Administration&rdquo; (<samp><span class="file">R-admin</span></samp>). 
<li>&ldquo;R Internals&rdquo; (<samp><span class="file">R-ints</span></samp>)
is a guide to R's internal structures. 
(Added in R 2.4.0.) 
</ul>

   <p>Books on R include

   <blockquote>
P. Dalgaard (2002), &ldquo;Introductory Statistics with R&rdquo;, Springer: New
York, ISBN 0-387-95475-9, <a href="http://www.biostat.ku.dk/~pd/ISwR.html">http://www.biostat.ku.dk/~pd/ISwR.html</a>.

        <p>J. Fox (2002), &ldquo;An R and <span class="sc">S-Plus</span> Companion to Applied Regression&rdquo;,
Sage Publications, ISBN 0-761-92280-6 (softcover) or 0-761-92279-2
(hardcover),
<a href="http://socserv.socsci.mcmaster.ca/jfox/Books/Companion/">http://socserv.socsci.mcmaster.ca/jfox/Books/Companion/</a>.

        <p>J. Maindonald and J. Braun (2003), &ldquo;Data Analysis and Graphics Using R:
An Example-Based Approach&rdquo;, Cambridge University Press, ISBN
0-521-81336-0, <a href="http://wwwmaths.anu.edu.au/~johnm/">http://wwwmaths.anu.edu.au/~johnm/</a>.

        <p>S. M. Iacus and G. Masarotto (2002), &ldquo;Laboratorio di statistica con
R&rdquo;, McGraw-Hill, ISBN 88-386-6084-0 (in Italian),<br>
<a href="http://www.ateneonline.it/LibroAteneo.asp?item_id=1436">http://www.ateneonline.it/LibroAteneo.asp?item_id=1436</a>.

        <p>P. Murrell (2005), &ldquo;R Graphics&rdquo;, Chapman &amp; Hall/CRC, ISBN:
1-584-88486-X,
<a href="http://www.stat.auckland.ac.nz/~paul/RGraphics/rgraphics.html">http://www.stat.auckland.ac.nz/~paul/RGraphics/rgraphics.html</a>.

   </blockquote>

<p class="noindent">The book

   <blockquote>
W. N. Venables and B. D. Ripley (2002), &ldquo;Modern Applied Statistics with
S&rdquo; (4th edition).  Springer, ISBN 0-387-95457-0
</blockquote>

<p class="noindent">has a home page at <a href="http://www.stats.ox.ac.uk/pub/MASS4/">http://www.stats.ox.ac.uk/pub/MASS4/</a> providing
additional material.  Its companion is

   <blockquote>
W. N. Venables and B. D. Ripley (2000), &ldquo;S Programming&rdquo;.  Springer,
ISBN 0-387-98966-8
</blockquote>

<p class="noindent">and provides an in-depth guide to writing software in the S language
which forms the basis of both the commercial <span class="sc">S-Plus</span> and the Open
Source R data analysis software systems.  See
<a href="http://www.stats.ox.ac.uk/pub/MASS3/Sprog/">http://www.stats.ox.ac.uk/pub/MASS3/Sprog/</a> for more information.

   <p>In addition to material written specifically or explicitly for R,
documentation for S/<span class="sc">S-Plus</span> (see <a href="#R-and-S">R and S</a>) can be used in
combination with this <acronym>FAQ</acronym> (see <a href="#What-are-the-differences-between-R-and-S_003f">What are the differences between R and S?</a>).  Introductory books include

   <blockquote>
P. Spector (1994), &ldquo;An introduction to S and <span class="sc">S-Plus</span>&rdquo;, Duxbury Press.

        <p>A. Krause and M. Olsen (2005), &ldquo;The Basics of <span class="sc">S-Plus</span>&rdquo; (4th edition). 
Springer, ISBN 0-387-26109-5. 
</blockquote>

   <p>The book

   <blockquote>
J. C. Pinheiro and D. M. Bates (2000), &ldquo;Mixed-Effects Models in S and
<span class="sc">S-Plus</span>&rdquo;, Springer, ISBN 0-387-98957-0
</blockquote>

<p class="noindent">provides a comprehensive guide to the use of the <strong>nlme</strong> package
for linear and nonlinear mixed-effects models. 
<!-- This has a home page at @url{http://nlme.stat.wisc.edu/MEMSS/}. -->

   <p>As an example of how R can be used in teaching an advanced introductory
statistics course, see

   <blockquote>
D. Nolan and T. Speed (2000), &ldquo;Stat Labs: Mathematical Statistics
Through Applications&rdquo;, Springer Texts in Statistics, ISBN
0-387-98974-9
</blockquote>

<p class="noindent">This integrates theory of statistics with the practice of statistics
through a collection of case studies (&ldquo;labs&rdquo;), and uses R to analyze
the data.  More information can be found at
<a href="http://www.stat.Berkeley.EDU/users/statlabs/">http://www.stat.Berkeley.EDU/users/statlabs/</a>.

   <p>Last, but not least, Ross' and Robert's experience in designing and
implementing R is described in Ihaka &amp; Gentleman (1996), &ldquo;R: A Language
for Data Analysis and Graphics&rdquo;,
<a href="http://www.amstat.org/publications/jcgs/"><em>Journal of Computational and Graphical Statistics</em></a>, <strong>5</strong>, 299&ndash;314.

   <p>An annotated bibliography (BibTeX format) of R-related publications
which includes most of the above references can be found at

<pre class="display">     <a href="http://www.R-project.org/doc/bib/R.bib">http://www.R-project.org/doc/bib/R.bib</a>
</pre>
   <div class="node">
<p><hr>
<a name="Citing-R"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#What-mailing-lists-exist-for-R_003f">What mailing lists exist for R?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#What-documentation-exists-for-R_003f">What documentation exists for R?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Basics">R Basics</a>

</div>

<h3 class="section">2.8 Citing R</h3>

<p>To cite R in publications, use

<pre class="example">     @Manual{,
       title        = {R: A Language and Environment for Statistical
                       Computing},
       author       = {{R Development Core Team}},
       organization = {R Foundation for Statistical Computing},
       address      = {Vienna, Austria},
       year         = 2008,
       note         = {{ISBN} 3-900051-07-0},
       url          = {http://www.R-project.org}
     }
</pre>
   <p>Citation strings (or BibTeX entries) for R and R packages can also be
obtained by <code>citation()</code>.

<div class="node">
<p><hr>
<a name="What-mailing-lists-exist-for-R%3f"></a>
<a name="What-mailing-lists-exist-for-R_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#What-is-CRAN_003f">What is CRAN?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Citing-R">Citing R</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Basics">R Basics</a>

</div>

<h3 class="section">2.9 What mailing lists exist for R?</h3>

<p>Thanks to <a href="mailto:Martin.Maechler@R-project.org">Martin Maechler</a>, there
are four mailing lists devoted to R.

     <dl>
<dt><code>R-announce</code><dd>A moderated list for major announcements about the development of R and
the availability of new code. 
<br><dt><code>R-packages</code><dd>A moderated list for announcements on the availability of new or
enhanced contributed packages. 
<br><dt><code>R-help</code><dd>The `main' R mailing list, for discussion about problems and solutions
using R, announcements (not covered by `R-announce' and `R-packages')
about the development of R and the availability of new code. 
<!-- enhancements and patches to the source code and documentation of R, -->
<!-- comparison and compatibility with S and @SPLUS{}, and for the posting of -->
<!-- nice examples and benchmarks. -->
<br><dt><code>R-devel</code><dd>This list is for questions and discussion about code development in R. 
<!-- discussions about the future of R, proposals of new functionality, and -->
<!-- pre-testing of new versions.  It is meant for those who maintain an -->
<!-- active position in the development of R. -->
</dl>

<p class="noindent">Please read the <a href="http://www.R-project.org/posting-guide.html">posting guide</a> <em>before</em> sending anything to any mailing list.

   <p>Note in particular that R-help is intended to be comprehensible to
people who want to use R to solve problems but who are not necessarily
interested in or knowledgeable about programming.  Questions likely to
prompt discussion unintelligible to non-programmers (e.g., questions
involving C or C++) should go to R-devel.

   <p>Convenient access to information on these lists, subscription, and
archives is provided by the web interface at
<a href="http://stat.ethz.ch/mailman/listinfo/">http://stat.ethz.ch/mailman/listinfo/</a>.  One can also subscribe
(or unsubscribe) via email, e.g. to R-help by sending `<samp><span class="samp">subscribe</span></samp>'
(or `<samp><span class="samp">unsubscribe</span></samp>') in the <em>body</em> of the message (not in the
subject!) to <a href="mailto:R-help-request@lists.R-project.org">R-help-request@lists.R-project.org</a>.

   <p>Send email to <a href="mailto:R-help@lists.R-project.org">R-help@lists.R-project.org</a> to send a message to
everyone on the R-help mailing list.  Subscription and posting to the
other lists is done analogously, with `<samp><span class="samp">R-help</span></samp>' replaced by
`<samp><span class="samp">R-announce</span></samp>', `<samp><span class="samp">R-packages</span></samp>', and `<samp><span class="samp">R-devel</span></samp>', respectively. 
Note that the R-announce and R-packages lists are gatewayed into R-help. 
Hence, you should subscribe to either of them only in case you are not
subscribed to R-help.

   <p>It is recommended that you send mail to R-help rather than only to the R
Core developers (who are also subscribed to the list, of course).  This
may save them precious time they can use for constantly improving R, and
will typically also result in much quicker feedback for yourself.

   <p>Of course, in the case of bug reports it would be very helpful to have
code which reliably reproduces the problem.  Also, make sure that you
include information on the system and version of R being used.  See
<a href="#R-Bugs">R Bugs</a> for more details.

   <p>See <a href="http://www.R-project.org/mail.html">http://www.R-project.org/mail.html</a> for more information on
the R mailing lists.

   <p>The R Core Team can be reached at <a href="mailto:R-core@lists.R-project.org">R-core@lists.R-project.org</a>
for comments and reports.

   <p>Many of the R project's mailing lists are also available via
<a href="http://gmane.org">Gmane</a>, from which they can be read with a web
browser, using an NNTP news reader, or via RSS feeds.  See
<a href="http://dir.gmane.org/index.php?prefix=gmane.comp.lang.r.">http://dir.gmane.org/index.php?prefix=gmane.comp.lang.r.</a> for
the available mailing lists, and <a href="http://www.gmane.org/rss.php">http://www.gmane.org/rss.php</a> for
details on RSS feeds.

<div class="node">
<p><hr>
<a name="What-is-CRAN%3f"></a>
<a name="What-is-CRAN_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Can-I-use-R-for-commercial-purposes_003f">Can I use R for commercial purposes?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#What-mailing-lists-exist-for-R_003f">What mailing lists exist for R?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Basics">R Basics</a>

</div>

<h3 class="section">2.10 What is <acronym>CRAN</acronym>?</h3>

<p>The &ldquo;Comprehensive R Archive Network&rdquo; (<acronym>CRAN</acronym>) is a collection of
sites which carry identical material, consisting of the R
distribution(s), the contributed extensions, documentation for R, and
binaries.

   <p>The <acronym>CRAN</acronym> master site at Wirtschaftsuniversit&auml;t Wien, Austria, can
be found at the <acronym>URL</acronym>

   <blockquote>
<!-- @multitable @columnfractions .45 .30 -->
<!-- @item -->
<a href="http://CRAN.R-project.org/">http://CRAN.R-project.org/</a>
<!-- @tab (Austria) -->
<!-- @end multitable -->
</blockquote>

<p class="noindent"><!-- (which is the same as @url{http://cran.at.R-project.org/}) -->
Daily mirrors are available at <acronym>URL</acronym>s including

   <blockquote>
        <p><table summary=""><tr align="left"><td valign="top" width="45%"><a href="http://cran.at.R-project.org/">http://cran.at.R-project.org/</a>
</td><td valign="top" width="40%">(WU Wien, Austria)
<br></td></tr><tr align="left"><td valign="top" width="45%"><a href="http://cran.au.R-project.org/">http://cran.au.R-project.org/</a>
</td><td valign="top" width="40%">(PlanetMirror, Australia)
<br></td></tr><tr align="left"><td valign="top" width="45%"><a href="http://cran.br.R-project.org/">http://cran.br.R-project.org/</a>
</td><td valign="top" width="40%">(Universidade Federal do Paran&aacute;, Brazil)
<br></td></tr><tr align="left"><td valign="top" width="45%"><a href="http://cran.ch.R-project.org/">http://cran.ch.R-project.org/</a>
</td><td valign="top" width="40%">(ETH Z&uuml;rich, Switzerland)
<br></td></tr><tr align="left"><td valign="top" width="45%"><a href="http://cran.dk.R-project.org/">http://cran.dk.R-project.org/</a>
</td><td valign="top" width="40%">(SunSITE, Denmark)
<br></td></tr><tr align="left"><td valign="top" width="45%"><a href="http://cran.es.R-project.org/">http://cran.es.R-project.org/</a>
</td><td valign="top" width="40%">(Spanish National Research Network, Madrid, Spain)
<br></td></tr><tr align="left"><td valign="top" width="45%"><a href="http://cran.fr.R-project.org/">http://cran.fr.R-project.org/</a>
</td><td valign="top" width="40%">(INRA, Toulouse, France)
<!-- @item @url{http://cran.hu.R-project.org/} -->
<!-- @tab (Semmelweis U, Hungary) -->
<br></td></tr><tr align="left"><td valign="top" width="45%"><a href="http://cran.pt.R-project.org/">http://cran.pt.R-project.org/</a>
</td><td valign="top" width="40%">(Universidade do Porto, Portugal)
<br></td></tr><tr align="left"><td valign="top" width="45%"><a href="http://cran.uk.R-project.org/">http://cran.uk.R-project.org/</a>
</td><td valign="top" width="40%">(U of Bristol, United Kingdom)
<!-- @item @url{http://cran.us.R-project.org/} -->
<!-- @tab (pair Networks, USA) -->
<br></td></tr><tr align="left"><td valign="top" width="45%"><a href="http://cran.za.R-project.org/">http://cran.za.R-project.org/</a>
</td><td valign="top" width="40%">(Rhodes U, South Africa)
        <br></td></tr></table>
</blockquote>

<p class="noindent">See <a href="http://CRAN.R-project.org/mirrors.html">http://CRAN.R-project.org/mirrors.html</a> for a complete list of
mirrors.  Please use the <acronym>CRAN</acronym> site closest to you to reduce network
load.

   <p>From <acronym>CRAN</acronym>, you can obtain the latest official release of R, daily
snapshots of R (copies of the current source trees), as gzipped and
bzipped tar files, a wealth of additional contributed code, as well as
prebuilt binaries for various operating systems (Linux, Mac OS Classic,
Mac OS X, and MS Windows).  <acronym>CRAN</acronym> also provides access to
documentation on R, existing mailing lists and the R Bug Tracking
system.

   <p>To &ldquo;submit&rdquo; to <acronym>CRAN</acronym>, simply upload to
<a href="ftp://CRAN.R-project.org/incoming/">ftp://CRAN.R-project.org/incoming/</a> and send an email to
<a href="mailto:CRAN@R-project.org">CRAN@R-project.org</a>.  Note that <acronym>CRAN</acronym> generally does not
accept submissions of precompiled binaries due to security reasons.  In
particular, binary packages for Windows and Mac OS X are provided by the
respective binary package maintainers.

   <blockquote>
<b>Note:</b> It is very important that you indicate the copyright (license)
information (<acronym>GPL-2</acronym>, <acronym>GPL-3</acronym>, <acronym>BSD</acronym>, Artistic,
<small class="dots">...</small>) in your submission. 
</blockquote>

   <p>Please always use the <acronym>URL</acronym> of the master site when referring to
<acronym>CRAN</acronym>.

<div class="node">
<p><hr>
<a name="Can-I-use-R-for-commercial-purposes%3f"></a>
<a name="Can-I-use-R-for-commercial-purposes_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Why-is-R-named-R_003f">Why is R named R?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#What-is-CRAN_003f">What is CRAN?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Basics">R Basics</a>

</div>

<h3 class="section">2.11 Can I use R for commercial purposes?</h3>

<p>R is released under the <a href="http://www.gnu.org/copyleft/gpl.html"><acronym>GNU</acronym> General Public License (GPL)</a>.  If you have any questions
regarding the legality of using R in any particular situation you should
bring it up with your legal counsel.  We are in no position to offer
legal advice.

   <p>It is the opinion of the R Core Team that one can use R for commercial
purposes (e.g., in business or in consulting).  The GPL, like all Open
Source licenses, permits all and any use of the package.  It only
restricts distribution of R or of other programs containing code from R. 
This is made clear in clause 6 (&ldquo;No Discrimination Against Fields of
Endeavor&rdquo;) of the <a href="http://www.opensource.org/docs/definition.html">Open Source Definition</a>:

   <blockquote>
The license must not restrict anyone from making use of the program in a
specific field of endeavor.  For example, it may not restrict the
program from being used in a business, or from being used for genetic
research. 
</blockquote>

<p class="noindent">It is also explicitly stated in clause 0 of the GPL, which says in part

   <blockquote>
Activities other than copying, distribution and modification are not
covered by this License; they are outside its scope.  The act of running
the Program is not restricted, and the output from the Program is
covered only if its contents constitute a work based on the Program. 
</blockquote>

   <p>Most add-on packages, including all recommended ones, also explicitly
allow commercial use in this way.  A few packages are restricted to
&ldquo;non-commercial use&rdquo;; you should contact the author to clarify whether
these may be used or seek the advice of your legal counsel.

   <p>None of the discussion in this section constitutes legal advice.  The R
Core Team does not provide legal advice under any circumstances.

<div class="node">
<p><hr>
<a name="Why-is-R-named-R%3f"></a>
<a name="Why-is-R-named-R_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#What-is-the-R-Foundation_003f">What is the R Foundation?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Can-I-use-R-for-commercial-purposes_003f">Can I use R for commercial purposes?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Basics">R Basics</a>

</div>

<h3 class="section">2.12 Why is R named R?</h3>

<p>The name is partly based on the (first) names of the first two R authors
(Robert Gentleman and Ross Ihaka), and partly a play on the name of the
Bell Labs language `S' (see <a href="#What-is-S_003f">What is S?</a>).

<!-- At the time the name was coined no one expected that the software would -->
<!-- get used outside of Auckland, so it seemed ok to make a joke of it. -->
<div class="node">
<p><hr>
<a name="What-is-the-R-Foundation%3f"></a>
<a name="What-is-the-R-Foundation_003f"></a>
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Why-is-R-named-R_003f">Why is R named R?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Basics">R Basics</a>

</div>

<h3 class="section">2.13 What is the R Foundation?</h3>

<p>The R Foundation is a not for profit organization working in the public
interest.  It was founded by the members of the R Core Team in order to
provide support for the R project and other innovations in statistical
computing, provide a reference point for individuals, institutions or
commercial enterprises that want to support or interact with the R
development community, and to hold and administer the copyright of R
software and documentation.  See
<a href="http://www.R-project.org/foundation/">http://www.R-project.org/foundation/</a> for more information.

<div class="node">
<p><hr>
<a name="R-and-S"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#R-Web-Interfaces">R Web Interfaces</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#R-Basics">R Basics</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Top">Top</a>

</div>

<h2 class="chapter">3 R and S</h2>

<ul class="menu">
<li><a accesskey="1" href="#What-is-S_003f">What is S?</a>
<li><a accesskey="2" href="#What-is-S_002dPLUS_003f">What is S-PLUS?</a>
<li><a accesskey="3" href="#What-are-the-differences-between-R-and-S_003f">What are the differences between R and S?</a>
<li><a accesskey="4" href="#Is-there-anything-R-can-do-that-S_002dPLUS-cannot_003f">Is there anything R can do that S-PLUS cannot?</a>
<li><a accesskey="5" href="#What-is-R_002dplus_003f">What is R-plus?</a>
</ul>

<div class="node">
<p><hr>
<a name="What-is-S%3f"></a>
<a name="What-is-S_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#What-is-S_002dPLUS_003f">What is S-PLUS?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#R-and-S">R and S</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-and-S">R and S</a>

</div>

<h3 class="section">3.1 What is S?</h3>

<p>S is a very high level language and an environment for data analysis and
graphics.  In 1998, the Association for Computing Machinery
(<acronym>ACM</acronym>) presented its Software System Award to John M. Chambers,
the principal designer of S, for

   <blockquote>
the S system, which has forever altered the way people analyze,
visualize, and manipulate data <small class="dots">...</small>

        <p>S is an elegant, widely accepted, and enduring software system, with
conceptual integrity, thanks to the insight, taste, and effort of John
Chambers. 
</blockquote>

   <p>The evolution of the S language is characterized by four books by John
Chambers and coauthors, which are also the primary references for S.

     <ul>
<li>Richard A. Becker and John M. Chambers (1984), &ldquo;S.  An Interactive
Environment for Data Analysis and Graphics,&rdquo; Monterey: Wadsworth and
Brooks/Cole.

     <p>This is also referred to as the &ldquo;<em>Brown Book</em>&rdquo;, and of historical
interest only.

     <li>Richard A. Becker, John M. Chambers and Allan R. Wilks (1988), &ldquo;The New
S Language,&rdquo; London: Chapman &amp; Hall.

     <p>This book is often called the &ldquo;<em>Blue Book</em>&rdquo;, and introduced what
is now known as S version 2.

     <li>John M. Chambers and Trevor J. Hastie (1992), &ldquo;Statistical Models in
S,&rdquo;  London: Chapman &amp; Hall.

     <p>This is also called the &ldquo;<em>White Book</em>&rdquo;, and introduced S version
3, which added structures to facilitate statistical modeling in S.

     <li>John M. Chambers (1998), &ldquo;Programming with Data,&rdquo; New York: Springer,
ISBN 0-387-98503-4
(<a href="http://cm.bell-labs.com/cm/ms/departments/sia/Sbook/">http://cm.bell-labs.com/cm/ms/departments/sia/Sbook/</a>).

     <p>This &ldquo;<em>Green Book</em>&rdquo; describes version 4 of S, a major revision of
S designed by John Chambers to improve its usefulness at every stage of
the programming process. 
</ul>

   <p>See <a href="http://cm.bell-labs.com/cm/ms/departments/sia/S/history.html">http://cm.bell-labs.com/cm/ms/departments/sia/S/history.html</a>
for further information on &ldquo;Stages in the Evolution of S&rdquo;.

   <p>There is a huge amount of user-contributed code for S, available at the
<a href="http://lib.stat.cmu.edu/S/">S Repository</a> at <acronym>CMU</acronym>.

<!-- The @url{http://lib.stat.cmu.edu/S/faq, ``Frequently Asked Questions -->
<!-- about S''} contains further information about S, but is not -->
<!-- up-to-date. -->
<div class="node">
<p><hr>
<a name="What-is-S-PLUS%3f"></a>
<a name="What-is-S_002dPLUS_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#What-are-the-differences-between-R-and-S_003f">What are the differences between R and S?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#What-is-S_003f">What is S?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-and-S">R and S</a>

</div>

<h3 class="section">3.2 What is <span class="sc">S-Plus</span>?</h3>

<p><span class="sc">S-Plus</span> is a value-added version of S sold by Insightful Corporation. 
Based on the S language, <span class="sc">S-Plus</span> provides functionality in a wide
variety of areas, including robust regression, modern non-parametric
regression, time series, survival analysis, multivariate analysis,
classical statistical tests, quality control, and graphics drivers. 
Add-on modules add additional capabilities.

   <p>See the <a href="http://www.insightful.com/products/splus/">Insightful <span class="sc">S-Plus</span> page</a> for further information.

<div class="node">
<p><hr>
<a name="What-are-the-differences-between-R-and-S%3f"></a>
<a name="What-are-the-differences-between-R-and-S_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Is-there-anything-R-can-do-that-S_002dPLUS-cannot_003f">Is there anything R can do that S-PLUS cannot?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#What-is-S_002dPLUS_003f">What is S-PLUS?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-and-S">R and S</a>

</div>

<h3 class="section">3.3 What are the differences between R and S?</h3>

<p>We can regard S as a language with three current implementations or
&ldquo;engines&rdquo;, the &ldquo;old S engine&rdquo; (S version 3; <span class="sc">S-Plus</span> 3.x and 4.x),
the &ldquo;new S engine&rdquo; (S version 4; <span class="sc">S-Plus</span> 5.x and above), and R. 
Given this understanding, asking for &ldquo;the differences between R and S&rdquo;
really amounts to asking for the specifics of the R implementation of
the S language, i.e., the difference between the R and S <em>engines</em>.

   <p>For the remainder of this section, &ldquo;S&rdquo; refers to the S engines and not
the S language.

<ul class="menu">
<li><a accesskey="1" href="#Lexical-scoping">Lexical scoping</a>
<li><a accesskey="2" href="#Models">Models</a>
<li><a accesskey="3" href="#Others">Others</a>
</ul>

<div class="node">
<p><hr>
<a name="Lexical-scoping"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Models">Models</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#What-are-the-differences-between-R-and-S_003f">What are the differences between R and S?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#What-are-the-differences-between-R-and-S_003f">What are the differences between R and S?</a>

</div>

<h4 class="subsection">3.3.1 Lexical scoping</h4>

<p>Contrary to other implementations of the S language, R has adopted an
evaluation model in which nested function definitions are lexically
scoped.  This is analogous to the evalutation model in Scheme.

   <p>This difference becomes manifest when <em>free</em> variables occur in a
function.  Free variables are those which are neither formal parameters
(occurring in the argument list of the function) nor local variables
(created by assigning to them in the body of the function).  In S, the
values of free variables are determined by a set of global variables
(similar to C, there is only local and global scope).  In R, they are
determined by the environment in which the function was created.

   <p>Consider the following function:

<pre class="example">     cube &lt;- function(n) {
       sq &lt;- function() n * n
       n * sq()
     }
</pre>
   <p>Under S, <code>sq()</code> does not &ldquo;know&rdquo; about the variable <code>n</code>
unless it is defined globally:

<pre class="example">     S&gt; cube(2)
     Error in sq():  Object "n" not found
     Dumped
     S&gt; n &lt;- 3
     S&gt; cube(2)
     [1] 18
</pre>
   <p>In R, the &ldquo;environment&rdquo; created when <code>cube()</code> was invoked is
also looked in:

<pre class="example">     R&gt; cube(2)
     [1] 8
</pre>
   <!-- The following more `realistic' example illustrating the differences in -->
<!-- scoping is due to @email{tlumley@@u.washington.edu, Thomas Lumley}. -->
<!-- The function -->
<!-- @example -->
<!-- jackknife.lm <- function(lmobj) @{ -->
<!-- n <- length(resid(lmobj)) -->
<!-- jval <- sapply(1:n, function(i) coef(update(lmobj, subset = -i))) -->
<!-- (n - 1) * (n - 1) * var(jval) / n -->
<!-- @} -->
<!-- @end example -->
<!-- @noindent -->
<!-- does something useful in R, but does not work in S.  In order to make it -->
<!-- work in S you need to explicitly pass the linear model object into the -->
<!-- function nested in @code{apply()}.  If you don't and you are lucky you -->
<!-- will get @samp{Error: Object "lmobj" not found}.  If you are unlucky -->
<!-- enough to have a linear model called @code{lmobj} in your global -->
<!-- environment you will get the wrong answer with no warning. -->
<!-- The following version works in S. -->
<!-- @example -->
<!-- jackknife.S.lm <- function(lmobj) @{ -->
<!-- n <- length(resid(lmobj)) -->
<!-- jval <- sapply(1:n, -->
<!-- function(i, lmobj) coef(update(lmobj, subset = -i)), -->
<!-- lmobj = lmobj) -->
<!-- (n - 1) * (n - 1) * var(jval) / n -->
<!-- @} -->
<!-- @end example -->
<!-- (The S version was written independently by Thomas and at least three of -->
<!-- his fellow students over the past couple of years, causing literally -->
<!-- hours of confusion on each occasion.) -->
<p>As a more &ldquo;interesting&rdquo; real-world problem, suppose you want to write
a function which returns the density function of the r-th order
statistic from a sample of size n from a (continuous)
distribution.  For simplicity, we shall use both the cdf and pdf of the
distribution as explicit arguments.  (Example compiled from various
postings by Luke Tierney.)

   <p>The <span class="sc">S-Plus</span> documentation for <code>call()</code> basically suggests the
following:

<pre class="example">     dorder &lt;- function(n, r, pfun, dfun) {
       f &lt;- function(x) NULL
       con &lt;- round(exp(lgamma(n + 1) - lgamma(r) - lgamma(n - r + 1)))
       PF &lt;- call(substitute(pfun), as.name("x"))
       DF &lt;- call(substitute(dfun), as.name("x"))
       f[[length(f)]] &lt;-
         call("*", con,
              call("*", call("^", PF, r - 1),
                   call("*", call("^", call("-", 1, PF), n - r),
                        DF)))
       f
     }
</pre>
   <p class="noindent">Rather tricky, isn't it?  The code uses the fact that in S,
functions are just lists of special mode with the function body as the
last argument, and hence does not work in R (one could make the idea
work, though).

   <p>A version which makes heavy use of <code>substitute()</code> and seems to work
under both S and R is

<pre class="example">     dorder &lt;- function(n, r, pfun, dfun) {
       con &lt;- round(exp(lgamma(n + 1) - lgamma(r) - lgamma(n - r + 1)))
       eval(substitute(function(x) K * PF(x)^a * (1 - PF(x))^b * DF(x),
                       list(PF = substitute(pfun), DF = substitute(dfun),
                            a = r - 1, b = n - r, K = con)))
     }
</pre>
   <p class="noindent">(the <code>eval()</code> is not needed in S).

   <p>However, in R there is a much easier solution:

<pre class="example">     dorder &lt;- function(n, r, pfun, dfun) {
       con &lt;- round(exp(lgamma(n + 1) - lgamma(r) - lgamma(n - r + 1)))
       function(x) {
         con * pfun(x)^(r - 1) * (1 - pfun(x))^(n - r) * dfun(x)
       }
     }
</pre>
   <p class="noindent">This seems to be the &ldquo;natural&rdquo; implementation, and it works because
the free variables in the returned function can be looked up in the
defining environment (this is lexical scope).

   <p>Note that what you really need is the function <em>closure</em>, i.e., the
body along with all variable bindings needed for evaluating it.  Since
in the above version, the free variables in the value function are not
modified, you can actually use it in S as well if you abstract out the
closure operation into a function <code>MC()</code> (for &ldquo;make closure&rdquo;):

<pre class="example">     dorder &lt;- function(n, r, pfun, dfun) {
       con &lt;- round(exp(lgamma(n + 1) - lgamma(r) - lgamma(n - r + 1)))
       MC(function(x) {
            con * pfun(x)^(r - 1) * (1 - pfun(x))^(n - r) * dfun(x)
          },
          list(con = con, pfun = pfun, dfun = dfun, r = r, n = n))
     }
</pre>
   <p>Given the appropriate definitions of the closure operator, this works in
both R and S, and is much &ldquo;cleaner&rdquo; than a substitute/eval solution
(or one which overrules the default scoping rules by using explicit
access to evaluation frames, as is of course possible in both R and S).

   <p>For R, <code>MC()</code> simply is

<pre class="example">     MC &lt;- function(f, env) f
</pre>
   <p class="noindent">(lexical scope!), a version for S is

<pre class="example">     MC &lt;- function(f, env = NULL) {
       env &lt;- as.list(env)
       if (mode(f) != "function")
         stop(paste("not a function:", f))
       if (length(env) &gt; 0 &amp;&amp; any(names(env) == ""))
         stop(paste("not all arguments are named:", env))
       fargs &lt;- if(length(f) &gt; 1) f[1:(length(f) - 1)] else NULL
       fargs &lt;- c(fargs, env)
       if (any(duplicated(names(fargs))))
         stop(paste("duplicated arguments:", paste(names(fargs)),
              collapse = ", "))
       fbody &lt;- f[length(f)]
       cf &lt;- c(fargs, fbody)
       mode(cf) &lt;- "function"
       return(cf)
     }
</pre>
   <p>Similarly, most optimization (or zero-finding) routines need some
arguments to be optimized over and have other parameters that depend on
the data but are fixed with respect to optimization.  With R scoping
rules, this is a trivial problem; simply make up the function with the
required definitions in the same environment and scoping takes care of
it.  With S, one solution is to add an extra parameter to the function
and to the optimizer to pass in these extras, which however can only
work if the optimizer supports this.

   <p>Nested lexically scoped functions allow using function closures and
maintaining local state.  A simple example (taken from Abelson and
Sussman) is obtained by typing <kbd>demo("scoping")</kbd> at the R prompt. 
Further information is provided in the standard R reference &ldquo;R: A
Language for Data Analysis and Graphics&rdquo; (see <a href="#What-documentation-exists-for-R_003f">What documentation exists for R?</a>) and in Robert Gentleman and Ross Ihaka (2000), &ldquo;Lexical
Scope and Statistical Computing&rdquo;,
<a href="http://www.amstat.org/publications/jcgs/"><em>Journal of Computational and Graphical Statistics</em></a>, <strong>9</strong>, 491&ndash;508.

   <p>Nested lexically scoped functions also imply a further major difference. 
Whereas S stores all objects as separate files in a directory somewhere
(usually <samp><span class="file">.Data</span></samp> under the current directory), R does not.  All
objects in R are stored internally.  When R is started up it grabs a
piece of memory and uses it to store the objects.  R performs its own
memory management of this piece of memory, growing and shrinking its
size as needed.  Having everything in memory is necessary because it is
not really possible to externally maintain all relevant &ldquo;environments&rdquo;
of symbol/value pairs.  This difference also seems to make R
<em>faster</em> than S.

   <p>The down side is that if R crashes you will lose all the work for the
current session.  Saving and restoring the memory &ldquo;images&rdquo; (the
functions and data stored in R's internal memory at any time) can be a
bit slow, especially if they are big.  In S this does not happen,
because everything is saved in disk files and if you crash nothing is
likely to happen to them.  (In fact, one might conjecture that the S
developers felt that the price of changing their approach to persistent
storage just to accommodate lexical scope was far too expensive.) 
Hence, when doing important work, you might consider saving often (see
<a href="#How-can-I-save-my-workspace_003f">How can I save my workspace?</a>) to safeguard against possible
crashes.  Other possibilities are logging your sessions, or have your R
commands stored in text files which can be read in using
<code>source()</code>.

   <blockquote>
<b>Note:</b> If you run R from within Emacs (see <a href="#R-and-Emacs">R and Emacs</a>), you can save the
contents of the interaction buffer to a file and conveniently manipulate
it using <code>ess-transcript-mode</code>, as well as save source copies of
all functions and data used. 
</blockquote>

<div class="node">
<p><hr>
<a name="Models"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Others">Others</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Lexical-scoping">Lexical scoping</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#What-are-the-differences-between-R-and-S_003f">What are the differences between R and S?</a>

</div>

<h4 class="subsection">3.3.2 Models</h4>

<p>There are some differences in the modeling code, such as

     <ul>
<li>Whereas in S, you would use <code>lm(y ~ x^3)</code> to regress <code>y</code> on
<code>x^3</code>, in R, you have to insulate powers of numeric vectors (using
<code>I()</code>), i.e., you have to use <code>lm(y ~ I(x^3))</code>. 
<li>The glm family objects are implemented differently in R and S.  The same
functionality is available but the components have different names. 
<li>Option <code>na.action</code> is set to <code>"na.omit"</code> by default in R,
but not set in S. 
<li>Terms objects are stored differently.  In S a terms object is an
expression with attributes, in R it is a formula with attributes.  The
attributes have the same names but are mostly stored differently. 
<li>Finally, in R <code>y ~ x + 0</code> is an alternative to <code>y ~ x - 1</code> for
specifying a model with no intercept.  Models with no parameters at all
can be specified by <code>y ~ 0</code>. 
</ul>

<div class="node">
<p><hr>
<a name="Others"></a>
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Models">Models</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#What-are-the-differences-between-R-and-S_003f">What are the differences between R and S?</a>

</div>

<h4 class="subsection">3.3.3 Others</h4>

<p>Apart from lexical scoping and its implications, R follows the S
language definition in the Blue and White Books as much as possible, and
hence really is an &ldquo;implementation&rdquo; of S.  There are some intentional
differences where the behavior of S is considered &ldquo;not clean&rdquo;.  In
general, the rationale is that R should help you detect programming
errors, while at the same time being as compatible as possible with S.

   <p>Some known differences are the following.

     <ul>
<li>In R, if <code>x</code> is a list, then <code>x[i] &lt;- NULL</code> and <code>x[[i]]
&lt;- NULL</code> remove the specified elements from <code>x</code>.  The first of
these is incompatible with S, where it is a no-op.  (Note that you can
set elements to <code>NULL</code> using <code>x[i] &lt;- list(NULL)</code>.)

     <!-- @item -->
<!-- In R @code{x[-4]} fails if @code{x} is not @code{NULL} but has fewer -->
<!-- than 4 elements.  In S it has no effect. -->
<li>In S, the functions named <code>.First</code> and <code>.Last</code> in the
<samp><span class="file">.Data</span></samp> directory can be used for customizing, as they are executed
at the very beginning and end of a session, respectively.

     <p>In R, the startup mechanism is as follows.  Unless <samp><span class="option">--no-environ</span></samp>
was given on the command line, R searches for site and user files to
process for setting environment variables.  Then, R searches for a
site-wide startup profile unless the command line option
<samp><span class="option">--no-site-file</span></samp> was given.  This code is loaded in package
<strong>base</strong>.  Then, unless <samp><span class="option">--no-init-file</span></samp> was given, R
searches for a user profile file, and sources it into the user
workspace.  It then loads a saved image of the user workspace from
<samp><span class="file">.RData</span></samp> in case there is one (unless <samp><span class="option">--no-restore-data</span></samp> or
<samp><span class="option">--no-restore</span></samp> were specified).  Next, a function <code>.First()</code>
is run if found on the search path.  Finally, function <code>.First.sys</code>
in the <strong>base</strong> package is run.  When terminating an R session, by
default a function <code>.Last</code> is run if found on the search path,
followed by <code>.Last.sys</code>.  If needed, the functions <code>.First()</code>
and <code>.Last()</code> should be defined in the appropriate startup
profiles.  See the help pages for <code>.First</code> and <code>.Last</code> for
more details.

     <li>In R, <code>T</code> and <code>F</code> are just variables being set to <code>TRUE</code>
and <code>FALSE</code>, respectively, but are not reserved words as in S and
hence can be overwritten by the user.  (This helps e.g. when you have
factors with levels <code>"T"</code> or <code>"F"</code>.)  Hence, when writing code
you should always use <code>TRUE</code> and <code>FALSE</code>.

     <li>In R, <code>dyn.load()</code> can only load <em>shared objects</em>, as created
for example by <kbd>R CMD SHLIB</kbd>.

     <li>In R, <code>attach()</code> currently only works for lists and data frames,
but not for directories.  (In fact, <code>attach()</code> also works for R
data files created with <code>save()</code>, which is analogous to attaching
directories in S.)  Also, you cannot attach at position 1.

     <li>Categories do not exist in R, and never will as they are deprecated now
in S.  Use factors instead.

     <li>In R, <code>For()</code> loops are not necessary and hence not supported.

     <li>In R, <code>assign()</code> uses the argument <samp><span class="option">envir=</span></samp> rather than
<samp><span class="option">where=</span></samp> as in S.

     <li>The random number generators are different, and the seeds have different
length.

     <li>R passes integer objects to C as <code>int *</code> rather than <code>long *</code>
as in S.

     <li>R has no single precision storage mode.  However, as of version 0.65.1,
there is a single precision interface to C/FORTRAN subroutines.

     <li>By default, <code>ls()</code> returns the names of the objects in the current
(under R) and global (under S) environment, respectively.  For example,
given

     <pre class="example">          x &lt;- 1; fun &lt;- function() {y &lt;- 1; ls()}
     </pre>
     <p class="noindent">then <code>fun()</code> returns <code>"y"</code> in R and <code>"x"</code> (together with
the rest of the global environment) in S.

     <li>R allows for zero-extent matrices (and arrays, i.e., some elements of
the <code>dim</code> attribute vector can be 0).  This has been determined a
useful feature as it helps reducing the need for special-case tests for
empty subsets.  For example, if <code>x</code> is a matrix, <code>x[, FALSE]</code>
is not <code>NULL</code> but a &ldquo;matrix&rdquo; with 0 columns.  Hence, such objects
need to be tested for by checking whether their <code>length()</code> is zero
(which works in both R and S), and not using <code>is.null()</code>.

     <li>Named vectors are considered vectors in R but not in S (e.g.,
<code>is.vector(c(a = 1:3))</code> returns <code>FALSE</code> in S and <code>TRUE</code>
in R).

     <li>Data frames are not considered as matrices in R (i.e., if <code>DF</code> is a
data frame, then <code>is.matrix(DF)</code> returns <code>FALSE</code> in R and
<code>TRUE</code> in S).

     <li>R by default uses treatment contrasts in the unordered case, whereas S
uses the Helmert ones.  This is a deliberate difference reflecting the
opinion that treatment contrasts are more natural.

     <li>In R, the argument of a replacement function which corresponds to the
right hand side must be named `<samp><span class="samp">value</span></samp>'.  E.g., <code>f(a) &lt;- b</code> is
evaluated as <code>a &lt;- "f&lt;-"(a, value = b)</code>.  S always takes the last
argument, irrespective of its name.

     <li>In S, <code>substitute()</code> searches for names for substitution in the
given expression in three places: the actual and the default arguments
of the matching call, and the local frame (in that order).  R looks in
the local frame only, with the special rule to use a &ldquo;promise&rdquo; if a
variable is not evaluated.  Since the local frame is initialized with
the actual arguments or the default expressions, this is usually
equivalent to S, until assignment takes place.

     <li>In S, the index variable in a <code>for()</code> loop is local to the inside
of the loop.  In R it is local to the environment where the <code>for()</code>
statement is executed.

     <li>In S, <code>tapply(simplify=TRUE)</code> returns a vector where R returns a
one-dimensional array (which can have named dimnames).

     <li>In S(-<span class="sc">Plus</span>) the C locale is used, whereas in R the current
operating system locale is used for determining which characters are
alphanumeric and how they are sorted.  This affects the set of valid
names for R objects (for example accented chars may be allowed in R) and
ordering in sorts and comparisons (such as whether <code>"aA" &lt; "Bb"</code> is
true or false).  From version 1.2.0 the locale can be (re-)set in R by
the <code>Sys.setlocale()</code> function.

     <li>In S, <code>missing(</code><var>arg</var><code>)</code> remains <code>TRUE</code> if <var>arg</var> is
subsequently modified; in R it doesn't.

     <li>From R version 1.3.0, <code>data.frame</code> strips <code>I()</code> when creating
(column) names.

     <li>In R, the string <code>"NA"</code> is not treated as a missing value in a
character variable.  Use <code>as.character(NA)</code> to create a missing
character value.

     <li>R disallows repeated formal arguments in function calls.

     <li>In S, <code>dump()</code>, <code>dput()</code> and <code>deparse()</code> are essentially
different interfaces to the same code.  In R from version 2.0.0, this is
only true if the same <code>control</code> argument is used, but by default it
is not.  By default <code>dump()</code> tries to write code that will evaluate
to reproduce the object, whereas <code>dput()</code> and <code>deparse()</code>
default to options for producing deparsed code that is readable.

     <li>In R, indexing a vector, matrix, array or data frame with <code>[</code> using
a character vector index looks only for exact matches (whereas <code>[[</code>
and <code>$</code> allow partial matches).  In S, <code>[</code> allows partial
matches.

     <li>S has a two-argument version of <code>atan</code> and no <code>atan2</code>.  A call
in S such as <code>atan(x1, x2)</code> is equivalent to R's <code>atan2(x1,
x2)</code>.  However, beware of named arguments since S's <code>atan(x = a, y
= b)</code> is equivalent to R's <code>atan2(y = a, x = b)</code> with the meanings
of <code>x</code> and <code>y</code> interchanged.  (R used to have undocumented
support for a two-argument <code>atan</code> with positional arguments, but
this has been withdrawn to avoid further confusion.)

     <li>Numeric constants with no fractional and exponent (i.e., only integer)
part are taken as integer in <span class="sc">S-Plus</span> 6.x or later, but as double in R.

   </ul>

   <p>There are also differences which are not intentional, and result from
missing or incorrect code in R.  The developers would appreciate hearing
about any deficiencies you may find (in a written report fully
documenting the difference as you see it).  Of course, it would be
useful if you were to implement the change yourself and make sure it
works.

<div class="node">
<p><hr>
<a name="Is-there-anything-R-can-do-that-S-PLUS-cannot%3f"></a>
<a name="Is-there-anything-R-can-do-that-S_002dPLUS-cannot_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#What-is-R_002dplus_003f">What is R-plus?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#What-are-the-differences-between-R-and-S_003f">What are the differences between R and S?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-and-S">R and S</a>

</div>

<h3 class="section">3.4 Is there anything R can do that <span class="sc">S-Plus</span> cannot?</h3>

<p>Since almost anything you can do in R has source code that you could
port to <span class="sc">S-Plus</span> with little effort there will never be much you can do
in R that you couldn't do in <span class="sc">S-Plus</span> if you wanted to.  (Note that
using lexical scoping may simplify matters considerably, though.)

   <p>R offers several graphics features that <span class="sc">S-Plus</span> does not, such as finer
handling of line types, more convenient color handling (via palettes),
gamma correction for color, and, most importantly, mathematical
annotation in plot texts, via input expressions reminiscent of TeX
constructs.  See the help page for <code>plotmath</code>, which features an
impressive on-line example.  More details can be found in Paul Murrell
and Ross Ihaka (2000), &ldquo;An Approach to Providing Mathematical
Annotation in Plots&rdquo;, <a href="http://www.amstat.org/publications/jcgs/"><em>Journal of Computational and Graphical Statistics</em></a>, <strong>9</strong>,
582&ndash;599.

<div class="node">
<p><hr>
<a name="What-is-R-plus%3f"></a>
<a name="What-is-R_002dplus_003f"></a>
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Is-there-anything-R-can-do-that-S_002dPLUS-cannot_003f">Is there anything R can do that S-PLUS cannot?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-and-S">R and S</a>

</div>

<h3 class="section">3.5 What is R-plus?</h3>

<p>For a very long time, there was no such thing.

   <p><a href="http://www.xlsolutions-corp.com/">XLSolutions Corporation</a> is
currently beta testing a commercially supported version of R named R+
(read R plus).

   <p>In addition, <a href="http://www.revolution-computing.com/">REvolution Computing</a> has released
<a href="http://www.revolution-computing.com/rpro.html">RPro</a>, an
enterprise-class statistical analysis system based on R, suitable for
deployment in professional, commercial and regulated environments.

   <p><a href="http://www.random-technologies-llc.com/">Random Technologies</a>
offers <a href="http://random-technologies-llc.com/products/RStat/rstat">RStat</a>, an enterprise-strength statistical computing environment which
combines R with enterprise-level validation, documentation, software
support, and consulting services, as well as related R-based products.

<div class="node">
<p><hr>
<a name="R-Web-Interfaces"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#R-Add_002dOn-Packages">R Add-On Packages</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#R-and-S">R and S</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Top">Top</a>

</div>

<h2 class="chapter">4 R Web Interfaces</h2>

<p><strong>Rweb</strong> is developed and maintained by
<a href="mailto:jeff@math.montana.edu">Jeff Banfield</a>.  The
<a href="http://www.math.montana.edu/Rweb/">Rweb Home Page</a> provides access
to all three versions of Rweb&mdash;a simple text entry form that returns
output and graphs, a more sophisticated Javascript version that provides
a multiple window environment, and a set of point and click modules that
are useful for introductory statistics courses and require no knowledge
of the R language.  All of the Rweb versions can analyze Web accessible
datasets if a <acronym>URL</acronym> is provided.

   <p>The paper &ldquo;Rweb: Web-based Statistical Analysis&rdquo;, providing a detailed
explanation of the different versions of Rweb and an overview of how
Rweb works, was published in the Journal of Statistical Software
(<a href="http://www.jstatsoft.org/v04/i01/">http://www.jstatsoft.org/v04/i01/</a>).

   <p><a href="mailto:ulfi@cs.tu-berlin.de">Ulf Bartel</a> has developed
<strong>R-Online</strong>, a simple on-line programming environment for R which
intends to make the first steps in statistical programming with R
(especially with time series) as easy as possible.  There is no need for
a local installation since the only requirement for the user is a
JavaScript capable browser.  See <a href="http://osvisions.com/r-online/">http://osvisions.com/r-online/</a>
for more information.

   <p><strong>Rcgi</strong> is a CGI WWW interface to R by <a href="mailto:mjr@dsl.pipex.com">MJ Ray</a>.  It had the ability to use &ldquo;embedded code&rdquo;: you could mix
user input and code, allowing the <acronym>HTML</acronym> author to do anything from
load in data sets to enter most of the commands for users without
writing CGI scripts.  Graphical output was possible in PostScript or GIF
formats and the executed code was presented to the user for revision. 
However, it is not clear if the project is still active. 
<!-- URL no longer seems to work ... -->
<!-- See @url{http://stats.mth.uea.ac.uk/Rcgi/} for more information. -->
Currently, a modified version of <strong>Rcgi</strong> by
<a href="mailto:mai@ms.uky.edu">Mai Zhou</a> (actually, two versions: one with
(bitmap) graphics and one without) as well as the original code are
available from <a href="http://www.ms.uky.edu/~statweb/">http://www.ms.uky.edu/~statweb/</a>.

   <p>CGI-based web access to R is also provided at
<a href="http://hermes.sdu.dk/cgi-bin/go/">http://hermes.sdu.dk/cgi-bin/go/</a>.  There are many additional
examples of web interfaces to R which basically allow to submit R code
to a remote server, see for example the collection of links available from
<a href="http://biostat.mc.vanderbilt.edu/twiki/bin/view/Main/StatCompCourse">http://biostat.mc.vanderbilt.edu/twiki/bin/view/Main/StatCompCourse</a>.

   <p><a href="http://www.warwick.ac.uk/go/dfirth">David Firth</a> has written
<strong>CGIwithR</strong>, an R add-on package available from <acronym>CRAN</acronym>.  It
provides some simple extensions to R to facilitate running R scripts
through the CGI interface to a web server, and allows submission of data
using both GET and POST methods.  It is easily installed using Apache
under Linux and in principle should run on any platform that supports R
and a web server provided that the installer has the necessary security
permissions.  David's paper &ldquo;CGIwithR: Facilities for Processing Web
Forms Using R&rdquo; was published in the Journal of Statistical Software
(<a href="http://www.jstatsoft.org/v08/i10/">http://www.jstatsoft.org/v08/i10/</a>).  The package is now
maintained by <a href="mailto:duncan@wald.ucdavis.edu">Duncan Temple Lang</a> and
has a web page at <a href="http://www.omegahat.org/CGIwithR/">http://www.omegahat.org/CGIwithR/</a>.

   <p><a href="http://www.rpad.org/Rpad">Rpad</a>, developed and actively maintained
by Tom Short, provides a sophisticated environment which combines some
of the features of the previous approaches with quite a bit of
Javascript, allowing for a <acronym>GUI</acronym>-like behavior (with sortable
tables, clickable graphics, editable output), etc.

   <p>Jeff Horner is working on the R/Apache Integration Project which embeds
the R interpreter inside Apache 2 (and beyond).  A tutorial and
presentation are available from the project web page at
<a href="http://biostat.mc.vanderbilt.edu/twiki/bin/view/Main/RApacheProject">http://biostat.mc.vanderbilt.edu/twiki/bin/view/Main/RApacheProject</a>.

   <p><a href="http://stats.math.uni-augsburg.de/Rserve/">Rserve</a> is a project
actively developed by Simon Urbanek.  It implements a TCP/IP server which
allows other programs to use facilities of R.  Clients are available from
the web site for Java and C++ (and could be written for other languages
that support TCP/IP sockets).

   <p><a href="http://openstatserver.org/index.html">OpenStatServer</a> is being
developed by a team lead by Greg Warnes; it aims &ldquo;to provide clean access
to computational modules defined in a variety of computational
environments (R, SAS, Matlab, etc) via a single well-defined client
interface&rdquo; and to turn computational services into web services.

   <p>Two projects use PHP to provide a web interface to R. 
<a href="http://steve-chen.net/R_PHP/">R_PHP_Online</a> by Steve Chen (though
it is unclear if this project is still active) is somewhat similar to
the above Rcgi and Rweb.  <a href="http://dssm.unipa.it/R-php/?cmd=home">R-php</a> is actively developed by Alfredo Pontillo and Angelo Mineo and
provides both a web interface to R and a set of pre-specified analyses
that need no R code input.

   <p><a href="http://www.bioconductor.org/">webbioc</a> is &ldquo;an integrated web
interface for doing microarray analysis using several of the
Bioconductor packages&rdquo; and is designed to be installed at local sites
as a shared computing resource.

   <p>Finally, <a href="http://rwui.cryst.bbk.ac.uk">Rwui</a> is a web application to
to create user-friendly web interfaces for R scripts.  All code for the
web interface is created automatically.  There is no need for the user
to do any extra scripting or learn any new scripting techniques.

<div class="node">
<p><hr>
<a name="R-Add-On-Packages"></a>
<a name="R-Add_002dOn-Packages"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#R-and-Emacs">R and Emacs</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#R-Web-Interfaces">R Web Interfaces</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Top">Top</a>

</div>

<h2 class="chapter">5 R Add-On Packages</h2>

<ul class="menu">
<li><a accesskey="1" href="#Which-add_002don-packages-exist-for-R_003f">Which add-on packages exist for R?</a>
<li><a accesskey="2" href="#How-can-add_002don-packages-be-installed_003f">How can add-on packages be installed?</a>
<li><a accesskey="3" href="#How-can-add_002don-packages-be-used_003f">How can add-on packages be used?</a>
<li><a accesskey="4" href="#How-can-add_002don-packages-be-removed_003f">How can add-on packages be removed?</a>
<li><a accesskey="5" href="#How-can-I-create-an-R-package_003f">How can I create an R package?</a>
<li><a accesskey="6" href="#How-can-I-contribute-to-R_003f">How can I contribute to R?</a>
</ul>

<div class="node">
<p><hr>
<a name="Which-add-on-packages-exist-for-R%3f"></a>
<a name="Which-add_002don-packages-exist-for-R_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-add_002don-packages-be-installed_003f">How can add-on packages be installed?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#R-Add_002dOn-Packages">R Add-On Packages</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Add_002dOn-Packages">R Add-On Packages</a>

</div>

<h3 class="section">5.1 Which add-on packages exist for R?</h3>

<ul class="menu">
<li><a accesskey="1" href="#Add_002don-packages-in-R">Add-on packages in R</a>
<li><a accesskey="2" href="#Add_002don-packages-from-CRAN">Add-on packages from CRAN</a>
<li><a accesskey="3" href="#Add_002don-packages-from-Omegahat">Add-on packages from Omegahat</a>
<li><a accesskey="4" href="#Add_002don-packages-from-Bioconductor">Add-on packages from Bioconductor</a>
<li><a accesskey="5" href="#Other-add_002don-packages">Other add-on packages</a>
</ul>

<div class="node">
<p><hr>
<a name="Add-on-packages-in-R"></a>
<a name="Add_002don-packages-in-R"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Add_002don-packages-from-CRAN">Add-on packages from CRAN</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Which-add_002don-packages-exist-for-R_003f">Which add-on packages exist for R?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Which-add_002don-packages-exist-for-R_003f">Which add-on packages exist for R?</a>

</div>

<h4 class="subsection">5.1.1 Add-on packages in R</h4>

<p>The R distribution comes with the following packages:

     <dl>
<!-- <FIXME> -->
<!-- 3.0.0 -->
<dt><strong>base</strong><dd>Base R functions (and datasets before R 2.0.0). 
<br><dt><strong>datasets</strong><dd>Base R datasets (added in R 2.0.0). 
<br><dt><strong>grDevices</strong><dd>Graphics devices for base and grid graphics (added in R 2.0.0). 
<!-- </FIXME> -->
<br><dt><strong>graphics</strong><dd>R functions for base graphics. 
<br><dt><strong>grid</strong><dd>A rewrite of the graphics layout capabilities, plus some support for
interaction. 
<br><dt><strong>methods</strong><dd>Formally defined methods and classes for R objects, plus other
programming tools, as described in the Green Book. 
<br><dt><strong>splines</strong><dd>Regression spline functions and classes. 
<br><dt><strong>stats</strong><dd>R statistical functions. 
<br><dt><strong>stats4</strong><dd>Statistical functions using S4 classes. 
<br><dt><strong>tcltk</strong><dd>Interface and language bindings to Tcl/Tk <acronym>GUI</acronym> elements. 
<br><dt><strong>tools</strong><dd>Tools for package development and administration. 
<br><dt><strong>utils</strong><dd>R utility functions. 
</dl>
   These &ldquo;base packages&rdquo; were substantially reorganized in R 1.9.0.  The
former <strong>base</strong> was split into the four packages <strong>base</strong>,
<strong>graphics</strong>, <strong>stats</strong>, and <strong>utils</strong>.  Packages
<strong>ctest</strong>, <strong>eda</strong>, <strong>modreg</strong>, <strong>mva</strong>,
<strong>nls</strong>, <strong>stepfun</strong> and <strong>ts</strong> were merged into
<strong>stats</strong>, package <strong>lqs</strong> returned to the recommended package
<strong>MASS</strong>, and package <strong>mle</strong> moved to <strong>stats4</strong>.

<div class="node">
<p><hr>
<a name="Add-on-packages-from-CRAN"></a>
<a name="Add_002don-packages-from-CRAN"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Add_002don-packages-from-Omegahat">Add-on packages from Omegahat</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Add_002don-packages-in-R">Add-on packages in R</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Which-add_002don-packages-exist-for-R_003f">Which add-on packages exist for R?</a>

</div>

<h4 class="subsection">5.1.2 Add-on packages from <acronym>CRAN</acronym></h4>

<p>The following packages are available from the <acronym>CRAN</acronym> <samp><span class="file">src/contrib</span></samp>
area.  (Packages denoted as <em>Recommended</em> are to be included in all
binary distributions of R.)

     <dl>
<dt><strong>ADaCGH</strong><dd>Analysis of data from aCGH experiments. 
<br><dt><strong>AER</strong><dd>Functions, data sets, examples and vignettes for the book &ldquo;Applied
Econometrics with R&rdquo; by Christian Kleiber and Achim Zeileis, 2008,
Springer-Verlag, New York. 
<br><dt><strong>AIS</strong><dd>Tools to look at the data (&ldquo;Ad Inidicia Spectata&rdquo;). 
<br><dt><strong>ALS</strong><dd>Multivariate curve resolution alternating least squares (MCR-ALS). 
<br><dt><strong>AMORE</strong><dd>A MORE flexible neural network package, providing the TAO robust neural
network algorithm. 
<br><dt><strong>ARES</strong><dd>Allelic richness estimation, with extrapolation beyond the sample size. 
<br><dt><strong>AcceptanceSampling</strong><dd>Creation and evaluation of acceptance sampling plans,
<br><dt><strong>AdMit</strong><dd>Adaptive mixture of Student t distributions. 
<br><dt><strong>AdaptFit</strong><dd>Adaptive semiparametic regression. 
<br><dt><strong>AlgDesign</strong><dd>Algorithmic experimental designs.  Calculates exact and approximate
theory experimental designs for D, A, and I criteria. 
<br><dt><strong>Amelia</strong><dd>Amelia II: a program for missing data. 
<br><dt><strong>AnalyzeFMRI</strong><dd>Functions for I/O, visualisation and analysis of functional Magnetic
Resonance Imaging (fMRI) datasets stored in the ANALYZE format. 
<br><dt><strong>ArDec</strong><dd>Time series autoregressive decomposition. 
<br><dt><strong>BACCO</strong><dd>Bayesian Analysis of Computer Code Output.  Contains
<strong>approximator</strong>, <strong>calibrator</strong>, and <strong>emulator</strong>, for
Bayesian prediction of complex computer codes, calibration of computer
models, and emulation of computer programs, respectively. 
<br><dt><strong>BARD</strong><dd>Better Automated ReDistricting. 
<br><dt><strong>BAYSTAR</strong><dd>Bayesian analysis of threshold autoregressive models. 
<br><dt><strong>BB</strong><dd>Barzilai-Borwein spectral methods for solving nonlinear system of
equations, and for optimizing nonlinear objective functions subject to
simple constraints. 
<br><dt><strong>BCE</strong><dd>Bayesian Composition Estimator for sample (taxonomic) composition from
biomarker data. 
<br><dt><strong>BHH2</strong><dd>Functions and data sets reproducing some examples in &ldquo;Statistics for
Experimenters II&rdquo; by G. E. P. Box, J. S. Hunter, and W. C. Hunter,
2005, John Wiley and Sons. 
<br><dt><strong>BLCOP</strong><dd>Black-Litterman and copula-opinion pooling frameworks. 
<br><dt><strong>BMA</strong><dd>Bayesian Model Averaging for linear models, generalizable linear models
and survival models (Cox regression). 
<br><dt><strong>BPHO</strong><dd>Bayesian Prediction with High-order Interactions. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2008-05-19 as requested by the maintainer: -->
<!-- Now lives in CRANextras ... -->
<!-- @item BRugs -->
<!-- OpenBUGS and its R interface BRugs. -->
<!-- </COMMENT> -->
<br><dt><strong>BaM</strong><dd>Functions and datasets for &ldquo;Bayesian Methods: A Social and Behavioral
Sciences Approach&rdquo; (2nd edition) by Jeff Gill, 2007, CRC Press. 
<br><dt><strong>BayHaz</strong><dd>Functions for Bayesian Hazard rate estimation. 
<br><dt><strong>BayesTree</strong><dd>Bayesian methods for tree based models. 
<br><dt><strong>BayesValidate</strong><dd>Bayesian software validation using posterior quantiles. 
<br><dt><strong>Bchron</strong><dd>Create chronologies based on radiocarbon and non-radiocarbon dated
depths. 
<br><dt><strong>Bhat</strong><dd>Functions for general likelihood exploration (MLE, MCMC, CIs). 
<br><dt><strong>BiasedUrn</strong><dd>Biased urn model distributions. 
<br><dt><strong>BioIDMapper</strong><dd>Mapping between BioIDs. 
<br><dt><strong>Biodem</strong><dd>A number of functions for biodemographycal analysis. 
<br><dt><strong>BiodiversityR</strong><dd>GUI for biodiversity and community ecology analysis. 
<br><dt><strong>BiplotGUI</strong><dd>Interactive biplots in R. 
<br><dt><strong>Bolstad</strong><dd>Functions and data sets for the book &ldquo;Introduction to Bayesian
Statistics&rdquo; by W. M. Bolstad, 2004, John Wiley and Sons. 
<br><dt><strong>BootCL</strong><dd>Bootstrapping test for chromosomal localization. 
<br><dt><strong>BootPR</strong><dd>Bootstrap prediction intervals and bias-corrected forecasting. 
<br><dt><strong>BradleyTerry</strong><dd>Specify and fit the Bradley-Terry model and structured versions. 
<br><dt><strong>Brobdingnag</strong><dd>Very large numbers in R. 
<br><dt><strong>BSDA</strong><dd>Data sets for the book &ldquo;Basic Statistics and Data Analysis&rdquo; by
L. J. Kitchens, 2003, Duxbury. 
<br><dt><strong>BsMD</strong><dd>Bayes screening and model discrimination follow-up designs. 
<br><dt><strong>CADStat</strong><dd>A <acronym>GUI</acronym> to several statistical methods for biological
inferences. 
<br><dt><strong>CCA</strong><dd>Canonical correlation analysis. 
<br><dt><strong>CDNmoney</strong><dd>Components of Canadian monetary aggregates. 
<br><dt><strong>CGIwithR</strong><dd>Facilities for the use of R to write CGI scripts. 
<br><dt><strong>CHsharp</strong><dd>Choi and Hall clustering in 3d. 
<br><dt><strong>CORREP</strong><dd>Multivariate correlation estimation. 
<br><dt><strong>COZIGAM</strong><dd>Constrained Zero-Inflated Generalized Additive Model. 
<br><dt><strong>CPE</strong><dd>Concordance probability estimates in survival analysis. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2008-02-21 as requested by the maintainer. -->
<!-- Replaced by Bchron. -->
<!-- @item CPGchron -->
<!-- Create radiocarbon-dated depth chronologies. -->
<!-- </COMMENT> -->
<br><dt><strong>CTFS</strong><dd>The CTFS large plot forest dynamics analyses. 
<br><dt><strong>CVThresh</strong><dd>Level-dependent Cross-Validation Thresholding. 
<br><dt><strong>Cairo</strong><dd>Graphics device using cairographics library for creating high-quality
PNG, PDF, SVG, PostScript output and interactive display devices such as
X11. 
<br><dt><strong>CarbonEL</strong><dd>Carbon Event Loop. 
<br><dt><strong>CellularAutomaton</strong><dd>One-dimensional cellular automata. 
<br><dt><strong>ChainLadder</strong><dd>Mack- and Munich-chain-ladder methods for insurance claims reserving. 
<br><dt><strong>CircStats</strong><dd>Circular Statistics, from &ldquo;Topics in Circular Statistics&rdquo; by S. Rao
Jammalamadaka and A. SenGupta, 2001, World Scientific. 
<br><dt><strong>CoCo</strong><dd>Graphical modeling for contingency tables using CoCo. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2005-05-08 as requested by -->
<!-- Stephane Dray <dray@biomserv.univ-lyon1.fr>. -->
<!-- @item CoCoAn -->
<!-- Constrained Correspondence Analysis. -->
<!-- </COMMENT> -->
<br><dt><strong>ComPairWise</strong><dd>Compare phylogenetic or population genetic data alignments. 
<br><dt><strong>CombMSC</strong><dd>Combined Model Selection Criteria. 
<br><dt><strong>CompetingRiskFrailty</strong><dd>Competing risk model with frailties for right censored survival data. 
<br><dt><strong>Containers</strong><dd>Object-oriented data structures including stack, queue, and binary
search tree. 
<br><dt><strong>ConvergenceConcepts</strong><dd>Seeing convergence concepts in action. 
<br><dt><strong>CoxBoost</strong><dd>Cox survival models by likelihood based boosting. 
<br><dt><strong>CreditMetrics</strong><dd>Functions for calculating the CreditMetrics risk model. 
<br><dt><strong>CvM2SL2Test</strong><dd>Cramer-von Mises two sample tests. 
<br><dt><strong>DAAG</strong><dd>Various data sets used in examples and exercises in &ldquo;Data Analysis and
Graphics Using R&rdquo; by John H. Maindonald and W. John Brown, 2003. 
<br><dt><strong>DAAGbio</strong><dd>Data sets and functions, for demonstrations with expression arrays. 
<br><dt><strong>DAAGxtras</strong><dd>Data sets and functions additional to <strong>DAAG</strong>. 
<br><dt><strong>DBI</strong><dd>A common database interface (DBI) class and method definitions.  All
classes in this package are virtual and need to be extended by the
various DBMS implementations. 
<br><dt><strong>DCluster</strong><dd>A set of functions for the detection of spatial clusters of diseases
using count data. 
<br><dt><strong>DDHFm</strong><dd>Variance stabilization by Data-Driven Haar-Fisz (for microarrays). 
<br><dt><strong>DEA</strong><dd>Data Envelopment Analysis. 
<br><dt><strong>DEoptim</strong><dd>Differential Evolution Optimization. 
<br><dt><strong>DICOM</strong><dd>Import and manipulate medical imaging data using the Digital Imaging and
Communications in Medicine (DICOM) Standard. 
<br><dt><strong>DPpackage</strong><dd>Semiparametric Bayesian analysis using Dirichlet process priors. 
<br><dt><strong>Davies</strong><dd>Functions for the Davies quantile function and the Generalized Lambda
distribution. 
<br><dt><strong>Defaults</strong><dd>Create global function defaults. 
<br><dt><strong>DescribeDisplay</strong><dd>R interface to the DescribeDisplay GGobi plugin. 
<br><dt><strong>Design</strong><dd>Regression modeling, testing, estimation, validation, graphics,
prediction, and typesetting by storing enhanced model design attributes
in the fit.  Design is a collection of about 180 functions that assist
and streamline modeling, especially for biostatistical and epidemiologic
applications.  It also contains new functions for binary and ordinal
logistic regression models and the Buckley-James multiple regression
model for right-censored responses, and implements penalized maximum
likelihood estimation for logistic and ordinary linear models.  Design
works with almost any regression model, but it was especially written to
work with logistic regression, Cox regression, accelerated failure time
models, ordinary linear models, and the Buckley-James model. 
<br><dt><strong>Devore5</strong><dd>Data sets and sample analyses from &ldquo;Probability and Statistics for
Engineering and the Sciences (5th ed)&rdquo; by Jay L. Devore, 2000, Duxbury. 
<br><dt><strong>Devore6</strong><dd>Data sets and sample analyses from &ldquo;Probability and Statistics for
Engineering and the Sciences (6th ed)&rdquo; by Jay L. Devore, 2003, Duxbury. 
<br><dt><strong>Devore7</strong><dd>Data sets and sample analyses from &ldquo;Probability and Statistics for
Engineering and the Sciences (7th ed)&rdquo; by Jay L. Devore, 2008, Thomson. 
<br><dt><strong>DiagnosisMed</strong><dd>Diagnostic test accuracy evaluation for medical professionals. 
<br><dt><strong>DierckxSpline</strong><dd>R companion to &ldquo;Curve and Surface Fitting with Splines&rdquo; by Paul
Dierckx, 1993, Oxford University Press. 
<br><dt><strong>EDR</strong><dd>Estimation of the effective dimension reduction (EDR) space. 
<br><dt><strong>EMC</strong><dd>Evolutionary Monte Carlo (EMC) algorithm. 
<br><dt><strong>EMCC</strong><dd>Evolutionary Monte Carlo (EMC) methods for clustering. 
<br><dt><strong>EMD</strong><dd>Empirical mode decomposition and Hilbert spectral analysis. 
<br><dt><strong>EMV</strong><dd>Estimation of missing values in a matrix by a k-th nearest
neighboors algorithm. 
<br><dt><strong>ETC</strong><dd>Tests and simultaneous confidence intervals for equivalence to control. 
<br><dt><strong>EbayesThresh</strong><dd>Empirical Bayes thresholding and related methods. 
<br><dt><strong>Ecdat</strong><dd>Data sets from econometrics textbooks. 
<br><dt><strong>EffectiveDose</strong><dd>Estimate the effective dose level for quantal bioassay data by
nonparametric techniques. 
<br><dt><strong>ElemStatLearn</strong><dd>Data sets, functions and examples from the book &ldquo;The Elements of
Statistical Learning: Data Mining, Inference, and Prediction&rdquo; by Trevor
Hastie, Robert Tibshirani and Jerome Friedman (2001), Springer. 
<br><dt><strong>Epi</strong><dd>Statistical analysis in epidemiology, with functions for demographic and
epidemiological analysis in the Lexis diagram. 
<br><dt><strong>FAiR</strong><dd>Factor Analysis in R, using genetic algorithms. 
<br><dt><strong>FGN</strong><dd>Fractional Gaussian Noise model fitting. 
<br><dt><strong>FITSio</strong><dd>FITS (Flexible Image Transport System) utilities. 
<br><dt><strong>FKBL</strong><dd>Fuzzy Knowledge Base Learning. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2007-11-17 as it fails R CMD check. -->
<!-- @item FLCore -->
<!-- Core package of FLR, fisheries modeling in R. -->
<!-- </COMMENT> -->
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2007-11-17 as it depends on FLCore. -->
<!-- @item FLEDA -->
<!-- Exploratory Data Analysis for FLR. -->
<!-- </COMMENT> -->
<br><dt><strong>FTICRMS</strong><dd>Analysis of Fourier Transform-Ion Cyclotron Resonance Mass Spectrometry
data. 
<br><dt><strong>FactoMineR</strong><dd>Factor analysis and data mining with R. 
<br><dt><strong>Fahrmeir</strong><dd>Data from the book &ldquo;Multivariate Statistical Modelling Based
on Generalized Linear Models&rdquo; by Ludwig Fahrmeir and Gerhard Tutz
(1994),  Springer. 
<br><dt><strong>FieldSim</strong><dd>Random fields simulations. 
<br><dt><strong>FinTS</strong><dd>Companion to the book &ldquo;Analysis of Financial Time Series&rdquo; (2nd
edition) by Ruey Tsay (2005), Wiley. 
<br><dt><strong>FitAR</strong><dd>Subset AR model fitting. 
<br><dt><strong>Flury</strong><dd>Data sets from from &ldquo;A First Course in Multivariate Statistics&rdquo; by
Bernard Flury (1997), Springer. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2007-11-17 as it fails R CMD check. -->
<!-- @item FortranCallsR -->
<!-- Simple Fortran/C/R interface example. -->
<!-- </COMMENT> -->
<br><dt><strong>FrF2</strong><dd>Analysis of fractional factorial designs with 2-level factors. 
<br><dt><strong>FracSim</strong><dd>Simulation of one- and two-dimensional fractional and multifractional
Levy motions. 
<br><dt><strong>FunCluster</strong><dd>Functional profiling of cDNA microarray expression data. 
<br><dt><strong>FunNet</strong><dd>Functional analysis of gene co-expression networks. 
<br><dt><strong>G1DBN</strong><dd>Dynamic Bayesian Network inference using 1st order conditional
dependencies. 
<br><dt><strong>GAMBoost</strong><dd>Generalized additive models by likelihood based boosting. 
<br><dt><strong>GDD</strong><dd>Platform and X11 independent device for creating bitmaps (png, gif and
jpeg) using the GD graphics library. 
<br><dt><strong>GEOmap</strong><dd>Topographic and geologic mapping. 
<br><dt><strong>GLDEX</strong><dd>Fit RS and FMKL generalised lambda distributions using discretized and
maximum likelihood methods. 
<br><dt><strong>GOSim</strong><dd>Computation of functional similarities between GO terms and gene
products. 
<br><dt><strong>GPArotation</strong><dd>Gradient Projection Algorithm rotation for factor analysis. 
<br><dt><strong>GRASS</strong><dd>An interface between the GRASS geographical information system and R,
based on starting R from within the GRASS environment and chosen
LOCATION_NAME and MAPSET.  Wrapper and helper functions are provided for
a range of R functions to match the interface metadata structures. 
<br><dt><strong>GSA</strong><dd>Gene set analysis. 
<br><dt><strong>GSM</strong><dd>Gamma Shape Mixture. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2007-11-17 as it fails R CMD check. -->
<!-- @item GammaTest -->
<!-- Gamma Test data analysis. -->
<!-- </COMMENT> -->
<br><dt><strong>GenABEL</strong><dd>Genome-wide SNP association analysis. 
<br><dt><strong>GenKern</strong><dd>Functions for generating and manipulating generalised binned kernel
density estimates. 
<br><dt><strong>GeneCycle</strong><dd>Identification of periodically expressed genes. 
<br><dt><strong>GeneF</strong><dd>Generalized F-statistics. 
<br><dt><strong>GeneNT</strong><dd>Relevance or Dependency network and signaling pathway discovery. 
<br><dt><strong>GeneNet</strong><dd>Modeling and inferring gene networks. 
<br><dt><strong>GeneTS</strong><dd>A package for analysing multiple gene expression time series data. 
Currently, implements methods for cell cycle analysis and for inferring
large sparse graphical Gaussian models. 
<br><dt><strong>Geneland</strong><dd>MCMC inference from individual genetic data based on a spatial
statistical model. 
<br><dt><strong>GeoXp</strong><dd>Interactive exploratory spatial data analysis. 
<br><dt><strong>GillespieSSA</strong><dd>Gillespie's Stochastic Simulation Algorithm (SSA). 
<br><dt><strong>GroupSeq</strong><dd>Computations related to group-seqential boundaries. 
<br><dt><strong>HAPim</strong><dd>Methods for QTL detection and fine mapping. 
<br><dt><strong>HH</strong><dd>Support software for &ldquo;Statistical Analysis and Data Display&rdquo; by
Richard M. Heiberger and Burt Holland, Springer, 2005. 
<br><dt><strong>HI</strong><dd>Simulation from distributions supported by nested hyperplanes. 
<br><dt><strong>HSAUR</strong><dd>Functions, data sets, analyses and examples from the book &ldquo;A Handbook
of Statistical Analyses Using R&rdquo; by Brian S. Everitt and Torsten
Hothorn (2006), Chapman &amp; Hall/CRC. 
<br><dt><strong>HTMLapplets</strong><dd>Functions inserting dynamic scatterplots and grids in documents
generated by <strong>R2HTML</strong>. 
<br><dt><strong>HFWutils</strong><dd>Utilities by H. Felix Wittmann: Excel connections, string matching, and
passing by reference. 
<br><dt><strong>HardyWeinberg</strong><dd>Graphical tests for Hardy-Weinberg equilibrium. 
<br><dt><strong>HiddenMarkov</strong><dd>Hidden Markov Models. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2008-05-09 (now contained in 'empiricalBayes' -->
<!-- bundle. -->
<!-- @item HighProbability -->
<!-- Estimation of the alternative hypotheses having frequentist or Bayesian -->
<!-- probabilities at least as great as a specified threshold, given a list -->
<!-- of @math{p}-values. -->
<!-- </COMMENT> -->
<br><dt><strong>Hmisc</strong><dd>Functions useful for data analysis, high-level graphics, utility
operations, functions for computing sample size and power, importing
datasets, imputing missing values, advanced table making, variable
clustering, character string manipulation, conversion of S objects to
LaTeX code, recoding variables, and bootstrap repeated measures
analysis. 
<br><dt><strong>HydroMe</strong><dd>Estimation of soil hydraulic parameters from experimental data. 
<br><dt><strong>HyperbolicDist</strong><dd>Basic functions for the hyperbolic distribution: probability density
function, distribution function, quantile function, a routine for
generating observations from the hyperbolic, and a function for fitting
the hyperbolic distribution to data. 
<br><dt><strong>IBrokers</strong><dd>R API to Interactive Brokers Trader Workstation. 
<br><dt><strong>ICE</strong><dd>Iterated Conditional Expectation: kernel estimators for
interval-censored data. 
<br><dt><strong>ICEinfer</strong><dd>Incremental Cost-Effectiveness (ICE) statistical inference (from
two unbiased samples). 
<br><dt><strong>ICS</strong><dd>ICS/ICA computation based on two scatter matrices. 
<br><dt><strong>ICSNP</strong><dd>Tools for multivariate nonparametrics. 
<br><dt><strong>IDPmisc</strong><dd>Utilities from the Institute of Data Analyses and Process Design,
IDP/ZHW. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2008-02-21 as requested by the maintainer. -->
<!-- Replaced by RcmdrPlugin.IPSUR. -->
<!-- @item IPSUR -->
<!-- Accompanies ``Introduction to Probability and Statistics Using R'' by -->
<!-- G. Andy Chang and G. Jay Kerns (in progress). -->
<!-- </COMMENT> -->
<br><dt><strong>ISA</strong><dd>Functions to support &ldquo;Introduzione alla Statistica Applicata con esempi
in R&rdquo; by Federico M. Stefanini, Pearson Education Milano, 2007. 
<br><dt><strong>ISOcodes</strong><dd>ISO language, territory, currency, script and character codes. 
<br><dt><strong>ISwR</strong><dd>Data sets for &ldquo;Introductory Statistics with R&rdquo; by Peter Dalgaard,
2002, Springer. 
<br><dt><strong>Icens</strong><dd>Functions for computing the NPMLE for censored and truncated data. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2007-11-17 as it fails R CMD check. -->
<!-- @item InfNet -->
<!-- Function that simulates an epidemic in a network of contacts. -->
<!-- </COMMENT> -->
<br><dt><strong>Iso</strong><dd>Functions to perform isotonic regression. 
<br><dt><strong>JADE</strong><dd>JADE and ICA performance criteria. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2008-04-22 (Orphaned/2.6.0). -->
<!-- @item JLLprod -->
<!-- Nonparametric estimation of homothetic and generalized homothetic -->
<!-- production functions. -->
<!-- </COMMENT> -->
<br><dt><strong>JGR</strong><dd>Java Gui for R. 
<br><dt><strong>JM</strong><dd>Joint Modeling of longitudinal and survival data. 
<br><dt><strong>JavaGD</strong><dd>Java Graphics Device. 
<br><dt><strong>JointGLM</strong><dd>Joint modeling of mean and dispersion through two interlinked GLM's. 
<em>Defunct</em> in favor of <strong>JointModeling</strong>. 
<br><dt><strong>JointModeling</strong><dd>Joint modeling of mean and dispersion. 
<br><dt><strong>JudgeIt</strong><dd>Calculates bias, responsiveness, and other characteristics of two-party
electoral systems, with district-level electoral and other data. 
<br><dt><strong>KMsurv</strong><dd>Data sets and functions for &ldquo;Survival Analysis, Techniques for Censored
and Truncated Data&rdquo; by Klein and Moeschberger, 1997, Springer. 
<br><dt><strong>Kendall</strong><dd>Kendall rank correlation and Mann-Kendall trend test. 
<br><dt><strong>KernSmooth</strong><dd>Functions for kernel smoothing (and density estimation) corresponding to
the book &ldquo;Kernel Smoothing&rdquo; by M. P. Wand and M. C. Jones, 1995. 
<em>Recommended</em>. 
<br><dt><strong>LDheatmap</strong><dd>Heat maps of linkage disequilibrium measures. 
<br><dt><strong>LDtests</strong><dd>Exact tests for linkage disequilibrium and Hardy-Weinberg equilibrium. 
<br><dt><strong>LIStest</strong><dd>Longest increasing subsequence independence test. 
<br><dt><strong>LLAhclust</strong><dd>Hierarchical clustering of variables or objects based on the likelihood
linkage analysis method. 
<br><dt><strong>LLN</strong><dd>Learning with latent networks. 
<br><dt><strong>LMGene</strong><dd>Date transformation and identification of differentially expressed genes
in gene expression arrays. 
<br><dt><strong>LearnBayes</strong><dd>Functions for Learning Bayesian Inference. 
<br><dt><strong>Lmoments</strong><dd>Estimation of L-moments and the parameters of normal and Cauchy
polynomial quantile mixtures. 
<br><dt><strong>LogConcDEAD</strong><dd>Maximum likelihood estimation of a log-concave density. 
<br><dt><strong>LogicReg</strong><dd>Routines for Logic Regression. 
<br><dt><strong>LoopAnalyst</strong><dd>A collection of tools to conduct Levins' Loop Analysis. 
<br><dt><strong>LowRankQP</strong><dd>Low Rank Quadratic Programming: QP problems where the hessian is
represented as the product of two matrices. 
<br><dt><strong>MASS</strong><dd>Functions and datasets from the main package of Venables and Ripley,
&ldquo;Modern Applied Statistics with S&rdquo;.  Contained in the <samp><span class="file">VR</span></samp>
bundle.  <em>Recommended</em>. 
<br><dt><strong>MBA</strong><dd>Multilevel B-spline Approximation. 
<br><dt><strong>MBESS</strong><dd>Methods for the Behavioral, Educational, and Social Sciences. 
<br><dt><strong>MCAPS</strong><dd>Weather and air pollution data, risk estimates, and other information
from the Medicare Air Pollution Study (MCAPS) of 204 U.S. counties,
1999&ndash;2002. 
<br><dt><strong>MCMCpack</strong><dd>Markov chain Monte Carlo (MCMC) package: functions for posterior
simulation for a number of statistical models. 
<br><dt><strong>MCPAN</strong><dd>Multiple comparisons using normal approximation. 
<br><dt><strong>MCPMod</strong><dd>Design and analysis of dose-finding studies. 
<br><dt><strong>MChtest</strong><dd>Monte Carlo hypothesis tests. 
<br><dt><strong>MDD</strong><dd>Minimum Detectable Difference calculations for several continuous and
binary endpoints. 
<br><dt><strong>MEMSS</strong><dd>Data sets and sample analyses from &ldquo;Mixed-effects Models in S and
S-PLUS&rdquo; by J. Pinheiro and D. Bates, 2000, Springer. 
<br><dt><strong>MFDA</strong><dd>Model Based Functional Data Analysis. 
<br><dt><strong>MKLE</strong><dd>Maximum kernel likelihood estimation. 
<br><dt><strong>MLDA</strong><dd>Methylation Linear Discriminant Analysis (MLDA). 
<br><dt><strong>MLDS</strong><dd>Maximum Likelihood Difference Scaling. 
<br><dt><strong>MLEcens</strong><dd>Computation of the MLE for bivariate (interval) censored data. 
<br><dt><strong>MMG</strong><dd>Mixture Model on Graphs. 
<br><dt><strong>MNP</strong><dd>Fitting Bayesian Multinomial Probit models via Markov chain Monte Carlo. 
Along with the standard Multinomial Probit model, it can also fit models
with different choice sets for each observation and complete or partial
ordering of all the available alternatives. 
<br><dt><strong>MPV</strong><dd>Data sets from the book &ldquo;Introduction to Linear Regression Analysis&rdquo;
by D. C. Montgomery, E. A. Peck, and C. G. Vining, 2001, John Wiley and
Sons. 
<br><dt><strong>MSBVAR</strong><dd>Bayesian vector autoregression models, impulse responses and
forecasting. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2006-04-20 as requested by the maintainer. -->
<!-- @item Malmig -->
<!-- An implementation of Malecot migration model together with a number of -->
<!-- related functions. -->
<!-- </COMMENT> -->
<br><dt><strong>MarkedPointProcess</strong><dd>Non-parametric analysis of the marks of marked point processes. 
<br><dt><strong>MasterBayes</strong><dd>Maximum likelihood and Markov chain Monte Carlo methods for pedigree
reconstruction, analysis and simulation. 
<br><dt><strong>MatchIt</strong><dd>Select matched samples of the original treated and control groups with
similar covariate distributions. 
<br><dt><strong>Matching</strong><dd>Multivariate and propensity score matching with formal tests of balance. 
<br><dt><strong>Matrix</strong><dd>A Matrix package. 
<br><dt><strong>Metabonomic</strong><dd><acronym>GUI</acronym> for metabonomic analysis. 
<br><dt><strong>MiscPsycho</strong><dd>Miscellaneous Psychometrics. 
<br><dt><strong>MultEq</strong><dd>Equivalence tests and simultaneous confidence intervals for multiple
endpoints. 
<br><dt><strong>NADA</strong><dd>Methods described in &ldquo;Nondetects And Data Analysis: Statistics for
Censored Environmental Data&rdquo; by Dennis R. Helsel, 2004, John Wiley and
Sons. 
<br><dt><strong>NISTnls</strong><dd>A set of test nonlinear least squares examples from <acronym>NIST</acronym>, the
U.S. National Institute for Standards and Technology. 
<br><dt><strong>NMMAPSlite</strong><dd>U.S. National Morbidity, Mortality, and Air Pollution Study data lite. 
<br><dt><strong>NORMT3</strong><dd>Evaluates complex erf, erfc and density of sum of Gaussian and Student's
t. 
<br><dt><strong>NRAIA</strong><dd>Data sets with sample code from &ldquo;Nonlinear Regression Analysis and Its
Applications&rdquo; by Doug Bates and Donald Watts, 1988, Wiley. 
<br><dt><strong>NestedCohort</strong><dd>Survival analysis for cohorts with missing covariate information. 
<br><dt><strong>NetIndices</strong><dd>Estimates network indices, including trophic structure of foodwebs. 
<br><dt><strong>ORMDR</strong><dd>Odds ratio based multivactor-dimensionality reduction method for
detecting gene-gene interactions. 
<br><dt><strong>Oarray</strong><dd>Arrays with arbitrary offsets. 
<br><dt><strong>PARccs</strong><dd>Estimation of partial attributable risks (PAR) from case-control data. 
<br><dt><strong>PASWR</strong><dd>Data and functions for the book &ldquo;Probability and Statistics with R&rdquo; by
M. D. Ugarte, A. F. Militino and A. T. Arnholt, 2008, Chapman &amp; Hall/CRC. 
<br><dt><strong>PBSmapping</strong><dd>Software evolved from fisheries research conducted at the Pacific
Biological Station (PBS) in Nanaimo, British Columbia, Canada.  Draws
maps and implements other GIS procedures. 
<br><dt><strong>PBSmodelling</strong><dd>Software to facilitate the design, testing, and operation of computer
models. 
<br><dt><strong>PET</strong><dd>Simulation and reconstruction of PET images. 
<br><dt><strong>PHYLOGR</strong><dd>Manipulation and analysis of phylogenetically simulated data sets (as
obtained from PDSIMUL in package PDAP) and phylogenetically-based
analyses using GLS. 
<br><dt><strong>PK</strong><dd>Estimation of pharmacokinetic parameters. 
<br><dt><strong>PKfit</strong><dd>A nonlinear regression (including a genetic algorithm) program designed
to deal with curve fitting for pharmacokinetics. 
<br><dt><dd>Unified computational interfaces for pop PK. 
<br><dt><strong>POT</strong><dd>Generalized Pareto distribution and Peaks Over Threshold. 
<br><dt><strong>PSAgraphics</strong><dd>Propensity Score Analysis Graphics. 
<br><dt><strong>PSM</strong><dd>Non-linear mixed-effects modeling using stochastic differential
equations. 
<br><dt><strong>PTAk</strong><dd>A multiway method to decompose a tensor (array) of any order, as a
generalisation of SVD also supporting non-identity metrics and
penalisations.  Also includes some other multiway methods. 
<br><dt><strong>Peaks</strong><dd>Spectrum manipulation: background estimation, Markov smoothing,
deconvolution and peaks search functions. 
<br><dt><strong>PearsonICA</strong><dd>Independent component analysis using score functions from the Pearson
system. 
<br><dt><strong>PerformanceAnalytics</strong><dd>Econometric tools for performance and risk analysis. 
<br><dt><strong>PhySim</strong><dd>Phylogenetic tree simulation. 
<br><dt><strong>PolynomF</strong><dd>Univariate polynomials. 
<br><dt><strong>Pomic</strong><dd>Pattern oriented modeling information criterion. 
<br><dt><strong>PredictiveRegression</strong><dd>Prediction intervals for three basic statistical models. 
<br><dt><strong>PresenceAbsence</strong><dd>Presence-absence model evaluation. 
<br><dt><strong>ProbForecastGOP</strong><dd>Probabilistic weather field forecasts using the Geostatistical Output
Perturbation method introduced by Gel, Raftery and Gneiting (2004). 
<br><dt><strong>ProbeR</strong><dd>Reliability for gene expression from Affymetrix chip. 
<br><dt><strong>ProfessR</strong><dd>Programs to determine student grades and create examinations from
question banks. 
<br><dt><strong>PtProcess</strong><dd>Time dependent point process modeling. 
<br><dt><strong>PwrGSD</strong><dd>Power in a Group Sequential Design. 
<br><dt><strong>QCA</strong><dd>Qualitative Comparative Analysis for crisp sets. 
<br><dt><strong>QCAGUI</strong><dd>QCA Graphical User Interface. 
<br><dt><strong>QRMlib</strong><dd>Code to examine Quantitative Risk Management concepts. 
<br><dt><strong>QuantPsyc</strong><dd>Quantitative Psychology tools. 
<br><dt><strong>R.cache</strong><dd>Fast and light-weight caching of objects. 
<br><dt><strong>R.huge</strong><dd>Methods for accessing huge amounts of data. 
<br><dt><strong>R.matlab</strong><dd>Read and write of MAT files together with R-to-Matlab connectivity. 
<br><dt><strong>R.methodsS3</strong><dd>Utility functions for defining S3 methods. 
<br><dt><strong>R.oo</strong><dd>R object-oriented programming with or without references. 
<br><dt><strong>R.rsp</strong><dd>R server pages. 
<br><dt><strong>R.utils</strong><dd>Utility classes and methods useful when programming in R and developing
R packages. 
<br><dt><strong>R2HTML</strong><dd>Functions for exporting R objects &amp; graphics in an <acronym>HTML</acronym> document. 
<br><dt><strong>R2WinBUGS</strong><dd>Running WinBUGS from R: call a BUGS model, summarize inferences and
convergence in a table and graph, and save the simulations in arrays for
easy access in R. 
<br><dt><strong>RArcInfo</strong><dd>Functions to import Arc/Info V7.x coverages and data. 
<br><dt><strong>RBGL</strong><dd>Interface to the boost C++ graph library. 
<br><dt><strong>RBloomberg</strong><dd>Fetch data from a Bloomberg API using COM. 
<br><dt><strong>RColorBrewer</strong><dd>ColorBrewer palettes for drawing nice maps shaded according to a
variable. 
<br><dt><strong>RDieHarder</strong><dd>R interface to the dieharder random number generator test suite. 
<br><dt><strong>RExcelInstaller</strong><dd>Integration of R and Excel under MS Windows. 
<br><dt><strong>RFA</strong><dd>Regional Frequency Analysis. 
<br><dt><strong>RFOC</strong><dd>Graphics for spherical distributions and earthquake focal mechanisms. 
<br><dt><strong>RFreak</strong><dd>An R interface to a modified version of the Free Evolutionary Algorithm
Kit FrEAK.g
<br><dt><strong>RGtk2</strong><dd>Facilities for programming graphical interfaces using Gtk (the Gimp Tool
Kit) version 2. 
<!-- <COMMENT> -->
<!-- Moved to Archive on 2007-05-29. -->
<!-- @item RGtk2.10 -->
<!-- R bindings for Gtk 2.10.x. -->
<!-- </COMMENT> -->
<br><dt><strong>RGrace</strong><dd>Mouse/menu driven interactive plotting application. 
<br><dt><strong>RGraphics</strong><dd>Data and functions from the book &ldquo;R Graphics&rdquo; by Paul Murrell, 2005,
Chapman &amp; Hall/CRC. 
<br><dt><strong>RHmm</strong><dd>Hidden Markov Model simulations and estimations. 
<br><dt><strong>RII</strong><dd>Estimation of the relative index of inequality for interval-censored
data using natural cubic splines. 
<br><dt><strong>RJDBC</strong><dd>Access to databases through the JDBC interface. 
<br><dt><strong>RJaCGH</strong><dd>Reversible Jump MCMC for the analysis of CGH arrays. 
<br><dt><strong>RKEA</strong><dd>R/KEA interface for extracting keyphrases from text documents. 
<br><dt><strong>RLMM</strong><dd>A genotype calling algorithm for Affymetrix SNP arrays. 
<br><dt><strong>RLRsim</strong><dd>Exact (Restricted) Likelihood Ratio tests for mixed and additive
models. 
<br><dt><strong>RLadyBug</strong><dd>Analysis of infectious diseases using stochastic epidemic models. 
<br><dt><strong>RM2</strong><dd>Revenue management and pricing. 
<br><dt><strong>RMySQL</strong><dd>An interface between R and the MySQL database system. 
<br><dt><strong>RNetCDF</strong><dd>An interface to Unidata's NetCDF library functions (version 3) and
furthermore access to Unidata's udunits calendar conversions. 
<br><dt><strong>ROCR</strong><dd>Visualizing the performance of scoring classifiers. 
<br><dt><strong>RODBC</strong><dd>An <acronym>ODBC</acronym> database interface. 
<br><dt><strong>ROptEst</strong><dd>Optimally robust estimation. 
<br><dt><strong>ROptEstTS</strong><dd>Optimally robust estimation for regression-type models. 
<br><dt><strong>ROracle</strong><dd>Oracle Database Interface driver for R.  Uses the ProC/C++ embedded SQL. 
<br><dt><strong>RPMG</strong><dd>Poor Man's Gui: create interactive R analysis sessions. 
<br><dt><strong>RPyGeo</strong><dd>ArcGIS Geoprocessing in R via Python. 
<br><dt><strong>RQuantLib</strong><dd>Provides access to (some) of the QuantLib functions from within R;
currently limited to some Option pricing and analysis functions.  The
QuantLib project aims to provide a comprehensive software framework for
quantitative finance. 
<br><dt><strong>RSAGA</strong><dd>SAGA geoprocessing and terrain analysis in R. 
<br><dt><strong>RSEIS</strong><dd>Seismic time series analysis tools. 
<br><dt><strong>RSQLite</strong><dd>Database Interface R driver for SQLite.  Embeds the SQLite database
engine in R. 
<br><dt><strong>RScaLAPACK</strong><dd>An interface to ScaLAPACK functions from R. 
<br><dt><strong>RSVGTipsDevice</strong><dd>An R <acronym>SVG</acronym> graphics device with dynamic tips and hyperlinks. 
<br><dt><strong>RSeqMeth</strong><dd>analysis of Sequenom EpiTYPER data. 
<br><dt><strong>RSvgDevice</strong><dd>A graphics device for R that uses the new w3.org <acronym>XML</acronym> standard for
Scalable Vector Graphics. 
<br><dt><strong>RTOMO</strong><dd>Visualization for seismic tomography. 
<br><dt><strong>RTisean</strong><dd>R interface to Tisean algorithms. 
<br><dt><strong>RUnit</strong><dd>Functions implementing a standard Unit Testing framework, with
additional code inspection and report generation tools. 
<br><dt><strong>RWeka</strong><dd>An R interface to Weka, a rich collection of machine learning algorithms
for data mining tasks. 
<br><dt><strong>RWinEdt</strong><dd>A plug in for using WinEdt as an editor for R. 
<br><dt><strong>RXshrink</strong><dd>Maximum Likelihood Shrinkage via Ridge or Least Angle Regression. 
<br><dt><strong>RadioSonde</strong><dd>A collection of programs for reading and plotting SKEW-T,log p diagrams
and wind profiles for data collected by radiosondes (the typical weather
balloon-borne instrument). 
<br><dt><strong>RandVar</strong><dd>Implementation of random variables by means of S4 classes and methods. 
<br><dt><strong>RandomFields</strong><dd>Creating random fields using various methods. 
<br><dt><strong>RankAggreg</strong><dd>Weighted rank aggregation. 
<br><dt><strong>RaschSampler</strong><dd>Sampling binary matrices with fixed margins. 
<br><dt><strong>Rcapture</strong><dd>Loglinear models in capture-recapture experiments. 
<br><dt><strong>Rcmdr</strong><dd>A platform-independent basic-statistics <acronym>GUI</acronym> (graphical user
interface) for R, based on the <strong>tcltk</strong> package. 
<br><dt><strong>RcmdrPlugin.Export</strong><dd>Graphically export objects to LaTeX or <acronym>HTML</acronym>. 
<br><dt><strong>RcmdrPlugin.FactoMineR</strong><dd>Rcmdr plug-in for the <strong>FactoMineR</strong> package. 
<br><dt><strong>RcmdrPlugin.HH</strong><dd>Rcmdr support for the introductory course at Temple University. 
<br><dt><strong>RcmdrPlugin.IPSUR</strong><dd>Rcmdr plugin for &ldquo;Introduction to Probability and Statistics Using R&rdquo;. 
<br><dt><strong>RcmdrPlugin.TeachingDemos</strong><dd>Rcmdr Teaching Demos plug-in. 
<br><dt><strong>RcmdrPlugin.epack</strong><dd>Rcmdr epack demos plug-in. 
<br><dt><strong>Rcplex</strong><dd>R interface to CPLEX solvers for linear, quadratic, and (linear and
quadratic) mixed integer programs. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2007-09-25 as requested by the maintainer. -->
<!-- @item RcppTemplate -->
<!-- An illustration of the use of the Rcpp R/C++ interface library. -->
<!-- </COMMENT> -->
<br><dt><strong>Read.isi</strong><dd>Access old data saved in fixed-width format based on ISI-formatted
codebooks. 
<br><dt><strong>Reliability</strong><dd>Functions for estimating parameters in software reliability models. 
<br><dt><strong>ResearchMethods</strong><dd>Using GUIs to help teach statistics to non-statistics students. 
<br><dt><strong>ResistorArray</strong><dd>Electrical properties of resistor networks. 
<br><dt><strong>Rfwdmv</strong><dd>Forward Search for Multivariate Data. 
<br><dt><strong>Rglpk</strong><dd>R/GNU Linear Programming Kit interface. 
<br><dt><strong>RiboSort</strong><dd>Classification and analysis of microbial community profiles. 
<br><dt><strong>Rigroup</strong><dd>Provides small integer group functions. 
<br><dt><strong>Rlab</strong><dd>Functions and data sets for the NCSU ST370 class. 
<br><dt><strong>Rlsf</strong><dd>Interface to the LSF queuing system. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2004-08-11. -->
<!-- @item RmSQL -->
<!-- An interface between R and the mSQL database system. -->
<!-- </COMMENT> -->
<br><dt><strong>Rmdr</strong><dd>R-Multifactor Dimensionality Reduction. 
<br><dt><strong>Rmetrics</strong><dd>Financial engineering and computational finance. 
<br><dt><strong>Rmpi</strong><dd>An interface (wrapper) to MPI (Message-Passing Interface) APIs.  It also
provides an interactive R slave environment in which distributed
statistical computing can be carried out. 
<!-- @item Rnotes -->
<!-- The data sets for the exercises in ``An Introduction to R'' -->
<!-- (@pxref{What documentation exists for R?}). -->
<br><dt><strong>RobLox</strong><dd>Optimally robust influence curves for location and scale. 
<br><dt><strong>RobRex</strong><dd>Optimally robust influence curves for regression and scale. 
<br><dt><strong>Rpad</strong><dd>Utility functions for the Rpad workbook-style interface. 
<br><dt><strong>Rsac</strong><dd>Seismic tools for R. 
<br><dt><strong>Rserve</strong><dd>A socket server (TCP/IP or local sockets) which allows binary requests
to be sent to R. 
<!-- REMOVED 2001-12-08 -->
<!-- @item Rstreams -->
<!-- Binary file stream support functions. -->
<br><dt><strong>Rsundials</strong><dd>SUite of Nonlinear DIfferential ALgebraic equations Solvers in R. 
<br><dt><strong>Rsymphony</strong><dd>An R interface to the SYMPHONY mixed integer linear program (MILP)
solver. 
<br><dt><strong>Runuran</strong><dd>Interface to the UNU.RAN library for Universal Non-Uniform RANdom
variate generators. 
<br><dt><strong>Rvelslant</strong><dd>Downhole seismic analysis. 
<br><dt><strong>Rwave</strong><dd>An environment for the time-frequency analysis of 1-D signals (and
especially for the wavelet and Gabor transforms of noisy signals), based
on the book &ldquo;Practical Time-Frequency Analysis: Gabor and Wavelet
Transforms with an Implementation in S&rdquo; by Rene Carmona, Wen L. Hwang
and Bruno Torresani, 1998, Academic Press. 
<br><dt><strong>Ryacas</strong><dd>An R interfaces to the yacas computer algebra system. 
<!-- <COMMENT> -->
<!-- Removed on 2006-10-06 as requested by the maintainer (now @BioC). -->
<!-- @item SAGx -->
<!-- Retrieval, preparation and analysis of data from the Affymetrix -->
<!-- GeneChip. -->
<!-- </COMMENT> -->
<br><dt><strong>SASPECT</strong><dd>Significant AnalysiS of PEptide CounTs. 
<br><dt><strong>SASmixed</strong><dd>Data sets and sample linear mixed effects analyses corresponding to the
examples in &ldquo;SAS System for Mixed Models&rdquo; by R. C. Littell,
G. A. Milliken, W. W. Stroup and R. D. Wolfinger, 1996, SAS Institute. 
<br><dt><strong>SASxport</strong><dd>Read and write SAS XPORT files. 
<br><dt><strong>SDDA</strong><dd>Stepwise Diagonal Discriminant Analysis. 
<br><dt><strong>SIN</strong><dd>A SINful approach to selection of Gaussian Graphical Markov Models. 
<br><dt><strong>SLmisc</strong><dd>Miscellaneous Functions for analysis of gene expression data at SIRS-Lab
GmbH. 
<br><dt><strong>SMC</strong><dd>Sequential Monte Carlo (SMC) Algorithm. 
<br><dt><strong>SMPracticals</strong><dd>Data sets and a few functions for use with the practicals outlined in
Appendix A of the book &ldquo;Statistical Models&rdquo; by Anthony Davison, 2003,
Cambridge University Press. 
<br><dt><strong>SNPMaP</strong><dd>SNP Microarrays and Pooling in R. 
<br><dt><strong>SNPMaP.cdm</strong><dd>Annotation for SNP microarrays and pooling in R. 
<br><dt><strong>SNPassoc</strong><dd>SNP-based whole genome association studies. 
<br><dt><strong>SNPmaxsel</strong><dd>Maximally selected statistics for SNP data. 
<br><dt><strong>SRPM</strong><dd>Shared Reproducibility Package Management. 
<br><dt><strong>SQLiteMap</strong><dd>Manage vector graphical maps using SQLite. 
<br><dt><strong>SQLiteDF</strong><dd>Stores data frames and matrices in SQLite tables. 
<br><dt><strong>SciViews</strong><dd>A bundle of packages to implement a full reusable <acronym>GUI</acronym> API for
R.  Contains <strong>svGUI</strong> with the main <acronym>GUI</acronym> features,
<strong>svDialogs</strong> for the dialog boxes, <strong>svIO</strong> for data
import/export, <strong>svMisc</strong> with miscellaneous supporting functions,
and <strong>svViews</strong> providing views and report features (views are
<acronym>HTML</acronym> presentations of the content of R objects, combining
text, tables and graphs in the same document). 
<br><dt><strong>SiZer</strong><dd>Significant Zero crossings. 
<br><dt><strong>SemiPar</strong><dd>Functions for semiparametric regression analysis, to complement the
book &ldquo;Semiparametric Regression&rdquo; by R. Ruppert, M. P. Wand, and
R. J. Carroll, 2003, Cambridge University Press. 
<br><dt><strong>SenSrivastava</strong><dd>Collection of datasets from &ldquo;Regression Analysis, Theory, Methods and
Applications&rdquo; by A. Sen and M. Srivastava, 1990, Springer. 
<br><dt><strong>SensoMineR</strong><dd>Sensory data analysis. 
<br><dt><strong>SeqKnn</strong><dd>Sequential KNN imputation. 
<br><dt><strong>SharedHT2</strong><dd>Shared Hotelling T^2 test for small sample microarray
experiments. 
<br><dt><strong>SimHap</strong><dd>A comprehensive modeling framework for epidemiological outcomes and a
multiple-imputation approach to haplotypic analysis of population-based
data. 
<br><dt><strong>Snowball</strong><dd>Snowball stemmers. 
<br><dt><strong>SoDA</strong><dd>Utilities and examples from the book &ldquo;Software for Data Analysis:
Programming with R&rdquo; by John Chambers, Springer, 2008. 
<br><dt><strong>SoPhy</strong><dd>Soil Physics Tools: simulation of water flux and solute transport in
soil. 
<br><dt><strong>SparseLogReg</strong><dd>Sparse logistic regression. 
<br><dt><strong>SparseM</strong><dd>Basic linear algebra for sparse matrices. 
<br><dt><strong>SpatialNP</strong><dd>Multivariate nonparametric methods based on spatial signs and ranks. 
<br><dt><strong>SpherWave</strong><dd>Spherical Wavelets and SW-based spatially adaptive methods. 
<br><dt><strong>StatDA</strong><dd>Statistical analysis for environmental data, a companion to the book
&ldquo;Statistical Data Analysis Explained: Applied Environmental Statistics
with R&rdquo; by C. Reimann, P. Filzmoser, R. G. Garrett, and R. Dutter,
2008, John Wiley and Sons. 
<br><dt><strong>StatDataML</strong><dd>Read and write StatDataML. 
<!-- <COMMENT> -->
<!-- Removed on 2007-11-17 due to license problems. -->
<!-- @item StoppingRules -->
<!-- Stopping rules for microarray classifiers. -->
<!-- </COMMENT> -->
<br><dt><strong>StreamMetabolism</strong><dd>Calculation of single station metabolism from diurnal oxygen curves. 
<br><dt><strong>SuppDists</strong><dd>Ten distributions supplementing those built into R (Inverse Gauss,
Kruskal-Wallis, Kendall's Tau, Friedman's chi squared, Spearman's rho,
maximum F ratio, the Pearson product moment correlation coefficiant,
Johnson distributions, normal scores and generalized hypergeometric
distributions). 
<br><dt><strong>SwissAir</strong><dd>Air quality data of Switzerland for one year in 30 min resolution. 
<br><dt><strong>SyNet</strong><dd>Inference and analysis of sympatry networks. 
<br><dt><strong>Synth</strong><dd>Causal inference using the synthetic control group method. 
<br><dt><strong>TIMP</strong><dd>A problem solving environment for fitting superposition models. 
<br><dt><strong>TRAMPR</strong><dd>Terminal Restriction Fragment Length Polymorphism (TRFLP) Analysis and
Matching Package for R. 
<br><dt><strong>TSA</strong><dd>Functions and datasets detailed in the book &ldquo;Time Series Analysis
With Applications in R&rdquo; (3rd edition) by Jonathan Cryer and Kung-Sik
Chan, 2008, Springer. 
<br><dt><strong>TSHRC</strong><dd>Two Stage Hazard Rate Comparison. 
<br><dt><strong>TSMySQL</strong><dd>Time Series Database Interface extensions for MySQL. 
<br><dt><strong>TSP</strong><dd>Traveling Salesperson Problem (TSP). 
<br><dt><strong>TSSQLite</strong><dd>Time Series Database Interface extensions for SQLite. 
<br><dt><strong>TSdbi</strong><dd>Time Series Database Interface. 
<br><dt><strong>TSpadi</strong><dd>Connect to a time series database (e.g., Fame) via PADI (Protocol for
Application Database Interface), using the <strong>TSdbi</strong>
infrastructure. 
<br><dt><strong>TTR</strong><dd>Technical Trading Rules. 
<br><dt><strong>TWIX</strong><dd>Trees WIth eXtra splits. 
<br><dt><strong>TeachingDemos</strong><dd>A set of demonstration functions that can be used in a classroom to
demonstrate statistical concepts, or on your own to better understand
the concepts or the programming. 
<br><dt><strong>TraMineR</strong><dd>Sequences and trajectories mining for social scientists. 
<br><dt><strong>TwoWaySurvival</strong><dd>Additive two-way hazards modeling of right censored survival data. 
<br><dt><strong>TwslmSpikeWeight</strong><dd>Normalization of cDNA microarray data with the two-way semilinear model
(TW-SLM). 
<br><dt><strong>UNF</strong><dd>Tools for creating universal numeric fingerprints for data. 
<br><dt><strong>USPS</strong><dd>Unsupervised and Supervised methods of Propensity Score adjustment for
bias. 
<br><dt><strong>Umacs</strong><dd>Universal MArkov Chain Sampler. 
<br><dt><strong>UsingR</strong><dd>Data sets to accompany the textbook &ldquo;Using R for Introductory
Statistics&rdquo; by J. Verzani, 2005, Chapman &amp; Hall/CRC. 
<br><dt><strong>VDCutil</strong><dd>Utilities supporting VDC, an open source digital library system for
quantitative data. 
<br><dt><strong>VGAM</strong><dd>Vector Generalized Linear and Additive Models. 
<br><dt><strong>VIM</strong><dd>Visualization and Imputation of Missing Values. 
<br><dt><strong>VLMC</strong><dd>Functions, classes &amp; methods for estimation, prediction, and simulation
(bootstrap) of VLMC (Variable Length Markov Chain) models. 
<br><dt><strong>VaR</strong><dd>Methods for calculation of Value at Risk (VaR). 
<br><dt><strong>VhayuR</strong><dd>R interface to the Vhayu Velocity high volume fast financial market data
archival and analysis products. 
<br><dt><strong>WINRPACK</strong><dd>Reads in WIN pickfile and waveform files. 
<br><dt><strong>WWGbook</strong><dd>Functions and datasets for the book &ldquo;Linear Mixed Models: A Practical
Guide Using Statistical Software&rdquo; by B. West, K. Welch, and A. Galecki,
2006, Chapman &amp; Hall/CRC. 
<br><dt><strong>WeedMap</strong><dd>Map of weed intensity. 
<br><dt><strong>WhatIf</strong><dd>Software for evaluating counterfactuals. 
<br><dt><strong>XML</strong><dd>Tools for reading <acronym>XML</acronym> documents and DTDs. 
<br><dt><strong>XReg</strong><dd>Extreme regression. 
<br><dt><strong>YaleToolkit</strong><dd>Data exploration tools from Yale University. 
<br><dt><strong>ZIGP</strong><dd>Zero Inflated Generalized Poisson (ZIGP) regression models. 
<br><dt><strong>Zelig</strong><dd>Everyone's statistical software: an easy-to-use program that can
estimate, and help interpret the results of, an enormous range of
statistical models. 
<br><dt><strong>aaMI</strong><dd>Mutual information for protein sequence alignments. 
<br><dt><strong>abind</strong><dd>Combine multi-dimensional arrays. 
<br><dt><strong>accuracy</strong><dd>A suite of tools designed to test and improve the accuracy of
statistical computation. 
<br><dt><strong>acepack</strong><dd>ACE (Alternating Conditional Expectations) and AVAS (Additivity and
VAriance Stabilization for regression) methods for selecting regression
transformations. 
<br><dt><strong>actuar</strong><dd>Functions related to actuarial science applications. 
<br><dt><strong>ada</strong><dd>Performs boosting algorithms for a binary response. 
<br><dt><strong>adabag</strong><dd>Adaboost.M1 and Bagging. 
<br><dt><strong>adapt</strong><dd>Adaptive quadrature in up to 20 dimensions. 
<br><dt><strong>ade4</strong><dd>Multivariate data analysis and graphical display. 
<br><dt><strong>ade4TkGUI</strong><dd>Tcl/Tk Graphical User Interface for <strong>ade4</strong>. 
<br><dt><strong>adegenet</strong><dd>Genetic data handling for multivariate analysis using <strong>ade4</strong>. 
<br><dt><strong>adehabitat</strong><dd>A collection of tools for the analysis of habitat selection by animals. 
<br><dt><strong>adimpro</strong><dd>Adaptive smoothing of digital images. 
<br><dt><strong>adk</strong><dd>Anderson-Darling K-sample test and combinations of such tests. 
<br><dt><strong>adlift</strong><dd>Adaptive Wavelet transforms for signal denoising. 
<br><dt><strong>ads</strong><dd>Spatial point patterns analysis. 
<br><dt><strong>agce</strong><dd>Analysis of growth curve experiments. 
<br><dt><strong>agricolae</strong><dd>Statistical procedures for agricultural research. 
<br><dt><strong>agsemisc</strong><dd>Miscellaneous plotting and utility functions. 
<br><dt><strong>akima</strong><dd>Linear or cubic spline interpolation for irregularly gridded data. 
<br><dt><strong>allelic</strong><dd>A fast, unbiased and exact allelic exact test. 
<br><dt><strong>alr3</strong><dd>Methods and data to accompany the textbook &ldquo;Applied Linear Regression&rdquo;
by S. Weisberg, 2005, Wiley. 
<br><dt><strong>amap</strong><dd>Another Multidimensional Analysis Package. 
<br><dt><strong>anacor</strong><dd>Simple and Canonical Correspondence Analysis. 
<br><dt><strong>analogue</strong><dd>Analogue methods for palaeoecology. 
<br><dt><strong>anapuce</strong><dd>Tools for microarray data analysis. 
<br><dt><strong>animation</strong><dd>Demonstrate animations in statistics. 
<br><dt><strong>anm</strong><dd>Analog model for statistical/empirical downscaling. 
<br><dt><strong>aod</strong><dd>Analysis of Overdispersed Data. 
<br><dt><strong>apTreeshape</strong><dd>Analyses of phylogenetic treeshape. 
<br><dt><strong>ape</strong><dd>Analyses of Phylogenetics and Evolution, providing functions for reading
and plotting phylogenetic trees in parenthetic format (standard Newick
format), analyses of comparative data in a phylogenetic framework,
analyses of diversification and macroevolution, computing distances from
allelic and nucleotide data, reading nucleotide sequences from GenBank
via internet, and several tools such as Mantel's test, computation of
minimum spanning tree, or the population parameter theta based on
various approaches. 
<br><dt><strong>aplpack</strong><dd>Another PLot PACKage: stem.leaf, bagplot, faces, spin3R, <small class="dots">...</small>. 
<br><dt><strong>apsrtable</strong><dd>American Political Science Review style table formatting. 
<br><dt><strong>argosfilter</strong><dd>Argos locations filter. 
<br><dt><strong>arm</strong><dd>Data Analysis using Regression and Multilevel/hierarchical models. 
<br><dt><strong>arrayImpute</strong><dd>Missing imputation for microarray data. 
<br><dt><strong>arrayMissPattern</strong><dd>Exploratory analysis of missing patterns for microarray data. 
<br><dt><strong>ars</strong><dd>Adaptive Rejection Sampling. 
<br><dt><strong>arules</strong><dd>Mining association rules and frequent itemsets with R. 
<br><dt><strong>arulesSequences</strong><dd>Mining frequent sequences. 
<br><dt><strong>ash</strong><dd>David Scott's ASH routines for 1D and 2D density estimation. 
<br><dt><strong>aspace</strong><dd>Estimating centrographic statistics and computational geometries from
spatial point patterns. 
<br><dt><strong>assist</strong><dd>A suite of functions implementing smoothing splines. 
<br><dt><strong>aster</strong><dd>Functions and datasets for Aster modeling (forest graph exponential
family conditional or unconditional canonical statistic models for life
history trait modeling). 
<br><dt><strong>asypow</strong><dd>A set of routines that calculate power and related quantities utilizing
asymptotic likelihood ratio methods. 
<br><dt><strong>asuR</strong><dd>Functions and data sets for a lecture in &ldquo;Advanced Statistics using
R&rdquo;. 
<br><dt><strong>aws</strong><dd>Functions to perform adaptive weights smoothing. 
<br><dt><strong>backfitRichards</strong><dd>Backfitted independent values of Richards curves. 
<br><dt><strong>backtest</strong><dd>Exploring portfolio-based hypotheses about financial instruments. 
<br><dt><strong>bark</strong><dd>Bayesian Additive Regression Kernels. 
<br><dt><strong>bayesGARCH</strong><dd>Bayesian estimation of the GARCH(1,1) model with Student's t
innovations. 
<br><dt><strong>bayesSurv</strong><dd>Bayesian survival regression with flexible error and (later on also
random effects) distributions. 
<br><dt><strong>bayescount</strong><dd>Bayesian analysis of count distributions with JAGS. 
<br><dt><strong>bayesm</strong><dd>Bayes Inference for Marketing/Micro-econometrics. 
<br><dt><strong>bayesmix</strong><dd>Bayesian mixture models of univariate Gaussian distributions using JAGS. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2006-04-28 (2.3.0 Error; no response). -->
<!-- @item baymvb -->
<!-- Bayesian analysis of multivariate binary data. -->
<!-- </COMMENT> -->
<br><dt><strong>bbmle</strong><dd>Modifications and extensions of <strong>stats4</strong> MLE code. 
<br><dt><strong>bcp</strong><dd>Bayesian Change Point based on the Barry and Hartigan product partition
model. 
<br><dt><strong>bear</strong><dd>Bioavability and bioequivalence data analysis with crossover design. 
<br><dt><strong>benchden</strong><dd>28 benchmark densities from Berlinet/Devroye (1994). 
<br><dt><strong>bentcableAR</strong><dd>Bent-cable regression for independent data or autoregressive time
series. 
<br><dt><strong>betaper</strong><dd>Distance decay of similarity among biological inventories in the face of
taxonomic uncertainty. 
<br><dt><strong>betareg</strong><dd>Beta regression for modeling rates and proportions. 
<br><dt><strong>bs</strong><dd>Utilities for the Birnbaum-Saunders distribution. 
<br><dt><strong>biOps</strong><dd>Basic image operations and image processing. 
<br><dt><strong>biOpsGUI</strong><dd><acronym>GUI</acronym> for Basic image operations. 
<br><dt><strong>biclust</strong><dd>BiCluster algorithms. 
<br><dt><strong>bicreduc</strong><dd>Reduction algorithm for the NPMLE for the distribution function of
bivariate interval-censored data. 
<br><dt><strong>bifactorial</strong><dd>Inferences for bi- and trifactorial trial designs. 
<br><dt><strong>biglm</strong><dd>Linear regression for data too large to fit in memory. 
<br><dt><strong>bigmemory</strong><dd>Manage massive matrices in R using C++, with UNIX support for shared
memory. 
<br><dt><strong>bim</strong><dd>Bayesian interval mapping diagnostics:  functions to interpret QTLCart
and Bmapqtl samples. 
<br><dt><strong>binGroup</strong><dd>Evaluation and experimental design for binomial group testing. 
<br><dt><strong>binMto</strong><dd>Asymptotic simultaneous confidence intervals for many-to-one comparisons
of proportions. 
<br><dt><strong>bindata</strong><dd>Generation of correlated artificial binary data. 
<br><dt><strong>binom</strong><dd>Binomial confidence intervals for several parameterizations. 
<br><dt><strong>bio.infer</strong><dd>Compute biological inferences. 
<br><dt><strong>biopara</strong><dd>Self-contained parallel system for R. 
<br><dt><strong>bipartite</strong><dd>Visualises bipartite networks and calculates some ecological indices. 
<br><dt><strong>birch</strong><dd>Dealing with very large datasets using BIRCH. 
<br><dt><strong>bise</strong><dd>Auxiliary functions for phenological data analysis. 
<br><dt><strong>bitops</strong><dd>Functions for Bitwise operations on integer vectors. 
<br><dt><strong>bivpois</strong><dd>Bivariate Poisson models using the EM algorithm. 
<br><dt><strong>blighty</strong><dd>Function for drawing the coastline of the United Kingdom. 
<br><dt><strong>blockTools</strong><dd>Block, randomly assign, and diagnose potential problems between units in
randomized experiments. 
<br><dt><strong>blockrand</strong><dd>Randomization for block random clinical trials. 
<br><dt><strong>bmd</strong><dd>Benchmark dose analysis for dose-response data. 
<br><dt><strong>bnlearn</strong><dd>Bayesian network structure learning. 
<br><dt><strong>boa</strong><dd>Bayesian Output Analysis Program for MCMC. 
<br><dt><strong>boolean</strong><dd>Boolean logit and probit: a procedure for testing Boolean hypotheses. 
<br><dt><strong>boost</strong><dd>Boosting methods for real and simulated data, featuring `BagBoost',
`LogitBoost', `AdaBoost', and `L2Boost'. 
<br><dt><strong>boot</strong><dd>Functions and datasets for bootstrapping from the book &ldquo;Bootstrap
Methods and Their Applications&rdquo; by A. C. Davison and D. V. Hinkley,
1997, Cambridge University Press.  <em>Recommended</em>. 
<br><dt><strong>bootStepAIC</strong><dd>Model selection by bootstrapping the <code>stepAIC()</code> procedure. 
<br><dt><strong>bootstrap</strong><dd>Software (bootstrap, cross-validation, jackknife), data and errata for
the book &ldquo;An Introduction to the Bootstrap&rdquo; by B. Efron and
R. Tibshirani, 1993, Chapman and Hall. 
<br><dt><strong>bpca</strong><dd>Biplot of multivariate data based on Principal Components Analysis. 
<br><dt><strong>bqtl</strong><dd>QTL mapping toolkit for inbred crosses and recombinant inbred lines. 
Includes maximum likelihood and Bayesian tools. 
<br><dt><strong>brainwaver</strong><dd>Basic wavelet analysis of multivariate time series with a vizualisation
and parametrization using graph theory. 
<br><dt><strong>brew</strong><dd>Templating framework for report generation. 
<br><dt><strong>brglm</strong><dd>Bias-reduction in binomial-response GLMs. 
<br><dt><strong>brlr</strong><dd>Bias-reduced logistic regression: fits logistic regression models by
maximum penalized likelihood. 
<br><dt><strong>bspec</strong><dd>Bayesian inference on the (discrete) power spectrum of time series. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2008-05-02 as requested by the maintainer. -->
<!-- @item butler -->
<!-- Unit testing, profiling and benchmarking for R. -->
<!-- </COMMENT> -->
<br><dt><strong>bvls</strong><dd>The Stark-Parker algorithm for bounded-variable least squares. 
<br><dt><strong>ca</strong><dd>Simple, multiple and joint Correspondence Analysis. 
<br><dt><strong>caMassClass</strong><dd>Processing and Classification of protein mass spectra (SELDI) data. 
<br><dt><strong>caTools</strong><dd>Miscellaneous utility functions, including reading/writing ENVI binary
files, a LogitBoost classifier, and a base64 encoder/decoder. 
<br><dt><strong>cacheSweave</strong><dd>Tools for caching Sweave computations. 
<br><dt><strong>cacher</strong><dd>Tools for caching and distributing statistical analyses. 
<br><dt><strong>cairoDevice</strong><dd>Loadable CAIRO/GTK device driver. 
<br><dt><strong>calib</strong><dd>Statistical tool for calibration of plate based bioassays. 
<br><dt><strong>calibrate</strong><dd>Calibration of biplot axes. 
<br><dt><strong>candisc</strong><dd>Generalized canonical discriminant analysis. 
<br><dt><strong>car</strong><dd>Companion to Applied Regression, containing functions for applied
regession, linear models, and generalized linear models, with an
emphasis on regression diagnostics, particularly graphical diagnostic
methods. 
<br><dt><strong>caret</strong><dd>Classification and REgression Training. 
<br><dt><strong>caretLSF</strong><dd>Classification and REgression Training, LSF style. 
<br><dt><strong>caretNWS</strong><dd>Classification and REgression Training in parallel using NetworkSpaces. 
<br><dt><strong>cat</strong><dd>Analysis of categorical-variable datasets with missing values. 
<br><dt><strong>catmap</strong><dd>Case-control and TDT meta-analysis package. 
<br><dt><strong>catspec</strong><dd>Special models for categorical variables. 
<br><dt><strong>cba</strong><dd>Clustering for Business Analytics, including implementations of Proximus
and Rock. 
<br><dt><strong>cclust</strong><dd>Convex clustering methods, including k-means algorithm, on-line
update algorithm (Hard Competitive Learning) and Neural Gas algorithm
(Soft Competitive Learning) and calculation of several indexes for
finding the number of clusters in a data set. 
<br><dt><strong>celsius</strong><dd>Retrieve Affymetrix microarray measurements and metadata from Celsius. 
<br><dt><strong>cem</strong><dd>The coarsened exact matching algorithm (and many extensions). 
<br><dt><strong>cfa</strong><dd>Analysis of configuration frequencies. 
<br><dt><strong>cggd</strong><dd>Continuous Generalized Gradient Descent. 
<br><dt><strong>cgh</strong><dd>Analysis of microarray comparative genome hybridisation data using the
Smith-Waterman algorithm. 
<br><dt><strong>cghFLasso</strong><dd>Hot spot detecting for CGH array data with fused lasso regression. 
<br><dt><strong>chplot</strong><dd>Augmented convex hull plots: informative and nice plots for grouped
bivariate data. 
<br><dt><strong>changeLOS</strong><dd>Change in length of hospital stay (LOS). 
<br><dt><strong>cheb</strong><dd>Discrete linear Chebyshev approximation. 
<br><dt><strong>chemCal</strong><dd>Calibration functions for analytical chemistry. 
<br><dt><strong>chemometrics</strong><dd>Companion to the book &ldquo;Introduction to Multivariate Statistical
Analysis in Chemometrics&rdquo; by K. Varmuza and P. Filzmoser, CRC Press, to
appear. 
<br><dt><strong>choplump</strong><dd>Choplump tests (permutation tests for comparing two groups with some
positive but many zero responses). 
<br><dt><strong>chron</strong><dd>A package for working with chronological objects (times and dates). 
<br><dt><strong>cir</strong><dd>Nonparametric estimation of monotone functions via isotonic regression
and centered isotonic regression. 
<br><dt><strong>circular</strong><dd>Circular statistics, from &ldquo;Topics in Circular Statistics&rdquo; by Rao
Jammalamadaka and A. SenGupta, 2001, World Scientific. 
<br><dt><strong>clValid</strong><dd>Statistical and biological validation of clustering results. 
<br><dt><strong>clac</strong><dd>Clust Along Chromosomes, a method to call gains/losses in CGH array
data. 
<br><dt><strong>class</strong><dd>Functions for classification (k-nearest neighbor and LVQ). 
Contained in the <samp><span class="file">VR</span></samp> bundle.  <em>Recommended</em>. 
<br><dt><strong>classGraph</strong><dd>Construct graph of S4 class hierarchies. 
<br><dt><strong>classInt</strong><dd>Choose univariate class intervals for mapping or other graphics
purposes. 
<br><dt><strong>classPP</strong><dd>Projection Pursuit for supervised classification. 
<br><dt><strong>classifly</strong><dd>Explore classification models in high dimensions. 
<br><dt><strong>clim.pact</strong><dd>Climate analysis and downscaling for monthly and daily data. 
<br><dt><strong>climatol</strong><dd>Functions to fill missing data in climatological (monthly) series and to
test their homogeneity, plus functions to draw wind-rose and
Walter&amp;Lieth diagrams. 
<!-- <COMMENT> -->
<!-- Archived on 2007-03-13 as requested by the maintainer. -->
<!-- @item clines -->
<!-- Calculates Contour Lines. -->
<!-- </COMMENT> -->
<br><dt><strong>clinfun</strong><dd>Utilities for clinical study design and data analyses. 
<br><dt><strong>clue</strong><dd>CLUster Ensembles. 
<br><dt><strong>clustTool</strong><dd><acronym>GUI</acronym> for clustering data with spatial information. 
<br><dt><strong>cluster</strong><dd>Functions for cluster analysis.  <em>Recommended</em>. 
<br><dt><strong>clusterGeneration</strong><dd>Random cluster generation (with specified degree of separation). 
<br><dt><strong>clusterRepro</strong><dd>Reproducibility of gene expression clusters. 
<br><dt><strong>clusterSim</strong><dd>Searching for optimal clustering procedure for a data set. 
<br><dt><strong>clusterfly</strong><dd>Explore clustering interactively using R and GGobi. 
<br><dt><strong>clustvarsel</strong><dd>Variable selection for model-based clustering. 
<br><dt><strong>clv</strong><dd>Cluster validation techniques. 
<br><dt><strong>cmprsk</strong><dd>Estimation, testing and regression modeling of subdistribution functions
in competing risks. 
<br><dt><strong>cobs99</strong><dd>Constrained B-splines: outdated 1999 version. 
<br><dt><strong>cobs</strong><dd>Constrained B-splines: qualitatively constrained (regression) smoothing
via linear programming and sparse matrices. 
<br><dt><strong>cocorresp</strong><dd>Co-correspondence analysis ordination methods for community ecology. 
<br><dt><strong>coda</strong><dd>Output analysis and diagnostics for Markov Chain Monte Carlo (MCMC)
simulations. 
<br><dt><strong>codetools</strong><dd>Code analysis tools.  <em>Recommended</em> for R 2.5.0 or later. 
<br><dt><strong>coin</strong><dd>COnditional INference procedures for the general independence problem
including two-sample, K-sample, correlation, censored, ordered
and multivariate problems. 
<br><dt><strong>colorRamp</strong><dd>Builds single and double gradient color maps. 
<br><dt><strong>colorspace</strong><dd>Mapping between assorted color spaces. 
<br><dt><strong>combinat</strong><dd>Combinatorics utilities. 
<br><dt><strong>compHclust</strong><dd>Complementary hierarchical clustering. 
<br><dt><strong>compOverlapCorr</strong><dd>Comparing overlapping correlation coefficients. 
<br><dt><strong>compare</strong><dd>Comparing objects for differences. 
<br><dt><strong>compoisson</strong><dd>Conway-Maxwell-Poisson distribution. 
<br><dt><strong>compositions</strong><dd>Functions for the consistent analysis of compositional data (e.g.,
portions of substances) and positive numbers (e.g., concentrations). 
<br><dt><strong>concor</strong><dd>Concordance, providing &ldquo;SVD by blocks&rdquo;. 
<br><dt><strong>concord</strong><dd>Measures of concordance and reliability. 
<br><dt><strong>conf.design</strong><dd>A series of simple tools for constructing and manipulating confounded
and fractional factorial designs. 
<br><dt><strong>connectedness</strong><dd>Find disconnected sets for two-way classification. 
<br><dt><strong>contfrac</strong><dd>Continued fractions. 
<br><dt><strong>contrast</strong><dd>A collection of contrast methods. 
<br><dt><strong>copas</strong><dd>Statistical methods to model and adjust for bias in meta-analysis. 
<br><dt><strong>copula</strong><dd>Classes of commonly used copulas (including elliptical and Archimedian),
and methods for density, distribution, random number generators, and
plotting. 
<br><dt><strong>corpora</strong><dd>Utility functions for the statistical analysis of corpus frequency
data. 
<br><dt><strong>corpcor</strong><dd>Efficient estimation of covariance and (partial) correlation. 
<br><dt><strong>corrgram</strong><dd>Plot a correlogram. 
<br><dt><strong>corrperm</strong><dd>Permutation tests of correlation with repeated measurements. 
<br><dt><strong>covRobust</strong><dd>Robust covariance estimation via nearest neighbor cleaning. 
<br><dt><strong>coxphf</strong><dd>Cox regression with Firth's penalized likelihood. 
<br><dt><strong>coxphw</strong><dd>Weighted estimation for Cox regression. 
<br><dt><strong>coxrobust</strong><dd>Robust Estimation in the Cox proportional hazards regression model. 
<br><dt><strong>cramer</strong><dd>Routine for the multivariate nonparametric Cramer test. 
<br><dt><strong>crank</strong><dd>Functions for completing and recalculating rankings. 
<br><dt><strong>crossdes</strong><dd>Functions for the construction and randomization of balanced carryover
balanced designs, to check given designs for balance, and for simulation
studies on the validity of two randomization procedures. 
<br><dt><strong>crosshybDetector</strong><dd>Detection of cross-hybridization events in microarray experiments. 
<br><dt><strong>crq</strong><dd>Quantile regression for randomly censored data. 
<br><dt><strong>cslogistic</strong><dd>Likelihood and posterior analysis of conditionally specified logistic
regression models. 
<br><dt><strong>cts</strong><dd>Continuous time autoregressive models and the Kalman filter. 
<br><dt><strong>ctv</strong><dd>Server-side and client-side tools for <acronym>CRAN</acronym> task views. 
<br><dt><strong>cwhmisc</strong><dd>Miscellaneous functions by Christian W. Hoffmann. 
<br><dt><strong>cyclones</strong><dd>Cyclone identification. 
<!-- <COMMENT> -->
<!-- Moved to Archive on 2006-05-19 as requested by the maintainer. -->
<!-- @item data.table -->
<!-- Just like a data frame but without rownames. -->
<!-- </COMMENT> -->
<br><dt><strong>dataframes2xls</strong><dd>Write data frames to <samp><span class="file">.xls</span></samp> files. 
<br><dt><strong>date</strong><dd>Functions for dealing with dates.  The most useful of them accepts a
vector of input dates in any of the forms `<samp><span class="samp">8/30/53</span></samp>',
`<samp><span class="samp">30Aug53</span></samp>', `<samp><span class="samp">30 August 1953</span></samp>', <small class="dots">...</small>, `<samp><span class="samp">August 30 53</span></samp>', or
any mixture of these. 
<br><dt><strong>dblcens</strong><dd>Calculates the NPMLE of the survival distribution for doubly censored
data. 
<br><dt><strong>ddesolve</strong><dd>Solver for Delay Differential Equations. 
<br><dt><strong>ddst</strong><dd>Data driven smooth Neyman test. 
<br><dt><strong>deSolve</strong><dd>General solvers for ordinary differential equations (ODE) and for
differential algebraic equations (DAE). 
<br><dt><strong>deal</strong><dd>Bayesian networks with continuous and/or discrete variables can be
learned and compared from data. 
<br><dt><strong>debug</strong><dd>Debugger for R functions, with code display, graceful error recovery,
line-numbered conditional breakpoints, access to exit code, flow
control, and full keyboard input. 
<br><dt><strong>degreenet</strong><dd>Models for skewed count distributions relevant to networks. 
<br><dt><strong>deldir</strong><dd>Calculates the  Delaunay triangulation and the Dirichlet or Voronoi
tesselation (with respect to the entire plane) of a planar point set. 
<br><dt><strong>delt</strong><dd>Estimation of multivariate densities with adaptive histograms. 
<br><dt><strong>demogR</strong><dd>Analysis of age-structured demographic models. 
<br><dt><strong>denpro</strong><dd>Visualization of multivariate density functions and estimates with level
set trees and shape trees, and visualization of multivariate data with
tail trees. 
<br><dt><strong>denstrip</strong><dd>Density strips and other methods for compactly illustrating
distributions. 
<br><dt><strong>depmix</strong><dd>Dependent Mixture Models: fit (multi-group) mixtures of latent Markov
models on mixed categorical and continuous (time series) data. 
<br><dt><strong>depmixS4</strong><dd>Dependent Mixture Models: fit latent (hidden) Markov models on mixed
categorical and continuous (time series) data. 
<br><dt><strong>desirability</strong><dd>Desirabiliy function optimization and ranking. 
<br><dt><strong>dfcrm</strong><dd>Dose-finding by the continual reassessment method. 
<br><dt><strong>dglm</strong><dd>Double generalized linear models. 
<br><dt><strong>diagram</strong><dd>Functions for visualising simple graphs (networks) and plotting flow
diagrams. 
<br><dt><strong>diamonds</strong><dd>Functions for illustrating aperture-4 diamond partitions in the plane,
or on the surface of an octahedron or icosahedron, for use as analysis
or sampling grids. 
<br><dt><strong>dice</strong><dd>Calculate probabilities of various dice-rolling events. 
<br><dt><strong>dichromat</strong><dd>Color schemes for dichromats: collapse red-green distinctions to
simulate the effects of colour-blindness. 
<br><dt><strong>diffractometry</strong><dd>Baseline identification and peak decomposition for x-ray
diffractograms. 
<br><dt><strong>digest</strong><dd>Two functions for the creation of &ldquo;hash&rdquo; digests of arbitrary R
objects using the md5 and sha-1 algorithms permitting easy comparison of
R language objects. 
<br><dt><strong>diptest</strong><dd>Compute Hartigan's dip test statistic for unimodality. 
<br><dt><strong>dispmod</strong><dd>Functions for modelling dispersion in GLMs. 
<br><dt><strong>distr</strong><dd>An object orientated implementation of distributions and some additional
functionality. 
<br><dt><strong>distrDoc</strong><dd>Documentation for packages <strong>distr</strong>, <strong>distrEx</strong>,
<strong>distrSim</strong>, and <strong>distrTEst</strong>. 
<br><dt><strong>distrEx</strong><dd>Extensions of package <strong>distr</strong>. 
<br><dt><strong>distrSim</strong><dd>Simulation classes based on package <strong>distr</strong>. 
<br><dt><strong>distrTEst</strong><dd>Estimation and Testing classes based on package <strong>distr</strong>. 
<br><dt><strong>distributions</strong><dd>Probability distributions based on TI-83 Plus. 
<br><dt><strong>diveMove</strong><dd>Dive analysis and calibration. 
<br><dt><strong>dlm</strong><dd>Maximum likelihood and Bayesian analysis of Dynamic Linear Models. 
<br><dt><strong>doBy</strong><dd>Facilities for groupwise computations. 
<br><dt><strong>dplR</strong><dd>Dendrochronology Program Library in R. 
<br><dt><strong>dprep</strong><dd>Data preprocessing and visualization functions for classification. 
<br><dt><strong>dr</strong><dd>Functions, methods, and datasets for fitting dimension reduction
regression, including pHd and inverse regression methods SIR and SAVE. 
<br><dt><strong>drc</strong><dd>Non-linear regression analysis for multiple curves with focus on
concentration-response, dose-response and time-response curves. 
<br><dt><strong>drm</strong><dd>Regression and association models for clustered categorical responses. 
<br><dt><strong>drfit</strong><dd>Dose-response data evaluation. 
<br><dt><strong>dse</strong><dd>Dynamic System Estimation, a multivariate time series package bundle. 
Contains <strong>dse1</strong> (the base system, including multivariate ARMA and
state space models) and <strong>dse2</strong> (extensions for evaluating
estimation techniques, forecasting, and for evaluating forecasting
model). 
<br><dt><strong>dti</strong><dd>DTI (Diffusion Tensor Image) analysis. 
<br><dt><strong>dtt</strong><dd>Discrete Trigonometric Transforms. 
<br><dt><strong>dtw</strong><dd>Dynamic Time Warping algorithms. 
<br><dt><strong>dyn</strong><dd>Time series regression. 
<br><dt><strong>dynamicGraph</strong><dd>Interactive graphical tool for manipulating graphs. 
<br><dt><strong>dynamicTreeCut</strong><dd>Methods for detection of clusters in hierarchical clustering
dendrograms. 
<br><dt><strong>dynamo</strong><dd>Estimation, simulation, regularization and prediction of univariate
dynamic models including ARMA, ARMA-GARCH, ACD, and MEM. 
<br><dt><strong>dynlm</strong><dd>Dynamic linear models and time series regression. 
<br><dt><strong>e1071</strong><dd>Miscellaneous functions used at the Department of Statistics at TU Wien
(E1071), including moments, short-time Fourier transforms, Independent
Component Analysis, Latent Class Analysis, support vector machines, and
fuzzy clustering, shortest path computation, bagged clustering, and some
more. 
<br><dt><strong>eRm</strong><dd>Estimating extended Rasch models. 
<br><dt><strong>earth</strong><dd>Earth: multivariate adaptive regression spline models. 
<br><dt><strong>eba</strong><dd>Fitting and testing probabilistic choice models, especially the BTL,
elimination-by-aspects (EBA), and preference tree (Pretree) models. 
<br><dt><strong>ecespa</strong><dd>Functions and data for spatial point pattern analysis. 
<br><dt><strong>eco</strong><dd>Fitting Bayesian models of ecological inference in 2 by 2 tables. 
<br><dt><strong>ecodist</strong><dd>Dissimilarity-based functions for ecological analysis. 
<br><dt><strong>ecolMod</strong><dd>Figures, data sets and examples from the book &ldquo;A Practical Guide to
Ecological Modelling &mdash; Using R as a Simulation Platform&rdquo; by Karline
Soetaert and Peter M. J. Herman, 2008, Springer. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2007-11-29 as it fails R CMD check. -->
<!-- @item edci -->
<!-- Edge Detection and Clustering in Images. -->
<!-- </COMMENT> -->
<br><dt><strong>effects</strong><dd>Graphical and tabular effect displays, e.g., of interactions, for linear
and generalised linear models. 
<br><dt><strong>eha</strong><dd>A package for survival and event history analysis. 
<br><dt><strong>eiPack</strong><dd>Ecological inference and higher-dimension data management. 
<br><dt><strong>eigenmodel</strong><dd>Semiparametric factor and regression models for symmetric relational
data. 
<br><dt><strong>elasticnet</strong><dd>Elastic net regularization and variable selection. 
<br><dt><strong>ellipse</strong><dd>Package for drawing ellipses and ellipse-like confidence regions. 
<br><dt><strong>elliptic</strong><dd>A suite of elliptic and related functions including Weierstrass and
Jacobi forms. 
<br><dt><strong>elrm</strong><dd>Exact Logistic Regression via MCMC. 
<br><dt><strong>emdbook</strong><dd>Data sets and auxiliary functions for &ldquo;Ecological Models and Data&rdquo; by
Ben Bolker (work in progress). 
<br><dt><strong>emme2</strong><dd>Functions to read from and write to an EMME/2 databank. 
<br><dt><strong>empiricalBayes</strong><dd>A bundle for dealing with extreme multiple testing problems by
estimating local false discovery rates.  Contains packages
<strong>localFDR</strong> and <strong>HighProbability</strong>. 
<br><dt><strong>emplik</strong><dd>Empirical likelihood ratio for means/quantiles/hazards from possibly
right censored data. 
<br><dt><strong>emu</strong><dd>Interface to the Emu speech database system. 
<br><dt><strong>energy</strong><dd>E-statistics (energy) tests for comparing distributions: multivariate
normality, Poisson test, multivariate k-sample test for equal
distributions, hierarchical clustering by e-distances. 
<br><dt><strong>ensembleBMA</strong><dd>Probabilistic forecasting using Bayesian Model Averaging of ensembles
using a mixture of normal distributions. 
<br><dt><strong>epiR</strong><dd>Functions for analyzing epidemiological data. 
<br><dt><strong>epibasix</strong><dd>Elementary functions for epidemiological analysis. 
<br><dt><strong>epicalc</strong><dd>Epidemiological calculator. 
<br><dt><strong>epitools</strong><dd>Basic tools for applied epidemiology. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2008-04-22 (Orphaned/2.6.0). -->
<!-- @item epsi -->
<!-- Edge Preserving Smoothing for Images. -->
<!-- </COMMENT> -->
<br><dt><strong>equivalence</strong><dd>Tests and graphics for assessing tests of equivalence. 
<br><dt><strong>ergm</strong><dd>An integrated set of tools to analyze and simulate networks based on
exponential-family random graph models (ERGM). 
<br><dt><strong>etm</strong><dd>Empirical Transition Matrix. 
<br><dt><strong>evd</strong><dd>Functions for extreme value distributions.  Extends simulation,
distribution, quantile and density functions to univariate, bivariate
and (for simulation) multivariate parametric extreme value
distributions, and provides fitting functions which calculate maximum
likelihood estimates for univariate and bivariate models. 
<br><dt><strong>evdbayes</strong><dd>Functions for the bayesian analysis of extreme value models, using MCMC
methods. 
<br><dt><strong>evir</strong><dd>Extreme Values in R: Functions for extreme value theory, which may be
divided into the following groups; exploratory data analysis, block
maxima, peaks over thresholds (univariate and bivariate), point
processes, gev/gpd distributions. 
<br><dt><strong>exactLoglinTest</strong><dd>Monte Carlo exact tests for log-linear models. 
<br><dt><strong>exactRankTests</strong><dd>Computes exact p-values and quantiles using an implementation of
the Streitberg/Roehmel shift algorithm. 
<br><dt><strong>exactmaxsel</strong><dd>Exact methods for maximally selected statistics for binary response
variables. 
<br><dt><strong>experiment</strong><dd>Designing and analyzing randomized experiments. 
<br><dt><strong>expert</strong><dd>Modeling of data using expert opinion. 
<br><dt><strong>extRemes</strong><dd>Extreme value toolkit. 
<br><dt><strong>fArma</strong><dd>The Rmetrics module for &ldquo;ARMA Time Series Modelling&rdquo;. 
<br><dt><strong>fAsianOptions</strong><dd>The Rmetrics module for &ldquo;Option Valuation&rdquo;. 
<br><dt><strong>fAssets</strong><dd>The Rmetrics module for &ldquo;Assets Selection and Modelling&rdquo;. 
<br><dt><strong>fBasics</strong><dd>The Rmetrics module for &ldquo;Markets and Basic Statistics&rdquo;. 
<br><dt><strong>fBonds</strong><dd>The Rmetrics module for &ldquo;Bonds and Interest Rate Models&rdquo;. 
<br><dt><strong>fCalendar</strong><dd>The Rmetrics module for &ldquo;Chronological and Calendarical Objects&rdquo;. 
<br><dt><strong>fCopulae</strong><dd>The Rmetrics module for &ldquo;Dependence Structures with Copulas&rdquo;. 
<br><dt><strong>fEcofin</strong><dd>The Rmetrics module for &ldquo;Economic and Financial Data Sets&rdquo;. 
<br><dt><strong>fExoticOptions</strong><dd>The Rmetrics module for &ldquo;Option Valuation&rdquo;. 
<br><dt><strong>fExtremes</strong><dd>The Rmetrics module for &ldquo;Extreme Financial Market Data&rdquo;. 
<br><dt><strong>fGarch</strong><dd>The Rmetrics module for &ldquo;Autoregressive Conditional Heteroskedastic
Modelling&rdquo;. 
<br><dt><strong>fImport</strong><dd>The Rmetrics module for &ldquo;Chronological and Calendarical Objects&rdquo;. 
<br><dt><strong>fMultivar</strong><dd>The Rmetrics module for &ldquo;Multivariate Market Analysis&rdquo;. 
<br><dt><strong>fNonlinear</strong><dd>The Rmetrics module for &ldquo;Nonlinear and Chaotic Time Series Modelling&rdquo;. 
<br><dt><strong>fOptions</strong><dd>The Rmetrics module for &ldquo;Basics of Option Valuation&rdquo;. 
<br><dt><strong>fPortfolio</strong><dd>The Rmetrics module for &ldquo;Portfolio Selection and Optimization&rdquo;. 
<br><dt><strong>fRegression</strong><dd>The Rmetrics module for &ldquo;Regression Based Decision and Prediction&rdquo;. 
<br><dt><strong>fSeries</strong><dd>The Rmetrics module for &ldquo;Financial Time Series Objects&rdquo;. 
<br><dt><strong>fTrading</strong><dd>The Rmetrics module for &ldquo;Technical Trading Analysis&rdquo;. 
<br><dt><strong>fUnitRoots</strong><dd>The Rmetrics module for &ldquo;The Dynamical Process Behind Markets&rdquo;. 
<br><dt><strong>fUtilities</strong><dd>The Rmetrics module for &ldquo;Rmetrics Function Utilities&rdquo;. 
<br><dt><strong>fame</strong><dd>Interface for FAME time series database. 
<br><dt><strong>far</strong><dd>Modelization for Functional AutoRegressive processes. 
<br><dt><strong>faraway</strong><dd>Functions and datasets for books by Julian Faraway. 
<br><dt><strong>fastICA</strong><dd>Implementation of FastICA algorithm to perform Independent Component
Analysis (ICA) and Projection Pursuit. 
<br><dt><strong>fbati</strong><dd>Gene by environment interaction tests. 
<br><dt><strong>fda</strong><dd>Functional Data Analysis: analysis of data where the basic observation
is a function of some sort. 
<br><dt><strong>fdim</strong><dd>Functions for calculating fractal dimension. 
<br><dt><strong>fdrtool</strong><dd>Estimation and control of (local) False Discovery Rates. 
<br><dt><strong>feature</strong><dd>Feature significance for multivariate kernel density estimation. 
<br><dt><strong>femmeR</strong><dd>Output analysis of FEMME model results. 
<br><dt><strong>ff</strong><dd>Flat file database designed for large vectors and multi-dimensional
arrays. 
<br><dt><strong>ffmanova</strong><dd>Fifty-fifty MANOVA. 
<br><dt><strong>fgac</strong><dd>Families of Generalized Archimedean Copulas. 
<br><dt><strong>fgui</strong><dd>Function <acronym>GUI</acronym>. 
<br><dt><strong>fields</strong><dd>A collection of programs for curve and function fitting with an emphasis
on spatial data.  The major methods implemented include cubic and thin
plate splines, universal Kriging and Kriging for large data sets.  The
main feature is that any covariance function implemented in R can be
used for spatial prediction. 
<br><dt><strong>filehash</strong><dd>Simple file-based hash table. 
<br><dt><strong>filehashSQLite</strong><dd>Simple key-value database using SQLite as the backend. 
<br><dt><strong>financial</strong><dd>Solving financial problems in R. 
<br><dt><strong>fingerprint</strong><dd>Functions to operate on binary fingerprint data. 
<br><dt><strong>flexclust</strong><dd>Flexible cluster algorithms. 
<br><dt><strong>flexmix</strong><dd>Flexible Mixture Modeling: a general framework for finite mixtures of
regression models using the EM algorithm. 
<br><dt><strong>fmri</strong><dd>Functions for the analysis of fMRI experiments. 
<br><dt><strong>forecasting</strong><dd>A bundle with functions and datasets for forecasting.  Contains
<strong>forecast</strong> (time series forecasting), <strong>fma</strong> (data sets
from the book &ldquo;Forecasting: Methods and Applications&rdquo; by Makridakis,
Wheelwright &amp; Hyndman, 1998), and <strong>Mcomp</strong> (data from the
M-competitions). 
<br><dt><strong>foreign</strong><dd>Functions for reading and writing data stored by statistical software
like Minitab, S, SAS, SPSS, Stata, Systat, etc.  <em>Recommended</em>. 
<br><dt><strong>forensic</strong><dd>Statistical methods in forensic genetics. 
<br><dt><strong>fork</strong><dd>Functions for handling multiple processes: simple wrappers around the
Unix process management API calls. 
<br><dt><strong>fortunes</strong><dd>R fortunes. 
<br><dt><strong>forward</strong><dd>Forward search approach to robust analysis in linear and generalized
linear regression models. 
<br><dt><strong>fpc</strong><dd>Fixed point clusters, clusterwise regression and discriminant plots. 
<br><dt><strong>fpca</strong><dd>Restricted MLE for Functional Principal Components Analysis. 
<br><dt><strong>fracdiff</strong><dd>Maximum likelihood estimation of the parameters of a fractionally
differenced ARIMA(p,d,q) model (Haslett and Raftery, Applied
Statistics, 1989). 
<br><dt><strong>fractal</strong><dd>Insightful fractal time series modeling and analysis. 
<br><dt><strong>frailtypack</strong><dd>Fit a shared gamma frailty model and Cox proportional hazards model
using a Penalized Likelihood on the hazard function. 
<br><dt><strong>fso</strong><dd>Fuzzy set ordination. 
<br><dt><strong>ftnonpar</strong><dd>Features and strings for nonparametric regression. 
<br><dt><strong>fts</strong><dd>Fast operations for time series objects via an interface to tslib (a
C++ time series library). 
<br><dt><strong>fuzzyFDR</strong><dd>Exact calculation of fuzzy decision rules for multiple testing. 
<br><dt><strong>fuzzyOP</strong><dd>Fuzzy numbers and the main mathematical operations on these. 
<br><dt><strong>fuzzyRankTests</strong><dd>Fuzzy rank tests and confidence intervals. 
<br><dt><strong>fxregime</strong><dd>Frankel-Wei regression and structural change tools for estimating,
testing, dating and monitoring (de facto) exchange rate regimes. 
<br><dt><strong>g.data</strong><dd>Create and maintain delayed-data packages (DDP's). 
<br><dt><strong>gRain</strong><dd>Probability propagation in graphical independence networks. 
<br><dt><strong>gRbase</strong><dd>A package for graphical modelling in R.  Defines S4 classes for
graphical meta data and graphical models, and illustrates how
hierarchical log-linear models may be implemented and combined with
<strong>dynamicGraph</strong>. 
<br><dt><strong>gRc</strong><dd>Inference in graphical Gaussian models with edge and vertex symmetries. 
<!-- <COMMENT> -->
<!-- Moved to the Archive (gRcox => gRc) as requested by the maintainer. -->
<!-- @item gRcox -->
<!-- Inference in graphical gaussian models with edge and vertex -->
<!-- symmetries. -->
<!-- </COMMENT> -->
<br><dt><strong>gWidgets</strong><dd>gWidgets API for building toolkit-independent, interactive
<acronym>GUI</acronym>s. 
<br><dt><strong>gWidgetsRGtk2</strong><dd>Toolkit implementation of <strong>gWidgets</strong> for <strong>RGtk2</strong>. 
<br><dt><strong>gWidgetsWWW</strong><dd>Toolkit implementation of gWidgets for www. 
<br><dt><strong>gWidgetsrJava</strong><dd>Toolkit implementation of <strong>gWidgets</strong> for <strong>rJava</strong>. 
<br><dt><strong>gWidgetstcltk</strong><dd>Toolkit implementation of <strong>gWidgets</strong> for <strong>tcltk</strong>. 
<br><dt><strong>gafit</strong><dd>Genetic algorithm for curve fitting. 
<br><dt><strong>gam</strong><dd>Functions for fitting and working with Generalized Additive Models, as
described in chapter 7 of the White Book, and in &ldquo;Generalized Additive
Models&rdquo; by T. Hastie and R. Tibshirani (1990). 
<br><dt><strong>gamair</strong><dd>Data sets used in the book &ldquo;Generalized Additive Models: An
Introduction with R&rdquo; by S. Wood (2006). 
<br><dt><strong>gamlss</strong><dd>Functions to fit Generalized Additive Models for Location Scale and
Shape. 
<br><dt><strong>gamlss.cens</strong><dd>A GAMLSS add on package for censored data. 
<br><dt><strong>gamlss.dist</strong><dd>Extra distributions for GAMLSS modeling. 
<br><dt><strong>gamlss.mx</strong><dd>A GAMLSS add on package for fitting mixture distributions. 
<br><dt><strong>gamlss.nl</strong><dd>A GAMLSS add on package for fitting non linear parametric models. 
<br><dt><strong>gamlss.tr</strong><dd>A GAMLSS add on for generating and fitting truncated (gamlss.family)
distributions. 
<br><dt><strong>gap</strong><dd>Genetic analysis package for both population and family data. 
<!-- Includes @strong{pathmix} for path analysis of pairs of relatives, -->
<!-- and @strong{pointer} for complex segregation analysis. -->
<br><dt><strong>gbev</strong><dd>Gradient Boosted regression trees with Errors-in-Variables. 
<br><dt><strong>gbm</strong><dd>Generalized Boosted Regression Models: implements extensions to Freund
and Schapire's AdaBoost algorithm and J. Friedman's gradient boosting
machine.  Includes regression methods for least squares, absolute loss,
logistic, Poisson, Cox proportional hazards partial likelihood, and
AdaBoost exponential loss. 
<br><dt><strong>gbs</strong><dd>Generalized Birnbaum-Saunders distributions. 
<br><dt><strong>gcl</strong><dd>Compute a fuzzy rules or tree classifier from data. 
<br><dt><strong>gclus</strong><dd>Clustering Graphics.  Orders panels in scatterplot matrices and parallel
coordinate displays by some merit index. 
<br><dt><strong>gcmrec</strong><dd>Parameters estimation of the general semiparametric model for recurrent
event data proposed by Pe&ntilde;a and Hollander. 
<br><dt><strong>gdata</strong><dd>Various functions to manipulate data. 
<br><dt><strong>gee</strong><dd>An implementation of the Liang/Zeger generalized estimating equation
approach to GLMs for dependent data. 
<br><dt><strong>geepack</strong><dd>Generalized estimating equations solver for parameters in mean, scale,
and correlation structures, through mean link, scale link, and
correlation link.  Can also handle clustered categorical responses. 
<br><dt><strong>geiger</strong><dd>Analysis of evolutionary diversification. 
<br><dt><strong>genalg</strong><dd>R based genetic algorithm for binary and floating point chromosomes. 
<br><dt><strong>gene2pathway</strong><dd>Prediction of KEGG pathway membership for individual genes based on
InterPro domain signatures. 
<br><dt><strong>genetics</strong><dd>Classes and methods for handling genetic data.  Includes classes to
represent genotypes and haplotypes at single markers up to multiple
markers on multiple chromosomes, and functions for allele frequencies,
flagging homo/heterozygotes, flagging carriers of certain alleles,
computing disequlibrium, testing Hardy-Weinberg equilibrium, <small class="dots">...</small>
<br><dt><strong>geoR</strong><dd>Functions to perform geostatistical data analysis including model-based
methods. 
<br><dt><strong>geoRglm</strong><dd>Functions for inference in generalised linear spatial models. 
<br><dt><strong>geomapdata</strong><dd>Data for topographic and geologic mapping. 
<br><dt><strong>geometry</strong><dd>Mesh generation and surface tesselation, based on the Qhull library. 
<br><dt><strong>geozoo</strong><dd>Definition of geometric objects and display via <strong>rggobi</strong>. 
<br><dt><strong>getopt</strong><dd>C-like getopt behavior for R scripts. 
<br><dt><strong>ggm</strong><dd>Functions for defining directed acyclic graphs and undirected graphs,
finding induced graphs and fitting Gaussian Markov models. 
<br><dt><strong>ggplot</strong><dd>Grammar of graphics based plots for R. 
<br><dt><strong>ggplot2</strong><dd>An implementation of the Grammar of Graphics in R. 
<br><dt><strong>ghyp</strong><dd>Univariate and multivariate generalized hyperbolic distributions. 
<br><dt><strong>giRaph</strong><dd>Data structures and algorithms for computations on graphs. 
<br><dt><strong>gibbs.met</strong><dd>Naive Gibbs sampling with Metropolis steps. 
<br><dt><strong>glasso</strong><dd>Graphical lasso. 
<br><dt><strong>gld</strong><dd>Basic functions for the generalised (Tukey) lambda distribution. 
<br><dt><strong>gllm</strong><dd>Routines for log-linear models of incomplete contingency tables,
including some latent class models via EM and Fisher scoring approaches. 
<br><dt><strong>glmc</strong><dd>Fitting Generalized Linear Models subject to Constraints. 
<br><dt><strong>glmmAK</strong><dd>Generalized Linear Mixed Models. 
<br><dt><strong>glmmML</strong><dd>A Maximum Likelihood approach to generalized linear models with random
intercept. 
<br><dt><strong>glmnet</strong><dd>Lasso and elastic-net regularized generalized linear models. 
<br><dt><strong>glmpath</strong><dd>L1 regularization path for Generalized Linear Models. 
<br><dt><strong>glpk</strong><dd>Interface to the <acronym>GNU</acronym> Linear Programming Kit (GLPK). 
<br><dt><strong>gmaps</strong><dd>Wrapper and auxiliary functions for the <strong>maps</strong> package to work
with the grid graphics system. 
<br><dt><strong>gmodels</strong><dd>Various functions to manipulate models. 
<br><dt><strong>gmp</strong><dd>Arithmetic &ldquo;without limitations&rdquo; using the <acronym>GNU</acronym> Multiple
Precision library. 
<br><dt><strong>gmt</strong><dd>Interface between the GMT 4.0 map-making software and R. 
<br><dt><strong>gnm</strong><dd>Functions to specify and fit generalized nonlinear models, including
models with multiplicative interaction terms such as the UNIDIFF model
from sociology and the AMMI model from crop science. 
<br><dt><strong>goalprog</strong><dd>Weighted and lexicographical goal programming and optimization. 
<br><dt><strong>gpclib</strong><dd>General polygon clipping routines for R based on Alan Murta's C
library. 
<br><dt><strong>gpls</strong><dd>Classification using generalized partial least squares for two-group and
multi-group (more than 2 group) classification. 
<br><dt><strong>gplots</strong><dd>Various functions to draw plots. 
<br><dt><strong>grImport</strong><dd>Importing vector graphics. 
<br><dt><strong>grade</strong><dd>Binary grading functions. 
<br><dt><strong>granova</strong><dd>Graphical Analysis of Variance. 
<br><dt><strong>graph</strong><dd>Handling of graph data structures. 
<br><dt><strong>grasp</strong><dd>Generalized Regression Analysis and Spatial Predictions for R. 
<br><dt><strong>gregmisc</strong><dd>Miscellaneous functions written/maintained by Gregory R. Warnes. 
<br><dt><strong>gridBase</strong><dd>Integration of base and grid graphics. 
<br><dt><strong>grnnR</strong><dd>A Generalized Regression Neural Network. 
<br><dt><strong>grouped</strong><dd>Regression models for grouped and coarse data, under the Coarsened At
Random assumption. 
<br><dt><strong>grplasso</strong><dd>Fit user specified models with group lasso penalty. 
<br><dt><strong>gsarima</strong><dd>functions for Generalized SARIMA time series simulation. 
<br><dt><strong>gsl</strong><dd>Wrapper for special functions of the Gnu Scientific Library (GSL). 
<br><dt><strong>gss</strong><dd>A comprehensive package for structural multivariate function estimation
using smoothing splines. 
<br><dt><strong>gstat</strong><dd>multivariable geostatistical modelling, prediction and simulation. 
Includes code for variogram modelling; simple, ordinary and universal
point or block (co)kriging, sequential Gaussian or indicator
(co)simulation, and map plotting functions. 
<br><dt><strong>gsubfn</strong><dd>Miscellaneous string utilities. 
<!-- <COMMENT> -->
<!-- Moved to Archive on 2008-03-04 as requested by the maintainer. -->
<!-- @item gtkDevice -->
<!-- GTK graphics device driver that may be used independently of the R-GNOME -->
<!-- interface and can be used to create R devices as embedded components in -->
<!-- a @acronym{GUI} using a Gtk drawing area widget, e.g., using RGtk. -->
<!-- </COMMENT> -->
<br><dt><strong>gtools</strong><dd>Various functions to help manipulate data. 
<br><dt><strong>gumbel</strong><dd>Functions for the Gumbel-Hougaard copula. 
<br><dt><strong>gvlma</strong><dd>Global Validation of Linear Models Assumptions. 
<br><dt><strong>hapassoc</strong><dd>Likelihood inference of trait associations with SNP haplotypes and other
attributes using the EM Algorithm. 
<br><dt><strong>haplo.ccs</strong><dd>Estimate haplotype relative risks in case-control data. 
<!-- <COMMENT> -->
<!-- Moved to Archive on 2005-10-05 as requested by the maintainer. -->
<!-- @item haplo.score -->
<!-- Score tests for association of traits with haplotypes when linkage phase -->
<!-- is ambiguous. -->
<!-- </COMMENT> -->
<br><dt><strong>haplo.stats</strong><dd>Statistical analysis of haplotypes with traits and covariates when
linkage phase is ambiguous. 
<br><dt><strong>hapsim</strong><dd>Haplotype data simulation. 
<br><dt><strong>hbim</strong><dd>Hill/Bliss Independence Model for combination vaccines. 
<br><dt><strong>hddplot</strong><dd>Use known groups in high-dimensional data to derive scores for plots. 
<br><dt><strong>hdeco</strong><dd>Hierarchical DECOmposition of entropy for categorical map comparisons. 
<br><dt><strong>hdf5</strong><dd>Interface to the <acronym>NCSA</acronym> HDF5 library. 
<br><dt><strong>hdrcde</strong><dd>Highest Density Regions and Conditional Density Estimation. 
<br><dt><strong>heatmap.plus</strong><dd>Heatmap with sensible behavior. 
<br><dt><strong>helloJavaWorld</strong><dd>A demonstration how to interface to a jar file that resides inside an R
package. 
<br><dt><strong>heplots</strong><dd>Visualizing tests in multivariate linear models. 
<br><dt><strong>hett</strong><dd>Functions for the fitting and summarizing of heteroscedastic
t-regression. 
<br><dt><strong>hexView</strong><dd>Viewing binary files. 
<br><dt><strong>hier.part</strong><dd>Hierarchical Partitioning: variance partition of a multivariate data
set. 
<br><dt><strong>hierfstat</strong><dd>Estimation of hierarchical F-statistics from haploid or diploid genetic
data with any numbers of levels in the hierarchy, and tests for the
significance of each F and variance components. 
<br><dt><strong>hints</strong><dd>Provide hints on what to do next. 
<br><dt><strong>hmm.discnp</strong><dd>Hidden Markov models with discrete non-parametric observation
distributions. 
<br><dt><strong>hoa</strong><dd>A bundle of packages for higher order likelihood-based inference. 
Contains <strong>cond</strong> for approximate conditional inference for
logistic and loglinear models, <strong>csampling</strong> for conditional
simulation in regression-scale models, <strong>marg</strong> for approximate
marginal inference for regression-scale models, and <strong>nlreg</strong> for
higher order inference for nonlinear heteroscedastic models. 
<br><dt><strong>homals</strong><dd>Homogeneity Analysis (HOMALS) package with optional Tcl/Tk interface. 
<br><dt><strong>homtest</strong><dd>Homogeneity tests for regional frequency analysis. 
<br><dt><strong>hopach</strong><dd>Hierarchical Ordered Partitioning and Collapsing Hybrid (HOPACH). 
<br><dt><strong>hot</strong><dd>Computation on micro-arrays. 
<br><dt><strong>howmany</strong><dd>A lower bound for the number of correct rejections. 
<br><dt><strong>hsmm</strong><dd>Hidden Semi Markov Models. 
<br><dt><strong>httpRequest</strong><dd>Implements HTTP Request protocols (GET, POST, and multipart POST
requests). 
<br><dt><strong>hwde</strong><dd>Models and tests for departure from Hardy-Weinberg equilibrium and
independence between loci. 
<br><dt><strong>hwriter</strong><dd>Easy-to-use and versatile functions to output R objects in <acronym>HTML</acronym>
format. 
<br><dt><strong>hybridHclust</strong><dd>Hybrid hierarchical clustering via mutual clusters. 
<br><dt><strong>hydrogeo</strong><dd>Groundwater data presentation and interpretation. 
<br><dt><strong>hydrosanity</strong><dd>Graphical user interface for exploring hydrological time series. 
<br><dt><strong>hypergeo</strong><dd>The hypergeometric function over the whole complex plane. 
<br><dt><strong>ibdreg</strong><dd>Regression methods for IBD linkage with covariates. 
<br><dt><strong>ic.infer</strong><dd>Inequality constrained inference in linear normal situations. 
<br><dt><strong>ic50</strong><dd>Evaluation of compound screens. 
<br><dt><strong>icomp</strong><dd>Calculates the ICOMP criterion and its variations. 
<br><dt><strong>identity</strong><dd>Jacquard condensed coefficients of identity. 
<br><dt><strong>ifa</strong><dd>Independent Factor Analysis. 
<br><dt><strong>ifs</strong><dd>Iterated Function Systems distribution function estimator. 
<br><dt><strong>ifultools</strong><dd>Insightful research tools. 
<br><dt><strong>ig</strong><dd>Robust and classical versions of the inverse Gaussian distribution. 
<br><dt><strong>igraph</strong><dd>Routines for simple graphs. 
<br><dt><strong>iid.test</strong><dd>Testing whether data is independent and identically distributed. 
<br><dt><strong>impute</strong><dd>Imputation for microarray data (currently KNN only). 
<br><dt><strong>ineq</strong><dd>Inequality, concentration and poverty measures, and Lorenz curves
(empirical and theoretic). 
<br><dt><strong>inetwork</strong><dd>Network analysis and plotting. 
<br><dt><strong>inline</strong><dd>Inline C/C++ function calls from R. 
<br><dt><strong>intcox</strong><dd>Implementation of the Iterated Convex Minorant Algorithm for the Cox
proportional hazard model for interval censored event data. 
<!-- @item integrate -->
<!-- Adaptive quadrature in up to 20 dimensions. -->
<br><dt><strong>iplots</strong><dd>Interactive graphics for R. 
<br><dt><strong>ipred</strong><dd>Improved predictive models by direct and indirect bootstrap aggregation
in classification and regression as well as resampling based estimators
of prediction error. 
<br><dt><strong>irr</strong><dd>Coefficients of Interrater Reliability and Agreement for quantitative,
ordinal and nominal data. 
<br><dt><strong>irtoys</strong><dd>Simple interface to the estimation and plotting of IRT models. 
<br><dt><strong>ismev</strong><dd>Functions to support the computations carried out in &ldquo;An Introduction
to Statistical Modeling of Extreme Values;' by S. Coles, 2001, Springer. 
The functions may be divided into the following groups; maxima/minima,
order statistics, peaks over thresholds and point processes. 
<br><dt><strong>its</strong><dd>An S4 class for handling irregular time series. 
<br><dt><strong>ivivc</strong><dd>In vitro in vivo correlation (IVIVC) modeling. 
<br><dt><strong>jit</strong><dd>Just-in-time compiler. 
<br><dt><strong>kappalab</strong><dd>The &ldquo;laboratory for capacities&rdquo;, an S4 tool box for capacity (or
non-additive measure, fuzzy measure) and integral manipulation on a
finite setting. 
<br><dt><strong>kerfdr</strong><dd>Semi-parametric kernel-based approach to local fdr estimations. 
<br><dt><strong>kernelPop</strong><dd>Spatially explicit population genetic simulations. 
<br><dt><strong>kernlab</strong><dd>Kernel-based machine learning methods including support vector machines. 
<br><dt><strong>kin.cohort</strong><dd>Analysis of kin-cohort studies. 
<br><dt><strong>kinship</strong><dd>Mixed-effects Cox models, sparse matrices, and modeling data from large
pedigrees. 
<br><dt><strong>kknn</strong><dd>Weighted k-nearest neighbors classification and regression. 
<br><dt><strong>klaR</strong><dd>Miscellaneous functions for classification and visualization developed
at the Department of Statistics, University of Dortmund. 
<br><dt><strong>klin</strong><dd>Linear equations with Kronecker structure. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2008-04-22 (Orphaned/2.6.0). -->
<!-- @item km.ci -->
<!-- Confidence intervals for the Kaplan-Meier estimator. -->
<!-- </COMMENT> -->
<br><dt><strong>kml</strong><dd>K-Means for Longitudinal data. 
<br><dt><strong>knnFinder</strong><dd>Fast nearest neighbor search. 
<br><dt><strong>knnTree</strong><dd>Construct or predict with k-nearest-neighbor classifiers, using
cross-validation to select k, choose variables (by forward or
backwards selection), and choose scaling (from among no scaling, scaling
each column by its SD, or scaling each column by its MAD).  The finished
classifier will consist of a classification tree with one such
k-nn classifier in each leaf. 
<br><dt><strong>knncat</strong><dd>Nearest-neighbor classification with categorical variables. 
<br><dt><strong>knnflex</strong><dd>A more flexible k-NN. 
<br><dt><strong>knorm</strong><dd>Microarray data from multiple biologically interrelated experiments. 
<br><dt><strong>kohonen</strong><dd>Supervised and unsupervised self-organising maps. 
<br><dt><strong>ks</strong><dd>Kernel smoothing: bandwidth matrices for kernel density estimators and
kernel discriminant analysis for bivariate data. 
<br><dt><strong>kza</strong><dd>Kolmogorov-Zurbenko Adpative filter for locating change points in a time
series. 
<br><dt><strong>kzft</strong><dd>Kolmogorov-Zurbenko Fourier Transform and application. 
<br><dt><strong>kzs</strong><dd>Kolmogorov-Zurbenko Spline. 
<br><dt><strong>labdsv</strong><dd>Laboratory for Dynamic Synthetic Vegephenomenology. 
<br><dt><strong>labstatR</strong><dd>Functions for the book &ldquo;Laboratorio di statistica con R&rdquo; by
S. M. Iacus and G. Masarotto, 2002, McGraw-Hill.  Function names and
documentation in Italian. 
<br><dt><strong>laercio</strong><dd>Tests to compare means. 
<br><dt><strong>lago</strong><dd>LAGO for rare target detection. 
<br><dt><strong>lancet.iraqmortality</strong><dd>Surveys of Iraq mortality published in The Lancet. 
<br><dt><strong>languageR</strong><dd>Data sets and functions for the book &ldquo;Analyzing Linguistic Data: A
practical introduction to statistics&rdquo; by R. H. Baayen, 2007, Cambridge:
Cambridge University Press. 
<br><dt><strong>lars</strong><dd>Least Angle Regression, Lasso and Forward Stagewise: efficient
procedures for fitting an entire lasso sequence with the cost of a
single least squares fit. 
<br><dt><strong>laser</strong><dd>Likelihood Analysis of Speciation/Extinction Rates from phylogenies. 
<br><dt><strong>lasso2</strong><dd>Routines and documentation for solving regression problems while
imposing an L1 constraint on the estimates, based on the algorithm of
Osborne et al. (1998). 
<br><dt><strong>latentnet</strong><dd>Latent position and cluster models for statistical networks. 
<br><dt><strong>latentnetHRT</strong><dd>Latent position and cluster models for statistical networks,
implementing the original specification by Handcock, Raftery and Tantrum. 
<br><dt><strong>lattice</strong><dd>Lattice graphics, an implementation of Trellis Graphics functions. 
<em>Recommended</em>. 
<br><dt><strong>latticeExtra</strong><dd>Generic functions and standard methods for Trellis-based displays. 
<br><dt><strong>lawstat</strong><dd>Statistical tests widely utilized in biostatistics, public policy and
law. 
<br><dt><strong>lazy</strong><dd>Lazy learning for local regression. 
<br><dt><strong>ldDesign</strong><dd>Design of experiments for detection of linkage disequilibrium,
<br><dt><strong>lda.cv</strong><dd>Cross-validation for linear discriminant analysis. 
<br><dt><strong>ldbounds</strong><dd>Lan-DeMets method for group sequential boundaries. 
<br><dt><strong>leaps</strong><dd>A package which performs an exhaustive search for the best subsets of a
given set of potential regressors, using a branch-and-bound algorithm,
and also performs searches using a number of less time-consuming
techniques. 
<br><dt><strong>lga</strong><dd>Tools for Linear Grouping Analysis (LGA). 
<br><dt><strong>lgtdl</strong><dd>A set of methods for longitudinal data objects. 
<br><dt><strong>lhs</strong><dd>Latin Hypercube Samples. 
<br><dt><strong>limSolve</strong><dd>Solving linear inverse models. 
<!-- <COMMENT> -->
<!-- Moved to Archive on 2007-10-03 as requested by the maintainer. -->
<!-- @item limma -->
<!-- LInear Models for MicroArray data. -->
<!-- </COMMENT> -->
<br><dt><strong>linprog</strong><dd>Solve linear programming/linear optimization problems by using the
simplex algorithm. 
<br><dt><strong>ljr</strong><dd>Logistic Joinpoint Regression. 
<br><dt><strong>lme4</strong><dd>Fit linear and generalized linear mixed-effects models. 
<br><dt><strong>lmeSplines</strong><dd>Fit smoothing spline terms in Gaussian linear and nonlinear
mixed-effects models. 
<br><dt><strong>lmm</strong><dd>Linear mixed models. 
<br><dt><strong>lmom</strong><dd>L-moments. 
<br><dt><strong>lmomco</strong><dd>L-moments and L-comoments. 
<br><dt><strong>lmtest</strong><dd>A collection of tests on the assumptions of linear regression models
from the book &ldquo;The linear regression model under test&rdquo; by W. Kraemer
and H. Sonnberger, 1986, Physica. 
<br><dt><strong>lnMLE</strong><dd>Marginally specified logistic normal models for longitudinal binary
data. 
<br><dt><strong>locfdr</strong><dd>Computation of local false discovery rates. 
<br><dt><strong>locfit</strong><dd>Local Regression, likelihood and density estimation. 
<br><dt><strong>locpol</strong><dd>Kernel local polynomial regression. 
<br><dt><strong>lodplot</strong><dd>Assorted plots of location score versus genetic map position. 
<br><dt><strong>logcondens</strong><dd>Estimate a log-concave probability density from i.i.d. observations. 
<br><dt><strong>logilasso</strong><dd>Analysis of sparse contingency tables with penalization approaches. 
<br><dt><strong>logistf</strong><dd>Firth's bias reduced logistic regression approach with penalized profile
likelihood based confidence intervals for parameter estimates. 
<br><dt><strong>loglognorm</strong><dd>Double log normal distribution functions. 
<br><dt><strong>logregperm</strong><dd>Inference in Logistic Regression using permutation tests. 
<br><dt><strong>logspline</strong><dd>Logspline density estimation. 
<br><dt><strong>lokern</strong><dd>Kernel regression smoothing with adaptive local or global plug-in
bandwidth selection. 
<br><dt><strong>longmemo</strong><dd>Datasets and Functionality from the textbook &ldquo;Statistics for
Long-Memory Processes&rdquo; by J. Beran, 1994, Chapman &amp; Hall. 
<br><dt><strong>lpSolve</strong><dd>Functions that solve general linear/integer problems, assignment
problems, and transportation problems via interfacing Lp_solve. 
<br><dt><strong>lpSolveAPI</strong><dd>An R interface to the lp_solve library API. 
<br><dt><strong>lpc</strong><dd>Lassoed principal components for testing significance of features. 
<br><dt><strong>lpridge</strong><dd>Local polynomial (ridge) regression. 
<br><dt><strong>lsa</strong><dd>Latent Semantic Analysis. 
<br><dt><strong>lspls</strong><dd>LS-PLS (least squares &mdash; partial least squares) models. 
<br><dt><strong>lss</strong><dd>Accelerated failure time model to right censored data based on
least-squares principle. 
<br><dt><strong>ltm</strong><dd>Analysis of multivariate Bernoulli data using latent trait models
(including the Rasch model) under the Item Response Theory approach. 
<br><dt><strong>ltsa</strong><dd>Linear Time Series Analysis. 
<br><dt><strong>luca</strong><dd>Likelihood Under Covariate Assumptions (LUCA). 
<br><dt><strong>lvplot</strong><dd>Letter-value box plots. 
<br><dt><strong>mAr</strong><dd>Estimation of multivariate AR models through a computationally efficient
stepwise least-squares algorithm. 
<br><dt><strong>mFilter</strong><dd>Miscellenous time series filters. 
<br><dt><strong>maanova</strong><dd>Analysis of N-dye Micro Array experiments using mixed model
effect.  Contains anlysis of variance, permutation and bootstrap,
cluster and consensus tree. 
<br><dt><strong>magic</strong><dd>A variety of methods for creating magic squares of any order greater
than 2, and various magic hypercubes. 
<br><dt><strong>mapLD</strong><dd>Linkage Disequilibrium mapping. 
<br><dt><strong>mapdata</strong><dd>Supplement to package <strong>maps</strong>, providing the larger and/or
higher-resolution databases. 
<br><dt><strong>mapproj</strong><dd>Map Projections: converts latitude/longitude into projected coordinates. 
<br><dt><strong>maps</strong><dd>Draw geographical maps.  Projection code and larger maps are in separate
packages. 
<br><dt><strong>maptools</strong><dd>Set of tools for manipulating and reading geographic data, in particular
ESRI shapefiles. 
<br><dt><strong>maptree</strong><dd>Functions with example data for graphing and mapping models from
hierarchical clustering and classification and regression trees. 
<br><dt><strong>marginTree</strong><dd>Margin trees for high-dimensional classification. 
<br><dt><strong>marginalmodelplots</strong><dd>Marginal model plots for linear and generalized linear models. 
<br><dt><strong>mathgraph</strong><dd>Tools for constructing and manipulating objects from a class of directed
and undirected graphs. 
<br><dt><strong>matlab</strong><dd>Emulate MATLAB code using R. 
<br><dt><strong>matrixcalc</strong><dd>Collection of functions for matrix differential calculus. 
<br><dt><strong>maxLik</strong><dd>Tools for Maximum Likelihood Estimation. 
<br><dt><strong>maxstat</strong><dd>Maximally selected rank and Gauss statistics with several p-value
approximations. 
<br><dt><strong>mblm</strong><dd>Median-based Linear models, using Theil-Sen single or Siegel repeated
medians. 
<br><dt><strong>mboost</strong><dd>Gradient boosting for fitting generalized linear, additive and
interaction models. 
<br><dt><strong>mcgibbsit</strong><dd>Warnes and Raftery's MCGibbsit MCMC diagnostic. 
<br><dt><strong>mclust</strong><dd>Model-based clustering and normal mixture modeling including Bayesian
regularization. 
<!-- <COMMENT> -->
<!-- Moved to Archive on 2004-02-19 as requested by the maintainer. -->
<!-- @item mclust1998 -->
<!-- Model-based cluster analysis: the 1998 version of MCLUST. -->
<!-- </COMMENT> -->
<br><dt><strong>mclust02</strong><dd>Model-based cluster analysis: the 2002 version of MCLUST. 
<br><dt><strong>mcmc</strong><dd>Functions for Markov Chain Monte Carlo (MCMC). 
<br><dt><strong>mco</strong><dd>Multi criteria optimization algorithms. 
<br><dt><strong>mda</strong><dd>Code for mixture discriminant analysis (MDA), flexible discriminant
analysis (FDA), penalized discriminant analysis (PDA), multivariate
additive regression splines (MARS), adaptive back-fitting splines
(BRUTO), and penalized regression. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2005-10-05 (Orphaned/2.2.0). -->
<!-- @item meanscore -->
<!-- Mean Score method for missing covariate data in logistic regression -->
<!-- models. -->
<!-- </COMMENT> -->
<br><dt><strong>meboot</strong><dd>Maximum entropy bootstrap for time series. 
<br><dt><strong>mefa</strong><dd>Faunistic count data handling and reporting. 
<br><dt><strong>meifly</strong><dd>Interactive model exploration using GGobi. 
<br><dt><strong>memisc</strong><dd>Miscellaneous Tools for data management, simulation, and presentation of
estimates. 
<br><dt><strong>merror</strong><dd>Accuracy and precision of measurements. 
<br><dt><strong>meta</strong><dd>Fixed and random effects meta-analysis, with functions for tests of
bias, forest and funnel plot. 
<br><dt><strong>mfp</strong><dd>Multiple Fractional Polynomials. 
<br><dt><strong>mgcv</strong><dd>Routines for GAMs and other genralized ridge regression problems with
multiple smoothing parameter selection by GCV or UBRE. 
<em>Recommended</em>. 
<!-- <COMMENT> -->
<!-- Removed on 2006-10-11 as requested by the maintainer. -->
<!-- @item micEcdat -->
<!-- Data sets for micro-econometrics. -->
<!-- </COMMENT> -->
<br><dt><strong>micEcon</strong><dd>Tools for microeconomic analysis and microeconomic modelling. 
<br><dt><strong>mice</strong><dd>Multivariate Imputation by Chained Equations. 
<br><dt><strong>mimR</strong><dd>An R interface to MIM for graphical modeling in R. 
<br><dt><strong>minet</strong><dd>Mutual Information NETwork. 
<br><dt><strong>minpack.lm</strong><dd>R interface for two functions from the MINPACK least squares
optimization library, solving the nonlinear least squares problem by a
modification of the Levenberg-Marquardt algorithm. 
<br><dt><strong>misc3d</strong><dd>A collection of miscellaneous 3d plots, including rgl-based isosurfaces. 
<br><dt><strong>mitools</strong><dd>Tools to perform analyses and combine results from multiple-imputation
datasets. 
<br><dt><strong>mix</strong><dd>Estimation/multiple imputation programs for mixed categorical and
continuous data. 
<br><dt><strong>mixPHM</strong><dd>Mixtures of proportional hazard models. 
<br><dt><strong>mixdist</strong><dd>Finite mixture distribution models. 
<br><dt><strong>mixreg</strong><dd>Functions to fit mixtures of regressions. 
<br><dt><strong>mixstock</strong><dd>Mixed stock analysis functions. 
<br><dt><strong>mixtools</strong><dd>Tools for mixture models. 
<br><dt><strong>mlCopulaSelection</strong><dd>Copula selection and fitting using maximum likelihood. 
<br><dt><strong>mlbench</strong><dd>A collection of artificial and real-world machine learning benchmark
problems, including the Boston housing data. 
<br><dt><strong>mlegp</strong><dd>Maximum Likelihood Estimates of Gaussian Processes. 
<br><dt><strong>mlica</strong><dd>Independent Component Analysis using Maximum Likelihood. 
<br><dt><strong>mlmRev</strong><dd>Examples from Multilevel Modelling Software Review. 
<br><dt><strong>mlmmm</strong><dd>Maximum likelihood estimation under multivariate linear mixed models
with missing values. 
<br><dt><strong>mlogit</strong><dd>Estimation of the multinomial logit model with choice specific
variables. 
<br><dt><strong>mmlcr</strong><dd>Mixed-mode latent class regression (also known as mixed-mode mixture
model regression or mixed-mode mixture regression models) which can
handle both longitudinal and one-time responses. 
<br><dt><strong>mnormt</strong><dd>The multivariate normal and t distributions. 
<br><dt><strong>moc</strong><dd>Fits a variety of mixtures models for multivariate observations with
user-difined distributions and curves. 
<br><dt><strong>modeest</strong><dd>Mode estimation and Chernoff distribution. 
<br><dt><strong>modehunt</strong><dd>Multiscale analysis for density functions. 
<br><dt><strong>modeltools</strong><dd>A collection of tools to deal with statistical models. 
<br><dt><strong>moduleColor</strong><dd>Methods for color labeling, calculation of eigengenes, and merging of
closely related modules. 
<br><dt><strong>mokken</strong><dd>Mokken Scale Analysis for test and questionnaire data. 
<br><dt><strong>mombf</strong><dd>Moment and inverse moment Bayes factors. 
<br><dt><strong>moments</strong><dd>Moments, skewness, kurtosis and related tests. 
<br><dt><strong>monomvn</strong><dd>Estimation for multivariate normal data with monotone missingness. 
<br><dt><strong>monreg</strong><dd>Estimation of monotone regression and variance functions in
nonparametric models. 
<br><dt><strong>monoProc</strong><dd>Strictly monotone smoothing procedure. 
<br><dt><strong>moonsun</strong><dd>Basic astronomical calculations. 
<br><dt><strong>mota</strong><dd>Mean Optimal Transformation Approach. 
<br><dt><strong>mpm</strong><dd>Spectral map analysis. 
<br><dt><strong>mprobit</strong><dd>Multivariate probit model for binary/ordinal response. 
<br><dt><strong>mra</strong><dd>Analysis of capture-recapture data. 
<br><dt><strong>mratios</strong><dd>Inferences for ratios of coefficients in the general linear model. 
<br><dt><strong>mrdrc</strong><dd>Model-robust concentration-response analysis. 
<br><dt><strong>msBreast</strong><dd>Protein mass spectra dataset from a breast cancer study. 
<br><dt><strong>msDilution</strong><dd>Protein mass spectra dataset from a dilution experiment. 
<br><dt><strong>msProcess</strong><dd>Tools for protein mass spectra processing including data preparation,
denoising, noise estimation, baseline correction, intensity
normalization, peak detection, peak alignment, peak quantification, and
various functionalities for data ingestion/conversion, mass calibration,
data quality assessment, and protein mass spectra simulation. 
<br><dt><strong>msProstate</strong><dd>Protein mass spectra dataset from a prostate cancer study. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2005-07-27 as requested by Witold Eryk Wolski -->
<!-- <W.E.Wolski@ncl.ac.uk> (package now in BioC). -->
<!-- @item mscalib -->
<!-- Calibration and filtering of MALDI-TOF Peptide Mass Fingerprint data. -->
<!-- </COMMENT> -->
<br><dt><strong>msm</strong><dd>Functions for fitting continuous-time Markov multi-state models to
categorical processes observed at arbitrary times, optionally with
misclassified responses, and covariates on transition or
misclassification rates. 
<br><dt><strong>muS2RC</strong><dd>S-plus to R Compatibility for package <strong>muStat</strong>. 
<br><dt><strong>muStat</strong><dd>Prentice rank sum test and McNemar test. 
<br><dt><strong>muUtil</strong><dd>Utility functions for package <strong>muStat</strong>. 
<br><dt><strong>muhaz</strong><dd>Hazard function estimation in survival analysis. 
<br><dt><strong>multcomp</strong><dd>Multiple comparison procedures for the one-way layout. 
<br><dt><strong>multcompView</strong><dd>Visualizations of paired comparisons. 
<!-- <COMMENT> -->
<!-- Removed from CRAN on 2004-10-20 due to possible legal problems. -->
<!-- @item multidim -->
<!-- Multidimensional descriptive statistics: factorial methods and -->
<!-- classification. -->
<!-- </COMMENT> -->
<br><dt><strong>multic</strong><dd>Quantitative linkage analysis tools using the variance components
approach. 
<br><dt><strong>multilevel</strong><dd>Analysis of multilevel data by organizational and social psychologists. 
<br><dt><strong>multinomRob</strong><dd>Overdispersed multinomial regression using robust (LQD and tanh)
estimation. 
<br><dt><strong>multipol</strong><dd>Utilities to manipulate multivariate polynomials. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2004-09-30. -->
<!-- @item multiv -->
<!-- Functions for hierarchical clustering, partitioning, bond energy -->
<!-- algorithm, Sammon mapping, PCA and correspondence analysis. -->
<!-- </COMMENT> -->
<br><dt><strong>multtest</strong><dd>Resampling-based multiple hypothesis testing. 
<br><dt><strong>mvbutils</strong><dd>Utilities by Mark V. Bravington for project organization, editing and
backup, sourcing, documentation (formal and informal), package
preparation, macro functions, and more. 
<br><dt><strong>mvna</strong><dd>Nelson-Aalen estimator of the cumulative hazard in multistate models. 
<br><dt><strong>mvnmle</strong><dd>ML estimation for multivariate normal data with missing values. 
<br><dt><strong>mvnormtest</strong><dd>Generalization of the Shapiro-Wilk test for multivariate variables. 
<br><dt><strong>mvtnormpcs</strong><dd>Multivariate Normal and T Distribution functions of Dunnett
(1989). 
<br><dt><strong>mvoutlier</strong><dd>Multivariate outlier detection based on robust estimates of location and
covariance structure. 
<br><dt><strong>mvpart</strong><dd>Multivariate partitioning. 
<br><dt><strong>mvtnorm</strong><dd>Multivariate normal and t distributions. 
<br><dt><strong>nFDR</strong><dd>Nonparametric Estimate of FDR Based on Bernstein polynomials. 
<br><dt><strong>nFactors</strong><dd>Non-graphical solution to the Cattell Scree Test. 
<br><dt><strong>ncdf</strong><dd>Interface to Unidata netCDF data files. 
<br><dt><strong>ncf</strong><dd>Spatial nonparametric covariance functions. 
<br><dt><strong>ncomplete</strong><dd>Functions to perform the regression depth method (RDM) to binary
regression to approximate the minimum number of observations that can be
removed such that the reduced data set has complete separation. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2008-04-22 (Orphaned/2.6.0). -->
<!-- @item ncvar -->
<!-- High-level R interface to netCDF datasets. -->
<!-- </COMMENT> -->
<br><dt><strong>negenes</strong><dd>Estimating the number of essential genes in a genome on the basis of
data from a random transposon mutagenesis experiment, through the use of
a Gibbs sampler. 
<!-- <COMMENT> -->
<!-- Moved to the Archive as suggested by the maintainer: we now have -->
<!-- ncdf, and other CRAN packages no longer use netCDF. -->
<!-- @item netCDF -->
<!-- Read data from netCDF files. -->
<!-- </COMMENT> -->
<br><dt><strong>netmodels</strong><dd>Tools for the study of scale free and small world networks. 
<br><dt><strong>network</strong><dd>Tools to create and modify network objects, which can represent a range
of relational data types. 
<br><dt><strong>networksis</strong><dd>Simulate bipartite graphs with fixed marginals through sequential
importance sampling. 
<br><dt><strong>neural</strong><dd>RBF and MLP neural networks with graphical user interface. 
<br><dt><strong>neuralnet</strong><dd>Training of neural networks. 
<!-- <COMMENT> -->
<!-- Renamed to allelic on 2006-05-24 as requested by the maintainer. -->
<!-- @item newallelic -->
<!-- A fast, unbiased and exact allelic exact test. -->
<!-- </COMMENT> -->
<br><dt><strong>nice</strong><dd>Get or set UNIX priority (niceness) of running R process. 
<br><dt><strong>nlme</strong><dd>Fit and compare Gaussian linear and nonlinear mixed-effects models. 
<em>Recommended</em>. 
<br><dt><strong>nlmeODE</strong><dd>Combine the <strong>nlme</strong> and <strong>odesolve</strong> packages for
mixed-effects modelling using differential equations. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2006-04-09 as requested by R Koenker. -->
<!-- @item nlrq -->
<!-- Nonlinear quantile regression routines.  @emph{Defunct}. -->
<!-- </COMMENT> -->
<br><dt><strong>nlrwr</strong><dd>Non-linear regression with R. 
<br><dt><strong>nls2</strong><dd>Non-linear regression with brute force. 
<br><dt><strong>nlstools</strong><dd>Tools for nonlinear regression diagnostics. 
<br><dt><strong>nlt</strong><dd>A nondecimated lifting transform for signal denoising. 
<br><dt><strong>nlts</strong><dd>(Non)linear time series analysis. 
<br><dt><strong>nltm</strong><dd>NonLinear Transformation Models for survival analysis. 
<br><dt><strong>nnet</strong><dd>Software for single hidden layer perceptrons (&ldquo;feed-forward neural
networks&rdquo;), and for multinomial log-linear models.  Contained in the
<samp><span class="file">VR</span></samp> bundle.  <em>Recommended</em>. 
<br><dt><strong>nnls</strong><dd>The Lawson-Hanson NNLS algorithm for non-negative least squares. 
<br><dt><strong>noia</strong><dd>Implementation of the Natural and Orthogonal InterAction (NOIA) model. 
<br><dt><strong>nonbinROC</strong><dd>ROC-type analysis for non-binary gold standards. 
<br><dt><strong>nor1mix</strong><dd>One-dimensional normal mixture models classes, for, e.g., density
estimation or clustering algorithms research and teaching; providing the
widely used Marron-Wand densities. 
<br><dt><strong>norm</strong><dd>Analysis of multivariate normal datasets with missing values. 
<br><dt><strong>normwm.test</strong><dd>Normality and white noise testing. 
<br><dt><strong>normalp</strong><dd>A collection of utilities for normal of order p distributions
(General Error Distributions). 
<br><dt><strong>nortest</strong><dd>Five omnibus tests for the composite hypothesis of normality. 
<br><dt><strong>noverlap</strong><dd>Functions to perform the regression depth method (RDM) to binary
regression to approximate the amount of overlap, i.e., the minimal
number of observations that need to be removed such that the reduced
data set has no longer overlap. 
<br><dt><strong>np</strong><dd>Nonparametric kernel smoothing methods for mixed datatypes. 
<br><dt><strong>nparcomp</strong><dd>Nonparametric relative contrast effects. 
<br><dt><strong>npde</strong><dd>Normalized prediction distribution errors for nonlinear mixed-effect
models. 
<br><dt><strong>nplplot</strong><dd>Plotting non-parametric LOD scores from multiple input files. 
<br><dt><strong>npmc</strong><dd>Nonparametric Multiple Comparisons:  provides simultaneous rank test
procedures for the one-way layout without presuming a certain
distribution. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2006-04-09 as requested by R Koenker. -->
<!-- @item nprq -->
<!-- Nonparametric quantile regression.  @emph{Defunct}. -->
<!-- </COMMENT> -->
<br><dt><strong>nsRFA</strong><dd>Non-supervised Regional Frequency Analysis. 
<br><dt><strong>numDeriv</strong><dd>Accurate numerical derivatives. 
<br><dt><strong>nws</strong><dd>Functions for NetWorkSpaces and Sleigh. 
<br><dt><strong>obsSens</strong><dd>Sensitivity analysis for observational studies. 
<br><dt><strong>oc</strong><dd>Optimal Classification roll call analysis. 
<br><dt><strong>oce</strong><dd>Analysis of oceanographic data. 
<br><dt><strong>odesolve</strong><dd>An interface for the Ordinary Differential Equation (ODE) solver lsoda. 
ODEs are expressed as R functions. 
<br><dt><strong>odfWeave</strong><dd>Sweave processing of Open Document Format (ODF) files. 
<br><dt><strong>ofw</strong><dd>Optimal Feature Weighting algorithm. 
<br><dt><strong>onemap</strong><dd>Analysis of molecular marker data from non-model systems to
simultaneously estimate linkage and linkage phases (genetic map
construction). 
<br><dt><strong>onion</strong><dd>A collection of routines to manipulate and visualize quaternions and
octonions. 
<br><dt><strong>openNLP</strong><dd>An interface to <a href="http://opennlp.sourceforge.net/">openNLP</a>, a
collection of natural language processing tools including a sentence
detector, tokenizer, pos-tagger, shallow and full syntactic parser, and
named-entity detector, using the <a href="http://maxent.sourceforge.net/">Maxent</a> Java package for training and using maximum entropy models. 
<br><dt><strong>openNLPmodels</strong><dd>English and spanish models for <strong>openNLP</strong>. 
<br><dt><strong>optmatch</strong><dd>Functions to perform optimal matching, particularly full matching. 
<br><dt><strong>orientlib</strong><dd>Representations, conversions and display of orientation SO(3) data. 
<br><dt><strong>orloca</strong><dd>Operations Research LOCational Analysis models. 
<br><dt><strong>orloca.es</strong><dd>Spanish version of <strong>orloca</strong> package. 
<br><dt><strong>orth</strong><dd>Multivariate logistic regressions using orthogonalized residuals. 
<br><dt><strong>orthopolynom</strong><dd>Functions for orthogonal and orthonormal polynomials. 
<br><dt><strong>ouch</strong><dd>Ornstein-Uhlenbeck models for phylogenetic comparative hypotheses. 
<br><dt><strong>outliers</strong><dd>A collection of some tests commonly used for identifying outliers. 
<br><dt><strong>oz</strong><dd>Functions for plotting Australia's coastline and state boundaries. 
<br><dt><strong>pARtial</strong><dd>(Partial) attributable risk estimates, corresponding variance estimates
and confidence intervals. 
<br><dt><strong>pack</strong><dd>Create and manipulate raw vectors. 
<br><dt><strong>pairwiseCI</strong><dd>Calculate and plot unadjusted confidence intervals for two sample
comparisons. 
<br><dt><strong>paleoTS</strong><dd>Modeling evolution in paleontological time-series. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2008-05-25 as requested by the maintainer. -->
<!-- (Functionality now integrated in paleoTS.) -->
<!-- @item paleoTSalt -->
<!-- Modeling evolution in paleontological time-series (alternate -->
<!-- paramaterization). -->
<!-- </COMMENT> -->
<br><dt><strong>paltran</strong><dd>Functions for paleolimnology. 
<br><dt><strong>pamr</strong><dd>Pam: Prediction Analysis for Microarrays. 
<br><dt><strong>pan</strong><dd>Multiple imputation for multivariate panel or clustered data. 
<br><dt><strong>panel</strong><dd>Functions and datasets for fitting models to Panel data. 
<br><dt><strong>papply</strong><dd>Parallel apply function using MPI. 
<br><dt><strong>paran</strong><dd>Horn's test of principal components/factors. 
<br><dt><strong>partitions</strong><dd>Additive partitions of integers. 
<br><dt><strong>partsm</strong><dd>Periodic AutoRegressive Time Series Models. 
<br><dt><strong>party</strong><dd>Unbiased recursive partitioning in a conditional inference framework. 
<br><dt><strong>pastecs</strong><dd>Package for Analysis of Space-Time Ecological Series. 
<br><dt><strong>pbatR</strong><dd>Frontend to PBAT to run within R. 
<br><dt><strong>pcaPP</strong><dd>Robust PCA by Projection Pursuit. 
<br><dt><strong>pcalg</strong><dd>Standard and robust estimation of the skeleton (ugraph) of a Directed
Acyclic Graph (DAG) via the PC algorithm. 
<br><dt><strong>pcse</strong><dd>Panel-Corrected Standard Error estimation. 
<br><dt><strong>pcurve</strong><dd>Fits a principal curve to a numeric multivariate dataset in arbitrary
dimensions.  Produces diagnostic plots.  Also calculates Bray-Curtis and
other distance matrices and performs multi-dimensional scaling and
principal component analyses. 
<br><dt><strong>pear</strong><dd>Periodic Autoregression Analysis. 
<br><dt><strong>pec</strong><dd>Prediction Error Curves for survival models. 
<br><dt><strong>penalized</strong><dd>Penalized estimation in generalized linear models. 
<br><dt><strong>peperr</strong><dd>Parallelised Estimation of Prediction ERRor. 
<br><dt><strong>permax</strong><dd>Functions intended to facilitate certain basic analyses of DNA array
data, especially with regard to comparing expression levels between two
types of tissue. 
<br><dt><strong>permtest</strong><dd>Permutation test to compare variability within and distance between two
groups. 
<br><dt><strong>perturb</strong><dd>Perturbation analysis for evaluating collinearity. 
<br><dt><strong>pga</strong><dd>An ensemble method for variable selection by carrying out Darwinian
evolution in parallel universes. 
<br><dt><strong>pgam</strong><dd>Poisson-Gamma Additive Models. 
<br><dt><strong>pgirmess</strong><dd>Functions for analysis and display of ecological and spatial data. 
<br><dt><strong>phangorn</strong><dd>Phylogenetic analysis in R. 
<br><dt><strong>pheno</strong><dd>Some easy-to-use functions for time series analyses of (plant-)
phenological data sets. 
<br><dt><strong>phpSerialize</strong><dd>Serialize R to PHP associative array. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2005-10-05 (Orphaned/2.2.0). -->
<!-- @item phyloarray -->
<!-- Software to process data from phylogenetic or identification -->
<!-- microarrays. -->
<!-- </COMMENT> -->
<br><dt><strong>picante</strong><dd>Tools for integrating phylogenies and ecology. 
<br><dt><strong>pinktoe</strong><dd>Converts S trees to <acronym>HTML</acronym>/Perl files for interactive tree traversal. 
<br><dt><strong>pixmap</strong><dd>Functions for import, export, plotting and other manipulations of
bitmapped images. 
<br><dt><strong>plRasch</strong><dd>Log linear by linear asscociation models. 
<br><dt><strong>playwith</strong><dd>A <acronym>GUI</acronym> for interactive plots using GTK+. 
<br><dt><strong>plink</strong><dd>Separate calibration linking methods. 
<br><dt><strong>plm</strong><dd>Linear models for panel data. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2008-06-19 as requested by the maintainer. -->
<!-- @item plotAndPlayGTK -->
<!-- A @acronym{GUI} for interactive plots using GTK. -->
<!-- </COMMENT> -->
<br><dt><strong>plotSEMM</strong><dd>Graphing nonlinear latent variable interactions in SEMM. 
<br><dt><strong>plotrix</strong><dd>Various useful functions for enhancing plots. 
<br><dt><strong>plugdensity</strong><dd>Kernel density estimation with global bandwidth selection via
&ldquo;plug-in&rdquo;. 
<br><dt><strong>pls</strong><dd>Partial Least Squares Regression (PLSR) and Principal Component
Regression (PCR). 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2006-05-09 as requested by the maintainer. -->
<!-- @item pls.pcr -->
<!-- Multivariate regression by PLS and PCR. -->
<!-- </COMMENT> -->
<br><dt><strong>plsgenomics</strong><dd>PLS analyses for genomics. 
<br><dt><strong>pmg</strong><dd>Poor Man's <acronym>GUI</acronym>. 
<br><dt><strong>pmml</strong><dd>Generate Predictive Modelling Markup Language (PMML) for various
models. 
<br><dt><strong>poLCA</strong><dd>POlytomous variable Latent Class Analysis. 
<br><dt><strong>poilog</strong><dd>Poisson lognormal and bivariate Poisson lognormal distribution. 
<!-- <COMMENT> -->
<!-- Outdated on CRAN along with 1.5.0 release as we do not have an active -->
<!-- maintainer to fix outstanding quality problems. -->
<!-- @item polymars -->
<!-- Polychotomous regression based on Multivariate Adaptive Regression -->
<!-- Splines. -->
<!-- </COMMENT> -->
<br><dt><strong>polspline</strong><dd>Routines for the polynomial spline fitting routines hazard regression,
hazard estimation with flexible tails, logspline, lspec, polyclass, and
polymars, by C. Kooperberg and co-authors. 
<br><dt><strong>polyapost</strong><dd>Simulating from the Polya posterior. 
<br><dt><strong>polycor</strong><dd>Polychoric and polyserial correlations. 
<br><dt><strong>polynom</strong><dd>A collection of functions to implement a class for univariate polynomial
manipulations. 
<br><dt><strong>pomp</strong><dd>Partially-observed Markov processes. 
<br><dt><strong>popbio</strong><dd>Construction and analysis of matrix population models. 
<br><dt><strong>popgen</strong><dd>Statistical and POPulation GENetics. 
<br><dt><strong>poplab</strong><dd>Population Lab, a tool for constructing a virtual electronic population
evolving over time. 
<br><dt><strong>portfolio</strong><dd>Classes for analyzing and implementing portfolios. 
<br><dt><strong>portfolioSim</strong><dd>Framework for simulating equity portfolio strategies. 
<br><dt><strong>powell</strong><dd>Optimizes a function using Powell's UObyQA algorithm. 
<br><dt><strong>powerpkg</strong><dd>Power analyses for the affected sib pair and the TDT design. 
<br><dt><strong>ppc</strong><dd>Sample classification of protein mass spectra by peak probabilty
contrasts. 
<br><dt><strong>ppls</strong><dd>Penalized Partial Least Squares. 
<br><dt><strong>pps</strong><dd>Functions to select samples using PPS (probability proportional to size)
sampling, for stratified simple random sampling, and to compute joint
inclusion probabilities for Sampford's method of PPS sampling. 
<br><dt><strong>prabclus</strong><dd>Distance based parametric bootstrap tests for clustering, mainly thought
for presence-absence data (clustering of species distribution maps). 
Jaccard and Kulczynski distance measures, clustering of MDS scores, and
nearest neighbor based noise detection. 
<br><dt><strong>predbayescor</strong><dd>Classification rule based on Bayesian naive Bayes models with feature
selection bias corrected. 
<br><dt><strong>predmixcor</strong><dd>Classification rule based on Bayesian mixture models with feature
selection bias corrected. 
<br><dt><strong>prettyR</strong><dd>Pretty descriptive stats. 
<br><dt><strong>prim</strong><dd>Patient Rule Induction Method (PRIM). 
<br><dt><strong>princurve</strong><dd>Fits a principal curve to a matrix of points in arbitrary dimension. 
<br><dt><strong>prob</strong><dd>Elementary probability on finite sample spaces. 
<br><dt><strong>prodlim</strong><dd>Product limit estimation. 
<br><dt><strong>profileModel</strong><dd>Tools for profiling inference functions for various model classes. 
<br><dt><strong>profr</strong><dd>Alternative display for profiling information. 
<br><dt><strong>proftools</strong><dd>Profile output processing tools for R. 
<br><dt><strong>proj4</strong><dd>A simple interface to the PROJ.4 cartographic projections library. 
<br><dt><strong>proptest</strong><dd>Tests of the proportional hazards assumption in the Cox model. 
<br><dt><strong>proto</strong><dd>An object oriented system using prototype or object-based (rather than
class-based) object oriented ideas. 
<br><dt><strong>proxy</strong><dd>Distance and similarity measures. 
<br><dt><strong>pscl</strong><dd>R in the Political Science Computational Laboratory, Stanford
University. 
<br><dt><strong>pseudo</strong><dd>Pseudo-observations. 
<br><dt><strong>pspline</strong><dd>Smoothing splines with penalties on order m derivatives. 
<br><dt><strong>psy</strong><dd>Various procedures used in psychometry: Kappa, ICC, Cronbach alpha,
screeplot, PCA and related methods. 
<br><dt><strong>psych</strong><dd>Procedures for personality and psychological research. 
<br><dt><strong>psychometric</strong><dd>Applied psychometric theory: functions useful for correlation theory,
meta-analysis (validity-generalization), reliability, item analysis,
inter-rater reliability, and classical utility. 
<br><dt><strong>psyphy</strong><dd>Functions for analyzing psychophysical data in R. 
<br><dt><strong>pwr</strong><dd>Basic functions for power analysis. 
<br><dt><strong>pwt</strong><dd>The Penn World Table providing purchasing power parity and national
income accounts converted to international prices for 168 countries for
some or all of the years 1950&ndash;2000. 
<br><dt><strong>pvclust</strong><dd>Hierarchical clustering with p-value. 
<br><dt><strong>qAnalyst</strong><dd>Variables and attributes control charts. 
<br><dt><strong>qcc</strong><dd>Quality Control Charts.  Shewhart quality control charts for continuous,
attribute and count data.  Cusum and EWMA charts.  Operating
characteristic curves.  Process capability analysis.  Pareto chart and
cause-and-effect chart. 
<br><dt><strong>qdg</strong><dd>Infer QTL-directed Dependency Graphs for phenotype networks. 
<br><dt><strong>qgen</strong><dd>Quantitative Genetics using R. 
<br><dt><strong>qlspack</strong><dd>Quasi least squares package. 
<br><dt><strong>qp</strong><dd>q-order partial correlation graph search algorithm. 
<br><dt><strong>qpcR</strong><dd>Modelling and analysis of real-time PCR data. 
<br><dt><strong>qtl</strong><dd>Analysis of experimental crosses to identify QTLs. 
<br><dt><strong>qtlDesign</strong><dd>Tools for the design of QTL experiments. 
<br><dt><strong>qtlbim</strong><dd>QTL Bayesian Interval Mapping. 
<br><dt><strong>quadprog</strong><dd>For solving quadratic programming problems. 
<br><dt><strong>qualV</strong><dd>Qualitative methods for the validation of models. 
<br><dt><strong>quantchem</strong><dd>Quantitative chemical analysis: calibration and evaluation of results. 
<br><dt><strong>quantmod</strong><dd>Quantitative financial modelling framework. 
<br><dt><strong>quantreg</strong><dd>Quantile regression and related methods. 
<br><dt><strong>quantregForest</strong><dd>Quantile Regression Forests, a tree-based ensemble method for estimation
of conditional quantiles. 
<br><dt><strong>qvalue</strong><dd>Q-value estimation for false discovery rate control. 
<br><dt><strong>qvcalc</strong><dd>Functions to compute quasi-variances and associated measures of
approximation error. 
<br><dt><strong>r2lUniv</strong><dd>R to LaTeX Univariate: perform basic analysis and generate
corresponding LaTeX code. 
<br><dt><strong>rJava</strong><dd>Low-level R to Java interface.  Allows creation of objects, calling
methods and accessing fields. 
<br><dt><strong>rPorta</strong><dd>An R interface to PORTA, a collection of routines for analyzing
polytopes and polyhedra. 
<br><dt><strong>race</strong><dd>Implementation of some racing methods for the empirical selection of the
best. 
<br><dt><strong>rake</strong><dd>Raking survey datasets by re-weighting. 
<br><dt><strong>ramps</strong><dd>Bayesian geostatistical modeling of Gaussian processes using a
reparameterized and marginalized posterior sampling (RAMPS) algorithm. 
<br><dt><strong>randaes</strong><dd>Random number generator based on AES cipher. 
<br><dt><strong>random</strong><dd>True random numbers using random.org. 
<br><dt><strong>randomLCA</strong><dd>Random effects Latent Class Analysis. 
<br><dt><strong>randomSurvivalForest</strong><dd>Ishwaran and Kogalur's random survival forest. 
<br><dt><strong>randomForest</strong><dd>Breiman's random forest classifier. 
<br><dt><strong>randtoolbox</strong><dd>Toolbox for pseudo and quasi random number generation. 
<br><dt><strong>rankreg</strong><dd>Rank regression estimator for the AFT model with right censored data. 
<br><dt><strong>rateratio.test</strong><dd>Exact rate ratio test. 
<!-- @item ratetables -->
<!-- US national and state mortality data (requires @strong{survival4} and -->
<!-- @strong{date}), for use with @strong{survival4}. -->
<br><dt><strong>rattle</strong><dd>A graphical user interface for data mining in R using GTK. 
<br><dt><strong>rbugs</strong><dd>Functions to prepare files needed for running BUGS in batch mode, and
running BUGS from R.  Support for Linux systems with Wine is emphasized. 
<br><dt><strong>rcdd</strong><dd>C Double Description for R, an interface to the CDD computational
geometry library. 
<br><dt><strong>rcdk</strong><dd>Interface to the CDK libraries, a Java framework for cheminformatics. 
<br><dt><strong>rcom</strong><dd>R COM Client Interface and internal COM Server. 
<br><dt><strong>rcompgen</strong><dd>Completion generator for R.  <em>Recommended</em> for R 2.5.0 or 2.6.0. 
<!-- <COMMENT> -->
<!-- Moved to the Archive o 2007-09-30 as requested by the maintainer. -->
<!-- @item rcompletion -->
<!-- TAB completion for R using Readline. -->
<!-- </COMMENT> -->
<br><dt><strong>rda</strong><dd>Shrunken Centroids Regularized Discriminant Analysis. 
<br><dt><strong>realized</strong><dd>Realized variance toolkit. 
<br><dt><strong>ref</strong><dd>Functions for creating references, reading from and writing ro
references and a memory efficient refdata type that transparently
encapsulates matrices and data frames. 
<br><dt><strong>regress</strong><dd>Fitting Gaussian linear models where the covariance structure is a
linear combination of known matrices by maximising the residual log
likelihood.  Can be used for multivariate models and random effects
models. 
<br><dt><strong>regsubseq</strong><dd>Detect and test regular sequences and subsequences. 
<br><dt><strong>regtest</strong><dd>Regression testing. 
<br><dt><strong>relaimpo</strong><dd>RELAtive IMPOrtance of regressors in linear models. 
<br><dt><strong>relations</strong><dd>Data structures for k-ary relations with arbitrary domains,
predicate functions, and fitters for consensus relations. 
<br><dt><strong>relax</strong><dd>Functions for report writing, presentation, and programming. 
<br><dt><strong>relaxo</strong><dd>Relaxed Lasso. 
<br><dt><strong>reldist</strong><dd>Functions for the comparison of distributions, including nonparametric
estimation of the relative distribution PDF and CDF and numerical
summaries as described in &ldquo;Relative Distribution Methods in the Social
Sciences&rdquo; by Mark S. Handcock and Martina Morris, 1999, Springer. 
<br><dt><strong>relimp</strong><dd>Functions to facilitate inference on the relative importance of
predictors in a linear or generalized linear model. 
<br><dt><strong>relsurv</strong><dd>Various functions for regression in relative survival. 
<br><dt><strong>repolr</strong><dd>Repeated measures proportional odds logistic regression. 
<br><dt><strong>reshape</strong><dd>Flexibly reshape data. 
<br><dt><strong>resper</strong><dd>Sampling from restricted permutations. 
<br><dt><strong>reweight</strong><dd>Adjustment of survey respondent weights. 
<br><dt><strong>rgcvpack</strong><dd>R interface for GCVPACK Fortran package. 
<br><dt><strong>rgdal</strong><dd>Provides bindings to Frank Warmerdam's Geospatial Data Abstraction
Library (GDAL). 
<br><dt><strong>rgenoud</strong><dd>R version of GENetic Optimization Using Derivatives. 
<br><dt><strong>rggm</strong><dd>Robustified methods for Gaussian Graphical Models. 
<br><dt><strong>rggobi</strong><dd>Interface between R and GGobi. 
<br><dt><strong>rgl</strong><dd>3D visualization device system (OpenGL). 
<br><dt><strong>rgr</strong><dd>The GSC (Geological Survey of Canada) applied geochemistry EDA package. 
<br><dt><strong>rhosp</strong><dd>Side effect risks in hospital: simulation and estimation. 
<br><dt><strong>richards</strong><dd>Richards curves. 
<br><dt><strong>rimage</strong><dd>Functions for image processing, including Sobel filter, rank filters,
fft, histogram equalization, and reading <acronym>JPEG</acronym> files. 
<br><dt><strong>rindex</strong><dd>Indexing for R. 
<br><dt><strong>risksetROC</strong><dd>Riskset ROC curve estimation from censored survival data. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2008-04-22 (Orphaned/2.6.0). -->
<!-- @item riv -->
<!-- Robust Instrumental Variables estimators based on high breakpoint point -->
<!-- S-estimators of multivariate location and scatter matrices. -->
<!-- </COMMENT> -->
<br><dt><strong>rjacobi</strong><dd>Jacobi polynomials and Gauss-Jacobi quadrature related operations. 
<br><dt><strong>rjags</strong><dd>Bayesian graphical models via an interface to the JAGS MCMC library. 
<br><dt><strong>rjson</strong><dd>JSON (JavaScript Object Notation) for R. 
<br><dt><strong>rlecuyer</strong><dd>R interface to RNG with multiple streams. 
<br><dt><strong>rmeta</strong><dd>Functions for simple fixed and random effects meta-analysis for
two-sample comparison of binary outcomes. 
<br><dt><strong>rmetasim</strong><dd>An interface between R and the metasim simulation engine.  Facilitates
the use of the metasim engine to build and run individual based
population genetics simulations. 
<br><dt><strong>robfilter</strong><dd>Robust time series filters. 
<!-- <COMMENT> -->
<!-- Moved to the Archive o 2007-09-30 as requested by the maintainer. -->
<!-- @item roblm -->
<!-- Robust regression estimators. -->
<!-- </COMMENT> -->
<br><dt><strong>robust</strong><dd>Insightful robust package. 
<br><dt><strong>robustbase</strong><dd>Basic Robust Statistics. 
<br><dt><strong>rootSolve</strong><dd>Nonlinear root finding, equilibrium and steady-state analysis of
ordinary differential equations. 
<br><dt><strong>rpanel</strong><dd>Simple interactive controls for R using the tcltk package. 
<br><dt><strong>rpart</strong><dd>Recursive PARTitioning and regression trees.  <em>Recommended</em>. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2008-04-22 (Orphaned/2.6.0). -->
<!-- @item rpart.permutation -->
<!-- Permutation tests of rpart models. -->
<!-- </COMMENT> -->
<br><dt><strong>rpubchem</strong><dd>R interface to the PubChem collection. 
<br><dt><strong>rpvm</strong><dd>R interface to PVM (Parallel Virtual Machine).  Provides interface to
PVM APIs, and examples and documentation for its use. 
<br><dt><strong>rqmcmb2</strong><dd>Markov chain marginal bootstrap for quantile regression. 
<br><dt><strong>rrcov</strong><dd>Functions for robust location and scatter estimation and robust
regression with high breakdown point. 
<br><dt><strong>rrp</strong><dd>Random Recursive Partitioning. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2008-03-25 as requested by the maintainer. -->
<!-- @item rsbml -->
<!-- R support for SBML (Systems Biology Markup Language), using libsbml. -->
<!-- </COMMENT> -->
<br><dt><strong>rscproxy</strong><dd>A portable C-style interface to R (StatConnector). 
<br><dt><strong>rsprng</strong><dd>Provides interface to SPRNG (Scalable Parallel Random Number Generators)
APIs, and examples and documentation for its use. 
<br><dt><strong>rstream</strong><dd>Unified object oriented interface for multiple independent streams of
random numbers from different sources. 
<br><dt><strong>rtiff</strong><dd>Read TIFF format images and return them as pixmap objects. 
<br><dt><strong>rv</strong><dd>Simulation-based random variable object class. 
<br><dt><strong>rwt</strong><dd>Rice Wavelet Toolbox wrapper, providing a set of functions for
performing digital signal processing. 
<br><dt><strong>s20x</strong><dd>Stats 20x functions. 
<br><dt><strong>sac</strong><dd>Semiparametric empirical likelihood ratio based test of changepoint with
one-change or epidemic alternatives with data-based model diagnostic. 
<br><dt><strong>sampfling</strong><dd>Implements a modified version of the Sampford sampling algorithm.  Given
a quantity assigned to each unit in the population, samples are drawn
with probability proportional to te product of the quantities of the
units included in the sample. 
<br><dt><strong>sampleSelection</strong><dd>Estimation of sample selection models. 
<br><dt><strong>sampling</strong><dd>A set of tools to select and to calibrate samples. 
<br><dt><strong>samr</strong><dd>Significance Analysis of Microarrays. 
<br><dt><strong>sandwich</strong><dd>Model-robust standard error estimators for time series and longitudinal
data. 
<br><dt><strong>sapa</strong><dd>Insightful Spectral Analysis for Physical Applications. 
<br><dt><strong>sbgcop</strong><dd>Semiparametric Bayesian Gaussian copula estimation. 
<br><dt><strong>sca</strong><dd>Simple Component Analysis. 
<br><dt><strong>scagnostics</strong><dd>Compute scagnostics. 
<br><dt><strong>scaleboot</strong><dd>Approximately unbiased p-values via multiscale bootstrap. 
<br><dt><strong>scape</strong><dd>functions to import and plot results from statistical catch-at-age
models, used in fisheries stock assessments. 
<br><dt><strong>scapeMCMC</strong><dd>Markov-chain Monte Carlo diagnostic plots, accompanying the
<strong>scape</strong> package. 
<br><dt><strong>scatterplot3d</strong><dd>Plots a three dimensional (3D) point cloud perspectively. 
<br><dt><strong>schoolmath</strong><dd>Functions and datasets for math used in school. 
<br><dt><strong>sciplot</strong><dd>Scientific graphing functions for factorial designs. 
<br><dt><strong>scope</strong><dd>Data manipulation using arbitrary row and column criteria. 
<br><dt><strong>scrime</strong><dd>Tools for the analysis of high-dimensional data developed/implemented at
the group &ldquo;Statistical Complexity Reduction In Molecular Epidemiology&rdquo;
(SCRIME), with main focus on SNP data. 
<br><dt><strong>scuba</strong><dd>Scuba diving calculations and decompression models. 
<br><dt><strong>sdcMicro</strong><dd>Statistical Disclosure Control methods for the generation of public and
scientific use files. 
<br><dt><strong>sde</strong><dd>Simulation and inference for Stochastic Differential Equations. 
<br><dt><strong>sdtalt</strong><dd>Signal Detection Theory measures and ALTernatives. 
<br><dt><strong>seacarb</strong><dd>Calculates parameters of the seawater carbonate system. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2005-10-05 (Orphaned/2.2.0). -->
<!-- @item seao -->
<!-- Simple Evolutionary Algorithm Optimization. -->
<!-- </COMMENT> -->
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2005-10-05 (Orphaned/2.2.0). -->
<!-- @item seao.gui -->
<!-- Simple Evolutionary Algorithm Optimization: graphical user interface. -->
<!-- </COMMENT> -->
<br><dt><strong>seas</strong><dd>Detailed seasonal plots of temperature and precipitation data. 
<br><dt><strong>seewave</strong><dd>Time wave analysis and graphical representation. 
<br><dt><strong>segclust</strong><dd>Segmentation and segmentation/clustering. 
<br><dt><strong>segmented</strong><dd>Functions to estimate break-points of segmented relationships in
regression models (GLMs). 
<br><dt><strong>sem</strong><dd>Functions for fitting general linear Structural Equation Models (with
observed and unobserved variables) by the method of maximum likelihood
using the RAM approach. 
<br><dt><strong>sendplot</strong><dd>Tool for sending interactive plots. 
<br><dt><strong>sensR</strong><dd>Thurstonian models for sensory discrimination. 
<br><dt><strong>sensitivity</strong><dd>Sensitivity analysis. 
<br><dt><strong>seqinr</strong><dd>Exploratory data analysis and data visualization for biological sequence
(DNA and protein) data. 
<br><dt><strong>seqmon</strong><dd>Sequential monitoring of clinical trials. 
<!-- <COMMENT> -->
<!-- Moved to the Archive as requested by the maintainer: serialize() et -->
<!-- al are available in R 1.8 or better. -->
<!-- @item serialize -->
<!-- Simple interfce for serializing to connections. -->
<!-- </COMMENT> -->
<br><dt><strong>seriation</strong><dd>Infrastructure for seriation. 
<br><dt><strong>session</strong><dd>Functions for interacting with, saving and restoring R sessions. 
<br><dt><strong>setRNG</strong><dd>Set (normal) random number generator and seed. 
<br><dt><strong>sets</strong><dd>Data structures and basic operations for ordinary sets, and
generalizations such as fuzzy sets, multisets, and fuzzy multisets. 
<br><dt><strong>sfsmisc</strong><dd>Utilities from Seminar fuer Statistik ETH Zurich. 
<br><dt><strong>sgeostat</strong><dd>An object-oriented framework for geostatistical modeling. 
<br><dt><strong>shape</strong><dd>Functions for plotting graphical shapes. 
<br><dt><strong>shapefiles</strong><dd>Functions to read and write ESRI shapefiles. 
<br><dt><strong>shapes</strong><dd>Routines for the statistical analysis of shapes, including procrustes
analysis, displaying shapes and principal components, testing for mean
shape difference, thin-plate spline transformation grids and edge
superimposition methods. 
<br><dt><strong>siar</strong><dd>Stable Isotope Analysis in R. 
<br><dt><strong>sigma2tools</strong><dd>Test of hypothesis about sigma2. 
<br><dt><strong>signal</strong><dd>A set of generally Matlab/Octave-compatible signal processing
functions. 
<br><dt><strong>signalextraction</strong><dd>Real-time signal extraction (Direct Filter Approach). 
<br><dt><strong>simba</strong><dd>Functions for similarity calculation of binary data. 
<br><dt><strong>simco</strong><dd>Import Structure files and deduce similarity coefficients from them. 
<br><dt><strong>simecol</strong><dd>SIMulation of ECOLogical (and other) dynamic systems. 
<br><dt><strong>simex</strong><dd>SIMEX and MCSIMEX algorithms for measurement error models. 
<br><dt><strong>similarityRichards</strong><dd>Similarity of Richards curves. 
<br><dt><strong>simpleboot</strong><dd>Simple bootstrap routines. 
<br><dt><strong>singlecase</strong><dd>Tests for single case studies in neuropsychology. 
<br><dt><strong>skewt</strong><dd>Density, distribution function, quantile function and random generation
for the skewed t distribution of Fernandez and Steel. 
<br><dt><strong>sm</strong><dd>Software linked to the book &ldquo;Applied Smoothing Techniques for Data
Analysis:  The Kernel Approach with <span class="sc">S-Plus</span> Illustrations&rdquo; by
A. W. Bowman and A. Azzalini, 1997, Oxford University Press. 
<br><dt><strong>sma</strong><dd>Functions for exploratory (statistical) microarray analysis. 
<br><dt><strong>smacof</strong><dd>Multidimensional scaling based on stress minimization by means of
majorization (smacof). 
<br><dt><strong>smatr</strong><dd>(Standardized) Major Axis estimation and Testing Routines. 
<br><dt><strong>smoothSurv</strong><dd>Survival regression with smoothed error distribution. 
<br><dt><strong>smoothtail</strong><dd>Smooth estimation of generalized Pareto distribution shape parameter. 
<br><dt><strong>sn</strong><dd>Functions for manipulating skew-normal probability distributions and for
fitting them to data, in the scalar and the multivariate case. 
<br><dt><strong>sna</strong><dd>A range of tools for social network analysis, including node and
graph-level indices, structural distance and covariance methods,
structural equivalence detection, p* modeling, and network
visualization. 
<!-- <COMMENT> -->
<!-- Removed ... -->
<!-- @item snns -->
<!-- An R interface to the Stuttgart Neural Networks Simulator (SNNS). -->
<!-- </COMMENT> -->
<br><dt><strong>snow</strong><dd>Simple Network of Workstations: support for simple parallel computing in
R. 
<br><dt><strong>snowFT</strong><dd>Fault Tolerant Simple Network of Workstations. 
<br><dt><strong>snowfall</strong><dd>Wrapper around <strong>snow</strong> for easier development of parallel R
programs. 
<br><dt><strong>snp.plotter</strong><dd>Plots of p-values using single SNP and/or haplotype data. 
<br><dt><strong>snpXpert</strong><dd>Tools to analyze SNP data. 
<br><dt><strong>som</strong><dd>Self-Organizing Maps (with application in gene clustering). 
<br><dt><strong>sound</strong><dd>A sound interface for R: Basic functions for dealing with <samp><span class="file">.wav</span></samp>
files and sound samples. 
<br><dt><strong>sp</strong><dd>A package that provides classes and methods for spatial data, including
utility functions for plotting data as maps, spatial selection, amd much
more. 
<br><dt><strong>spBayes</strong><dd>Fit Gaussian models with potentially complex hierarchical error
structures by Markov chain Monte Carlo (MCMC). 
<br><dt><strong>space</strong><dd>Sparse PArtial Correlation Estimation. 
<br><dt><strong>spam</strong><dd>SPArse Matrix algebra. 
<br><dt><strong>spatclus</strong><dd>Arbitrarily shaped multiple spatial cluster detection for case event
data. 
<br><dt><strong>spatgraphs</strong><dd>Graphs for 2-d point patterns. 
<br><dt><strong>spatial</strong><dd>Functions for kriging and point pattern analysis from &ldquo;Modern Applied
Statistics with S&rdquo; by W. Venables and B. Ripley.  Contained in the
<samp><span class="file">VR</span></samp> bundle.  <em>Recommended</em>. 
<br><dt><strong>spatialCovariance</strong><dd>Computation of spatial covariance matrices for data on rectangles using
one dimensional numerical integration and analytic results. 
<br><dt><strong>spatialkernel</strong><dd>Nonparameteric estimation of spatial segregation in a multivariate point
process. 
<br><dt><strong>spatstat</strong><dd>Data analysis and modelling of two-dimensional point patterns, including
multitype points and spatial covariates. 
<br><dt><strong>spc</strong><dd>Statistical Process Control: evaluation of control charts by means of
the zero-state, steady-state ARL (Average Run Length), setting up
control charts for given in-control ARL, and plotting of the related
figures. 
<br><dt><strong>spdep</strong><dd>A collection of functions to create spatial weights matrix objects from
polygon contiguities, from point patterns by distance and tesselations,
for summarising these objects, and for permitting their use in spatial
data analysis; a collection of tests for spatial autocorrelation,
including global Moran's I and Geary's C, local Moran's I, saddlepoint
approximations for global and local Moran's I; and functions for
estimating spatial simultaneous autoregressive (SAR) models.  (Was
formerly the three packages: <strong>spweights</strong>, <strong>sptests</strong>, and
<strong>spsarlm</strong>.) 
<br><dt><strong>spe</strong><dd>Stochastic Proximity Embedding. 
<br><dt><strong>spectralGP</strong><dd>Approximate Gaussian processes using the Fourier basis. 
<br><dt><strong>spectrino</strong><dd>Spectra organizer, visualization and data extraction from within R. 
<br><dt><strong>spgrass6</strong><dd>Interface between the GRASS 6.0 geographical information system and R. 
<br><dt><strong>spgwr</strong><dd>Geographically weighted regression. 
<br><dt><strong>splancs</strong><dd>Spatial and space-time point pattern analysis functions. 
<br><dt><strong>splus2R</strong><dd>Insightful package providing missing S-PLUS functionality in R. 
<!-- <COMMENT> -->
<!-- Deprecated in favor of new @strong{spdep} along with 1.5.0 release. -->
<!-- @item spsarlm -->
<!-- Functions for estimating spatial simultaneous autoregressive (SAR) -->
<!-- models, including sparse matrix methods for computing the Jacobian. -->
<!-- @item sptests -->
<!-- A collection of tests for spatial autocorrelation, including global -->
<!-- Moran's I and Geary's C. -->
<!-- @item spweights -->
<!-- A collection of functions to create spatial weights matrix objects from -->
<!-- polygon contiguities, from point patterns by distance and tesselations, -->
<!-- for summarising these objects, and for permitting their use in spatial -->
<!-- data analysis. -->
<!-- </COMMENT> -->
<br><dt><strong>spssDDI</strong><dd>Read SPSS system files and produce valid DDI version 3.0 documents. 
<br><dt><strong>spsurvey</strong><dd>Spatial survey design and analysis. 
<br><dt><strong>sqldf</strong><dd>Perform <acronym>SQL</acronym> selects on R data frames. 
<br><dt><strong>ssanv</strong><dd>Sample Size Adjusted for Nonadherence or Variability of input
parameters. 
<br><dt><strong>sspir</strong><dd>State SPace models In R. 
<br><dt><strong>sspline</strong><dd>Smoothing splines on the sphere. 
<br><dt><strong>st</strong><dd>Shrinkage t statistic. 
<br><dt><strong>staRt</strong><dd>Inferenza classica con TI-83 Plus. 
<br><dt><strong>stab</strong><dd>Data analysis of drug stability. 
<!-- @item stable -->
<!-- Density, distribution, quantile and hazard functions of a stable -->
<!-- variate; generalized linear models for the parameters of a stable -->
<!-- distribution. -->
<br><dt><strong>startupmsg</strong><dd>Utilities for start-up messages. 
<br><dt><strong>stashR</strong><dd>A Set of Tools for Administering SHared Repositories. 
<!-- <COMMENT> -->
<!-- Merged into foreign. -->
<!-- @item stataread -->
<!-- Read and write Stata v5 and v6 @file{.dta} files. -->
<!-- </COMMENT> -->
<br><dt><strong>statmod</strong><dd>Miscellaneous biostatistical modelling functions. 
<br><dt><strong>statnet</strong><dd>Software tools for the statistical modeling of network data. 
<br><dt><strong>stepPlr</strong><dd>L2 penalized logistic regression with a stepwise variable selection. 
<br><dt><strong>stepwise</strong><dd>A stepwise approach to identifying recombination breakpoints in a
sequence alignment. 
<br><dt><strong>stinepack</strong><dd>Stineman interpolation package. 
<br><dt><strong>stochasticGEM</strong><dd>Fitting Stochastic General Epidemic Models. 
<br><dt><strong>stochmod</strong><dd>Learning and inference algorithms for a variety of probabilistic
models. 
<br><dt><strong>stream.net</strong><dd>Building and analyzing binary stream networks. 
<br><dt><strong>strucchange</strong><dd>Various tests on structural change in linear regression models. 
<br><dt><strong>subplex</strong><dd>The subplex algorithm for unconstrained optimization. 
<br><dt><strong>subselect</strong><dd>A collection of functions which assess the quality of variable subsets
as surrogates for a full data set, and search for subsets which are
optimal under various criteria. 
<br><dt><strong>sudoku</strong><dd>Sudoku puzzle solver. 
<br><dt><strong>supclust</strong><dd>Methodology for supervised grouping of predictor variables. 
<br><dt><strong>superpc</strong><dd>Supervised principal components. 
<br><dt><strong>surv2sample</strong><dd>Two-sample tests for survival analysis. 
<br><dt><strong>survBayes</strong><dd>Fits a proportional hazards model to time to event data by a Bayesian
approach. 
<br><dt><strong>surveillance</strong><dd>Outbreak detection algorithms for surveillance data. 
<br><dt><strong>survey</strong><dd>Summary statistics, generalized linear models, and general maximum
likelihood estimation for stratified, cluster-sampled, unequally
weighted survey samples. 
<br><dt><strong>surveyNG</strong><dd>Complex survey samples: database interface, sparse matrices. 
<!-- @item survival4 -->
<!-- Functions for survival analysis, version 4 (requires -->
<!-- @strong{splines}). -->
<br><dt><strong>survival</strong><dd>Functions for survival analysis, including penalised likelihood. 
<em>Recommended</em>. 
<br><dt><strong>survivalROC</strong><dd>Time-dependent ROC curve estimation from censored survival data. 
<br><dt><strong>survrec</strong><dd>Survival analysis for recurrent event data. 
<br><dt><strong>svcR</strong><dd>A support vector machine technique for clustering. 
<br><dt><strong>svcm</strong><dd>2d and 3d Space-Varying Coefficient Models. 
<br><dt><strong>svmpath</strong><dd>Computes the entire regularization path for the two-class svm classifier
with essentialy the same cost as a single SVM fit. 
<br><dt><strong>systemfit</strong><dd>Contains functions for fitting simultaneous systems of equations using
Ordinary Least Sqaures (OLS), Two-Stage Least Squares (2SLS), and
Three-Stage Least Squares (3SLS). 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2007-04-11 as requested by the maintainer. -->
<!-- @item tapiR -->
<!-- Tools for accessing (UK) parliamentary information in R. -->
<!-- </COMMENT> -->
<br><dt><strong>taskPR</strong><dd>Task-Parallel R package. 
<br><dt><strong>tcltk2</strong><dd>A series of widgets and functions to supplement <strong>tcltk</strong>. 
<br><dt><strong>tdist</strong><dd>Computes the distribution of a linear combination of independent
Student's t variables. 
<br><dt><strong>tdm</strong><dd>A tool for Therapeutic Drug Monitoring. 
<br><dt><strong>tdthap</strong><dd>Transmission/disequilibrium tests for extended marker haplotypes. 
<br><dt><strong>tensor</strong><dd>Tensor product of arrays. 
<br><dt><strong>tensorA</strong><dd>Advanced tensors arithmetic with named indices. 
<br><dt><strong>termstrc</strong><dd>Term structure and credit spread estimation. 
<br><dt><strong>tframe</strong><dd>Time Frame coding kernel: functions for writing code that is independent
of the way time is represented. 
<br><dt><strong>tframePlus</strong><dd>Time Frame coding kernel extensions. 
<br><dt><strong>tgp</strong><dd>Bayesian regression and adaptive sampling with Treed Gaussian Process
models. 
<br><dt><strong>tileHMM</strong><dd>Hidden Markov Models for ChIP-on-Chip analysis. 
<br><dt><strong>time</strong><dd>Time tracking for developers. 
<br><dt><strong>timereg</strong><dd>Code and data sets for the book &ldquo;Dynamic Regression Models for Survival
Data&rdquo; by T.  Martinussen and T. Scheike, 2006, Springer Verlag, plus
more recent developments. 
<br><dt><strong>timsac</strong><dd>TIMe Series Analysis and Control package. 
<br><dt><strong>titan</strong><dd>Titration analysis for mass spectrometry data. 
<br><dt><strong>titecrm</strong><dd>TIme-To-Event Continual Reassessment Method and calibration tools. 
<br><dt><strong>tkrgl</strong><dd>TK widget tools for <strong>rgl</strong> package. 
<br><dt><strong>tkrplot</strong><dd>Simple mechanism for placing R graphics in a Tk widget. 
<br><dt><strong>tlnise</strong><dd>Two-level normal independent sampling estimation. 
<br><dt><strong>tm</strong><dd>A framework for text mining applications within R. 
<br><dt><strong>topmodel</strong><dd>An R implementation of TOPMODEL. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2008-05-25 as requested by the maintainer. -->
<!-- (Replaced by randtoolbox.) -->
<!-- @item torus -->
<!-- Torus algorithm for quasi random number generation. -->
<!-- </COMMENT> -->
<br><dt><strong>tossm</strong><dd>Testing Of Spatial Structure Methods. 
<br><dt><strong>tpr</strong><dd>Temporal Process Regression. 
<br><dt><strong>trackObjs</strong><dd>Track objects. 
<br><dt><strong>tradeCosts</strong><dd>Post-trade analysis of transaction costs. 
<br><dt><strong>tree</strong><dd>Classification and regression trees. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2008-04-22 (Orphaned/2.6.0). -->
<!-- @item treeglia -->
<!-- Stem analysis functions for volume increment and carbon uptake -->
<!-- assessment from tree-rings. -->
<!-- </COMMENT> -->
<br><dt><strong>triangle</strong><dd>Standard distribution functions for the triangle distribution. 
<br><dt><strong>trimcluster</strong><dd>Cluster analysis with trimming. 
<br><dt><strong>trip</strong><dd>Spatial analysis of animal track data. 
<br><dt><strong>tripEstimation</strong><dd>Metropolis sampler and supporting functions for estimating animal
movement from archival tags and satellite fixes. 
<br><dt><strong>tripack</strong><dd>A constrained two-dimensional Delaunay triangulation package. 
<br><dt><strong>truncgof</strong><dd>Goodness-of-fit tests allowing for left truncated data. 
<br><dt><strong>trust</strong><dd>Local optimization using two derivatives and trust regions. 
<br><dt><strong>tsDyn</strong><dd>Time series analysis based on dynamical systems theory. 
<br><dt><strong>tsModel</strong><dd>Time series modeling for air pollution and health. 
<br><dt><strong>tseries</strong><dd>Package for time series analysis with emphasis on non-linear modelling. 
<br><dt><strong>tseriesChaos</strong><dd>Routines for the analysis of non-linear time series. 
<br><dt><strong>tsfa</strong><dd>Time Series Factor Analysis. 
<br><dt><strong>tuneR</strong><dd>Collection of tools to analyze music, handle wave files, transcription,
etc. 
<br><dt><strong>tutoR</strong><dd>Student-friendly package to mask common functions. 
<br><dt><strong>twang</strong><dd>Toolkit for Weighting and Analysis of Nonequivalent Groups. 
<br><dt><strong>tweedie</strong><dd>Maximum likelihood computations for Tweedie exponential family models. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2005-10-05 (Orphaned/2.2.0). -->
<!-- @item twostage -->
<!-- Functions for optimal design of two-stage-studies using the Mean Score -->
<!-- method. -->
<!-- </COMMENT> -->
<br><dt><strong>twslm</strong><dd>A two-way semilinear model for normalization and analysis of cDNA
microarray data. 
<br><dt><strong>udunits</strong><dd>Interface to Unidata's routines to convert units. 
<br><dt><strong>ump</strong><dd>Uniformly Most Powerful tests. 
<br><dt><strong>unbalhaar</strong><dd>Function estimation via Unbalanced Haar wavelets. 
<br><dt><strong>uncompress</strong><dd>For uncompressing <samp><span class="file">.Z</span></samp> files. 
<br><dt><strong>untb</strong><dd>Ecological drift under the UNTB (Unified Neutral Theory of
Biodiversity). 
<br><dt><strong>urca</strong><dd>Unit root and cointegration tests for time series data. 
<br><dt><strong>urn</strong><dd>Functions for sampling without replacement (simulated urns). 
<br><dt><strong>uroot</strong><dd>Unit root tests and graphics for seasonal time series. 
<br><dt><strong>vabayelMix</strong><dd>Variational Bayesian mixture model. 
<br><dt><strong>varSelRF</strong><dd>Variable selection using random forests. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2008-04-22 (Orphaned/2.6.0). -->
<!-- @item vardiag -->
<!-- Interactive variogram diagnostics. -->
<!-- </COMMENT> -->
<br><dt><strong>varmixt</strong><dd>Mixture model on the variance for the analysis of gene expression data. 
<br><dt><strong>vars</strong><dd>VAR modeling. 
<br><dt><strong>vbmp</strong><dd>Variational Bayesian Multinomial Probit Regression. 
<br><dt><strong>vcd</strong><dd>Functions and data sets based on the book &ldquo;Visualizing Categorical
Data&rdquo; by Michael Friendly. 
<br><dt><strong>vegan</strong><dd>Various help functions for vegetation scientists and community
ecologists. 
<br><dt><strong>verification</strong><dd>Utilities for verification of discrete and probabilistic forecasts. 
<br><dt><strong>verify</strong><dd>Construction of test suites using verify objects. 
<br><dt><strong>vioplot</strong><dd>Violin plots, which are a combination of a box plot and a kernel density
plot. 
<br><dt><strong>vrtest</strong><dd>Variance ratio tests for weak-form market efficiency. 
<br><dt><strong>waveclock</strong><dd>Time-frequency analysis of cycling cell luminescence data. 
<br><dt><strong>waved</strong><dd>WaveD transform in R. 
<br><dt><strong>wavelets</strong><dd>Functions for computing wavelet filters, wavelet transforms and
multiresolution analyses. 
<br><dt><strong>waveslim</strong><dd>Basic wavelet routines for time series analysis. 
<br><dt><strong>wavethresh</strong><dd>Software to perform 1-d and 2-d wavelet statistics and transforms. 
<br><dt><strong>wccsom</strong><dd>SOM networks for comparing patterns with peak shifts. 
<br><dt><strong>wgaim</strong><dd>Whole Genome Average Interval Mapping for QTL detection using mixed
models. 
<br><dt><strong>wikibooks</strong><dd>Functions and datasets for the German WikiBook &ldquo;GNU R&rdquo;. 
<br><dt><strong>wle</strong><dd>Robust statistical inference via a weighted likelihood approach. 
<br><dt><strong>wmtsa</strong><dd>Insightful Wavelet Methods for Time Series Analysis. 
<br><dt><strong>wnominate</strong><dd>WNOMINATE roll call analysis software. 
<br><dt><strong>wombsoft</strong><dd>Wombling computation. 
<br><dt><strong>wordnet</strong><dd>WordNet interface. 
<br><dt><strong>write.snns</strong><dd>Function for exporting data to SNNS (Stuttgart Neural Network Simulator)
pattern files. 
<br><dt><strong>xgobi</strong><dd>Interface to the XGobi and XGvis programs for graphical data analysis. 
<br><dt><strong>xlsReadWrite</strong><dd>Natively read and write Excel files. 
<br><dt><strong>xtable</strong><dd>Export data to LaTeX and <acronym>HTML</acronym> tables. 
<br><dt><strong>xts</strong><dd>Extensible time series. 
<br><dt><strong>yaImpute</strong><dd>Performs popular nearest neighbor routines. 
<br><dt><strong>yaml</strong><dd>Methods to convert R to YAML and back. 
<br><dt><strong>yest</strong><dd>Gaussian independence models. 
<!-- <COMMENT> -->
<!-- Moved to the Archive on 2008-04-22 (Orphaned/2.6.0). -->
<!-- @item zicounts -->
<!-- Fit classical, zero-inflated and interval censored count data regression -->
<!-- models. -->
<!-- </COMMENT> -->
<br><dt><strong>zipfR</strong><dd>Statistical models for word frequency distributions. 
<br><dt><strong>zoeppritz</strong><dd>Zoeppritz equations: calculate and plot scattering coefficients of
seismic waves when they interact at an interface between two layers. 
<br><dt><strong>zoo</strong><dd>A class with methods for totally ordered indexed observations such as
irregular time series. 
</dl>

<p class="noindent">See <acronym>CRAN</acronym> <samp><span class="file">src/contrib/PACKAGES</span></samp> for more information.

   <p>Some <acronym>CRAN</acronym> packages that do not build out of the box on Windows,
require additional software, or are shipping third party libraries for
Windows cannot be made available on <acronym>CRAN</acronym> in form of a Windows binary
packages.  Nevertheless, some of these packages are available at the
&ldquo;CRAN extras&rdquo; repository at <a href="http://www.stats.ox.ac.uk/pub/RWin/">http://www.stats.ox.ac.uk/pub/RWin/</a>
kindly provided by Brian D. Ripley.  Note that this repository is a
default repository for recent versions of R for Windows.

   <p>There used to be a <acronym>CRAN</acronym> <samp><span class="file">src/contrib/Devel</span></samp> directory with
packages still &ldquo;under development&rdquo; or depending on features only
present in the current development versions of R.  This area is no
longer provided, with packages formerly in this area either in the
regular package area or the archive <samp><span class="file">src/contrib/Archive</span></samp>.

<!-- <COMMENT> -->
<!-- CRAN's src/contrib/Devel discontinued on 2007-04-14. -->
<!-- @table @strong -->
<!-- @c <COMMENT> -->
<!-- @c Moved to Attic on 2004-12-14 as requested by maintainer. -->
<!-- @c @item Dopt -->
<!-- @c Finding D-optimal experimental designs. -->
<!-- @c </COMMENT> -->
<!-- @c <COMMENT> -->
<!-- @c Moved to Devel for 1.8 as it fails R CMD check. -->
<!-- @item GLMMGibbs -->
<!-- Generalised Linear Mixed Models by Gibbs sampling. -->
<!-- @c </COMMENT> -->
<!-- @c <COMMENT> -->
<!-- @c Moved to Devel for 1.6 as it fails R CMD check. -->
<!-- @item RPgSQL -->
<!-- Provides methods for accessing data stored in PostgreSQL tables. -->
<!-- @c </COMMENT> -->
<!-- @c <COMMENT> -->
<!-- @c Moved to Attic for 1.5.1 as no longer needed. -->
<!-- @c (See also the info on rdbi.sourceforge.net.) -->
<!-- @c @item Rdbi -->
<!-- @c Generic framework for database access in R. -->
<!-- @c @item Rdbi.PgSQL -->
<!-- @c Provides methods for accessing data stored in PostgreSQL tables. -->
<!-- @c </COMMENT> -->
<!-- @c <COMMENT> -->
<!-- @c Moved to Attic for 1.5 (as this contains the summary functions). -->
<!-- @c @item colSums -->
<!-- @c Matrix summary functions in C. -->
<!-- @c </COMMENT> -->
<!-- @c <COMMENT> -->
<!-- @c Moved to Attic for 1.5.1 as no longer needed. -->
<!-- @c @item cxx -->
<!-- @c A small C++ test package. -->
<!-- @c </COMMENT> -->
<!-- @item dseplus -->
<!-- Extensions to @strong{dse}, the Dynamic Systems Estimation multivariate -->
<!-- time series package.  Contains PADI, juice and monitoring extensions. -->
<!-- @item ensemble -->
<!-- Ensembles of tree classifiers. -->
<!-- @c <COMMENT> -->
<!-- @c Moved to Devel for 1.5 as it fails R CMD check. -->
<!-- @c Moved to Attic for 1.5.1 as apparently no longer actively -->
<!-- @c maintained. -->
<!-- @c @item event.chart -->
<!-- @c Package for creating event charts. -->
<!-- @c </COMMENT> -->
<!-- @c <COMMENT> -->
<!-- @c Moved to Attic for 1.5.1 as no longer maintained. -->
<!-- @c @item hpower -->
<!-- @c A suite of functions to compute power and sample size for tests of the -->
<!-- @c general linear hypothesis. -->
<!-- @c </COMMENT> -->
<!-- @c <COMMENT> -->
<!-- @c Moved to Attic on 2003-09-12 as requested by the package maintainer, -->
<!-- @c "kjetil brinchmann halvorsen" <kjetil@@entelnet.bo>. -->
<!-- @c @item pls -->
<!-- @c Univariate Partial Least Squares Regression. -->
<!-- @c </COMMENT> -->
<!-- @c <COMMENT> -->
<!-- @c Moved to Attic for 1.5.1 as no longer needed. -->
<!-- @c @item regexp -->
<!-- @c Simple regular expression interface. -->
<!-- @c </COMMENT> -->
<!-- @item runStat -->
<!-- Running median and mean. -->
<!-- @end table -->
<!-- </COMMENT> -->
<div class="node">
<p><hr>
<a name="Add-on-packages-from-Omegahat"></a>
<a name="Add_002don-packages-from-Omegahat"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Add_002don-packages-from-Bioconductor">Add-on packages from Bioconductor</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Add_002don-packages-from-CRAN">Add-on packages from CRAN</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Which-add_002don-packages-exist-for-R_003f">Which add-on packages exist for R?</a>

</div>

<h4 class="subsection">5.1.3 Add-on packages from Omegahat</h4>

<p>The <a href="http://www.omegahat.org/">Omegahat Project for Statistical Computing</a> provides a variety of open-source software for statistical
applications, with special emphasis on web-based software, Java, the
Java virtual machine, and distributed computing.  A <acronym>CRAN</acronym> style
R package repository is available via <a href="http://www.omegahat.org/R/">http://www.omegahat.org/R/</a>.

   <p>Currently, there are the following packages.

     <dl>
<dt><strong>Aspell</strong><dd>An interface to facilities in the aspell library. 
<br><dt><strong>CGIwithR</strong><dd>Facilities for the use of R to write CGI scripts. 
<br><dt><strong>CORBA</strong><dd>Dynamic CORBA client/server facilities for R.  Connects to other
CORBA-aware applications developed in arbitrary languages, on different
machines and allows R functionality to be exported in the same way to
other applications. 
<br><dt><strong>Combinations</strong><dd>Compute the combinations of choosing r items from n
elements. 
<br><dt><strong>IDocs</strong><dd>Infrastructure for interactive documents. 
<br><dt><strong>OOP</strong><dd>OOP style classes and methods for R and <span class="sc">S-Plus</span>.  Object references and
class-based method definition are supported in the style of languages
such as Java and C++. 
<br><dt><strong>RCurl</strong><dd>Allows one to compose HTTP requests to fetch URIs, post forms, etc., and
process the results returned by the Web server. 
<br><dt><strong>RDCOMClient</strong><dd>Provides dynamic client-side access to (D)COM applications from within
R. 
<br><dt><strong>RDCOMEvents</strong><dd>Provides facilities to use R functions and objects as handlers for DCOM
events. 
<br><dt><strong>RDCOMServer</strong><dd>Facilities for exporting S objects and functions as COM objects. 
<br><dt><strong>REmbeddedPostgres</strong><dd>Allows R functions and objects to be used to implement SQL functions &mdash;
per-record, aggregate and trigger functions. 
<br><dt><strong>REventLoop</strong><dd>An abstract event loop mechanism that is toolkit independent and can be
used to to replace the R event loop. 
<br><dt><strong>Rexif</strong><dd>Extract meta-information from <acronym>JPEG</acronym> files. 
<br><dt><strong>RGdkPixbuf</strong><dd>S language functions to access the facilities in the GdkPixbuf library
for manipulating images. 
<br><dt><strong>RGnumeric</strong><dd>A plugin for the Gnumeric spreadsheet that allows R functions to be
called from cells within the sheet, automatic recalculation, etc. 
<br><dt><strong>RGtk</strong><dd>Facilities in the S language for programming graphical interfaces using
Gtk, the Gnome <acronym>GUI</acronym> toolkit. 
<br><dt><strong>RGtkBindingGenerator</strong><dd>A meta-package which generates C and R code to provide bindings to a
Gtk-based library. 
<br><dt><strong>RGtkExtra</strong><dd>A collection of S functions that provide an interface to the widgets in
the gtk+extra library such as the GtkSheet data-grid display, icon list,
file list and directory tree. 
<br><dt><strong>RGtkGlade</strong><dd>S language bindings providing an interface to Glade, the interactive
Gnome <acronym>GUI</acronym> creator. 
<br><dt><strong>RGtkHTML</strong><dd>A collection of S functions that provide an interface to creating and
controlling an <acronym>HTML</acronym> widget which can be used to display <acronym>HTML</acronym>
documents from files or content generated dynamically in S. 
<br><dt><strong>RGtkIPrimitives</strong><dd>A collection of low-level primitives for interactive use with R graphics
and the gtkDevice using <strong>RGtk</strong>. 
<br><dt><strong>RGtkViewers</strong><dd>A collection of tools for viewing different S objects, databases, class
and widget hierarchies, S source file contents, etc. 
<br><dt><strong>RJavaDevice</strong><dd>A graphics device for R that uses Java components and graphics. 
<acronym>API</acronym>s. 
<br><dt><strong>RMatlab</strong><dd>A bi-directional interface between R and Matlab. 
<br><dt><strong>RObjectTables</strong><dd>The C and S code allows one to define R objects to be used as elements
of the search path with their own semantics and facilities for reading
and writing variables.  The objects implement a simple interface via R
functions (either methods or closures) and can access external data,
e.g., in other applications, languages, formats, <small class="dots">...</small>
<br><dt><strong>RSMethods</strong><dd>An implementation of S version 4 methods and classes for R, consistent
with the basic material in &ldquo;Programming with Data&rdquo; by John
M. Chambers, 1998, Springer NY. 
<br><dt><strong>RSPerl</strong><dd>An interface from R to an embedded, persistent Perl interpreter,
allowing one to call arbitrary Perl subroutines, classes and methods. 
<br><dt><strong>RSPython</strong><dd>Allows Python programs to invoke S functions, methods, etc., and S code
to call Python functionality. 
<br><dt><strong>RXLisp</strong><dd>An interface to call XLisp-Stat functions from within R. 
<br><dt><strong>Rcompression</strong><dd>In-memory decompression for <acronym>GNU</acronym> zip and bzip2 formats. 
<br><dt><strong>Rlibstree</strong><dd>Suffix Trees in R via the libstree C library. 
<br><dt><strong>Rstem</strong><dd>Interface to Snowball implementation of Porter's word stemming
algorithm. 
<br><dt><strong>RwxWidgets</strong><dd>Facilities to program <acronym>GUI</acronym>s using wxWidgets in R. 
<br><dt><strong>Ryacas</strong><dd>R interface to yacas. 
<br><dt><strong>SASXML</strong><dd>Example for reading <acronym>XML</acronym> files in SAS 8.2 manner. 
<br><dt><strong>SJava</strong><dd>An interface from R to Java to create and call Java objects and
methods. 
<br><dt><strong>SLanguage</strong><dd>Functions and C support utilities to support S language programming
that can work in both R and <span class="sc">S-Plus</span>. 
<br><dt><strong>SNetscape</strong><dd>Plugin for Netscape and JavaScript. 
<br><dt><strong>SSOAP</strong><dd>A client interface to SOAP (Simple Object Access Protocol) servers from
within S. 
<br><dt><strong>SWinRegistry</strong><dd>Provides access from within R to read and write the Windows registry. 
<br><dt><strong>SWinTypeLibs</strong><dd>Provides ways to extract type information from type libraries and/or
DCOM objects that describes the methods, properties, etc., of an
interface. 
<br><dt><strong>SXalan</strong><dd>Process <acronym>XML</acronym> documents using <acronym>XSL</acronym> functions implemented in R and
dynamically substituting output from R. 
<br><dt><strong>Slcc</strong><dd>Parses C source code, allowing one to analyze and automatically generate
interfaces from S to that code, including the table of S-accessible
native symbols, parameter count and type information, S constructors
from C objects, call graphs, etc. 
<br><dt><strong>Sxslt</strong><dd>An extension module for libxslt, the <acronym>XML</acronym>-<acronym>XSL</acronym> document translator,
that allows <acronym>XSL</acronym> functions to be implemented via R functions. 
<br><dt><strong>XML</strong><dd>Tools for reading <acronym>XML</acronym> documents and DTDs. 
</dl>

<div class="node">
<p><hr>
<a name="Add-on-packages-from-Bioconductor"></a>
<a name="Add_002don-packages-from-Bioconductor"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Other-add_002don-packages">Other add-on packages</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Add_002don-packages-from-Omegahat">Add-on packages from Omegahat</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Which-add_002don-packages-exist-for-R_003f">Which add-on packages exist for R?</a>

</div>

<h4 class="subsection">5.1.4 Add-on packages from Bioconductor</h4>

<p>The <a href="http://www.bioconductor.org/">Bioconductor Project</a> produces an
open source software framework that will assist biologists and
statisticians working in bioinformatics, with primary emphasis on
inference using DNA microarrays.  A <acronym>CRAN</acronym> style R package
repository is available via <a href="http://www.bioconductor.org/">http://www.bioconductor.org/</a>.

   <p>The following R packages are contained in the current release of
Bioconductor, with more packages under development.

     <dl>
<dt><strong>ABarray</strong><dd>Microarray QA and statistical data analysis for Applied Biosystems
Genome Survey Micorarray (AB1700) gene expression data. 
<br><dt><strong>ACME</strong><dd>Algorithms for Calculating Microarray Enrichment (ACME). 
<br><dt><strong>AffyCompatible</strong><dd>Affymetrix GeneChip software compatibility
<br><dt><strong>AffyExpress</strong><dd>Affymetrix quality assessment and analysis tool. 
<br><dt><strong>AnnBuilder</strong><dd>Processing annotation date from public data repositories and building
annoation data packages or XML data documents using the source data. 
<br><dt><strong>AnnotationDbi</strong><dd>Annotation DataBase Interface. 
<br><dt><strong>BAC</strong><dd>Bayesian Analysis of Chip-chip experiment. 
<br><dt><strong>BCRANK</strong><dd>Predicting binding site consensus from ranked DNA sequences. 
<br><dt><strong>BGmix</strong><dd>Bayesian models for differential gene expression. 
<br><dt><strong>BSgenome</strong><dd>Infrastructure for Biostrings-based genome data packages. 
<!-- Still present in development branch but not in 2.2 release. -->
<!-- @item BeadExplorer -->
<!-- QC, normalization, annotation and exploration of Illumina BeadChip data. -->
<br><dt><strong>BioMVCClass</strong><dd>Model-View-Controller (MVC) classes that use <strong>Biobase</strong>. 
<br><dt><strong>Biobase</strong><dd>Object-oriented representation and manipulation of genomic data (S4
class structure). 
<br><dt><strong>BiocCaseStudies</strong><dd>Support for the Bioconductor Case Studies monograph. 
<br><dt><strong>Biostrings</strong><dd>Class definitions and generics for biological sequences along with
pattern matching algorithms. 
<br><dt><strong>BufferedMatrix</strong><dd>Microarray analysis methods that use BufferedMatrix objects. 
<br><dt><strong>BufferedMatrixMethods</strong><dd>A matrix data storage object held in temporary files. 
<br><dt><strong>CALIB</strong><dd>Calibration model for estimating absolute expression levels from
microarray data. 
<br><dt><strong>CGHcall</strong><dd>Calling aberrations for array CGH tumor profiles. 
<br><dt><strong>CORREP</strong><dd>Multivariate correlation estimation and statistical inference. 
<br><dt><strong>Category</strong><dd>A collection of tools for performing category analysis. 
<!-- Retired from Bioconductor as of 2.2 release. -->
<!-- @item ChromoViz -->
<!-- Draw gene expression profile onto chromosome using @acronym{SVG}. -->
<br><dt><strong>CoCiteStats</strong><dd>A collection of software tools for dealing with co-citation data. 
<br><dt><strong>DEDS</strong><dd>Differential Expression via Distance Summary for microarray data. 
<br><dt><strong>DNAcopy</strong><dd>Segments DNA copy number data using circular binary segmentation to
detect regions with abnormal copy number. 
<br><dt><strong>DynDoc</strong><dd>Functionality to create and interact with dynamic documents, vignettes,
and other navigable documents. 
<br><dt><strong>EBImage</strong><dd>Image processing and image analysis toolkit. 
<br><dt><strong>EBarrays</strong><dd>Empirical Bayes tools for the analysis of replicated microarray data
across multiple conditions. 
<br><dt><strong>GEOmetadb</strong><dd>A compilation of metadata from NCBI GEO. 
<br><dt><strong>GEOquery</strong><dd>Get data from NCBI Gene Expression Omnibus (GEO). 
<br><dt><strong>GGBase</strong><dd>Infrastructure for genetics of gene expression. 
<br><dt><strong>GGtools</strong><dd>Software and data for genetical genomics. 
<br><dt><strong>GLAD</strong><dd>Gain and Loss Analysis of DNA. 
<br><dt><strong>GOstats</strong><dd>Tools for manipulating GO and microarrays. 
<br><dt><strong>GSEABase</strong><dd>Gene set enrichment data structures and methods. 
<br><dt><strong>GSEAlm</strong><dd>Linear model toolset for Gene Set Enrichment Analysis. 
<br><dt><strong>GeneMeta</strong><dd>A collection of meta-analysis tools for analyzing high throughput
experimental data. 
<br><dt><strong>GeneR</strong><dd>Package manipulating nucleotidic sequences (Embl, Fasta, GenBank). 
<br><dt><strong>GeneRfold</strong><dd>R for genes and sequences, using viennaRNA package (fold). 
<br><dt><strong>GeneSelector</strong><dd>GeneSelector. 
<br><dt><strong>GeneSpring</strong><dd>Functions and class definitions to be able to read and write GeneSpring
specific data objects and convert them to Bioconductor objects. 
<!-- Retired from Bioconductor as of 2.2 release. -->
<!-- @item GeneTS -->
<!-- A package for analysing multiple gene expression time series data. -->
<!-- Currently, implements methods for cell cycle analysis and for inferring -->
<!-- large sparse graphical Gaussian models. -->
<br><dt><strong>GeneTraffic</strong><dd>GeneTraffic R integration functions. 
<br><dt><strong>GeneticsBase</strong><dd>Classes and functions for handling genetic data. 
<br><dt><strong>GeneticsDesign</strong><dd>Functions for designing genetics studies. 
<br><dt><strong>GeneticsPed</strong><dd>Pedigree and genetic relationship functions. 
<br><dt><strong>GenomeGraphs</strong><dd>Plotting genomic information from Ensembl. 
<br><dt><strong>GlobalAncova</strong><dd>Calculates a global test for differential gene expression between
groups. 
<br><dt><strong>GraphAT</strong><dd>Graph theoretic Association Tests. 
<br><dt><strong>GraphAlignment</strong><dd>GraphAlignment. 
<br><dt><strong>HEM</strong><dd>Heterogeneous Error Model for analysis of microarray data. 
<br><dt><strong>Harshlight</strong><dd>A &ldquo;corrective make-up&rdquo; program for microarray chips. 
<br><dt><strong>Heatplus</strong><dd>A heat map displaying covariates and coloring clusters. 
<br><dt><strong>Icens</strong><dd>Functions for computing the NPMLE for censored and truncated data. 
<br><dt><strong>KEGGSOAP</strong><dd>Client-side SOAP access KEGG. 
<br><dt><strong>LBE</strong><dd>Estimation of the false discovery rate. 
<br><dt><strong>LMGene</strong><dd>Analysis of microarray data using a linear model and glog data
transformation. 
<br><dt><strong>LPE</strong><dd>Significance analysis of microarray data with small number of replicates
using the Local Pooled Error (LPE) method. 
<br><dt><strong>MANOR</strong><dd>Micro-Array NORmalization. 
<!-- Still present in development branch but not in 2.2 release. -->
<!-- @item MCRestimate -->
<!-- Misclassification error estimation with cross-validation. -->
<br><dt><strong>MLInterfaces</strong><dd>Uniform interfaces to machine learning code for the exprSet class from
Bioconductor. 
<br><dt><strong>MVCClass</strong><dd>Model-View-Controller (MVC) classes. 
<br><dt><strong>MantelCorr</strong><dd>Compute Mantel Cluster Correlations. 
<br><dt><strong>MassSpecWavelet</strong><dd>Mass spectrum processing by wavelet-based algorithms. 
<br><dt><strong>MeasurementError.cor</strong><dd>Two-stage measurement error model for correlation estimation with
smaller bias than the usual sample correlation. 
<br><dt><strong>MergeMaid</strong><dd>Cross-study comparison of gene expression array data. 
<br><dt><strong>Mfuzz</strong><dd>Soft clustering of time series gene expression data. 
<br><dt><strong>MiPP</strong><dd>Misclassification Penalized Posterior Classification. 
<br><dt><strong>OCplus</strong><dd>Operating characteristics plus sample size and local fdr for microarray
experiments. 
<br><dt><strong>OLIN</strong><dd>Optimized Local Intensity-dependent Normalisation of two-color
microarrays. 
<br><dt><strong>OLINgui</strong><dd>Graphical user interface for <strong>OLIN</strong>. 
<br><dt><strong>OrderedList</strong><dd>Similarities of ordered gene lists. 
<br><dt><strong>OutlierD</strong><dd>Outlier detection using quantile regression on the M-A scatterplots of
high-throughput data. 
<br><dt><strong>PCpheno</strong><dd>Phenotypes and cellular organizational units. 
<br><dt><strong>PGSEA</strong><dd>Parametric Gene Set Enrichment Analysis. 
<br><dt><strong>PROcess</strong><dd>Ciphergen SELDI-TOF processing. 
<br><dt><strong>RBGL</strong><dd>An interface between the graph package and the Boost graph libraries,
allowing for fast manipulation of graph objects in R. 
<br><dt><strong>RBioinf</strong><dd>Support for R for Bioinformatics monograph. 
<br><dt><strong>RLMM</strong><dd>A genotype calling algorithm for Affymetrix SNP arrays. 
<br><dt><strong>RMAGEML</strong><dd>Functionality to handle MAGEML documents. 
<!-- Retired from Bioconductor as of 2.2 release. -->
<!-- @item RMAPPER -->
<!-- Interface to mapper.chip.org. -->
<br><dt><strong>ROC</strong><dd>Receiver Operating Characteristic (ROC) approach for identifying genes
that are differentially expressed in two types of samples. 
<br><dt><strong>RSNPper</strong><dd>Interface to chip.org::SNPper for SNP-related data. 
<br><dt><strong>RWebServices</strong><dd>Expose R functions as web services through Java/Axis/Apache. 
<br><dt><strong>RankProd</strong><dd>Rank Product method for identifying differentially expressed genes. 
<br><dt><strong>RbcBook1</strong><dd>Support for Springer monograph on Bioconductor. 
<br><dt><strong>Rdbi</strong><dd>Generic framework for database access in R. 
<br><dt><strong>RdbiPgSQL</strong><dd>Methods for accessing data stored in PostgreSQL tables. 
<br><dt><strong>Rdisop</strong><dd>Decomposition of isotopic patterns. 
<br><dt><strong>RefPlus</strong><dd>Functions for pre-processing Affymetrix data using the RMA+ and the
RMA++ methods. 
<br><dt><strong>Resourcerer</strong><dd>Read annotation data from TIGR Resourcerer or convert the annotation
data into Bioconductor data package. 
<br><dt><strong>Rgraphviz</strong><dd>An interface with Graphviz for plotting graph objects in R. 
<br><dt><strong>Ringo</strong><dd>R Investigation of NimbleGen Oligoarrays.. 
<br><dt><strong>Rintact</strong><dd>Interface to EBI Intact protein interaction data base. 
<br><dt><strong>Rredland</strong><dd>Interface to redland RDF utilities. 
<br><dt><strong>Rtreemix</strong><dd>Mutagenetic trees mixture models. 
<br><dt><strong>Ruuid</strong><dd>Creates Universally Unique ID values (UUIDs) in R. 
<!-- Still present in development branch but not in 2.2 release. -->
<!-- @item SAGElyzer -->
<!-- Locates genes based on SAGE tags. -->
<br><dt><strong>SAGx</strong><dd>Statistical Analysis of the GeneChip. 
<br><dt><strong>SBMLR</strong><dd>Systems Biology Markup Language (SBML) interface and biochemical system
analysis tools. 
<br><dt><strong>SLGI</strong><dd>Synthetic Lethal Genetic Interaction. 
<br><dt><strong>SLqPCR</strong><dd>Functions for analysis of real-time quantitative PCR data at SIRS-Lab
GmbH. 
<br><dt><strong>SMAP</strong><dd>A Segmental Maximum A Posteriori approach to array-CGH copy number
profiling. 
<br><dt><strong>SNPchip</strong><dd>Classes and methods for high throughput SNP chip data. 
<br><dt><strong>ScISI</strong><dd>In Silico Interactome. 
<br><dt><strong>SemSim</strong><dd>Gene ontology-based semantic similarity measures. 
<br><dt><strong>TypeInfo</strong><dd>Optional type specification prototype. 
<br><dt><strong>VanillaICE</strong><dd>Methods for fitting Hidden Markov Models to SNP chip data. 
<br><dt><strong>XDE</strong><dd>A Bayesian hierarchical model for cross-study analysis of differential
gene expression. 
<br><dt><strong>aCGH</strong><dd>Classes and functions for Array Comparative Genomic Hybridization data. 
<br><dt><strong>adSplit</strong><dd>Annotation-driven clustering. 
<br><dt><strong>affxparser</strong><dd>Package for parsing Affymetrix files (CDF, CEL, CHP, BPMAP, BAR). 
<br><dt><strong>affy</strong><dd>Methods for Affymetrix Oligonucleotide Arrays. 
<br><dt><strong>affyPLM</strong><dd>For fitting Probe Level Models. 
<br><dt><strong>affyPara</strong><dd>Parallelized preprocessing methods for Affymetrix Oligonucleotide
Arrays. 
<br><dt><strong>affyQCReport</strong><dd>QC Report Generation for affyBatch objects. 
<br><dt><strong>affycomp</strong><dd>Graphics toolbox for assessment of Affymetrix expression measures. 
<br><dt><strong>affycoretools</strong><dd>Functions useful for those doing repetitive analyses. 
<br><dt><strong>affyio</strong><dd>Tools for parsing Affymetrix data files. 
<br><dt><strong>affylmGUI</strong><dd>Graphical User Interface for affy analysis using package <strong>limma</strong>. 
<br><dt><strong>affypdnn</strong><dd>Probe Dependent Nearest Neighbors (PDNN) for the <strong>affy</strong> package. 
<br><dt><strong>altcdfenvs</strong><dd>Utilities to handle cdfenvs. 
<br><dt><strong>annaffy</strong><dd>Functions for handling data from Bioconductor Affymetrix annotation data
packages. 
<br><dt><strong>annotate</strong><dd>Associate experimental data in real time to biological metadata from web
databases such as GenBank, LocusLink and PubMed.  Process and store
query results.  Generate <acronym>HTML</acronym> reports of analyses. 
<br><dt><strong>annotationTools</strong><dd>Annotate microarrays and perform cross-species gene expression analyses
using flat file databases. 
<br><dt><strong>apComplex</strong><dd>Estimate protein complex membership using AP-MS protein data. 
<br><dt><strong>aroma.light</strong><dd>Light-weight methods for normalization and visualization of microarray
data using only basic R data types. 
<!-- Retired from Bioconductor as of 2.2 release. -->
<!-- @item arrayMagic -->
<!-- Utilities for quality control and processing for two-color cDNA -->
<!-- microarray data. -->
<br><dt><strong>arrayQuality</strong><dd>Performing print-run and array level quality assessment. 
<br><dt><strong>arrayQualityMetrics</strong><dd>Quality metrics on ExpressionSets. 
<br><dt><strong>beadarray</strong><dd>Quality control and low-level analysis of BeadArrays. 
<br><dt><strong>beadarraySNP</strong><dd>Normalization and reporting of Illumina SNP bead arrays. 
<br><dt><strong>bgafun</strong><dd>A method to identify specifity determining residues in protein
families. 
<br><dt><strong>bgx</strong><dd>Bayesian Gene eXpression. 
<br><dt><strong>bioDist</strong><dd>A collection of software tools for calculating distance measures. 
<br><dt><strong>biocGraph</strong><dd>Graph examples and use cases in Bioinformatics. 
<br><dt><strong>biocViews</strong><dd>Categorized views of R package repositories. 
<br><dt><strong>biomaRt</strong><dd>Interface to BioMart databases (e.g., Ensembl)
<br><dt><strong>bridge</strong><dd>Bayesian Robust Inference for Differential Gene Expression. 
<br><dt><strong>cellHTS</strong><dd>Analysis of cell-based screens. 
<br><dt><strong>cellHTS2</strong><dd>Analysis of cell-based screens &mdash; revised version of <strong>cellHTS</strong>. 
<br><dt><strong>cghMCR</strong><dd>Find chromosome regions showing common gains/losses. 
<br><dt><strong>clusterStab</strong><dd>Compute cluster stability scores for microarray data. 
<br><dt><strong>codelink</strong><dd>Manipulation of Codelink Bioarrays data. 
<br><dt><strong>convert</strong><dd>Convert Microarray Data Objects. 
<br><dt><strong>copa</strong><dd>Functions to perform cancer outlier profile analysis. 
<br><dt><strong>cosmo</strong><dd>Supervised detection of conserved motifs in DNA sequences. 
<br><dt><strong>cosmoGUI</strong><dd>GUI for constructing constraint sets used by the <strong>cosmo</strong>
package. 
<br><dt><strong>ctc</strong><dd>Tools to export and import Tree and Cluster to other programs. 
<br><dt><strong>daMA</strong><dd>Functions for the efficient design of factorial two-color microarray
experiments and for the statistical analysis of factorial microarray
data. 
<br><dt><strong>diffGeneAnalysis</strong><dd>Performs differential Gene expression Analysis. 
<br><dt><strong>ecolitk</strong><dd>Metadata and tools to work with E. coli. 
<br><dt><strong>edd</strong><dd>Expression density diagnostics: graphical methods and pattern
recognition algorithms for distribution shape classification. 
<br><dt><strong>exonmap</strong><dd>High level analysis of Affymetrix exon array data. 
<br><dt><strong>explorase</strong><dd>GUI for exploratory data analysis of systems biology data. 
<br><dt><strong>externalVector</strong><dd>Vector objects for R with external storage. 
<br><dt><strong>factDesign</strong><dd>A set of tools for analyzing data from factorial designed microarray
experiments.  The functions can be used to evaluate appropriate tests of
contrast and perform single outlier detection. 
<br><dt><strong>fbat</strong><dd>Family Based Association Tests for genetic data. 
<br><dt><strong>fdrame</strong><dd>FDR Adjustments of Microarray Experiments (FDR-AME). 
<br><dt><strong>flowClust</strong><dd>Clustering for flow cytometry. 
<br><dt><strong>flowCore</strong><dd>Basic structures for flow cytometry data. 
<br><dt><strong>flowQ</strong><dd>Qualitiy control for flow cytometry. 
<br><dt><strong>flowUtils</strong><dd>Utilities for flow cytometry. 
<br><dt><strong>flowViz</strong><dd>Visualization for flow cytometry. 
<br><dt><strong>gaga</strong><dd>GaGa hierarchical model for microarray data analysis. 
<br><dt><strong>gaggle</strong><dd>Broadcast data between R and Java bioinformatics programs. 
<br><dt><strong>gcrma</strong><dd>Background adjustment using sequence information. 
<br><dt><strong>genArise</strong><dd>A tool for dual color microarray data. 
<br><dt><strong>geneRecommender</strong><dd>A gene recommender algorithm to identify genes coexpressed with a query
set of genes. 
<br><dt><strong>genefilter</strong><dd>Tools for sequentially filtering genes using a wide variety of filtering
functions.  Example of filters include: number of missing value,
coefficient of variation of expression measures, ANOVA p-value,
Cox model p-values.  Sequential application of filtering
functions to genes. 
<br><dt><strong>geneplotter</strong><dd>Graphical tools for genomic data, for example for plotting expression
data along a chromosome or producing color images of expression data
matrices. 
<br><dt><strong>globaltest</strong><dd>Testing globally whether a group of genes is significantly related to
some clinical variable of interest. 
<br><dt><strong>goProfiles</strong><dd>Statistical analysis of functional profiles. 
<br><dt><strong>goTools</strong><dd>Functions for description/comparison of oligo ID list using the Gene
Ontology database. 
<br><dt><strong>gpls</strong><dd>Classification using generalized partial least squares for two-group and
multi-group classification. 
<br><dt><strong>graph</strong><dd>Classes and tools for creating and manipulating graphs within R. 
<br><dt><strong>hexbin</strong><dd>Binning functions, in particular hexagonal bins for graphing. 
<br><dt><strong>hopach</strong><dd>Hierarchical Ordered Partitioning and Collapsing Hybrid (HOPACH). 
<br><dt><strong>hypergraph</strong><dd>Capabilities for representing and manipulating hypergraphs. 
<!-- Retired from Bioconductor as of 2.2 release. -->
<!-- @item iSNetwork -->
<!-- Interactive network plots. -->
<!-- Retired from Bioconductor as of 2.2 release. -->
<!-- @item iSPlot -->
<!-- Link views that are based on the same data set. -->
<br><dt><strong>idiogram</strong><dd>Plotting genomic data by chromosomal location. 
<br><dt><strong>impute</strong><dd>Imputation for microarray data (currently KNN only). 
<br><dt><strong>keggorth</strong><dd>Graph support for KO, KEGG Orthology. 
<br><dt><strong>lapmix</strong><dd>Laplace mixture model in microarray experiments. 
<br><dt><strong>limma</strong><dd>Linear models for microarray data. 
<br><dt><strong>limmaGUI</strong><dd>Graphical User Interface for package <strong>limma</strong>. 
<br><dt><strong>logicFS</strong><dd>Identification of SNP interactions. 
<br><dt><strong>lumi</strong><dd>BeadArray specific methods for Illumina microarrays. 
<br><dt><strong>maCorrPlot</strong><dd>Visualize artificial correlation in microarray data. 
<br><dt><strong>maDB</strong><dd>Microarray database and utility functions for microarray analysis. 
<br><dt><strong>maSigPro</strong><dd>Significant gene expression profile differeneces in time course
microarray data. 
<br><dt><strong>maanova</strong><dd>Tools for analyzing micro array experiments. 
<br><dt><strong>macat</strong><dd>MicroArray Chromosome Analysis Tool. 
<br><dt><strong>made4</strong><dd>Multivariate analysis of microarray data using ADE4. 
<br><dt><strong>maigesPack</strong><dd>Functions to handle cDNA microarray data, including several methods of
data analysis. 
<br><dt><strong>makePlatformDesign</strong><dd>Creates the Platform Design environments (PDenvs) required by
<strong>oligo</strong>. 
<br><dt><strong>makecdfenv</strong><dd>Two functions.  One reads a Affymetrix chip description file (CDF) and
creates a hash table environment containing the location/probe set
membership mapping.  The other creates a package that automatically loads
that environment. 
<br><dt><strong>marray</strong><dd>Exploratory analysis for two-color spotted microarray data. 
<br><dt><strong>matchprobes</strong><dd>Tools for sequence matching of probes on arrays. 
<br><dt><strong>mdqc</strong><dd>Mahalanobis Distance Quality Control for microarrays. 
<br><dt><strong>metaArray</strong><dd>Integration of microarray data for meta-analysis. 
<br><dt><strong>multtest</strong><dd>Multiple testing procedures for controlling the family-wise error rate
(FWER) and the false discovery rate (FDR).  Tests can be based on
t- or F-statistics for one- and two-factor designs, and
permutation procedures are available to estimate adjusted
p-values. 
<br><dt><strong>nem</strong><dd>Nested Effects Models to reconstruct phenotypic hierarchies. 
<br><dt><strong>nnNorm</strong><dd>Spatial and intensity based normalization of cDNA microarray data based
on robust neural nets. 
<br><dt><strong>nudge</strong><dd>Normal Uniform Differential Gene Expression detection. 
<br><dt><strong>occugene</strong><dd>Functions for multinomial occupancy distribution. 
<br><dt><strong>oligo</strong><dd>Oligonucleotide arrays. 
<br><dt><strong>oligoClasses</strong><dd>Classes for high-throughput SNP arrays. 
<br><dt><strong>oneChannelGUI</strong><dd>Extend the capabilities of <strong>affylmGUI</strong>. 
<br><dt><strong>ontoTools</strong><dd>Graphs and sparse matrices for working with ontologies; formal objects
for nomenclatures with provenance management. 
<br><dt><strong>pamr</strong><dd><!-- CRAN -->
Pam: Prediction Analysis for Microarrays. 
<br><dt><strong>panp</strong><dd>Presence-Absence calls from Negative strand matching Probesets. 
<br><dt><strong>pathRender</strong><dd>Render molecular pathways. 
<br><dt><strong>pcaMethods</strong><dd>A collection of PCA methods. 
<br><dt><strong>pcot2</strong><dd>Principal coordinates and Hotelling's T-square method. 
<br><dt><strong>pdInfoBuilder</strong><dd>Platform design information package builder. 
<br><dt><strong>pdmclass</strong><dd>CLASSification of microarray samples using Penalized Discriminant
Methods. 
<br><dt><strong>pgUtils</strong><dd>Utility functions for PostgreSQL databases. 
<br><dt><strong>pickgene</strong><dd>Adaptive gene picking for microarray expression data analysis. 
<br><dt><strong>pkgDepTools</strong><dd>Package dependency tools. 
<br><dt><strong>plgem</strong><dd>Power Law Global Error Model. 
<br><dt><strong>plier</strong><dd>Implements the Affymetrix PLIER (Probe Logarithmic Error Intensity
Estimate) algorithm. 
<br><dt><strong>plw</strong><dd>Probe level Locally moderated Weighted t-tests. 
<br><dt><strong>ppiStats</strong><dd>Protein-Protein Interaction Statistical package. 
<br><dt><strong>prada</strong><dd>Tools for analyzing and navigating data from high-throughput phenotyping
experiments based on cellular assays and fluorescent detection. 
<br><dt><strong>preprocessCore</strong><dd>A collection of pre-processing functions. 
<br><dt><strong>puma</strong><dd>Propagating Uncertainty in Microarray Analysis. 
<br><dt><strong>quantsmooth</strong><dd>Quantile smoothing and genomic visualization of array data. 
<br><dt><strong>qvalue</strong><dd>Q-value estimation for false discovery rate control. 
<br><dt><strong>rHVDM</strong><dd>Hidden Variable Dynamic Modeling. 
<br><dt><strong>rama</strong><dd>Robust Analysis of MicroArrays: robust estimation of cDNA microarray
intensities with replicates using a Bayesian hierarchical model. 
<br><dt><strong>rbsurv</strong><dd>Robust likelihood-based survival modeling with microarray data. 
<br><dt><strong>reb</strong><dd>Regional Expression Biases. 
<br><dt><strong>rflowcyt</strong><dd>Statistical tools and data structures for analytic flow cytometry. 
<br><dt><strong>rsbml</strong><dd>R support for SBML, using libsbml. 
<br><dt><strong>rtracklayer</strong><dd>R interface to genome browsers and their annotation tracks. 
<br><dt><strong>safe</strong><dd>Significance Analysis of Function and Expression. 
<br><dt><strong>sagenhaft</strong><dd>Functions for reading and comparing SAGE (Serial Analysis of Gene
Expression) libraries. 
<br><dt><strong>seqLogo</strong><dd>Sequence logos for DNA sequence alignments. 
<br><dt><strong>sigPathway</strong><dd>Pathway analysis. 
<br><dt><strong>siggenes</strong><dd>Identifying differentially expressed genes and estimating the False
Discovery Rate (FDR) with both the Significance Analysis of Microarrays
(SAM) and the Empirical Bayes Analyses of Microarrays (EBAM). 
<br><dt><strong>simpleaffy</strong><dd>Very simple high level analysis of Affymetrix data. 
<br><dt><strong>simulatorAPMS</strong><dd>Computationally simulates the AP-MS technology. 
<br><dt><strong>sizepower</strong><dd>Sample size and power calculation in microrarray studies. 
<br><dt><strong>snapCGH</strong><dd>Segmentation, normalization and processing of aCGH data. 
<br><dt><strong>snpMatrix</strong><dd>The snp.matrix and X.snp.matrix classes. 
<br><dt><strong>spikeLI</strong><dd>Affymetrix Spike-in Langmuir Isotherm data analysis tool. 
<br><dt><strong>splicegear</strong><dd>A set of tools to work with alternative splicing. 
<br><dt><strong>splots</strong><dd>Visualization routines for high throughput screens. 
<br><dt><strong>spotSegmentation</strong><dd>Microarray spot segmentation and gridding for blocks of microarray
spots. 
<br><dt><strong>sscore</strong><dd>S-score algorithm for Affymetrix oligonucleotide microarrays. 
<br><dt><strong>ssize</strong><dd>Estimate microarry sample size. 
<br><dt><strong>stam</strong><dd>STructured Analysis of Microarray data. 
<br><dt><strong>stepNorm</strong><dd>Stepwise normalization functions for cDNA microarrays. 
<br><dt><strong>tilingArray</strong><dd>Analysis of tiling arrays. 
<br><dt><strong>timecourse</strong><dd>Statistical analysis for developmental microarray time course data. 
<br><dt><strong>tkWidgets</strong><dd>Widgets in Tcl/Tk that provide functionality for Bioconductor packages. 
<br><dt><strong>topGO</strong><dd>Enrichment analysis for Gene Ontology. 
<br><dt><strong>twilight</strong><dd>Estimation of local false discovery rate. 
<br><dt><strong>vbmp</strong><dd>Variational Bayesian Multinomial Probit regression. 
<br><dt><strong>vsn</strong><dd>Calibration and variance stabilizing transformations for both Affymetrix
and cDNA array data. 
<br><dt><strong>weaver</strong><dd>Tools and extensions for processing Sweave documents. 
<br><dt><strong>webbioc</strong><dd>Integrated web interface for doing microarray analysis using several of
the Bioconductor packages. 
<br><dt><strong>widgetInvoke</strong><dd>Evaluation widgets for functions. 
<br><dt><strong>widgetTools</strong><dd>Tools for creating Tcl/Tk widgets, i.e., small-scale graphical user
interfaces. 
<br><dt><strong>xcms</strong><dd>LC/MS and GC/MS data analysis: framework for processing and
visualization of chromatographically separated mass spectral data. 
<br><dt><strong>xps</strong><dd>Methods for processing and analysis of Affymetrix Oligonucleotide
Arrays. 
<br><dt><strong>yaqcaffy</strong><dd>Affymetrix expression data quality control and reproducibility
analysis. 
</dl>

<!-- These packages will eventually also be made available via @CRAN{} -->
<!-- as well. -->
<div class="node">
<p><hr>
<a name="Other-add-on-packages"></a>
<a name="Other-add_002don-packages"></a>
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Add_002don-packages-from-Bioconductor">Add-on packages from Bioconductor</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Which-add_002don-packages-exist-for-R_003f">Which add-on packages exist for R?</a>

</div>

<h4 class="subsection">5.1.5 Other add-on packages</h4>

<p><a href="mailto:jlindsey@gen.unimaas.nl">Jim Lindsey</a> has written a collection
of R packages for nonlinear regression and repeated measurements,
consisting of <strong>event</strong> (event history procedures and models),
<strong>gnlm</strong> (generalized nonlinear regression models), <strong>growth</strong>
(multivariate normal and elliptically-contoured repeated measurements
models), <strong>repeated</strong> (non-normal repeated measurements models),
<strong>rmutil</strong> (utilities for nonlinear regression and repeated
measurements), and <strong>stable</strong> (probability functions and
generalized regression models for stable distributions).  All analyses
in the new edition of his book &ldquo;Models for Repeated Measurements&rdquo;
(1999, Oxford University Press) were carried out using these packages. 
Jim has also started <strong>dna</strong>, a package with procedures for the
analysis of DNA sequences.  Jim's packages can be obtained from
<a href="http://popgen.unimaas.nl/~jlindsey/rcode.html">http://popgen.unimaas.nl/~jlindsey/rcode.html</a>.

   <p>More code has been posted to the R-help mailing list, and can be
obtained from the mailing list archive.

<div class="node">
<p><hr>
<a name="How-can-add-on-packages-be-installed%3f"></a>
<a name="How-can-add_002don-packages-be-installed_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-add_002don-packages-be-used_003f">How can add-on packages be used?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Which-add_002don-packages-exist-for-R_003f">Which add-on packages exist for R?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Add_002dOn-Packages">R Add-On Packages</a>

</div>

<h3 class="section">5.2 How can add-on packages be installed?</h3>

<p>(Unix only.)  The add-on packages on <acronym>CRAN</acronym> come as gzipped tar
files named <var>pkg</var><code>_</code><var>version</var><code>.tar.gz</code>, which may in fact be
&ldquo;bundles&rdquo; containing more than one package.  Provided that
<samp><span class="command">tar</span></samp> and <samp><span class="command">gzip</span></samp> are available on your system, type

<pre class="example">     $ R CMD INSTALL /path/to/<var>pkg</var>_<var>version</var>.tar.gz
</pre>
   <p class="noindent">at the shell prompt to install to the library tree rooted at the first
directory in your library search path (see the help page for
<code>.libPaths()</code> for details on how the search path is determined).

   <p>To install to another tree (e.g., your private one), use

<pre class="example">     $ R CMD INSTALL -l <var>lib</var> /path/to/<var>pkg</var>_<var>version</var>.tar.gz
</pre>
   <p class="noindent">where <var>lib</var> gives the path to the library tree to install to.

   <p>Even more conveniently, you can install and automatically update
packages from within R if you have access to repositories such as
<acronym>CRAN</acronym>.  See the help page for <code>available.packages()</code> for more
information.

<!-- <COMMENT> -->
<!-- This is really no longer quite accurate (R_LIBS_USER is preferred to -->
<!-- R_LIBS), and described in ?libPaths anyways ... hence comment out. -->
<!-- You can use several library trees of add-on packages.  The easiest way -->
<!-- to tell R to use these is via the environment variable @env{R_LIBS} -->
<!-- which should be a colon-separated list of directories at which R library -->
<!-- trees are rooted.  You do not have to specify the default tree in -->
<!-- @env{R_LIBS}.  E.g., to use a private tree in @file{$HOME/lib/R} and a -->
<!-- public site-wide tree in @file{/usr/local/lib/R-contrib}, put -->
<!-- @example -->
<!-- R_LIBS="$HOME/lib/R:/usr/local/lib/R-contrib"; export R_LIBS -->
<!-- @end example -->
<!-- @noindent -->
<!-- into your (Bourne) shell profile or even preferably, add the line -->
<!-- @example -->
<!-- R_LIBS="~/lib/R:/usr/local/lib/R-contrib" -->
<!-- @end example -->
<!-- @noindent -->
<!-- your environment (e.g., @file{~/.Renviron}) file.  (Note that no -->
<!-- @code{export} statement is needed or allowed in this file; see the -->
<!-- on-line help for @code{Startup} for more information.) -->
<div class="node">
<p><hr>
<a name="How-can-add-on-packages-be-used%3f"></a>
<a name="How-can-add_002don-packages-be-used_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-add_002don-packages-be-removed_003f">How can add-on packages be removed?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-add_002don-packages-be-installed_003f">How can add-on packages be installed?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Add_002dOn-Packages">R Add-On Packages</a>

</div>

<h3 class="section">5.3 How can add-on packages be used?</h3>

<p>To find out which additional packages are available on your system, type

<pre class="example">     library()
</pre>
   <p class="noindent">at the R prompt.

   <p>This produces something like

   <blockquote>
<p><table class="cartouche" summary="cartouche" border="1"><tr><td>
     <pre class="smallexample">          Packages in `/home/me/lib/R':
          
          mystuff       My own R functions, nicely packaged but not documented
          
          Packages in `/usr/local/lib/R/library':
          
          KernSmooth    Functions for kernel smoothing for Wand &amp; Jones (1995)
          MASS          Main Package of Venables and Ripley's MASS
          base          The R Base package
          boot          Bootstrap R (S-Plus) Functions (Canty)
          class         Functions for Classification
          cluster       Functions for clustering (by Rousseeuw et al.)
          codetools     Code Analysis Tools for R
          datasets      The R Datasets Package
          foreign       Read Data Stored by Minitab, S, SAS, SPSS, Stata, Systat,
                        dBase, ...
          grDevices     The R Graphics Devices and Support for Colours and Fonts
          graphics      The R Graphics Package
          grid          The Grid Graphics Package
          lattice       Lattice Graphics
          methods       Formal Methods and Classes
          mgcv          GAMs with GCV smoothness estimation and GAMMs by REML/PQ
          nlme          Linear and Nonlinear Mixed Effects Models
          nnet          Feed-forward Neural Networks and Multinomial Log-Linear
                        Models
          rpart         Recursive Partitioning
          spatial       Functions for Kriging and Point Pattern Analysis
          splines       Regression Spline Functions and Classes
          stats         The R Stats Package
          stats4        Statistical functions using S4 Classes
          survival      Survival analysis, including penalised likelihood
          tcltk         Tcl/Tk Interface
          tools         Tools for Package Development
          utils         The R Utils Package
     </pre>
        </td></tr></table>
   </blockquote>

   <p>You can &ldquo;load&rdquo; the installed package <var>pkg</var> by

<pre class="example">     library(<var>pkg</var>)
</pre>
   <p>You can then find out which functions it provides by typing one of

<pre class="example">     library(help = <var>pkg</var>)
     help(package = <var>pkg</var>)
</pre>
   <p>You can unload the loaded package <var>pkg</var> by

<pre class="example">     detach("package:<var>pkg</var>")
</pre>
   <div class="node">
<p><hr>
<a name="How-can-add-on-packages-be-removed%3f"></a>
<a name="How-can-add_002don-packages-be-removed_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-I-create-an-R-package_003f">How can I create an R package?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-add_002don-packages-be-used_003f">How can add-on packages be used?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Add_002dOn-Packages">R Add-On Packages</a>

</div>

<h3 class="section">5.4 How can add-on packages be removed?</h3>

<p>Use

<pre class="example">     $ R CMD REMOVE <var>pkg_1</var> ... <var>pkg_n</var>
</pre>
   <p class="noindent">to remove the packages <var>pkg_1</var>, <small class="dots">...</small>, <var>pkg_n</var> from the
library tree rooted at the first directory given in <samp><span class="env">R_LIBS</span></samp> if this
is set and non-null, and from the default library otherwise.  (Versions
of R prior to 1.3.0 removed from the default library by default.)

   <p>To remove from library <var>lib</var>, do

<pre class="example">     $ R CMD REMOVE -l <var>lib</var> <var>pkg_1</var> ... <var>pkg_n</var>
</pre>
   <div class="node">
<p><hr>
<a name="How-can-I-create-an-R-package%3f"></a>
<a name="How-can-I-create-an-R-package_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-I-contribute-to-R_003f">How can I contribute to R?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-add_002don-packages-be-removed_003f">How can add-on packages be removed?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Add_002dOn-Packages">R Add-On Packages</a>

</div>

<h3 class="section">5.5 How can I create an R package?</h3>

<p>A package consists of a subdirectory containing the files
<samp><span class="file">DESCRIPTION</span></samp> and <samp><span class="file">INDEX</span></samp>, and the subdirectories <samp><span class="file">R</span></samp>,
<samp><span class="file">data</span></samp>, <samp><span class="file">demo</span></samp>, <samp><span class="file">exec</span></samp>, <samp><span class="file">inst</span></samp>, <samp><span class="file">man</span></samp>,
<samp><span class="file">src</span></samp>, and <samp><span class="file">tests</span></samp> (some of which can be missing).  Optionally
the package can also contain script files <samp><span class="file">configure</span></samp> and
<samp><span class="file">cleanup</span></samp> which are executed before and after installation.

   <p>See <a href="R-exts.html#Creating-R-packages">Creating R packages</a>, for details.

   <p>R version 1.3.0 has added the function <code>package.skeleton()</code> which
will set up directories, save data and code, and create skeleton help
files for a set of R functions and datasets.

   <p>See <a href="#What-is-CRAN_003f">What is CRAN?</a>, for information on uploading a package to <acronym>CRAN</acronym>.

<div class="node">
<p><hr>
<a name="How-can-I-contribute-to-R%3f"></a>
<a name="How-can-I-contribute-to-R_003f"></a>
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-I-create-an-R-package_003f">How can I create an R package?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Add_002dOn-Packages">R Add-On Packages</a>

</div>

<h3 class="section">5.6 How can I contribute to R?</h3>

<p>R is in active development and there is always a risk of bugs creeping
in.  Also, the developers do not have access to all possible machines
capable of running R.  So, simply using it and communicating problems is
certainly of great value.

   <p>One place where functionality is still missing is the modeling software
as described in &ldquo;Statistical Models in S&rdquo; (see <a href="#What-is-S_003f">What is S?</a>); some
of the nonlinear modeling code is not there yet.

   <p>The <a href="http://developer.R-project.org/">R Developer Page</a> acts as an
intermediate repository for more or less finalized ideas and plans for
the R statistical system.  It contains (pointers to) TODO lists, RFCs,
various other writeups, ideas lists, and CVS miscellanea.

   <p>Many (more) of the packages available at the Statlib S Repository might
be worth porting to R.

   <p>If you are interested in working on any of these projects, please notify
<a href="mailto:Kurt.Hornik@R-project.org">Kurt Hornik</a>.

<div class="node">
<p><hr>
<a name="R-and-Emacs"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#R-Miscellanea">R Miscellanea</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#R-Add_002dOn-Packages">R Add-On Packages</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Top">Top</a>

</div>

<h2 class="chapter">6 R and Emacs</h2>

<ul class="menu">
<li><a accesskey="1" href="#Is-there-Emacs-support-for-R_003f">Is there Emacs support for R?</a>
<li><a accesskey="2" href="#Should-I-run-R-from-within-Emacs_003f">Should I run R from within Emacs?</a>
<li><a accesskey="3" href="#Debugging-R-from-within-Emacs">Debugging R from within Emacs</a>
</ul>

<div class="node">
<p><hr>
<a name="Is-there-Emacs-support-for-R%3f"></a>
<a name="Is-there-Emacs-support-for-R_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Should-I-run-R-from-within-Emacs_003f">Should I run R from within Emacs?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#R-and-Emacs">R and Emacs</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-and-Emacs">R and Emacs</a>

</div>

<h3 class="section">6.1 Is there Emacs support for R?</h3>

<p>There is an Emacs package called <acronym>ESS</acronym> (&ldquo;Emacs Speaks
Statistics&rdquo;) which provides a standard interface between statistical
programs and statistical processes.  It is intended to provide
assistance for interactive statistical programming and data analysis. 
Languages supported include: S dialects (R, S 3/4, and <span class="sc">S-Plus</span>
3.x/4.x/5.x/6.x/7.x), LispStat dialects (XLispStat, ViSta), SAS, Stata,
and BUGS.

   <p><acronym>ESS</acronym> grew out of the need for bug fixes and extensions to
S-mode 4.8 (which was a <acronym>GNU</acronym> Emacs interface to S/<span class="sc">S-Plus</span>
version 3 only).  The current set of developers desired support for
XEmacs, R, S4, and MS Windows.  In addition, with new modes being
developed for R, Stata, and SAS, it was felt that a unifying interface
and framework for the user interface would benefit both the user and the
developer, by helping both groups conform to standard Emacs usage.  The
end result is an increase in efficiency for statistical programming and
data analysis, over the usual tools.

   <p>R support contains code for editing R source code (syntactic indentation
and highlighting of source code, partial evaluations of code, loading
and error-checking of code, and source code revision maintenance) and
documentation (syntactic indentation and highlighting of source code,
sending examples to running <acronym>ESS</acronym> process, and previewing),
interacting with an inferior R process from within Emacs (command-line
editing, searchable command history, command-line completion of R object
and file names, quick access to object and search lists, transcript
recording, and an interface to the help system), and transcript
manipulation (recording and saving transcript files, manipulating and
editing saved transcripts, and re-evaluating commands from transcript
files).

   <p>The latest stable version of <acronym>ESS</acronym> are available via <acronym>CRAN</acronym> or
the <a href="http://ESS.R-project.org/">ESS web page</a>.  The <acronym>HTML</acronym> version
of the documentation can be found at <a href="http://stat.ethz.ch/ESS/">http://stat.ethz.ch/ESS/</a>.

   <p><acronym>ESS</acronym> comes with detailed installation instructions.

   <p>For help with <acronym>ESS</acronym>, send email to
<a href="mailto:ESS-help@stat.math.ethz.ch">ESS-help@stat.math.ethz.ch</a>.

   <p>Please send bug reports and suggestions on <acronym>ESS</acronym> to
<a href="mailto:ESS-bugs@stat.math.ethz.ch">ESS-bugs@stat.math.ethz.ch</a>.  The easiest way to do this from is
within Emacs by typing <kbd>M-x ess-submit-bug-report</kbd> or using the
[ESS] or [iESS] pulldown menus.

<div class="node">
<p><hr>
<a name="Should-I-run-R-from-within-Emacs%3f"></a>
<a name="Should-I-run-R-from-within-Emacs_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Debugging-R-from-within-Emacs">Debugging R from within Emacs</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Is-there-Emacs-support-for-R_003f">Is there Emacs support for R?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-and-Emacs">R and Emacs</a>

</div>

<h3 class="section">6.2 Should I run R from within Emacs?</h3>

<p>Yes, <em>definitely</em>.  Inferior R mode provides a readline/history
mechanism, object name completion, and syntax-based highlighting of the
interaction buffer using Font Lock mode, as well as a very convenient
interface to the R help system.

   <p>Of course, it also integrates nicely with the mechanisms for editing R
source using Emacs.  One can write code in one Emacs buffer and send
whole or parts of it for execution to R; this is helpful for both data
analysis and programming.  One can also seamlessly integrate with a
revision control system, in order to maintain a log of changes in your
programs and data, as well as to allow for the retrieval of past
versions of the code.

   <p>In addition, it allows you to keep a record of your session, which can
also be used for error recovery through the use of the transcript mode.

   <p>To specify command line arguments for the inferior R process, use
<kbd>C-u M-x R</kbd> for starting R.

<!-- This prompts you for the arguments; in particular, you can increase -->
<!-- the memory size this way (@pxref{Why does R run out of memory?}). -->
<div class="node">
<p><hr>
<a name="Debugging-R-from-within-Emacs"></a>
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Should-I-run-R-from-within-Emacs_003f">Should I run R from within Emacs?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-and-Emacs">R and Emacs</a>

</div>

<h3 class="section">6.3 Debugging R from within Emacs</h3>

<p>To debug R &ldquo;from within Emacs&rdquo;, there are several possibilities.  To
use the Emacs GUD (Grand Unified Debugger) library with the recommended
debugger GDB, type <kbd>M-x gdb</kbd> and give the path to the R
<em>binary</em> as argument.  At the <samp><span class="command">gdb</span></samp> prompt, set
<samp><span class="env">R_HOME</span></samp> and other environment variables as needed (using e.g. 
<kbd>set env R_HOME /path/to/R/</kbd>, but see also below), and start the
binary with the desired arguments (e.g., <kbd>run --quiet</kbd>).

   <p>If you have <acronym>ESS</acronym>, you can do <kbd>C-u M-x R &lt;RET&gt; - d
&lt;SPC&gt; g d b &lt;RET&gt;</kbd> to start an inferior R process with arguments
<samp><span class="option">-d gdb</span></samp>.

   <p>A third option is to start an inferior R process via <acronym>ESS</acronym>
(<kbd>M-x R</kbd>) and then start GUD (<kbd>M-x gdb</kbd>) giving the R binary
(using its full path name) as the program to debug.  Use the program
<samp><span class="command">ps</span></samp> to find the process number of the currently running R
process then use the <code>attach</code> command in gdb to attach it to that
process.  One advantage of this method is that you have separate
<code>*R*</code> and <code>*gud-gdb*</code> windows.  Within the <code>*R*</code> window
you have all the <acronym>ESS</acronym> facilities, such as object-name
completion, that we know and love.

   <p>When using GUD mode for debugging from within Emacs, you may find it
most convenient to use the directory with your code in it as the current
working directory and then make a symbolic link from that directory to
the R binary.  That way <samp><span class="file">.gdbinit</span></samp> can stay in the directory with
the code and be used to set up the environment and the search paths for
the source, e.g. as follows:

<pre class="example">     set env R_HOME /opt/R
     set env R_PAPERSIZE letter
     set env R_PRINTCMD lpr
     dir /opt/R/src/appl
     dir /opt/R/src/main
     dir /opt/R/src/nmath
     dir /opt/R/src/unix
</pre>
   <div class="node">
<p><hr>
<a name="R-Miscellanea"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#R-Programming">R Programming</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#R-and-Emacs">R and Emacs</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Top">Top</a>

</div>

<h2 class="chapter">7 R Miscellanea</h2>

<ul class="menu">
<li><a accesskey="1" href="#How-can-I-set-components-of-a-list-to-NULL_003f">How can I set components of a list to NULL?</a>
<li><a accesskey="2" href="#How-can-I-save-my-workspace_003f">How can I save my workspace?</a>
<li><a accesskey="3" href="#How-can-I-clean-up-my-workspace_003f">How can I clean up my workspace?</a>
<li><a accesskey="4" href="#How-can-I-get-eval_0028_0029-and-D_0028_0029-to-work_003f">How can I get eval() and D() to work?</a>
<li><a accesskey="5" href="#Why-do-my-matrices-lose-dimensions_003f">Why do my matrices lose dimensions?</a>
<li><a accesskey="6" href="#How-does-autoloading-work_003f">How does autoloading work?</a>
<li><a accesskey="7" href="#How-should-I-set-options_003f">How should I set options?</a>
<li><a accesskey="8" href="#How-do-file-names-work-in-Windows_003f">How do file names work in Windows?</a>
<li><a accesskey="9" href="#Why-does-plotting-give-a-color-allocation-error_003f">Why does plotting give a color allocation error?</a>
<li><a href="#How-do-I-convert-factors-to-numeric_003f">How do I convert factors to numeric?</a>
<li><a href="#Are-Trellis-displays-implemented-in-R_003f">Are Trellis displays implemented in R?</a>
<li><a href="#What-are-the-enclosing-and-parent-environments_003f">What are the enclosing and parent environments?</a>
<li><a href="#How-can-I-substitute-into-a-plot-label_003f">How can I substitute into a plot label?</a>
<li><a href="#What-are-valid-names_003f">What are valid names?</a>
<li><a href="#Are-GAMs-implemented-in-R_003f">Are GAMs implemented in R?</a>
<li><a href="#Why-is-the-output-not-printed-when-I-source_0028_0029-a-file_003f">Why is the output not printed when I source() a file?</a>
<li><a href="#Why-does-outer_0028_0029-behave-strangely-with-my-function_003f">Why does outer() behave strangely with my function?</a>
<li><a href="#Why-does-the-output-from-anova_0028_0029-depend-on-the-order-of-factors-in-the-model_003f">Why does the output from anova() depend on the order of factors in the model?</a>
<li><a href="#How-do-I-produce-PNG-graphics-in-batch-mode_003f">How do I produce PNG graphics in batch mode?</a>
<li><a href="#How-can-I-get-command-line-editing-to-work_003f">How can I get command line editing to work?</a>
<li><a href="#How-can-I-turn-a-string-into-a-variable_003f">How can I turn a string into a variable?</a>
<li><a href="#Why-do-lattice_002ftrellis-graphics-not-work_003f">Why do lattice/trellis graphics not work?</a>
<li><a href="#How-can-I-sort-the-rows-of-a-data-frame_003f">How can I sort the rows of a data frame?</a>
<li><a href="#Why-does-the-help_002estart_0028_0029-search-engine-not-work_003f">Why does the help.start() search engine not work?</a>
<li><a href="#Why-did-my-_002eRprofile-stop-working-when-I-updated-R_003f">Why did my .Rprofile stop working when I updated R?</a>
<li><a href="#Where-have-all-the-methods-gone_003f">Where have all the methods gone?</a>
<li><a href="#How-can-I-create-rotated-axis-labels_003f">How can I create rotated axis labels?</a>
<li><a href="#Why-is-read_002etable_0028_0029-so-inefficient_003f">Why is read.table() so inefficient?</a>
<li><a href="#What-is-the-difference-between-package-and-library_003f">What is the difference between package and library?</a>
<li><a href="#I-installed-a-package-but-the-functions-are-not-there">I installed a package but the functions are not there</a>
<li><a href="#Why-doesn_0027t-R-think-these-numbers-are-equal_003f">Why doesn't R think these numbers are equal?</a>
<li><a href="#How-can-I-capture-or-ignore-errors-in-a-long-simulation_003f">How can I capture or ignore errors in a long simulation?</a>
<li><a href="#Why-are-powers-of-negative-numbers-wrong_003f">Why are powers of negative numbers wrong?</a>
<li><a href="#How-can-I-save-the-result-of-each-iteration-in-a-loop-into-a-separate-file_003f">How can I save the result of each iteration in a loop into a separate file?</a>
<li><a href="#Why-are-p_002dvalues-not-displayed-when-using-lmer_0028_0029_003f">Why are p-values not displayed when using lmer()?</a>
<li><a href="#Why-are-there-unwanted-lines-between-polygons-in-PDF-output-viewed-in-Adobe-Reader_003f">Why are there unwanted lines between polygons in PDF output viewed in Adobe Reader?</a>
<li><a href="#Why-does-backslash-behave-strangely-inside-strings_003f">Why does backslash behave strangely inside strings?</a>
</ul>

<!-- @node Why does R run out of memory?, Why does sourcing a correct file fail?, R Miscellanea, R Miscellanea -->
<!-- @section Why does R run out of memory? -->
<!-- Versions of R prior to 1.2.0 used a @emph{static} memory model.  At -->
<!-- startup, R asked the operating system to reserve a fixed amount of -->
<!-- memory for it.  The size of this chunk could not be changed -->
<!-- subsequently.  Hence, it could happen that not enough memory was -->
<!-- allocated, e.g., when trying to read large data sets into R.  In such -->
<!-- cases, it was necessary to restart R with more memory available, as -->
<!-- controlled by the command line options @option{-nsize} and -->
<!-- @option{-vsize}. -->
<!-- R version 1.2.0 introduces a new ``generational'' garbage collector, -->
<!-- which will increase the memory available to R as needed.  Hence, user -->
<!-- intervention is no longer necessary for ensuring that enough memory is -->
<!-- available. -->
<!-- The new garbage collector does not move objects in memory, meaning that -->
<!-- it is possible for the free memory to become fragmented so that large -->
<!-- objects cannot be allocated even when there is apparently enough memory -->
<!-- for them. -->
<!-- @node Why does sourcing a correct file fail?, How can I set components of a list to NULL?, Why does R run out of memory?, R Miscellanea -->
<!-- @section Why does sourcing a correct file fail? -->
<!-- Versions of R prior to 1.2.1 may have had problems parsing files not -->
<!-- ending in a newline.  Earlier R versions had a similar problem when -->
<!-- reading in data files.  This should no longer happen. -->
<div class="node">
<p><hr>
<a name="How-can-I-set-components-of-a-list-to-NULL%3f"></a>
<a name="How-can-I-set-components-of-a-list-to-NULL_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-I-save-my-workspace_003f">How can I save my workspace?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#R-Miscellanea">R Miscellanea</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.1 How can I set components of a list to NULL?</h3>

<p>You can use

<pre class="example">     x[i] &lt;- list(NULL)
</pre>
   <p class="noindent">to set component <code>i</code> of the list <code>x</code> to <code>NULL</code>, similarly
for named components.  Do not set <code>x[i]</code> or <code>x[[i]]</code> to
<code>NULL</code>, because this will remove the corresponding component from
the list.

   <p>For dropping the row names of a matrix <code>x</code>, it may be easier to use
<code>rownames(x) &lt;- NULL</code>, similarly for column names.

<div class="node">
<p><hr>
<a name="How-can-I-save-my-workspace%3f"></a>
<a name="How-can-I-save-my-workspace_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-I-clean-up-my-workspace_003f">How can I clean up my workspace?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-I-set-components-of-a-list-to-NULL_003f">How can I set components of a list to NULL?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.2 How can I save my workspace?</h3>

<p><code>save.image()</code> saves the objects in the user's <code>.GlobalEnv</code> to
the file <samp><span class="file">.RData</span></samp> in the R startup directory.  (This is also what
happens after <kbd>q("yes")</kbd>.)  Using <code>save.image(</code><var>file</var><code>)</code> one
can save the image under a different name.

<div class="node">
<p><hr>
<a name="How-can-I-clean-up-my-workspace%3f"></a>
<a name="How-can-I-clean-up-my-workspace_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-I-get-eval_0028_0029-and-D_0028_0029-to-work_003f">How can I get eval() and D() to work?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-I-save-my-workspace_003f">How can I save my workspace?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.3 How can I clean up my workspace?</h3>

<p>To remove all objects in the currently active environment (typically
<code>.GlobalEnv</code>), you can do

<pre class="example">     rm(list = ls(all = TRUE))
</pre>
   <p class="noindent">(Without <samp><span class="option">all = TRUE</span></samp>, only the objects with names not starting
with a `<samp><span class="samp">.</span></samp>' are removed.)

<div class="node">
<p><hr>
<a name="How-can-I-get-eval()-and-D()-to-work%3f"></a>
<a name="How-can-I-get-eval_0028_0029-and-D_0028_0029-to-work_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Why-do-my-matrices-lose-dimensions_003f">Why do my matrices lose dimensions?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-I-clean-up-my-workspace_003f">How can I clean up my workspace?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.4 How can I get eval() and D() to work?</h3>

<p>Strange things will happen if you use <code>eval(print(x), envir = e)</code>
or <code>D(x^2, "x")</code>.  The first one will either tell you that
"<code>x</code>" is not found, or print the value of the wrong <code>x</code>. 
The other one will likely return zero if <code>x</code> exists, and an error
otherwise.

   <p>This is because in both cases, the first argument is evaluated in the
calling environment first.  The result (which should be an object of
mode <code>"expression"</code> or <code>"call"</code>) is then evaluated or
differentiated.  What you (most likely) really want is obtained by
&ldquo;quoting&rdquo; the first argument upon surrounding it with
<code>expression()</code>.  For example,

<pre class="example">     R&gt; D(expression(x^2), "x")
     2 * x
</pre>
   <p>Although this behavior may initially seem to be rather strange, is
perfectly logical.  The &ldquo;intuitive&rdquo; behavior could easily be
implemented, but problems would arise whenever the expression is
contained in a variable, passed as a parameter, or is the result of a
function call.  Consider for instance the semantics in cases like

<pre class="example">     D2 &lt;- function(e, n) D(D(e, n), n)
</pre>
   <p class="noindent">or

<pre class="example">     g &lt;- function(y) eval(substitute(y), sys.frame(sys.parent(n = 2)))
     g(a * b)
</pre>
   <p>See the help page for <code>deriv()</code> for more examples.

<div class="node">
<p><hr>
<a name="Why-do-my-matrices-lose-dimensions%3f"></a>
<a name="Why-do-my-matrices-lose-dimensions_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-does-autoloading-work_003f">How does autoloading work?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-I-get-eval_0028_0029-and-D_0028_0029-to-work_003f">How can I get eval() and D() to work?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.5 Why do my matrices lose dimensions?</h3>

<p>When a matrix with a single row or column is created by a subscripting
operation, e.g., <code>row &lt;- mat[2, ]</code>, it is by default turned into a
vector.  In a similar way if an array with dimension, say, 2&nbsp;x&nbsp;3&nbsp;x&nbsp;1&nbsp;x&nbsp;4<!-- /@w --> is created by subscripting it will be coerced into a 2&nbsp;x&nbsp;3&nbsp;x&nbsp;4<!-- /@w -->
array, losing the unnecessary dimension.  After much discussion this has
been determined to be a <em>feature</em>.

   <p>To prevent this happening, add the option <samp><span class="option">drop = FALSE</span></samp> to the
subscripting.  For example,

<pre class="example">     rowmatrix &lt;- mat[2, , drop = FALSE]  # <span class="roman">creates a row matrix</span>
     colmatrix &lt;- mat[, 2, drop = FALSE]  # <span class="roman">creates a column matrix</span>
     a &lt;- b[1, 1, 1, drop = FALSE]        # <span class="roman">creates a 1 x 1 x 1 array</span>
</pre>
   <p>The <samp><span class="option">drop = FALSE</span></samp> option should be used defensively when
programming.  For example, the statement

<pre class="example">     somerows &lt;- mat[index, ]
</pre>
   <p class="noindent">will return a vector rather than a matrix if <code>index</code> happens to
have length 1, causing errors later in the code.  It should probably be
rewritten as

<pre class="example">     somerows &lt;- mat[index, , drop = FALSE]
</pre>
   <div class="node">
<p><hr>
<a name="How-does-autoloading-work%3f"></a>
<a name="How-does-autoloading-work_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-should-I-set-options_003f">How should I set options?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Why-do-my-matrices-lose-dimensions_003f">Why do my matrices lose dimensions?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.6 How does autoloading work?</h3>

<p>R has a special environment called <code>.AutoloadEnv</code>.  Using
<kbd>autoload(</kbd><var>name</var><kbd>, </kbd><var>pkg</var><kbd>)</kbd>, where <var>name</var> and
<var>pkg</var> are strings giving the names of an object and the package
containing it, stores some information in this environment.  When R
tries to evaluate <var>name</var>, it loads the corresponding package
<var>pkg</var> and reevaluates <var>name</var> in the new package's
environment.

   <p>Using this mechanism makes R behave as if the package was loaded, but
does not occupy memory (yet).

   <p>See the help page for <code>autoload()</code> for a very nice example.

<div class="node">
<p><hr>
<a name="How-should-I-set-options%3f"></a>
<a name="How-should-I-set-options_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-do-file-names-work-in-Windows_003f">How do file names work in Windows?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-does-autoloading-work_003f">How does autoloading work?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.7 How should I set options?</h3>

<p>The function <code>options()</code> allows setting and examining a variety of
global &ldquo;options&rdquo; which affect the way in which R computes and displays
its results.  The variable <code>.Options</code> holds the current values of
these options, but should never directly be assigned to unless you want
to drive yourself crazy&mdash;simply pretend that it is a &ldquo;read-only&rdquo;
variable.

   <p>For example, given

<pre class="example">     test1 &lt;- function(x = pi, dig = 3) {
       oo &lt;- options(digits = dig); on.exit(options(oo));
       cat(.Options$digits, x, "\n")
     }
     test2 &lt;- function(x = pi, dig = 3) {
       .Options$digits &lt;- dig
       cat(.Options$digits, x, "\n")
     }
</pre>
   <p class="noindent">we obtain:

<pre class="example">     R&gt; test1()
     3 3.14
     R&gt; test2()
     3 3.141593
</pre>
   <p>What is really used is the <em>global</em> value of <code>.Options</code>, and
using <kbd>options(OPT = VAL)</kbd> correctly updates it.  Local copies of
<code>.Options</code>, either in <code>.GlobalEnv</code> or in a function
environment (frame), are just silently disregarded.

<div class="node">
<p><hr>
<a name="How-do-file-names-work-in-Windows%3f"></a>
<a name="How-do-file-names-work-in-Windows_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Why-does-plotting-give-a-color-allocation-error_003f">Why does plotting give a color allocation error?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-should-I-set-options_003f">How should I set options?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.8 How do file names work in Windows?</h3>

<p>As R uses C-style string handling, `<samp><span class="samp">\</span></samp>' is treated as an escape
character, so that for example one can enter a newline as `<samp><span class="samp">\n</span></samp>'. 
When you really need a `<samp><span class="samp">\</span></samp>', you have to escape it with another
`<samp><span class="samp">\</span></samp>'.

   <p>Thus, in filenames use something like <code>"c:\\data\\money.dat"</code>.  You
can also replace `<samp><span class="samp">\</span></samp>' by `<samp><span class="samp">/</span></samp>' (<code>"c:/data/money.dat"</code>).

<div class="node">
<p><hr>
<a name="Why-does-plotting-give-a-color-allocation-error%3f"></a>
<a name="Why-does-plotting-give-a-color-allocation-error_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-do-I-convert-factors-to-numeric_003f">How do I convert factors to numeric?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-do-file-names-work-in-Windows_003f">How do file names work in Windows?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.9 Why does plotting give a color allocation error?</h3>

<p>On an X11 device, plotting sometimes, e.g., when running
<code>demo("image")</code>, results in &ldquo;Error: color allocation error&rdquo;. 
This is an X problem, and only indirectly related to R.  It occurs when
applications started prior to R have used all the available colors. 
(How many colors are available depends on the X configuration; sometimes
only 256 colors can be used.)

   <p>One application which is notorious for &ldquo;eating&rdquo; colors is Netscape. 
If the problem occurs when Netscape is running, try (re)starting it with
either the <samp><span class="option">-no-install</span></samp> (to use the default colormap) or the
<samp><span class="option">-install</span></samp> (to install a private colormap) option.

   <p>You could also set the <code>colortype</code> of <code>X11()</code> to
<code>"pseudo.cube"</code> rather than the default <code>"pseudo"</code>.  See the
help page for <code>X11()</code> for more information.

<!-- @node Is R Y2K-compliant?, How do I convert factors to numeric?, Why does plotting give a color allocation error?, R Miscellanea -->
<!-- @section Is R Y2K-compliant? -->
<!-- We expect R to be Y2K compliant when compiled and run on a Y2K compliant -->
<!-- system.  In particular R does not internally represent or manipulate -->
<!-- dates as two-digit quantities.  However, no guarantee of Y2K compliance -->
<!-- is provided for R.  R is free software and comes with @emph{no warranty -->
<!-- whatsoever}. -->
<!-- R, like any other programming language, can be used to write programs -->
<!-- and manipulate data in ways that are not Y2K compliant. -->
<div class="node">
<p><hr>
<a name="How-do-I-convert-factors-to-numeric%3f"></a>
<a name="How-do-I-convert-factors-to-numeric_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Are-Trellis-displays-implemented-in-R_003f">Are Trellis displays implemented in R?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Why-does-plotting-give-a-color-allocation-error_003f">Why does plotting give a color allocation error?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.10 How do I convert factors to numeric?</h3>

<p>It may happen that when reading numeric data into R (usually, when
reading in a file), they come in as factors.  If <code>f</code> is such a
factor object, you can use

<pre class="example">     as.numeric(as.character(f))
</pre>
   <p class="noindent">to get the numbers back.  More efficient, but harder to remember, is

<pre class="example">     as.numeric(levels(f))[as.integer(f)]
</pre>
   <p>In any case, do not call <code>as.numeric()</code> or their likes directly for
the task at hand (as <code>as.numeric()</code> or <code>unclass()</code> give the
internal codes).

<div class="node">
<p><hr>
<a name="Are-Trellis-displays-implemented-in-R%3f"></a>
<a name="Are-Trellis-displays-implemented-in-R_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#What-are-the-enclosing-and-parent-environments_003f">What are the enclosing and parent environments?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-do-I-convert-factors-to-numeric_003f">How do I convert factors to numeric?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.11 Are Trellis displays implemented in R?</h3>

<p>The recommended package <strong>lattice</strong> (which is based on another
recommended package, <strong>grid</strong>) provides graphical functionality
that is compatible with most Trellis commands.

   <p>You could also look at <code>coplot()</code> and <code>dotchart()</code> which might
do at least some of what you want.  Note also that the R version of
<code>pairs()</code> is fairly general and provides most of the functionality
of <code>splom()</code>, and that R's default plot method has an argument
<code>asp</code> allowing to specify (and fix against device resizing) the
aspect ratio of the plot.

   <p>(Because the word &ldquo;Trellis&rdquo; has been claimed as a trademark we do not
use it in R.  The name &ldquo;lattice&rdquo; has been chosen for the R
equivalent.)

<div class="node">
<p><hr>
<a name="What-are-the-enclosing-and-parent-environments%3f"></a>
<a name="What-are-the-enclosing-and-parent-environments_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-I-substitute-into-a-plot-label_003f">How can I substitute into a plot label?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Are-Trellis-displays-implemented-in-R_003f">Are Trellis displays implemented in R?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.12 What are the enclosing and parent environments?</h3>

<p>Inside a function you may want to access variables in two additional
environments: the one that the function was defined in (&ldquo;enclosing&rdquo;),
and the one it was invoked in (&ldquo;parent&rdquo;).

   <p>If you create a function at the command line or load it in a package its
enclosing environment is the global workspace.  If you define a function
<code>f()</code> inside another function <code>g()</code> its enclosing environment
is the environment inside <code>g()</code>.  The enclosing environment for a
function is fixed when the function is created.  You can find out the
enclosing environment for a function <code>f()</code> using
<code>environment(f)</code>.

   <p>The &ldquo;parent&rdquo; environment, on the other hand, is defined when you
invoke a function.  If you invoke <code>lm()</code> at the command line its
parent environment is the global workspace, if you invoke it inside a
function <code>f()</code> then its parent environment is the environment
inside <code>f()</code>.  You can find out the parent environment for an
invocation of a function by using <code>parent.frame()</code> or
<code>sys.frame(sys.parent())</code>.

   <p>So for most user-visible functions the enclosing environment will be the
global workspace, since that is where most functions are defined.  The
parent environment will be wherever the function happens to be called
from.  If a function <code>f()</code> is defined inside another function
<code>g()</code> it will probably be used inside <code>g()</code> as well, so its
parent environment and enclosing environment will probably be the same.

   <p>Parent environments are important because things like model formulas
need to be evaluated in the environment the function was called from,
since that's where all the variables will be available.  This relies on
the parent environment being potentially different with each invocation.

   <p>Enclosing environments are important because a function can use
variables in the enclosing environment to share information with other
functions or with other invocations of itself (see the section on
lexical scoping).  This relies on the enclosing environment being the
same each time the function is invoked.  (In C this would be done with
static variables.)

   <p>Scoping <em>is</em> hard.  Looking at examples helps.  It is particularly
instructive to look at examples that work differently in R and S and try
to see why they differ.  One way to describe the scoping differences
between R and S is to say that in S the enclosing environment is
<em>always</em> the global workspace, but in R the enclosing environment
is wherever the function was created.

<div class="node">
<p><hr>
<a name="How-can-I-substitute-into-a-plot-label%3f"></a>
<a name="How-can-I-substitute-into-a-plot-label_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#What-are-valid-names_003f">What are valid names?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#What-are-the-enclosing-and-parent-environments_003f">What are the enclosing and parent environments?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.13 How can I substitute into a plot label?</h3>

<p>Often, it is desired to use the value of an R object in a plot label,
e.g., a title.  This is easily accomplished using <code>paste()</code> if the
label is a simple character string, but not always obvious in case the
label is an expression (for refined mathematical annotation).  In such a
case, either use <code>parse()</code> on your pasted character string or use
<code>substitute()</code> on an expression.  For example, if <code>ahat</code> is an
estimator of your parameter a of interest, use

<pre class="example">     title(substitute(hat(a) == ahat, list(ahat = ahat)))
</pre>
   <p class="noindent">(note that it is `<samp><span class="samp">==</span></samp>' and not `<samp><span class="samp">=</span></samp>').  Sometimes <code>bquote()</code>
gives a more compact form, e.g.,

<pre class="example">     title(bquote(hat(a) = .(ahat)))
</pre>
   <p class="noindent">where subexpressions enclosed in `<samp><span class="samp">.()</span></samp>' are replaced by their
values.

   <p>There are more worked examples in the mailing list achives.

<div class="node">
<p><hr>
<a name="What-are-valid-names%3f"></a>
<a name="What-are-valid-names_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Are-GAMs-implemented-in-R_003f">Are GAMs implemented in R?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-I-substitute-into-a-plot-label_003f">How can I substitute into a plot label?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.14 What are valid names?</h3>

<p>When creating data frames using <code>data.frame()</code> or
<code>read.table()</code>, R by default ensures that the variable names are
syntactically valid.  (The argument <samp><span class="option">check.names</span></samp> to these
functions controls whether variable names are checked and adjusted by
<code>make.names()</code> if needed.)

   <p>To understand what names are &ldquo;valid&rdquo;, one needs to take into account
that the term &ldquo;name&rdquo; is used in several different (but related) ways
in the language:

     <ol type=1 start=1>
<li>A <em>syntactic name</em> is a string the parser interprets as this type
of expression.  It consists of letters, numbers, and the dot and (for
version of R at least 1.9.0) underscore characters, and starts with
either a letter or a dot not followed by a number.  Reserved words are
not syntactic names. 
<li>An <em>object name</em> is a string associated with an object that is
assigned in an expression either by having the object name on the left
of an assignment operation or as an argument to the <code>assign()</code>
function.  It is usually a syntactic name as well, but can be any
non-empty string if it is quoted (and it is always quoted in the call to
<code>assign()</code>).

     <li>An <em>argument name</em> is what appears to the left of the equals sign
when supplying an argument in a function call (for example,
<code>f(trim=.5)</code>).  Argument names are also usually syntactic names,
but again can be anything if they are quoted.

     <li>An <em>element name</em> is a string that identifies a piece of an object
(a component of a list, for example.)  When it is used on the right of
the `<samp><span class="samp">$</span></samp>' operator, it must be a syntactic name, or quoted. 
Otherwise, element names can be any strings.  (When an object is used as
a database, as in a call to <code>eval()</code> or <code>attach()</code>, the
element names become object names.)

     <li>Finally, a <em>file name</em> is a string identifying a file in the
operating system for reading, writing, etc.  It really has nothing much
to do with names in the language, but it is traditional to call these
strings file &ldquo;names&rdquo;.
        </ol>

<div class="node">
<p><hr>
<a name="Are-GAMs-implemented-in-R%3f"></a>
<a name="Are-GAMs-implemented-in-R_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Why-is-the-output-not-printed-when-I-source_0028_0029-a-file_003f">Why is the output not printed when I source() a file?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#What-are-valid-names_003f">What are valid names?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.15 Are GAMs implemented in R?</h3>

<p>Package <strong>gam</strong> from <acronym>CRAN</acronym> implements all the Generalized
Additive Models (GAM) functionality as described in the GAM chapter of
the White Book.  In particular, it implements backfitting with both
local regression and smoothing splines, and is extendable.  There is a
<code>gam()</code> function for GAMs in package <strong>mgcv</strong>, but it is not
an exact clone of what is described in the White Book (no <code>lo()</code>
for example).  Package <strong>gss</strong> can fit spline-based GAMs too.  And
if you can accept regression splines you can use <code>glm()</code>.  For
gaussian GAMs you can use <code>bruto()</code> from package <strong>mda</strong>.

<div class="node">
<p><hr>
<a name="Why-is-the-output-not-printed-when-I-source()-a-file%3f"></a>
<a name="Why-is-the-output-not-printed-when-I-source_0028_0029-a-file_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Why-does-outer_0028_0029-behave-strangely-with-my-function_003f">Why does outer() behave strangely with my function?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Are-GAMs-implemented-in-R_003f">Are GAMs implemented in R?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.16 Why is the output not printed when I source() a file?</h3>

<p>Most R commands do not generate any output. The command

<pre class="example">     1+1
</pre>
   <p class="noindent">computes the value 2 and returns it; the command

<pre class="example">     summary(glm(y~x+z, family=binomial))
</pre>
   <p class="noindent">fits a logistic regression model, computes some summary information and
returns an object of class <code>"summary.glm"</code> (see <a href="#How-should-I-write-summary-methods_003f">How should I write summary methods?</a>).

   <p>If you type `<samp><span class="samp">1+1</span></samp>' or `<samp><span class="samp">summary(glm(y~x+z, family=binomial))</span></samp>' at
the command line the returned value is automatically printed (unless it
is <code>invisible()</code>), but in other circumstances, such as in a
<code>source()</code>d file or inside a function it isn't printed unless you
specifically print it.

   <p>To print the value use

<pre class="example">     print(1+1)
</pre>
   <p class="noindent">or

<pre class="example">     print(summary(glm(y~x+z, family=binomial)))
</pre>
   <p class="noindent">instead, or use <code>source(</code><var>file</var><code>, echo=TRUE)</code>.

<div class="node">
<p><hr>
<a name="Why-does-outer()-behave-strangely-with-my-function%3f"></a>
<a name="Why-does-outer_0028_0029-behave-strangely-with-my-function_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Why-does-the-output-from-anova_0028_0029-depend-on-the-order-of-factors-in-the-model_003f">Why does the output from anova() depend on the order of factors in the model?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Why-is-the-output-not-printed-when-I-source_0028_0029-a-file_003f">Why is the output not printed when I source() a file?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.17 Why does outer() behave strangely with my function?</h3>

<p>As the help for <code>outer()</code> indicates, it does not work on arbitrary
functions the way the <code>apply()</code> family does.  It requires functions
that are vectorized to work elementwise on arrays.  As you can see by
looking at the code, <code>outer(x, y, FUN)</code> creates two large vectors
containing every possible combination of elements of <code>x</code> and
<code>y</code> and then passes this to <code>FUN</code> all at once.  Your function
probably cannot handle two large vectors as parameters.

   <p>If you have a function that cannot handle two vectors but can handle two
scalars, then you can still use <code>outer()</code> but you will need to wrap
your function up first, to simulate vectorized behavior.  Suppose your
function is

<pre class="example">     foo &lt;- function(x, y, happy) {
       stopifnot(length(x) == 1, length(y) == 1) # scalars only!
       (x + y) * happy
     }
</pre>
   <p class="noindent">If you define the general function

<pre class="example">     wrapper &lt;- function(x, y, my.fun, ...) {
       sapply(seq(along = x), FUN = function(i) my.fun(x[i], y[i], ...))
     }
</pre>
   <p class="noindent">then you can use <code>outer()</code> by writing, e.g.,

<pre class="example">     outer(1:4, 1:2, FUN = wrapper, my.fun = foo, happy = 10)
</pre>
   <div class="node">
<p><hr>
<a name="Why-does-the-output-from-anova()-depend-on-the-order-of-factors-in-the-model%3f"></a>
<a name="Why-does-the-output-from-anova_0028_0029-depend-on-the-order-of-factors-in-the-model_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-do-I-produce-PNG-graphics-in-batch-mode_003f">How do I produce PNG graphics in batch mode?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Why-does-outer_0028_0029-behave-strangely-with-my-function_003f">Why does outer() behave strangely with my function?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.18 Why does the output from anova() depend on the order of factors in the model?</h3>

<p>In a model such as <code>~A+B+A:B</code>, R will report the difference in sums
of squares between the models <code>~1</code>, <code>~A</code>, <code>~A+B</code> and
<code>~A+B+A:B</code>.  If the model were <code>~B+A+A:B</code>, R would report
differences between <code>~1</code>, <code>~B</code>, <code>~A+B</code>, and
<code>~A+B+A:B</code> . In the first case the sum of squares for <code>A</code> is
comparing <code>~1</code> and <code>~A</code>, in the second case it is comparing
<code>~B</code> and <code>~B+A</code>.  In a non-orthogonal design (i.e., most
unbalanced designs) these comparisons are (conceptually and numerically)
different.

   <p>Some packages report instead the sums of squares based on comparing the
full model to the models with each factor removed one at a time (the
famous `Type III sums of squares' from SAS, for example).  These do not
depend on the order of factors in the model.  The question of which set
of sums of squares is the Right Thing provokes low-level holy wars on
R-help from time to time.

   <p>There is no need to be agitated about the particular sums of squares
that R reports.  You can compute your favorite sums of squares quite
easily.  Any two models can be compared with <code>anova(</code><var>model1</var><code>,
</code><var>model2</var><code>)</code>, and <code>drop1(</code><var>model1</var><code>)</code> will show the sums of
squares resulting from dropping single terms.

<div class="node">
<p><hr>
<a name="How-do-I-produce-PNG-graphics-in-batch-mode%3f"></a>
<a name="How-do-I-produce-PNG-graphics-in-batch-mode_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-I-get-command-line-editing-to-work_003f">How can I get command line editing to work?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Why-does-the-output-from-anova_0028_0029-depend-on-the-order-of-factors-in-the-model_003f">Why does the output from anova() depend on the order of factors in the model?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.19 How do I produce PNG graphics in batch mode?</h3>

<p>Under a Unix-alike, if your installation supports the
<code>type="cairo"</code> option to the <code>png()</code> device there should be no
problems, and the default settings should just work.  This option is not
available for versions of R prior to 2.7.0, or without support for
cairo.  From R 2.7.0 <code>png()</code> by default uses the Quartz device
on Mac OS X, and that too works in batch mode.

   <p>Earlier versions of the <code>png()</code> device uses the X11 driver, which is
a problem in batch mode or for remote operation.  If you have Ghostscript
you can use <code>bitmap()</code>, which produces a PostScript or PDF file then
converts it to any bitmap format supported by Ghostscript.  On some
installations this produces ugly output, on others it is perfectly
satisfactory.  Many systems now come with Xvfb from
<a href="http://www.x.org/Downloads.html">X.Org</a> (possibly as an optional
install), which is an X11 server that does not require a screen; and there
is the <strong>GDD</strong> package from <acronym>CRAN</acronym>, which produces PNG, JPEG and GIF
bitmaps without X11.

<div class="node">
<p><hr>
<a name="How-can-I-get-command-line-editing-to-work%3f"></a>
<a name="How-can-I-get-command-line-editing-to-work_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-I-turn-a-string-into-a-variable_003f">How can I turn a string into a variable?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-do-I-produce-PNG-graphics-in-batch-mode_003f">How do I produce PNG graphics in batch mode?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.20 How can I get command line editing to work?</h3>

<p>The Unix command-line interface to R can only provide the inbuilt
command line editor which allows recall, editing and re-submission of
prior commands provided that the <acronym>GNU</acronym> readline library is
available at the time R is configured for compilation.  Note that the
`development' version of readline including the appropriate headers is
needed: users of Linux binary distributions will need to install
packages such as <code>libreadline-dev</code> (Debian) or
<code>readline-devel</code> (Red Hat).

<div class="node">
<p><hr>
<a name="How-can-I-turn-a-string-into-a-variable%3f"></a>
<a name="How-can-I-turn-a-string-into-a-variable_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Why-do-lattice_002ftrellis-graphics-not-work_003f">Why do lattice/trellis graphics not work?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-I-get-command-line-editing-to-work_003f">How can I get command line editing to work?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.21 How can I turn a string into a variable?</h3>

<p>If you have

<pre class="example">     varname &lt;- c("a", "b", "d")
</pre>
   <p class="noindent">you can do

<pre class="example">     get(varname[1]) + 2
</pre>
   <p class="noindent">for

<pre class="example">     a + 2
</pre>
   <p class="noindent">or

<pre class="example">     assign(varname[1], 2 + 2)
</pre>
   <p class="noindent">for

<pre class="example">     a &lt;- 2 + 2
</pre>
   <p class="noindent">or

<pre class="example">     eval(substitute(lm(y ~ x + variable),
                     list(variable = as.name(varname[1]))
</pre>
   <p class="noindent">for

<pre class="example">     lm(y ~ x + a)
</pre>
   <p>At least in the first two cases it is often easier to just use a list,
and then you can easily index it by name

<pre class="example">     vars &lt;- list(a = 1:10, b = rnorm(100), d = LETTERS)
     vars[["a"]]
</pre>
   <p class="noindent">without any of this messing about.

<div class="node">
<p><hr>
<a name="Why-do-lattice%2ftrellis-graphics-not-work%3f"></a>
<a name="Why-do-lattice_002ftrellis-graphics-not-work_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-I-sort-the-rows-of-a-data-frame_003f">How can I sort the rows of a data frame?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-I-turn-a-string-into-a-variable_003f">How can I turn a string into a variable?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.22 Why do lattice/trellis graphics not work?</h3>

<p>The most likely reason is that you forgot to tell R to display the
graph.  Lattice functions such as <code>xyplot()</code> create a graph object,
but do not display it (the same is true of Trellis graphics in
<span class="sc">S-Plus</span>).  The <code>print()</code> method for the graph object produces the
actual display.  When you use these functions interactively at the
command line, the result is automatically printed, but in
<code>source()</code> or inside your own functions you will need an explicit
<code>print()</code> statement.

<div class="node">
<p><hr>
<a name="How-can-I-sort-the-rows-of-a-data-frame%3f"></a>
<a name="How-can-I-sort-the-rows-of-a-data-frame_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Why-does-the-help_002estart_0028_0029-search-engine-not-work_003f">Why does the help.start() search engine not work?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Why-do-lattice_002ftrellis-graphics-not-work_003f">Why do lattice/trellis graphics not work?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.23 How can I sort the rows of a data frame?</h3>

<p>To sort the rows within a data frame, with respect to the values in one
or more of the columns, simply use <code>order()</code> (e.g.,
<code>DF[order(DF$a, DF[["b"]]), ]</code> to sort the data frame <code>DF</code> on
columns named <code>a</code> and <code>b</code>).

<div class="node">
<p><hr>
<a name="Why-does-the-help.start()-search-engine-not-work%3f"></a>
<a name="Why-does-the-help_002estart_0028_0029-search-engine-not-work_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Why-did-my-_002eRprofile-stop-working-when-I-updated-R_003f">Why did my .Rprofile stop working when I updated R?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-I-sort-the-rows-of-a-data-frame_003f">How can I sort the rows of a data frame?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.24 Why does the help.start() search engine not work?</h3>

<p>The browser-based search engine in <code>help.start()</code> utilizes a Java
applet.  In order for this to function properly, a compatible version of
Java must installed on your system and linked to your browser, and both
Java <em>and</em> JavaScript need to be enabled in your browser.

   <p>There have been a number of compatibility issues with versions of Java
and of browsers. 
See <a href="R-admin.html#Enabling-search-in-HTML-help">Enabling search in HTML help</a>, for further details.

<div class="node">
<p><hr>
<a name="Why-did-my-.Rprofile-stop-working-when-I-updated-R%3f"></a>
<a name="Why-did-my-_002eRprofile-stop-working-when-I-updated-R_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Where-have-all-the-methods-gone_003f">Where have all the methods gone?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Why-does-the-help_002estart_0028_0029-search-engine-not-work_003f">Why does the help.start() search engine not work?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.25 Why did my .Rprofile stop working when I updated R?</h3>

<p>Did you read the <samp><span class="file">NEWS</span></samp> file?  For functions that are not in the
<strong>base</strong> package you need to specify the correct package namespace,
since the code will be run <em>before</em> the packages are loaded.  E.g.,

<pre class="example">     ps.options(horizontal = FALSE)
     help.start()
</pre>
   <p class="noindent">needs to be

<pre class="example">     grDevices::ps.options(horizontal = FALSE)
     utils::help.start()
</pre>
   <!-- <FIXME> -->
<!-- 3.0.0 -->
<p class="noindent">(<code>graphics::ps.options(horizontal = FALSE)</code> in R 1.9.x). 
<!-- </FIXME> -->

<div class="node">
<p><hr>
<a name="Where-have-all-the-methods-gone%3f"></a>
<a name="Where-have-all-the-methods-gone_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-I-create-rotated-axis-labels_003f">How can I create rotated axis labels?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Why-did-my-_002eRprofile-stop-working-when-I-updated-R_003f">Why did my .Rprofile stop working when I updated R?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.26 Where have all the methods gone?</h3>

<p>Many functions, particularly S3 methods, are now hidden in namespaces. 
This has the advantage that they cannot be called inadvertently with
arguments of the wrong class, but it makes them harder to view.

   <p>To see the code for an S3 method (e.g., <code>[.terms</code>) use

<pre class="example">     getS3method("[", "terms")
</pre>
   <p class="noindent">To see the code for an unexported function <code>foo()</code> in the namespace
of package <code>"bar"</code> use <code>bar:::foo</code>.  Don't use these
constructions to call unexported functions in your own code&mdash;they are
probably unexported for a reason and may change without warning.

<div class="node">
<p><hr>
<a name="How-can-I-create-rotated-axis-labels%3f"></a>
<a name="How-can-I-create-rotated-axis-labels_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Why-is-read_002etable_0028_0029-so-inefficient_003f">Why is read.table() so inefficient?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Where-have-all-the-methods-gone_003f">Where have all the methods gone?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.27 How can I create rotated axis labels?</h3>

<p>To rotate axis labels (using base graphics), you need to use
<code>text()</code>, rather than <code>mtext()</code>, as the latter does not
support <code>par("srt")</code>.

<pre class="example">     ## <span class="roman">Increase bottom margin to make room for rotated labels</span>
     par(mar = c(7, 4, 4, 2) + 0.1)
     ## <span class="roman">Create plot with no x axis and no x axis label</span>
     plot(1 : 8, xaxt = "n",  xlab = "")
     ## <span class="roman">Set up x axis with tick marks alone</span>
     axis(1, labels = FALSE)
     ## <span class="roman">Create some text labels</span>
     labels &lt;- paste("Label", 1:8, sep = " ")
     ## <span class="roman">Plot x axis labels at default tick marks</span>
     text(1:8, par("usr")[3] - 0.25, srt = 45, adj = 1,
          labels = labels, xpd = TRUE)
     ## <span class="roman">Plot x axis label at line 6 (of 7)</span>
     mtext(1, text = "X Axis Label", line = 6)
</pre>
   <p class="noindent">When plotting the x axis labels, we use <code>srt = 45</code> for text
rotation angle, <code>adj = 1</code> to place the right end of text at the
tick marks, and <code>xpd = TRUE</code> to allow for text outside the plot
region.  You can adjust the value of the <code>0.25</code> offset as required
to move the axis labels up or down relative to the x axis.  See
<code>?par</code> for more information.

   <p>Also see Figure 1 and associated code in Paul Murrell (2003),
&ldquo;Integrating grid Graphics Output with Base Graphics Output&rdquo;,
<em>R News</em>, <strong>3/2</strong>, 7&ndash;12.

<div class="node">
<p><hr>
<a name="Why-is-read.table()-so-inefficient%3f"></a>
<a name="Why-is-read_002etable_0028_0029-so-inefficient_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#What-is-the-difference-between-package-and-library_003f">What is the difference between package and library?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-I-create-rotated-axis-labels_003f">How can I create rotated axis labels?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.28 Why is read.table() so inefficient?</h3>

<p>By default, <code>read.table()</code> needs to read in everything as character
data, and then try to figure out which variables to convert to numerics
or factors.  For a large data set, this takes condiderable amounts of
time and memory.  Performance can substantially be improved by using the
<code>colClasses</code> argument to specify the classes to be assumed for the
columns of the table.

<div class="node">
<p><hr>
<a name="What-is-the-difference-between-package-and-library%3f"></a>
<a name="What-is-the-difference-between-package-and-library_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#I-installed-a-package-but-the-functions-are-not-there">I installed a package but the functions are not there</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Why-is-read_002etable_0028_0029-so-inefficient_003f">Why is read.table() so inefficient?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.29 What is the difference between package and library?</h3>

<p>A <dfn>package</dfn> is a standardized collection of material extending R,
e.g. providing code, data, or documentation.  A <dfn>library</dfn> is a
place (directory) where R knows to find packages it can use (i.e., which
were <dfn>installed</dfn>).  R is told to use a package (to &ldquo;load&rdquo; it and
add it to the search path) via calls to the function <code>library</code>. 
I.e., <code>library()</code> is employed to load a package from libraries
containing packages.

   <p>See <a href="#R-Add_002dOn-Packages">R Add-On Packages</a>, for more details.  See also Uwe Ligges (2003),
&ldquo;R Help Desk: Package Management&rdquo;, <em>R News</em>, <strong>3/3</strong>,
37&ndash;39.

<div class="node">
<p><hr>
<a name="I-installed-a-package-but-the-functions-are-not-there"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Why-doesn_0027t-R-think-these-numbers-are-equal_003f">Why doesn't R think these numbers are equal?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#What-is-the-difference-between-package-and-library_003f">What is the difference between package and library?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.30 I installed a package but the functions are not there</h3>

<p>To actually <em>use</em> the package, it needs to be <em>loaded</em> using
<code>library()</code>.

   <p>See <a href="#R-Add_002dOn-Packages">R Add-On Packages</a> and <a href="#What-is-the-difference-between-package-and-library_003f">What is the difference between package and library?</a> for more information.

<div class="node">
<p><hr>
<a name="Why-doesn't-R-think-these-numbers-are-equal%3f"></a>
<a name="Why-doesn_0027t-R-think-these-numbers-are-equal_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-I-capture-or-ignore-errors-in-a-long-simulation_003f">How can I capture or ignore errors in a long simulation?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#I-installed-a-package-but-the-functions-are-not-there">I installed a package but the functions are not there</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.31 Why doesn't R think these numbers are equal?</h3>

<p>The only numbers that can be represented exactly in R's numeric type
are integers and fractions whose denominator is a power of 2.  Other
numbers have to be rounded to (typically) 53 binary digits accuracy.  As a
result, two floating point numbers will not reliably be equal unless they
have been computed by the same algorithm, and not always even then.  For
example

<pre class="example">     R&gt; a &lt;- sqrt(2)
     R&gt; a * a == 2
     [1] FALSE
     R&gt; a * a - 2
     [1] 4.440892e-16
</pre>
   <p>The function <code>all.equal()</code> compares two objects using a numeric
tolerance of <code>.Machine$double.eps ^ 0.5</code>.  If you want much greater
accuracy than this you will need to consider error propagation
carefully.

   <p>For more information, see e.g. David Goldberg (1991), &ldquo;What Every
Computer Scientist Should Know About Floating-Point Arithmetic&rdquo;,
<em>ACM Computing Surveys</em>, <strong>23/1</strong>, 5&ndash;48, also available via
<a href="http://docs.sun.com/source/806-3568/ncg_goldberg.html">http://docs.sun.com/source/806-3568/ncg_goldberg.html</a>.

   <p>To quote from &ldquo;The Elements of Programming Style&rdquo; by Kernighan and
Plauger:

   <blockquote>
<em>10.0 times 0.1 is hardly ever 1.0</em>. 
</blockquote>

<div class="node">
<p><hr>
<a name="How-can-I-capture-or-ignore-errors-in-a-long-simulation%3f"></a>
<a name="How-can-I-capture-or-ignore-errors-in-a-long-simulation_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Why-are-powers-of-negative-numbers-wrong_003f">Why are powers of negative numbers wrong?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Why-doesn_0027t-R-think-these-numbers-are-equal_003f">Why doesn't R think these numbers are equal?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.32 How can I capture or ignore errors in a long simulation?</h3>

<p>Use <code>try()</code>, which returns an object of class <code>"try-error"</code>
instead of an error, or preferably <code>tryCatch()</code>, where the return
value can be configured more flexibly.  For example

<pre class="example">     beta[i,] &lt;- tryCatch(coef(lm(formula, data)),
                          error = function(e) rep(NaN, 4))
</pre>
   <p class="noindent">would return the coefficients if the <code>lm()</code> call succeeded and
would return <code>c(NaN, NaN, NaN, NaN)</code> if it failed (presumably there
are supposed to be 4 coefficients in this example).

<div class="node">
<p><hr>
<a name="Why-are-powers-of-negative-numbers-wrong%3f"></a>
<a name="Why-are-powers-of-negative-numbers-wrong_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-I-save-the-result-of-each-iteration-in-a-loop-into-a-separate-file_003f">How can I save the result of each iteration in a loop into a separate file?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-I-capture-or-ignore-errors-in-a-long-simulation_003f">How can I capture or ignore errors in a long simulation?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.33 Why are powers of negative numbers wrong?</h3>

<p>You are probably seeing something like

<pre class="example">     R&gt; -2^2
     [1] -4
</pre>
   <p class="noindent">and misunderstanding the precedence rules for expressions in R. 
Write

<pre class="example">     R&gt; (-2)^2
     [1] 4
</pre>
   <p class="noindent">to get the square of -2.

   <p>The precedence rules are documented in <code>?Syntax</code>, and to see how R
interprets  an expression you can look at the parse tree

<pre class="example">     R&gt; as.list(quote(-2^2))
     [[1]]
     `-`
     
     [[2]]
     2^2
</pre>
   <div class="node">
<p><hr>
<a name="How-can-I-save-the-result-of-each-iteration-in-a-loop-into-a-separate-file%3f"></a>
<a name="How-can-I-save-the-result-of-each-iteration-in-a-loop-into-a-separate-file_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Why-are-p_002dvalues-not-displayed-when-using-lmer_0028_0029_003f">Why are p-values not displayed when using lmer()?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Why-are-powers-of-negative-numbers-wrong_003f">Why are powers of negative numbers wrong?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.34 How can I save the result of each iteration in a loop into a separate file?</h3>

<p>One way is to use <code>paste()</code> (or <code>sprintf()</code>) to concatenate a
stem filename and the iteration number while <code>file.path()</code>
constructs the path.  For example, to save results into files
<samp><span class="file">result1.rda</span></samp>, <small class="dots">...</small>, <samp><span class="file">result100.rda</span></samp> in the subdirectory
<samp><span class="file">Results</span></samp> of the current working directory, one can use

<pre class="example">     for(i in 1:100) {
       ## Calculations constructing "some_object" ...
       fp &lt;- file.path("Results", paste("result", i, ".rda", sep = ""))
       save(list = "some_object", file = fp)
     }
</pre>
   <div class="node">
<p><hr>
<a name="Why-are-p-values-not-displayed-when-using-lmer()%3f"></a>
<a name="Why-are-p_002dvalues-not-displayed-when-using-lmer_0028_0029_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Why-are-there-unwanted-lines-between-polygons-in-PDF-output-viewed-in-Adobe-Reader_003f">Why are there unwanted lines between polygons in PDF output viewed in Adobe Reader?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-I-save-the-result-of-each-iteration-in-a-loop-into-a-separate-file_003f">How can I save the result of each iteration in a loop into a separate file?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.35 Why are p-values not displayed when using lmer()?</h3>

<p>Doug Bates has kindly provided an extensive response in a post to the
r-help list, which can be reviewed at
<a href="https://stat.ethz.ch/pipermail/r-help/2006-May/094765.html">https://stat.ethz.ch/pipermail/r-help/2006-May/094765.html</a>.

<div class="node">
<p><hr>
<a name="Why-are-there-unwanted-lines-between-polygons-in-PDF-output-viewed-in-Adobe-Reader%3f"></a>
<a name="Why-are-there-unwanted-lines-between-polygons-in-PDF-output-viewed-in-Adobe-Reader_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Why-does-backslash-behave-strangely-inside-strings_003f">Why does backslash behave strangely inside strings?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Why-are-p_002dvalues-not-displayed-when-using-lmer_0028_0029_003f">Why are p-values not displayed when using lmer()?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.36 Why are there unwanted lines between polygons in PDF output viewed in Adobe Reader?</h3>

<p>Output from <code>polygon()</code> (and other functions calling
<code>polygon()</code>) with the argument <code>border=NA</code> or
<code>border="transparent"</code> should suppress border lines between
polygons for all graphics devices.

   <p>PDF output from R can be made in many ways, both directly, and through
for example Postscript or Windows Metafiles converted to PDF in external
software. In Adobe Reader, the default setting for line art, such as
polygons, is to smooth, which produces the impression of thin borders. 
Adobe Reader does this both for PDF files written by R or through other
software.

   <p>This is irritating, especially when using Adobe Reader for presentation. 
The unwanted effect can be removed by turning off smoothing for line
art: use the `Edit | Preferences | Page Display | Smooth line art' menu
in Adobe Reader 7.0.

<div class="node">
<p><hr>
<a name="Why-does-backslash-behave-strangely-inside-strings%3f"></a>
<a name="Why-does-backslash-behave-strangely-inside-strings_003f"></a>
Previous:&nbsp;<a rel="previous" accesskey="p" href="#Why-are-there-unwanted-lines-between-polygons-in-PDF-output-viewed-in-Adobe-Reader_003f">Why are there unwanted lines between polygons in PDF output viewed in Adobe Reader?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Miscellanea">R Miscellanea</a>

</div>

<h3 class="section">7.37 Why does backslash behave strangely inside strings?</h3>

<p>This question most often comes up in relation to file names (see
<a href="#How-do-file-names-work-in-Windows_003f">How do file names work in Windows?</a>)  but it also happens that
people complain that they cannot seem to put a single `<samp><span class="samp">\</span></samp>' character
into a text string unless it happens to be followed by certain other
characters.

   <p>To understand this, you have to distinguish between character strings
and <em>representations</em> of character strings.  Mostly, the
representation in R is just the string with a single or double quote at
either end, but there are strings that cannot be represented that way,
e.g., strings that themselves contains the quote character.  So

<pre class="example">     &gt; str &lt;- "This \"text\" is quoted"
     &gt; str
     [1] "This \"text\" is quoted"
     &gt; cat(str, "\n")
     This "text" is quoted
</pre>
   <p class="noindent">The <em>escape sequences</em> `<samp><span class="samp">\"</span></samp>' and `<samp><span class="samp">\n</span></samp>' represent a double
quote and the newline character respectively. Printing text strings,
using <code>print()</code> or by typing the name at the prompt will use the
escape sequences too, but the <code>cat()</code> function will display the
string as-is. Notice that `<samp><span class="samp">"\n"</span></samp>' is a one-character string, not
two; the backslash is not actually in the string, it is just generated
in the printed representation.

<pre class="example">     &gt; nchar("\n")
     [1] 1
     &gt; substring("\n", 1, 1)
     [1] "\n"
</pre>
   <p>So how do you put a backslash in a string? For this, you have to
escape the escape character. I.e., you have to double the backslash. 
as in

<pre class="example">     &gt; cat("\\n", "\n")
     \n
</pre>
   <p>Some functions, particularly those involving regular expression
matching, themselves use metacharacters, which may need to be escaped by
the backslash mechanism.  In those cases you may need a <em>quadruple</em>
backslash to represent a single literal one.

   <p>In versions of R up to 2.4.1 an unknown escape sequence like `<samp><span class="samp">\p</span></samp>'
was quietly interpreted as just `<samp><span class="samp">p</span></samp>'.  Current versions of R emit a
warning.

<div class="node">
<p><hr>
<a name="R-Programming"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#R-Bugs">R Bugs</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#R-Miscellanea">R Miscellanea</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Top">Top</a>

</div>

<h2 class="chapter">8 R Programming</h2>

<ul class="menu">
<li><a accesskey="1" href="#How-should-I-write-summary-methods_003f">How should I write summary methods?</a>
<li><a accesskey="2" href="#How-can-I-debug-dynamically-loaded-code_003f">How can I debug dynamically loaded code?</a>
<li><a accesskey="3" href="#How-can-I-inspect-R-objects-when-debugging_003f">How can I inspect R objects when debugging?</a>
<li><a accesskey="4" href="#How-can-I-change-compilation-flags_003f">How can I change compilation flags?</a>
<li><a accesskey="5" href="#How-can-I-debug-S4-methods_003f">How can I debug S4 methods?</a>
</ul>

<div class="node">
<p><hr>
<a name="How-should-I-write-summary-methods%3f"></a>
<a name="How-should-I-write-summary-methods_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-I-debug-dynamically-loaded-code_003f">How can I debug dynamically loaded code?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#R-Programming">R Programming</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Programming">R Programming</a>

</div>

<h3 class="section">8.1 How should I write summary methods?</h3>

<p>Suppose you want to provide a summary method for class <code>"foo"</code>. 
Then <code>summary.foo()</code> should not print anything, but return an
object of class <code>"summary.foo"</code>, <em>and</em> you should write a
method <code>print.summary.foo()</code> which nicely prints the summary
information and invisibly returns its object.  This approach is
preferred over having <code>summary.foo()</code> print summary information and
return something useful, as sometimes you need to grab something
computed by <code>summary()</code> inside a function or similar.  In such
cases you don't want anything printed.

<div class="node">
<p><hr>
<a name="How-can-I-debug-dynamically-loaded-code%3f"></a>
<a name="How-can-I-debug-dynamically-loaded-code_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-I-inspect-R-objects-when-debugging_003f">How can I inspect R objects when debugging?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-should-I-write-summary-methods_003f">How should I write summary methods?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Programming">R Programming</a>

</div>

<h3 class="section">8.2 How can I debug dynamically loaded code?</h3>

<p>Roughly speaking, you need to start R inside the debugger, load the
code, send an interrupt, and then set the required breakpoints.

   <p>See <a href="R-exts.html#Finding-entry-points">Finding entry points in dynamically loaded code</a>.

<div class="node">
<p><hr>
<a name="How-can-I-inspect-R-objects-when-debugging%3f"></a>
<a name="How-can-I-inspect-R-objects-when-debugging_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-I-change-compilation-flags_003f">How can I change compilation flags?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-I-debug-dynamically-loaded-code_003f">How can I debug dynamically loaded code?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Programming">R Programming</a>

</div>

<h3 class="section">8.3 How can I inspect R objects when debugging?</h3>

<p>The most convenient way is to call <code>R_PV</code> from the symbolic
debugger.

   <p>See <a href="R-exts.html#Inspecting-R-objects">Inspecting R objects when debugging</a>.

<div class="node">
<p><hr>
<a name="How-can-I-change-compilation-flags%3f"></a>
<a name="How-can-I-change-compilation-flags_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-can-I-debug-S4-methods_003f">How can I debug S4 methods?</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-I-inspect-R-objects-when-debugging_003f">How can I inspect R objects when debugging?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Programming">R Programming</a>

</div>

<h3 class="section">8.4 How can I change compilation flags?</h3>

<p>Suppose you have C code file for dynloading into R, but you want to use
<code>R CMD SHLIB</code> with compilation flags other than the default ones
(which were determined when R was built).

   <p>Starting with R 2.1.0, users can provide personal Makevars configuration
files in <samp><span class="file">$<samp>HOME</samp>/.R</span></samp> to override the default flags. 
See <a href="R-admin.html#Add_002don-packages">Add-on packages</a>.

   <p>For earlier versions of R, you could change the file
<samp><var>R_HOME</var><span class="file">/etc/Makeconf</span></samp> to reflect your preferences, or (at
least for systems using <acronym>GNU</acronym> Make) override them by the
environment variable <samp><span class="env">MAKEFLAGS</span></samp>. 
See <a href="R-exts.html#Creating-shared-objects">Creating shared objects</a>.

<div class="node">
<p><hr>
<a name="How-can-I-debug-S4-methods%3f"></a>
<a name="How-can-I-debug-S4-methods_003f"></a>
Previous:&nbsp;<a rel="previous" accesskey="p" href="#How-can-I-change-compilation-flags_003f">How can I change compilation flags?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Programming">R Programming</a>

</div>

<h3 class="section">8.5 How can I debug S4 methods?</h3>

<p>Use the <code>trace()</code> function with argument <code>signature=</code> to add
calls to the browser or any other code to the method that will be
dispatched for the corresponding signature.  See <code>?trace</code> for
details.

<div class="node">
<p><hr>
<a name="R-Bugs"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#Acknowledgments">Acknowledgments</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#R-Programming">R Programming</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Top">Top</a>

</div>

<h2 class="chapter">9 R Bugs</h2>

<ul class="menu">
<li><a accesskey="1" href="#What-is-a-bug_003f">What is a bug?</a>
<li><a accesskey="2" href="#How-to-report-a-bug">How to report a bug</a>
</ul>

<div class="node">
<p><hr>
<a name="What-is-a-bug%3f"></a>
<a name="What-is-a-bug_003f"></a>
Next:&nbsp;<a rel="next" accesskey="n" href="#How-to-report-a-bug">How to report a bug</a>,
Previous:&nbsp;<a rel="previous" accesskey="p" href="#R-Bugs">R Bugs</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Bugs">R Bugs</a>

</div>

<h3 class="section">9.1 What is a bug?</h3>

<p>If R executes an illegal instruction, or dies with an operating system
error message that indicates a problem in the program (as opposed to
something like &ldquo;disk full&rdquo;), then it is certainly a bug.  If you call
<code>.C()</code>, <code>.Fortran()</code>, <code>.External()</code> or <code>.Call()</code> (or
<code>.Internal()</code>) yourself (or in a function you wrote), you can
always crash R by using wrong argument types (modes).  This is not a
bug.

   <p>Taking forever to complete a command can be a bug, but you must make
certain that it was really R's fault.  Some commands simply take a long
time.  If the input was such that you <em>know</em> it should have been
processed quickly, report a bug.  If you don't know whether the command
should take a long time, find out by looking in the manual or by asking
for assistance.

   <p>If a command you are familiar with causes an R error message in a case
where its usual definition ought to be reasonable, it is probably a bug. 
If a command does the wrong thing, that is a bug.  But be sure you know
for certain what it ought to have done.  If you aren't familiar with the
command, or don't know for certain how the command is supposed to work,
then it might actually be working right.  For example, people sometimes
think there is a bug in R's mathematics because they don't understand
how finite-precision arithmetic works.  Rather than jumping to
conclusions, show the problem to someone who knows for certain. 
Unexpected results of comparison of decimal numbers, for example
<code>0.28 * 100 != 28</code> or <code>0.1 + 0.2 != 0.3</code>, are not a bug. 
See <a href="#Why-doesn_0027t-R-think-these-numbers-are-equal_003f">Why doesn't R think these numbers are equal?</a>, for more details.

   <p>Finally, a command's intended definition may not be best for statistical
analysis.  This is a very important sort of problem, but it is also a
matter of judgment.  Also, it is easy to come to such a conclusion out
of ignorance of some of the existing features.  It is probably best not
to complain about such a problem until you have checked the
documentation in the usual ways, feel confident that you understand it,
and know for certain that what you want is not available.  If you are
not sure what the command is supposed to do after a careful reading of
the manual this indicates a bug in the manual.  The manual's job is to
make everything clear.  It is just as important to report documentation
bugs as program bugs.  However, we know that the introductory
documentation is seriously inadequate, so you don't need to report this.

   <p>If the online argument list of a function disagrees with the manual, one
of them must be wrong, so report the bug.

<div class="node">
<p><hr>
<a name="How-to-report-a-bug"></a>
Previous:&nbsp;<a rel="previous" accesskey="p" href="#What-is-a-bug_003f">What is a bug?</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#R-Bugs">R Bugs</a>

</div>

<h3 class="section">9.2 How to report a bug</h3>

<p>When you decide that there is a bug, it is important to report it and to
report it in a way which is useful.  What is most useful is an exact
description of what commands you type, starting with the shell command
to run R, until the problem happens.  Always include the version of R,
machine, and operating system that you are using; type <kbd>version</kbd> in
R to print this.

   <p>The most important principle in reporting a bug is to report
<em>facts</em>, not hypotheses or categorizations.  It is always easier to
report the facts, but people seem to prefer to strain to posit
explanations and report them instead.  If the explanations are based on
guesses about how R is implemented, they will be useless; others will
have to try to figure out what the facts must have been to lead to such
speculations.  Sometimes this is impossible.  But in any case, it is
unnecessary work for the ones trying to fix the problem.

   <p>For example, suppose that on a data set which you know to be quite large
the command

<pre class="example">     R&gt; data.frame(x, y, z, monday, tuesday)
</pre>
   <p class="noindent">never returns.  Do not report that <code>data.frame()</code> fails for large
data sets.  Perhaps it fails when a variable name is a day of the week. 
If this is so then when others got your report they would try out the
<code>data.frame()</code> command on a large data set, probably with no day of
the week variable name, and not see any problem.  There is no way in the
world that others could guess that they should try a day of the week
variable name.

   <p>Or perhaps the command fails because the last command you used was a
method for <code>"["()</code> that had a bug causing R's internal data
structures to be corrupted and making the <code>data.frame()</code> command
fail from then on.  This is why others need to know what other commands
you have typed (or read from your startup file).

   <p>It is very useful to try and find simple examples that produce
apparently the same bug, and somewhat useful to find simple examples
that might be expected to produce the bug but actually do not.  If you
want to debug the problem and find exactly what caused it, that is
wonderful.  You should still report the facts as well as any
explanations or solutions.  Please include an example that reproduces
the problem, preferably the simplest one you have found.

   <p>Invoking R with the <samp><span class="option">--vanilla</span></samp> option may help in isolating a
bug.  This ensures that the site profile and saved data files are not
read.

   <p>Before you actually submit a bug report, you should check whether the
bug has already been reported and/or fixed.  First, try the &ldquo;Search
Existing Reports&rdquo; facility in the Bug Tracking page at
<a href="http://bugs.R-project.org/">http://bugs.R-project.org/</a>.  Second, consult
<a href="https://svn.R-project.org/R/trunk/NEWS">https://svn.R-project.org/R/trunk/NEWS</a>, which records changes that
will appear in the <em>next</em> release of R, including some bug fixes
that do not appear in Bug Tracking.  (Windows users should additionally
consult
<a href="https://svn.R-project.org/R/trunk/src/gnuwin32/CHANGES">https://svn.R-project.org/R/trunk/src/gnuwin32/CHANGES</a>.) 
Third, if possible try the current r-patched or r-devel version of R. 
If a bug has already been reported or fixed, please do not submit
further bug reports on it.  Finally, check carefully whether the bug is
with R, or a contributed package.  Bug reports on contributed packages
should be sent first to the package maintainer, and only submitted to
the R-bugs repository by package maintainers, mentioning the package in
the subject line.

   <p>On Unix systems a bug report can be generated using the function
<code>bug.report()</code>.  This automatically includes the version
information and sends the bug to the correct address.  Alternatively the
bug report can be emailed to <a href="mailto:R-bugs@R-project.org">R-bugs@R-project.org</a> or submitted
to the Web page at <a href="http://bugs.R-project.org/">http://bugs.R-project.org/</a>.  Please try
including results of <code>sessionInfo()</code> in your bug report.

   <p>There is a section of the bug repository for suggestions for
enhancements for R labelled `<samp><span class="samp">wishlist</span></samp>'.  Suggestions can be
submitted in the same ways as bugs, but please ensure that the subject
line makes clear that this is for the wishlist and not a bug report, for
example by starting with `<samp><span class="samp">Wishlist:</span></samp>'.

   <p>Comments on and suggestions for the Windows port of R should be sent to
<a href="mailto:R-windows@R-project.org">R-windows@R-project.org</a>.

   <p>Corrections to and comments on message translation should be sent to the
last translator (listed at the top of the appropriate `<samp><span class="samp">.po</span></samp>' file)
or to the translation team as listed at
<a href="http://developer.R-project.org/TranslationTeams.html">http://developer.R-project.org/TranslationTeams.html</a>.

<div class="node">
<p><hr>
<a name="Acknowledgments"></a>
Previous:&nbsp;<a rel="previous" accesskey="p" href="#R-Bugs">R Bugs</a>,
Up:&nbsp;<a rel="up" accesskey="u" href="#Top">Top</a>

</div>

<h2 class="chapter">10 Acknowledgments</h2>

<p>Of course, many many thanks to Robert and Ross for the R system, and to
the package writers and porters for adding to it.

   <p>Special thanks go to Doug Bates, Peter Dalgaard, Paul Gilbert, Stefano
Iacus, Fritz Leisch, Jim Lindsey, Thomas Lumley, Martin Maechler, Brian
D. Ripley, Anthony Rossini, and Andreas Weingessel for their comments
which helped me improve this <acronym>FAQ</acronym>.

   <p>More to come soon <small class="dots">...</small>

</body></html>

